# 实例展示

	项目介绍：GVHD数据，有32个样本，测了宏基因组，16S和代谢数据。

## WGCNA demo

### load package
```{r message=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(xlsx)
library(WGCNA)
library(igraph)
library(phyloseq)
library(DT)

options(stringsAsFactors = FALSE)
```

### load data

- metagenomics
- metabolic
- metadata

```{r}
mgs_profile <- read.table("/share/projects/Analytics/analytics/MultiOmics/tools/HAllA/GvHD/metaphlan2_merged.tsv", header = TRUE, sep = "\t") # read metagenomic hirerachy table，only need ID column
metabolites_level <- read.xlsx("/share/projects/Analytics/analytics/MultiOmics/tools/HAllA/GvHD/ALL_sample_data.xlsx", sheetIndex = 1) # read metabolic hirerachy table
mgs <- read.table("/share/projects/Analytics/analytics/MultiOmics/tools/HAllA/GvHD/metagenomics.txt", header = TRUE, sep = "\t") # read metagenomic species level data
metabolites_data <- read.table("/share/projects/Analytics/analytics/MultiOmics/tools/HAllA/GvHD/metabolic.txt", header = TRUE, sep = "\t") # read metabolic data
metadata <- read.table("/share/projects/Analytics/analytics/MultiOmics/tools/HAllA/GvHD/metadata.txt", header = TRUE, sep = "\t") # read metadata
```

* mgs_profile need taxanomy level column, as shwon in the table below

```{r}
datatable(mgs_profile[1:10, 1:10])
```

* Metabolic level data are selected as the Class.I and Class.II column in the following analysis, the details can be adjusted according to the data reality

```{r}
datatable(metabolites_level[1:10, 1:10])
```

mgs species: row names are Features, column names are Sample

metabolites_data: row names are Features, column names are Sample

### restructured the hierarchy table

```{r}
# taxa hierarchy table
mgs_species <- mgs_profile %>%
  .[sapply(str_split(mgs_profile$ID, "\\|"), function(x) length(x) == 7), ] 
taxa_table <- str_split_fixed(mgs_species$ID,"\\|", 7) %>% as.data.frame()
colnames(taxa_table) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
rownames(taxa_table) <- str_replace(taxa_table$Species, "s__", "")

# metabolic hierarchy table
metabolites_level <- metabolites_level %>% select(Index, Class.I, Class.II)
metabolites_level <- metabolites_level %>% column_to_rownames("Index")
```

### mgs standardization

#### check data format

```{r}
cat("[mgs]", paste(c("Feature: ", "Sample: "), dim(mgs)))
cat("[metabolites]", paste0(c("Feature: ", "Sample: "), dim(metabolites_data)))
```

#### using CSS method

```{r}
data.metagenomeSeq <- metagenomeSeq::newMRexperiment(mgs) # Variables as rows, samples as columns
p <- metagenomeSeq::cumNormStat(data.metagenomeSeq)
data.cumnorm <- metagenomeSeq::cumNorm(data.metagenomeSeq, p=p)
mgs <- metagenomeSeq::MRcounts(data.cumnorm, norm=TRUE, log=TRUE) # log here is a +1 shifted log2

dim(mgs)
```

### mgs constructe module

```{r}
mgs_t <- mgs %>% t() # Samples as rows, Features as columns
```

#### select power

```{r message=FALSE, warning=FALSE}
powers <- c(1:10, seq(12,20,2)) # default

suppressWarnings(sft <- pickSoftThreshold(mgs_t, 
                                          powerVector = powers, 
                                          verbose = 1, 
                                          networkType = "signed",
                                          corFn= "bicor"))


# Find the soft thresholding power beta to which co-expression similarity is raised to calculate adjacency.
# based on the criterion of approximate scale-free topology.

idx <- min(which((-sign(sft$fitIndices[,3])*sft$fitIndices[,2]) > 0.90))
if(is.infinite(idx)){
  idx <- min(which((-sign(sft$fitIndices[,3])*sft$fitIndices[,2]) > 0.80))
  if(!is.infinite(idx)){
    st <- sft$fitIndices[idx,1]
  } else{
    idx <- which.max(-sign(sft$fitIndices[,3])*sft$fitIndices[,2])
    st <- sft$fitIndices[idx,1]
  }
} else{
  st <- sft$fitIndices[idx,1]
}


# Plot Scale independence measure and Mean connectivity measure

# Scale-free topology fit index as a function of the soft-thresholding power
data.frame(Indices = sft$fitIndices[,1],
           sfApprox = -sign(sft$fitIndices[,3])*sft$fitIndices[,2]) %>% 
  ggplot() + 
  geom_hline(yintercept = 0.9, color = "red", alpha = 0.6) + # corresponds to R^2 cut-off of 0.9
  geom_hline(yintercept = 0.8, color = "red", alpha = 0.2) + # corresponds to R^2 cut-off of 0.8
  geom_line(aes(x = Indices, y = sfApprox), color = "red", alpha = 0.1, size = 2.5) +
  geom_text(mapping = aes(x = Indices, y = sfApprox, label = Indices), color = "red", size = 4) +
  ggtitle("Scale independence") +
  xlab("Soft Threshold (power)") +
  ylab("SF Model Fit,signed R^2") +
  xlim(1,20) +
  ylim(-1,1) +
  geom_segment(aes(x = st, y = 0.25, xend = st, yend = sfApprox[idx]-0.05), 
               arrow = arrow(length = unit(0.2,"cm")), 
               size = 0.5)-> scale_independence_plot 
  
 


# Mean connectivity as a function of the soft-thresholding power

data.frame(Indices = sft$fitIndices[,1],
           meanApprox = sft$fitIndices[,5]) %>% 
  ggplot() + 
  geom_line(aes(x = Indices, y = meanApprox), color = "red", alpha = 0.1, size = 2.5) +
  geom_text(mapping = aes(x = Indices, y = meanApprox, label = Indices), color = "red", size = 4) +
  xlab("Soft Threshold (power)") +
  ylab("Mean Connectivity") +
  geom_segment(aes(x = st-0.4, 
                   y = sft$fitIndices$mean.k.[idx], 
                   xend = 0, 
                   yend = sft$fitIndices$mean.k.[idx]),
               arrow = arrow(length = unit(0.2,"cm")), 
               size = 0.4) +
  ggtitle(paste0("Mean connectivity: ", 
                 round(sft$fitIndices$mean.k.[idx],2))) -> mean_connectivity_plot


cowplot::plot_grid(scale_independence_plot, mean_connectivity_plot, ncol = 2, align = "h", labels = c("A", "B"), label_size = 15) -> si_mc_plot

si_mc_plot
```

The number closest to the 0.8 line is 18, and the constructed adjacency matrix most closely fits with scale-free topology. Therefore power is `r st`.

#### Block-wise network construction and module detection

The function `blockwiseModules` will first pre cluster with fast crude clustering method to cluster OTUs into blocks not exceeding the maximum, blocks may therefore not be fully optimal in the end.

Change the parameters here to better reflect your own data.
```{r message=FALSE, warning=FALSE}

if(!dir.exists("./result/")) {
  dir.create("./result/")
}

Run_analysis <- TRUE
if(Run_analysis){
  modules.mgs <- blockwiseModules(mgs_t,
                          power = st, 
                          networkType = "signed", 
                          TOMType = "signed",
                          corType = 'bicor',
                          maxPOutliers = 0.05,
                          deepSplit = 2,
                          pamStage = FALSE, 
                          pamRespectsDendro = TRUE,
                          mergeCutHeight = 0.25,
                          replaceMissingAdjacencies = TRUE,
                          minModuleSize = 5, # There are fewer otus than genes, and that many might not be connected
                          numericLabels = TRUE,
                          saveTOMs = TRUE,
                          saveTOMFileBase = "./result/modules.mgs", #"/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/GVHD/bookdown_GVHD/result/modules.mgs",
                          verbose = 1)
  
  rownames(modules.mgs$MEs) <- rownames(mgs_t)
  names(modules.mgs$colors) <- colnames(mgs_t)
  names(modules.mgs$unmergedColors) <- colnames(mgs_t)
  
  hubs.mgs <- chooseTopHubInEachModule(mgs_t, modules.mgs$colors)
}
```

#### mgs module details

```{r}
# Convert labels to colors for plotting
merged_colors <- labels2colors(modules.mgs$colors)
```

```{r include=FALSE}
n_modules <- unique(merged_colors) %>% length()

samples_good <- sum(modules.mgs$goodSamples) == length(modules.mgs$goodSamples)
OTUs_good <- sum(modules.mgs$goodGenes) == length(modules.mgs$goodGenes)

ME_good <- sum(modules.mgs$MEsOK) == length(modules.mgs$MEsOK)
```

`r ifelse(samples_good, "All samples are OK.","Not all samples are OK.")`  
`r ifelse(OTUs_good, "All OTUs are OK.","Not all OTUs are OK.")`  

There where `r n_modules` modules found.  
`r ifelse(ME_good, "All module eigenOTUs are OK.","Not all module eigenOTUs are OK.")`  

How many OTUs are there in each module?
```{r fig.height=5, fig.width=5}
table(modules.mgs$colors) %>% 
  as.data.frame() %>% 
  dplyr::rename(Module = Var1, Size = Freq) %>% 
  dplyr::mutate(Module_color = labels2colors(as.numeric(as.character(Module)))) -> module_size

module_size %>% 
  ggplot(aes(x = Module, y = Size, fill = Module)) +
  geom_col(color =  "#000000") +
  ggtitle("Number of OTUs in each module") +
  theme(legend.position = "none") + 
  scale_fill_manual(values = setNames(module_size$Module_color,module_size$Module)) +
  geom_text(aes(label = Size),vjust = 0.5, hjust = -0.18, size = 3.5) +
  ylim(0, max(module_size$Size)*1.1) +
  theme(plot.margin = margin(2, 2, 2, 2, "pt")) +
  coord_flip()-> module_size_barplot

module_size_barplot

```

cluster relationships between the module

```{r}
# labels2colors(modules.mgs$colors)

plotEigengeneNetworks(modules.mgs$MEs, "Eigengene adjacency heatmap",
marDendro = c(3,3,2,4),
marHeatmap = c(3,4,2,2), plotDendrograms = T,
xLabelsAngle = 90)
```

```{r}
table(modules.mgs$colors) %>% as.data.frame() -> res
res$`Module color` <- WGCNA::labels2colors(as.numeric(as.character(res$Var1)))
res <- res[, c(1,3,2)]
colnames(res) <- c("Module", "Module color", "Number of OTUs")
res %>% datatable()
```

Dendrogram and module colors

```{r fig.height=5, fig.width=10}
# Plot the dendrogram and the module colors underneath for each block
for(i in seq_along(modules.mgs$dendrograms)){
  plotDendroAndColors(modules.mgs$dendrograms[[i]], merged_colors[modules.mgs$blockGenes[[i]]],
                      "Module colors",
                      dendroLabels = FALSE, hang = 0.03,
                      addGuide = TRUE, guideHang = 0.05,
                      main = paste0("Cluster Dendrogram\n", 
                                    "for block ", 
                                    i,": ",
                                    length(modules.mgs$blockGenes[[i]]),
                                    " OTUs"))
}
```

Module (Eigengene) correlation

```{r message=FALSE}
MEs <- modules.mgs$MEs

# Module correlation to other modules
MEs_R <- bicor(MEs, MEs, maxPOutliers = 0.05)

idx.r <- which(rownames(MEs_R) == "ME0")
idx.c <- which(colnames(MEs_R) == "ME0")

MEs_R_noME0 <- MEs_R[-idx.r, -idx.c]
```


```{r fig.height=5, fig.width=10}
MEs_R[upper.tri(MEs_R_noME0)] %>% 
  as.data.frame() %>% 
  dplyr::rename("correlation" = ".") %>% 
  ggplot(aes(x=correlation)) + 
  geom_density() + 
  ggtitle(paste0("mgs","ME correlation density\n without ","mgs" ,"ME0")) -> MEs_R_density

pheatmap::pheatmap(MEs_R, color = colorRampPalette(c("Blue", "White", "Red"))(100),
                   silent = T, 
                   breaks = seq(-1,1,length.out = 101),
                   treeheight_row = 5, 
                   treeheight_col = 5,
                   main = paste0("mgs","ME correlation heatmap"),
                   labels_row = paste0("mgs", rownames(MEs_R)),
                   labels_col = paste0("mgs", colnames(MEs_R))) -> MEs_R_Corr

cowplot::plot_grid(MEs_R_density, MEs_R_Corr$gtable, labels = c("D", "E"), label_size = 15, rel_widths = c(0.6, 1)) -> density_eigen

density_eigen
```

```{r message=FALSE, warning=FALSE}
all(rownames(mgs_t) == rownames(MEs))
dim(mgs_t) %>% paste0(c(" samples", " OTUs"))
kME <- bicor(mgs_t, MEs, maxPOutliers = 0.05)
dim(kME) %>% paste0(c(" OTUs", " modules"))
```

Show a plot of the intra modular correlation; How the OTUs within a module correlates to the module eigengene.

```{r}
intra_cor <- c()
for (i in 1:ncol(mgs_t)) {
  m <- modules.mgs$colors[i]
  intra_cor[i] <- kME[i, paste0("ME", m)]
  if(m != 0){
    intra_cor[i] <- kME[i, paste0("ME", m)]
  } else{
    intra_cor[i] <- NA
  }
  
}

idx <- which(is.na(intra_cor))
intra_cor <- intra_cor[-idx]

plot(density(intra_cor), main = "Correlations with module-eigenOTU (within module correlation)\nNo ME0", xlim = c(-1,1))
```

Show the same thing, but for each module individually, and color by module color.

```{r}
# Corr within modules
corr_within_module <- function(mgs_t, modules, module_x = 1){
  idx.omics_data <- which(modules$colors == module_x)
  idx.me <- which(colnames(modules$MEs) == paste0("ME",module_x))
  kME_x <- bicor(mgs_t[,idx.omics_data], modules$MEs[,idx.me], maxPOutliers = 0.05)
  kME_x
}

ggplot.list <- list()

for(m in colnames(modules.mgs$MEs)){
  h <- as.numeric(sub("ME","", m))
  data.frame(x = suppressWarnings(corr_within_module(mgs_t = mgs_t, modules = modules.mgs, module_x = h))) %>% 
    ggplot() + 
    geom_density(aes(x = x), fill = labels2colors(h), color = "black", alpha = 0.5) + 
    xlim(-1, 1) +
    xlab("OTU correlation")+
    ggtitle(paste0("mgs",m)) -> da_plot
  
  ggplot.list[[m]] <- da_plot
}

ggplot.list <- ggplot.list[ggplot.list %>% names() %>% sub("ME", "", .) %>% as.numeric() %>% order()]
```

```{r fig.height=9, fig.width=8}
cowplot::plot_grid(plotlist = ggplot.list, ncol = 2) -> density_all_plot
density_all_plot
```

Combine to one plot

```{r fig.height=15, fig.width=12}
cowplot::plot_grid(si_mc_plot , density_eigen, ncol = 1, rel_heights = c(0.8,1)) -> part_1


cowplot::plot_grid(part_1, module_size_barplot, labels = c("", "C"), label_size = 15, rel_widths = c(1,0.5)) -> part_2


cowplot::plot_grid(part_2, density_all_plot, ncol = 1, rel_heights = c(0.8,1), labels = c("", "F"), label_size = 15)
```

Hub genes

For each module it is possible to pick a hub gene with the function `chooseTopHubInEachModule`.


```{r}
hubs.mgs %>% 
  as.data.frame() %>% 
  dplyr::rename("OTU_name" = ".") %>%
  tibble::rownames_to_column(var = "Module") -> hubOTUs



dplyr::left_join(hubOTUs, 
                 (taxa_table %>%
                    tibble::rownames_to_column(var = "OTU_name")), 
                 by = "OTU_name") -> hubOTUs

hubOTUs %>% datatable()
```

### metabolites constructe module

```{r}
metabolites_t <- metabolites_data %>% t()
```

#### select power

```{r message=FALSE, warning=FALSE}
powers <- c(1:10, seq(12,20,2))

suppressWarnings(sft <- pickSoftThreshold(metabolites_t, 
                                          powerVector = powers, 
                                          verbose = 1, 
                                          networkType = "signed",
                                          corFn= "bicor"))


# Find the soft thresholding power beta to which co-expression similarity is raised to calculate adjacency.
# based on the criterion of approximate scale-free topology.

idx <- min(which((-sign(sft$fitIndices[,3])*sft$fitIndices[,2]) > 0.90))
if(is.infinite(idx)){
  idx <- min(which((-sign(sft$fitIndices[,3])*sft$fitIndices[,2]) > 0.80))
  if(!is.infinite(idx)){
    st <- sft$fitIndices[idx,1]
  } else{
    idx <- which.max(-sign(sft$fitIndices[,3])*sft$fitIndices[,2])
    st <- sft$fitIndices[idx,1]
  }
} else{
  st <- sft$fitIndices[idx,1]
}


# Plot Scale independence measure and Mean connectivity measure

# Scale-free topology fit index as a function of the soft-thresholding power
data.frame(Indices = sft$fitIndices[,1],
           sfApprox = -sign(sft$fitIndices[,3])*sft$fitIndices[,2]) %>% 
  ggplot() + 
  geom_hline(yintercept = 0.9, color = "red", alpha = 0.6) + # corresponds to R^2 cut-off of 0.9
  geom_hline(yintercept = 0.8, color = "red", alpha = 0.2) + # corresponds to R^2 cut-off of 0.8
  geom_line(aes(x = Indices, y = sfApprox), color = "red", alpha = 0.1, size = 2.5) +
  geom_text(mapping = aes(x = Indices, y = sfApprox, label = Indices), color = "red", size = 4) +
  ggtitle("Scale independence") +
  xlab("Soft Threshold (power)") +
  ylab("SF Model Fit,signed R^2") +
  xlim(1,20) +
  ylim(-1,1) +
  geom_segment(aes(x = st, y = 0.25, xend = st, yend = sfApprox[idx]-0.05), 
               arrow = arrow(length = unit(0.2,"cm")), 
               size = 0.5)-> scale_independence_plot 
  
 


# Mean connectivity as a function of the soft-thresholding power

data.frame(Indices = sft$fitIndices[,1],
           meanApprox = sft$fitIndices[,5]) %>% 
  ggplot() + 
  geom_line(aes(x = Indices, y = meanApprox), color = "red", alpha = 0.1, size = 2.5) +
  geom_text(mapping = aes(x = Indices, y = meanApprox, label = Indices), color = "red", size = 4) +
  xlab("Soft Threshold (power)") +
  ylab("Mean Connectivity") +
  geom_segment(aes(x = st-0.4, 
                   y = sft$fitIndices$mean.k.[idx], 
                   xend = 0, 
                   yend = sft$fitIndices$mean.k.[idx]),
               arrow = arrow(length = unit(0.2,"cm")), 
               size = 0.4) +
  ggtitle(paste0("Mean connectivity: ", 
                 round(sft$fitIndices$mean.k.[idx],2))) -> mean_connectivity_plot


cowplot::plot_grid(scale_independence_plot, mean_connectivity_plot, ncol = 2, align = "h", labels = c("A", "B"), label_size = 15) -> si_mc_plot

si_mc_plot
```

The power is `r st`

#### Block-wise network construction and module detection

The function `blockwiseModules` will first pre cluster with fast crude clustering method to cluster OTUs into blocks not exceeding the maximum, blocks may therefore not be fully optimal in the end.

Change the parameters here to better reflect your own data.
```{r message=FALSE, warning=FALSE}
Run_analysis <- TRUE
if(Run_analysis){
  modules.metabolites <- blockwiseModules(metabolites_t,
                          power = st, 
                          networkType = "signed", 
                          TOMType = "signed",
                          corType = 'bicor',
                          maxPOutliers = 0.05,
                          deepSplit = 2,
                          pamStage = FALSE, 
                          pamRespectsDendro = TRUE,
                          mergeCutHeight = 0.25,
                          replaceMissingAdjacencies = TRUE,
                          minModuleSize = 5, # There are fewer otus than genes, and that many might not be connected
                          numericLabels = TRUE,
                          saveTOMs = TRUE,
                          saveTOMFileBase = "./result/module.metabolites", #"/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/GVHD/bookdown_GVHD/result/module.metabolites",
                          verbose = 1)
  
  rownames(modules.metabolites$MEs) <- rownames(metabolites_t)
  names(modules.metabolites$colors) <- colnames(metabolites_t)
  names(modules.metabolites$unmergedColors) <- colnames(metabolites_t)
  
  hubs.metabolites <- chooseTopHubInEachModule(metabolites_t, modules.metabolites$colors)
}
```

#### metabolites module details

```{r}
# Convert labels to colors for plotting
merged_colors <- labels2colors(modules.metabolites$colors)
```

```{r include=FALSE}
n_modules <- unique(merged_colors) %>% length()

samples_good <- sum(modules.metabolites$goodSamples) == length(modules.metabolites$goodSamples)
OTUs_good <- sum(modules.metabolites$goodGenes) == length(modules.metabolites$goodGenes)

ME_good <- sum(modules.metabolites$MEsOK) == length(modules.metabolites$MEsOK)
```

`r ifelse(samples_good, "All samples are OK.","Not all samples are OK.")`  
`r ifelse(OTUs_good, "All OTUs are OK.","Not all OTUs are OK.")`  

There where `r n_modules` modules found.  
`r ifelse(ME_good, "All module eigenOTUs are OK.","Not all module eigenOTUs are OK.")`  

How many Metabolites are there in each module?
```{r fig.height=5, fig.width=5}
table(modules.metabolites$colors) %>% 
  as.data.frame() %>% 
  dplyr::rename(Module = Var1, Size = Freq) %>% 
  dplyr::mutate(Module_color = labels2colors(as.numeric(as.character(Module)))) -> module_size

module_size %>% 
  ggplot(aes(x = Module, y = Size, fill = Module)) +
  geom_col(color =  "#000000") +
  ggtitle("Number of Metabolites in each module") +
  theme(legend.position = "none") + 
  scale_fill_manual(values = setNames(module_size$Module_color,module_size$Module)) +
  geom_text(aes(label = Size),vjust = 0.5, hjust = -0.18, size = 3.5) +
  ylim(0, max(module_size$Size)*1.1) +
  theme(plot.margin = margin(2, 2, 2, 2, "pt")) +
  coord_flip()-> module_size_barplot

module_size_barplot

```

cluster relationships between the module

```{r}
# labels2colors(modules.metabolites$colors)

plotEigengeneNetworks(modules.metabolites$MEs, "Eigengene adjacency heatmap",
marDendro = c(3,3,2,4),
marHeatmap = c(3,4,2,2), plotDendrograms = T,
xLabelsAngle = 90)
```

```{r eval=TRUE, include=TRUE}
table(modules.metabolites$colors) %>% as.data.frame() -> res
res$`Module color` <- WGCNA::labels2colors(as.numeric(as.character(res$Var1)))
res <- res[, c(1,3,2)]
colnames(res) <- c("Module", "Module color", "Number of metabolic")
res %>% datatable()
```

Dendrogram and module colors

```{r fig.height=5, fig.width=10}
# Plot the dendrogram and the module colors underneath for each block
for(i in seq_along(modules.metabolites$dendrograms)){
  plotDendroAndColors(modules.metabolites$dendrograms[[i]], merged_colors[modules.metabolites$blockGenes[[i]]],
                      "Module colors",
                      dendroLabels = FALSE, hang = 0.03,
                      addGuide = TRUE, guideHang = 0.05,
                      main = paste0("Cluster Dendrogram\n", 
                                    "for block ", 
                                    i,": ",
                                    length(modules.metabolites$blockGenes[[i]]),
                                    " metabolic"))
}
```

Module (Eigengene) correlation

```{r message=FALSE}
MEs <- modules.metabolites$MEs

# Module correlation to other modules
MEs_R <- bicor(MEs, MEs, maxPOutliers = 0.05)

idx.r <- which(rownames(MEs_R) == "ME0")
idx.c <- which(colnames(MEs_R) == "ME0")

MEs_R_noME0 <- MEs_R[-idx.r, -idx.c]
```

```{r fig.height=5, fig.width=10}
MEs_R[upper.tri(MEs_R_noME0)] %>% 
  as.data.frame() %>% 
  dplyr::rename("correlation" = ".") %>% 
  ggplot(aes(x=correlation)) + 
  geom_density() + 
  ggtitle(paste0("mbs","ME correlation density\n without ","mbs" ,"ME0")) -> MEs_R_density

pheatmap::pheatmap(MEs_R, color = colorRampPalette(c("Blue", "White", "Red"))(100),
                   silent = T, 
                   breaks = seq(-1,1,length.out = 101),
                   treeheight_row = 5, 
                   treeheight_col = 5,
                   main = paste0("mbs","ME correlation heatmap"),
                   labels_row = paste0("mbs", rownames(MEs_R)),
                   labels_col = paste0("mbs", colnames(MEs_R))) -> MEs_R_Corr

cowplot::plot_grid(MEs_R_density, MEs_R_Corr$gtable, labels = c("D", "E"), label_size = 15, rel_widths = c(0.6, 1)) -> density_eigen

density_eigen
```

```{r message=FALSE, warning=FALSE}
all(rownames(metabolites_t) == rownames(MEs))
dim(metabolites_t) %>% paste0(c(" samples", " metabolic"))
kME <- bicor(metabolites_t, MEs, maxPOutliers = 0.05)
dim(kME) %>% paste0(c(" metabolic", " modules"))
```

Show a plot of the intra modular correlation; How the OTUs within a module correlates to the module eigengene.

```{r}
intra_cor <- c()
for (i in 1:ncol(metabolites_t)) {
  m <- modules.metabolites$colors[i]
  intra_cor[i] <- kME[i, paste0("ME", m)]
  if(m != 0){
    intra_cor[i] <- kME[i, paste0("ME", m)]
  } else{
    intra_cor[i] <- NA
  }
  
}

idx <- which(is.na(intra_cor))
intra_cor <- intra_cor[-idx]

plot(density(intra_cor), main = "Correlations with module-eigenMBS (within module correlation)\nNo ME0", xlim = c(-1,1))
```

Show the same thing, but for each module individually, and color by module color.

```{r}
# Corr within modules
corr_within_module <- function(metabolites_t, modules, module_x = 1){
  idx.omics_data <- which(modules$colors == module_x)
  idx.me <- which(colnames(modules$MEs) == paste0("ME",module_x))
  kME_x <- bicor(metabolites_t[,idx.omics_data], modules$MEs[,idx.me], maxPOutliers = 0.05)
  kME_x
}

ggplot.list <- list()

for(m in colnames(modules.metabolites$MEs)){
  h <- as.numeric(sub("ME","", m))
  data.frame(x = suppressWarnings(corr_within_module(metabolites_t = metabolites_t, modules = modules.metabolites, module_x = h))) %>% 
    ggplot() + 
    geom_density(aes(x = x), fill = labels2colors(h), color = "black", alpha = 0.5) + 
    xlim(-1, 1) +
    xlab("metabolic correlation")+
    ggtitle(paste0("mbs",m)) -> da_plot
  
  ggplot.list[[m]] <- da_plot
}

ggplot.list <- ggplot.list[ggplot.list %>% names() %>% sub("ME", "", .) %>% as.numeric() %>% order()]
```

```{r fig.height=9, fig.width=8}
cowplot::plot_grid(plotlist = ggplot.list, ncol = 5) -> density_all_plot # ncol 可以根据module的大小调整
density_all_plot
```

Combine to one plot

```{r fig.height=15, fig.width=12}
cowplot::plot_grid(si_mc_plot , density_eigen, ncol = 1, rel_heights = c(0.8,1)) -> part_1


cowplot::plot_grid(part_1, module_size_barplot, labels = c("", "C"), label_size = 15, rel_widths = c(1,0.5)) -> part_2


cowplot::plot_grid(part_2, density_all_plot, ncol = 1, rel_heights = c(0.8,1), labels = c("", "F"), label_size = 15)
```

Hub metabolic

For each module it is possible to pick a hub metabolic with the function `chooseTopHubInEachModule`.


```{r}
hubs.metabolites %>% 
  as.data.frame() %>% 
  dplyr::rename("mbs_name" = ".") %>%
  tibble::rownames_to_column(var = "Module") -> hubMBS



dplyr::left_join(hubMBS, 
                 (metabolites_level %>%
                    tibble::rownames_to_column(var = "mbs_name")), 
                 by = "mbs_name") -> hubMBS

hubMBS %>% datatable()
```

### metabolites, mgs and metadata association

```{r}
X_eigengenes <- modules.metabolites$MEs # X_eigengenes is metabolic
Y_eigengenes <- modules.mgs$MEs # Y_eigengenes is mgs
```

```{r}
# Create a dendrogram of the metabolites eigengenes to organise the final plots.
X_ME_dendro <- hclust(as.dist(1 - WGCNA::bicor(X_eigengenes, maxPOutliers = 0.05)), method = "ward.D2")
```

```{r}
heatmap_colors <- colorRampPalette(c("#18b29f","#FFFFFF","#ac6721"), interpolate = "spline", space = "rgb")(51)

annotation_col <- metadata %>% 
  t() %>%
  as.data.frame() %>%
  select(V1_outcome, GVHD_type, Donor)
  

annotation_colors <- list(
  V1_outcome = c(`CR` = "#F08A46", `PR` = "#8EB470", `NR` = "#B7CFA4"),
  GVHD_type = c(`aGVHD` = "red", `cGVHD` = "green"),
  Donor = c(`DO1` = "paleturquoise", `DO2` = "palevioletred")
)

X_eigengenes_to_plot <- 
  dplyr::inner_join(annotation_col %>% 
               rownames_to_column(var = "sampleName"), 
             X_eigengenes %>% 
               rownames_to_column(var = "sampleName"), 
             by = "sampleName") %>%
  dplyr::arrange(V1_outcome, GVHD_type, Donor) %>%              # The order at which the columns should appear, given that there is no clustering.
  dplyr::select(sampleName, starts_with("ME")) %>% 
  tibble::column_to_rownames(var = "sampleName") %>% 
  t()

pheatmap::pheatmap(X_eigengenes_to_plot, 
                   cluster_cols = TRUE,
                   cluster_rows = X_ME_dendro,
                   treeheight_row = 20,
                   cutree_rows = 4,
                   cutree_cols = 4,
                   color = heatmap_colors,
                   fontsize = 10,
                   fontsize_col = 6,
                   annotation_colors = annotation_colors,
                   annotation_col = annotation_col, 
                   silent = F,
                   labels_row = paste0("mb", rownames(X_eigengenes_to_plot)),
                   main = paste("Metabolites Module 'expression'\n")) -> X_plot
```

##### Correlate modules from metabolic and metagenomics.

```{r}
p.value_matr <- corr.value_matr <- matrix(ncol = ncol(Y_eigengenes), 
                                          nrow = ncol(X_eigengenes), 
                                          dimnames = list(colnames(X_eigengenes), 
                                                          colnames(Y_eigengenes)))


for(i in 1:ncol(X_eigengenes)){
  for(j in 1:ncol(Y_eigengenes)){
    cor.res <- cor.test(X_eigengenes[,i], Y_eigengenes[,j])
    p.value_matr[i, j] <- cor.res$p.value
    corr.value_matr[i, j] <- cor.res$estimate
  }
}

# Correct for number of tests
p.value_matr.adjust <- p.adjust(p.value_matr, method = "fdr")
dim(p.value_matr.adjust) <- dim(p.value_matr)
dimnames(p.value_matr.adjust) <- list(colnames(X_eigengenes), colnames(Y_eigengenes))


# Add significance level.  
# One star means a p-value of less than 0.05; Two stars is less than 0.01, and three, is less than 0.001.

signif_matrix <- rep("", length(p.value_matr))
three_star <- which( p.value_matr <= 0.001)
signif_matrix[three_star] <- "***"
two_star <- which((p.value_matr <= 0.01) & (p.value_matr > 0.001))
signif_matrix[two_star] <- "**"
one_star <- which((p.value_matr <= 0.05) & (p.value_matr > 0.01))
signif_matrix[one_star] <- "*"
dim(signif_matrix) = dim(p.value_matr) # Give textMatrix the correct dimensions 


# Collect all results into a list.
Y_corr_X <- list(p_value = p.value_matr, 
                 p_value_adj = p.value_matr.adjust,
                 signif_matrix = signif_matrix,
                 correlation = corr.value_matr)
rm(p.value_matr, p.value_matr.adjust, signif_matrix, corr.value_matr)
```

```{r}
heatmap_colors <- colorRampPalette(rev(RColorBrewer::brewer.pal(n = 6, name ="RdBu")))(51)

pheatmap::pheatmap(Y_corr_X$correlation, 
                   color = heatmap_colors, 
                   treeheight_col = 0, 
                   treeheight_row = 0,  # will be shown on the transcriptomics ME heatmap
                   cluster_rows = X_ME_dendro,
                   cutree_rows = 4,
                   display_numbers = Y_corr_X$signif_matrix, 
                   fontsize_number = 10,
                   breaks = seq(from = -1, to = 1, length.out = 51), 
                   silent = F,
                   show_rownames = F,
                   labels_row = paste0("mg", rownames(Y_corr_X$correlation)),
                   labels_col = paste0("mg", colnames(Y_corr_X$correlation)),
                   main = "EigenOTUs") -> Y_corr_X_plot
```

##### Combine heatmaps

```{r}
cowplot::plot_grid(Y_corr_X_plot$gtable,
                   X_plot$gtable,
                   ncol = 2, 
                   rel_widths = c(dim(Y_eigengenes)[2]/3, 
                   dim(X_eigengenes)[1]/9),
                   align = "h") + ggplot2::theme(plot.margin = ggplot2::unit(c(3,0,2.5,1), "cm"))
```

### Visualization network

The strong correlation between mgME3 and mbME1 is the modules of interest, and we wondered if they contain those bacteria and metabolites, which ara published in the existing literature, or if there is a potential mining value.

So we visualize network in mgME3 and mbME1 (In the practical application，find the module you are interested in for visualization)

Visualize mgME3

```{r}
#load(file = "/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/GVHD/bookdown_GVHD/result/modules.mgs-block.1.RData")
load(file = "./result/modules.mgs-block.1.RData")

# The TOM is saved as a dist object and needs to be converted to a matrix
TOM <- as.matrix(TOM)
# Add OTU names to the TOM matrix. It is symmetrical so rownames = colnames
rownames(TOM) <- colnames(TOM) <- names(modules.mgs$colors)
```

Which taxonomic level should the graph be colored with in addition to modules?
```{r}
selected_taxa <- "Genus"
```

Convert module labels and taxonomy to hex colors
```{r}
taxonomy_info <- 
  taxa_table %>% 
  rownames_to_column(var = "OTU_name") %>%
  dplyr::select("OTU_name", selected_taxa) %>%
  mutate_all(.funs = list(as.character)) 

module_info <- 
  modules.mgs$colors %>% 
  as.data.frame() %>%
  rownames_to_column(var = "OTU_name") %>% 
  dplyr::rename(Module = ".") %>%
  filter(Module == 3)

# selected module, here is ME3
TOM <- TOM[module_info$OTU_name, module_info$OTU_name]

graph_info <- left_join(module_info, taxonomy_info, by = "OTU_name")

# Converts R-colors to hex colors
color2hex <- function(x){
  x <- col2rgb(x)
  rgb(x[1,], x[2,], x[3,], maxColorValue = 255)
}

# Add specific colors to the taxa
taxa_colors <- 
  graph_info %>% 
  dplyr::select(selected_taxa) %>% 
  unique() %>% 
  mutate(tax_color = colorRampPalette(RColorBrewer::brewer.pal(8, "Accent"))(nrow(.)))

graph_info_colors <- 
  left_join(graph_info, taxa_colors) %>% 
  mutate(module_color = WGCNA::labels2colors(Module)) %>% 
  mutate(module_color = color2hex(module_color)) %>% 
  mutate(module_color = paste0(module_color,70))
```

If all lines are too thick, reduce the strength (between 0 and 1).
The lower the number the weaker the lines.
```{r}
strength_adjust = 1
```


```{r}
g <- graph.adjacency(TOM, mode="undirected", weighted= TRUE)

#~https://stackoverflow.com/questions/28366329/how-to-scale-edge-colors-igraph
igraph::delete.edges(g, which(E(g)$weight <1))
E(g)$width <- E(g)$weight*strength_adjust + min(E(g)$weight)
E(g)$color <- "red"

set.seed(231) # Ensures the same layout given the same data.
l <- layout_with_fr(g, weights = E(g)$weight)


# Order graph_info_colors by the graph 
graph_info_colors <- graph_info_colors[which(graph_info_colors$OTU_name %in% V(g)$name),]

# Ensure that the names are in the same order
if(all(V(g)$name == graph_info_colors$OTU_name)){cat("\nOTU names match")}

# Add square shapes to hub OTUs
V(g)$shape <- ifelse(V(g)$name %in% hubs.mgs[-1], "square", "circle") #-1 means dont use module 0

# OTUs in modules have larger nodes
V(g)$size <- ifelse(graph_info_colors$Module != 0, 10, 5)

# And larger text
V(g)$label.cex <- ifelse(graph_info_colors$Module != 0, 0.8, 0.4)

# Remove everything but the number to increase readability
V(g)$name =  sub("OTU_", "", graph_info_colors$OTU_name)

```

Find distinct entires for the plot legends
```{r}
module_labels <- 
  graph_info_colors %>% 
  dplyr::select(Module, module_color) %>% 
  distinct() %>% 
  arrange(Module)

tax_labels <- 
  graph_info_colors %>% 
  dplyr::select(selected_taxa, tax_color) %>% 
  distinct()
```


Plot the graphs, leftmost is colored by module, rightmost is colored by taxonomic classification
```{r fig.height=10, fig.width=15}
par(mfrow = c(1,2))

par(mar = c(0,0,0,0))

# plot(g, layout = l, vertex.color = graph_info_colors$module_color)
# # legend("topleft", legend = paste0("mM", 0:(nrow(module_labels)-1)), fill=module_labels$module_color)
# legend("topleft", legend = paste0("mgsM", 3), fill=module_labels$module_color)

plot(g, layout = l, vertex.color = graph_info_colors$tax_color)
legend("topleft", legend = tax_labels$Genus, fill=tax_labels$tax_color)

par(mfrow = c(1,1))
```

Visualize mbME1

```{r}
#load(file = "/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/GVHD/bookdown_GVHD/result/module.metabolites-block.1.RData")
load(file = "./result/module.metabolites-block.1.RData")
# The TOM is saved as a dist object and needs to be converted to a matrix
TOM <- as.matrix(TOM)
# Add OTU names to the TOM matrix. It is symmetrical so rownames = colnames
rownames(TOM) <- colnames(TOM) <- names(modules.metabolites$colors)
```

Which metabolic level should the graph be colored with in addition to modules?
```{r}
selected_taxa <- "Class.II"
```

Convert module labels and taxonomy to hex colors
```{r}
taxonomy_info <- 
  metabolites_level %>% 
  rownames_to_column(var = "mbs_name") %>%
  dplyr::select("mbs_name", selected_taxa) %>%
  mutate_all(.funs = list(as.character)) 

module_info <- 
  modules.metabolites$colors %>% 
  as.data.frame() %>%
  rownames_to_column(var = "mbs_name") %>% 
  dplyr::rename(Module = ".") %>%
  filter(Module == 1)

# selected module, here is ME1
TOM <- TOM[module_info$mbs_name, module_info$mbs_name]

graph_info <- left_join(module_info, taxonomy_info, by = "mbs_name")

# Converts R-colors to hex colors
color2hex <- function(x){
  x <- col2rgb(x)
  rgb(x[1,], x[2,], x[3,], maxColorValue = 255)
}

# Add specific colors to the taxa
taxa_colors <- 
  graph_info %>% 
  dplyr::select(selected_taxa) %>% 
  unique() %>% 
  mutate(tax_color = colorRampPalette(RColorBrewer::brewer.pal(8, "Accent"))(nrow(.)))

graph_info_colors <- 
  left_join(graph_info, taxa_colors) %>% 
  mutate(module_color = WGCNA::labels2colors(Module)) %>% 
  mutate(module_color = color2hex(module_color)) %>% 
  mutate(module_color = paste0(module_color,70))
```

If all lines are too thick, reduce the strength (between 0 and 1).
The lower the number the weaker the lines.
```{r}
strength_adjust = 0.2
```


```{r}
g <- graph.adjacency(TOM, mode="undirected", weighted= TRUE)

#~https://stackoverflow.com/questions/28366329/how-to-scale-edge-colors-igraph
igraph::delete.edges(g, which(E(g)$weight <1))
E(g)$width <- E(g)$weight*strength_adjust + min(E(g)$weight)
E(g)$color <- "red"

set.seed(231) # Ensures the same layout given the same data.
l <- layout_with_fr(g, weights = E(g)$weight)


# Order graph_info_colors by the graph 
graph_info_colors <- graph_info_colors[which(graph_info_colors$mbs_name %in% V(g)$name),]

# Ensure that the names are in the same order
if(all(V(g)$name == graph_info_colors$mbs_name)){cat("\nmbs names match")}

# Add square shapes to hub OTUs
V(g)$shape <- ifelse(V(g)$name %in% hubs.metabolites[-1], "square", "circle") #-1 means dont use module 0

# OTUs in modules have larger nodes
V(g)$size <- ifelse(graph_info_colors$Module != 0, 10, 5)

# And larger text
V(g)$label.cex <- ifelse(graph_info_colors$Module != 0, 0.8, 0.4)

# Remove everything but the number to increase readability
V(g)$name =  sub("mbs_", "", graph_info_colors$mbs_name)

```

Find distinct entires for the plot legends
```{r}
module_labels <- 
  graph_info_colors %>% 
  dplyr::select(Module, module_color) %>% 
  distinct() %>% 
  arrange(Module)

tax_labels <- 
  graph_info_colors %>% 
  dplyr::select(selected_taxa, tax_color) %>% 
  distinct()
```


Plot the graphs, leftmost is colored by module, rightmost is colored by taxonomic classification
```{r fig.height=10, fig.width=15}
par(mfrow = c(1,2))

par(mar = c(0,0,0,0))

# plot(g, layout = l, vertex.color = graph_info_colors$module_color)
# # legend("topleft", legend = paste0("mM", 0:(nrow(module_labels)-1)), fill=module_labels$module_color)
# legend("topleft", legend = paste0("mbM", 1), fill=module_labels$module_color)

plot(g, layout = l, vertex.color = graph_info_colors$tax_color)
legend("topright", legend = tax_labels$Class.II, fill=tax_labels$tax_color)

par(mfrow = c(1,1))
```


### Session Info

```{r}
devtools::session_info()
```



## HAllA demo

### 登102集群: 执行下述步骤

```bash

conda activate /opt/conda/envs/Halla

halla -h

halla -x table_X -y table_Y -m correlation_methods -o output_file --fdr_alpha adjustP --fdr_method FDR_methods

```

### 结果展示

#### HAllA result after WGCNA

![Correlation with modules](./figure/GvHD_module.png)

此部分分析用的wgcna生成的宏基因组和代谢组 modules，因此在运行halla的时候需要加上 —alla 取消数据集内部cluster构建。
以上是用五种方法进行了关联分析，得出的结果不完全一样，序号的先后顺序代表显著性的强弱(adjust p-value)。从找到的模块关系中，对应回WGCNA里的Hub物种，用作此module的关系解释。可选择感兴趣/能说明问题的图表（关联方法）进行深度挖掘。

#### 常规数据集结果

![Correlation with modules](./figure/HAllA_GvHD.png)

此部分分析用的GvHD的宏基因组和代谢组的原始数据（spearman-correlation），热图中显著性的cluster均有数字或白点进行标注（过滤条件fdr < 0.05）；其中的数值显示是按照cluster 中最小的 adjust p-value 给定rank，序号越小，显著性越强。
