<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Example | Omics analysis</title>
  <meta name="description" content="Tutorial for OMics analysis" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Example | Omics analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Tutorial for OMics analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Example | Omics analysis" />
  
  <meta name="twitter:description" content="Tutorial for OMics analysis" />
  

<meta name="author" content="Zhengnong Zhu and Yifan Zhao" />


<meta name="date" content="2022-07-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="利用wgcna构建网络.html"/>
<link rel="next" href="halla-tutorial.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.14/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">OMics analysis workflow</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> OMics Analysis Template</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#wgcna-preface"><i class="fa fa-check"></i><b>1.1</b> WGCNA Preface</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#wgcna-introduction"><i class="fa fa-check"></i><b>1.2</b> WGCNA Introduction</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#wgcna-步骤"><i class="fa fa-check"></i><b>1.2.1</b> WGCNA 步骤</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#halla-introduction"><i class="fa fa-check"></i><b>1.3</b> HAllA Introduction</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="利用wgcna构建网络.html"><a href="利用wgcna构建网络.html"><i class="fa fa-check"></i><b>2</b> 利用WGCNA构建网络</a><ul>
<li class="chapter" data-level="2.1" data-path="利用wgcna构建网络.html"><a href="利用wgcna构建网络.html#s-测序数据"><i class="fa fa-check"></i><b>2.1</b> 16S 测序数据</a><ul>
<li class="chapter" data-level="2.1.1" data-path="利用wgcna构建网络.html"><a href="利用wgcna构建网络.html#输入数据格式"><i class="fa fa-check"></i><b>2.1.1</b> 输入数据格式</a></li>
<li class="chapter" data-level="2.1.2" data-path="利用wgcna构建网络.html"><a href="利用wgcna构建网络.html#构建network"><i class="fa fa-check"></i><b>2.1.2</b> 构建network</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="利用wgcna构建网络.html"><a href="利用wgcna构建网络.html#宏基因组测序数据"><i class="fa fa-check"></i><b>2.2</b> 宏基因组测序数据</a></li>
<li class="chapter" data-level="2.3" data-path="利用wgcna构建网络.html"><a href="利用wgcna构建网络.html#代谢数据"><i class="fa fa-check"></i><b>2.3</b> 代谢数据</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="example.html"><a href="example.html"><i class="fa fa-check"></i><b>3</b> Example</a><ul>
<li class="chapter" data-level="3.1" data-path="example.html"><a href="example.html#数据介绍"><i class="fa fa-check"></i><b>3.1</b> 数据介绍</a></li>
<li class="chapter" data-level="3.2" data-path="example.html"><a href="example.html#load-package"><i class="fa fa-check"></i><b>3.2</b> load package</a></li>
<li class="chapter" data-level="3.3" data-path="example.html"><a href="example.html#load-data"><i class="fa fa-check"></i><b>3.3</b> load data</a></li>
<li class="chapter" data-level="3.4" data-path="example.html"><a href="example.html#restructured-the-hierarchy-table"><i class="fa fa-check"></i><b>3.4</b> restructured the hierarchy table</a></li>
<li class="chapter" data-level="3.5" data-path="example.html"><a href="example.html#mgs-standardization"><i class="fa fa-check"></i><b>3.5</b> mgs standardization</a><ul>
<li class="chapter" data-level="3.5.1" data-path="example.html"><a href="example.html#check-data-format"><i class="fa fa-check"></i><b>3.5.1</b> check data format</a></li>
<li class="chapter" data-level="3.5.2" data-path="example.html"><a href="example.html#using-css-method"><i class="fa fa-check"></i><b>3.5.2</b> using CSS method</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="example.html"><a href="example.html#mgs-constructe-module"><i class="fa fa-check"></i><b>3.6</b> mgs constructe module</a><ul>
<li class="chapter" data-level="3.6.1" data-path="example.html"><a href="example.html#select-power"><i class="fa fa-check"></i><b>3.6.1</b> select power</a></li>
<li class="chapter" data-level="3.6.2" data-path="example.html"><a href="example.html#block-wise-network-construction-and-module-detection"><i class="fa fa-check"></i><b>3.6.2</b> Block-wise network construction and module detection</a></li>
<li class="chapter" data-level="3.6.3" data-path="example.html"><a href="example.html#mgs-module-details"><i class="fa fa-check"></i><b>3.6.3</b> mgs module details</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="example.html"><a href="example.html#metabolites-constructe-module"><i class="fa fa-check"></i><b>3.7</b> metabolites constructe module</a><ul>
<li class="chapter" data-level="3.7.1" data-path="example.html"><a href="example.html#select-power-1"><i class="fa fa-check"></i><b>3.7.1</b> select power</a></li>
<li class="chapter" data-level="3.7.2" data-path="example.html"><a href="example.html#block-wise-network-construction-and-module-detection-1"><i class="fa fa-check"></i><b>3.7.2</b> Block-wise network construction and module detection</a></li>
<li class="chapter" data-level="3.7.3" data-path="example.html"><a href="example.html#metabolites-module-details"><i class="fa fa-check"></i><b>3.7.3</b> metabolites module details</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="example.html"><a href="example.html#metabolites-mgs-and-metadata-association"><i class="fa fa-check"></i><b>3.8</b> metabolites, mgs and metadata association</a></li>
<li class="chapter" data-level="3.9" data-path="example.html"><a href="example.html#visualization-network"><i class="fa fa-check"></i><b>3.9</b> Visualization network</a></li>
<li class="chapter" data-level="3.10" data-path="example.html"><a href="example.html#session-info"><i class="fa fa-check"></i><b>3.10</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="halla-tutorial.html"><a href="halla-tutorial.html"><i class="fa fa-check"></i><b>4</b> Halla tutorial</a><ul>
<li class="chapter" data-level="4.1" data-path="halla-tutorial.html"><a href="halla-tutorial.html#introduction"><i class="fa fa-check"></i><b>4.1</b> Introduction</a></li>
<li class="chapter" data-level="4.2" data-path="halla-tutorial.html"><a href="halla-tutorial.html#halla-步骤"><i class="fa fa-check"></i><b>4.2</b> HAllA 步骤</a><ul>
<li class="chapter" data-level="4.2.1" data-path="halla-tutorial.html"><a href="halla-tutorial.html#登102集群-执行下述步骤"><i class="fa fa-check"></i><b>4.2.1</b> 登102集群: 执行下述步骤</a></li>
<li class="chapter" data-level="4.2.2" data-path="halla-tutorial.html"><a href="halla-tutorial.html#available-pairwise-distance-metrics"><i class="fa fa-check"></i><b>4.2.2</b> Available pairwise distance metrics:</a></li>
<li class="chapter" data-level="4.2.3" data-path="halla-tutorial.html"><a href="halla-tutorial.html#running-procedures"><i class="fa fa-check"></i><b>4.2.3</b> Running procedures</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="halla-tutorial.html"><a href="halla-tutorial.html#结果展示"><i class="fa fa-check"></i><b>4.3</b> 结果展示</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Omics analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="example" class="section level1 hasAnchor">
<h1><span class="header-section-number">Chapter 3</span> Example<a href="example.html#example" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="数据介绍" class="section level2 hasAnchor">
<h2><span class="header-section-number">3.1</span> 数据介绍<a href="example.html#数据介绍" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>GVHD数据，有32个样本，测了宏基因组和代谢数据。</p>
</div>
<div id="load-package" class="section level2 hasAnchor">
<h2><span class="header-section-number">3.2</span> load package<a href="example.html#load-package" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="example.html#cb60-1"></a>knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">echo =</span> <span class="ot">TRUE</span>)</span>
<span id="cb60-2"><a href="example.html#cb60-2"></a></span>
<span id="cb60-3"><a href="example.html#cb60-3"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb60-4"><a href="example.html#cb60-4"></a><span class="kw">library</span>(xlsx)</span>
<span id="cb60-5"><a href="example.html#cb60-5"></a><span class="kw">library</span>(WGCNA)</span>
<span id="cb60-6"><a href="example.html#cb60-6"></a><span class="kw">library</span>(igraph)</span>
<span id="cb60-7"><a href="example.html#cb60-7"></a><span class="kw">library</span>(phyloseq)</span>
<span id="cb60-8"><a href="example.html#cb60-8"></a><span class="kw">library</span>(DT)</span>
<span id="cb60-9"><a href="example.html#cb60-9"></a></span>
<span id="cb60-10"><a href="example.html#cb60-10"></a><span class="kw">options</span>(<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<div id="load-data" class="section level2 hasAnchor">
<h2><span class="header-section-number">3.3</span> load data<a href="example.html#load-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>metagenomics</li>
<li>metabolic</li>
<li>metadata</li>
</ul>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="example.html#cb61-1"></a>mgs_profile &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;/home/xuxiaomin/project/NanFangHospitalGvHD/02.MGS_output/metaphlan2_merged.tsv&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>) <span class="co"># read metagenomic hirerachy table，only need ID column</span></span>
<span id="cb61-2"><a href="example.html#cb61-2"></a>metabolites_level &lt;-<span class="st"> </span><span class="kw">read.xlsx</span>(<span class="st">&quot;/home/zhaoyifan/zhaoyifan_Xbiome1/project/GVHD/metabolites_TM/ALL_sample_data.xlsx&quot;</span>, <span class="dt">sheetIndex =</span> <span class="dv">1</span>) <span class="co"># read metabolic hirerachy table</span></span>
<span id="cb61-3"><a href="example.html#cb61-3"></a>mgs &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;/home/zhaoyifan/projects/Analytics/analytics/MultiOmics/tools/HAllA/GvHD/metagenomics.txt&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>) <span class="co"># read metagenomic species level data</span></span>
<span id="cb61-4"><a href="example.html#cb61-4"></a>metabolites_data &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;/home/zhaoyifan/projects/Analytics/analytics/MultiOmics/tools/HAllA/GvHD/metabolic.txt&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>) <span class="co"># read metabolic data</span></span>
<span id="cb61-5"><a href="example.html#cb61-5"></a>metadata &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;/home/zhaoyifan/projects/Analytics/analytics/MultiOmics/tools/HAllA/GvHD/metadata.txt&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>) <span class="co"># read metadata</span></span></code></pre></div>
<ul>
<li>mgs_profile need taxanomy level column, as shwon in the table below</li>
</ul>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="example.html#cb62-1"></a><span class="kw">datatable</span>(mgs_profile[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>])</span></code></pre></div>
<div id="htmlwidget-a799f89801ae5bc191c2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a799f89801ae5bc191c2">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10"],["k__Bacteria","k__Viruses","k__Bacteria|p__Firmicutes","k__Viruses|p__Viruses_noname","k__Bacteria|p__Proteobacteria","k__Bacteria|p__Actinobacteria","k__Bacteria|p__Firmicutes|c__Bacilli","k__Viruses|p__Viruses_noname|c__Viruses_noname","k__Bacteria|p__Proteobacteria|c__Gammaproteobacteria","k__Bacteria|p__Actinobacteria|c__Actinobacteria"],[81.95994,18.04006,81.77676,18.04006,0.09632,0.08687,81.77571,18.04006,0.09632,0.08687],[68.81374,31.18626,68.81374,31.18626,0,0,68.81374,31.18626,0,0],[100,0,74.83979,0,7.87894,0.79339,4.81587,0,7.87894,0.79339],[100,0,65.71979,0,6.42708,2.99347,0.36717,0,6.42708,2.99347],[99.78322,0.21678,4.0227,0.21678,94.11209,1.64844,2.32155,0.21678,93.9276,1.64844],[99.51076,0.48924,79.0703,0.48924,15.15951,5.28066,55.10263,0.48924,15.04486,5.28066],[99.8758,0.10647,95.93161,0.10647,0.28038,3.66381,25.02493,0.10647,0.27939,3.66381],[99.85559,0.14441,30.33094,0.14441,0.21575,0,30.33094,0.14441,0.02107,0],[99.92002,0.07998,64.19794,0.07998,8.38902,12.77622,27.07718,0.07998,8.38828,12.77622]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ID<\/th>\n      <th>X4367<\/th>\n      <th>X4368<\/th>\n      <th>X4369<\/th>\n      <th>X4370<\/th>\n      <th>X4371<\/th>\n      <th>X4372<\/th>\n      <th>X4373<\/th>\n      <th>X4374<\/th>\n      <th>X4375<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>Metabolic level data are selected as the Class.I and Class.II column in the following analysis, the details can be adjusted according to the data reality</li>
</ul>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="example.html#cb63-1"></a><span class="kw">datatable</span>(metabolites_level[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>])</span></code></pre></div>
<div id="htmlwidget-a82a301cedd8cedb9515" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a82a301cedd8cedb9515">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10"],["MEDL00066","MEDL00356","MEDL00369","MEDL00375","MEDL00392","MEDL00401","MEDL00404","MEDL00416","MEDL00442","MEDL00568"],["2-Hydroxy-2-methylbutyric acid","Phe-Asp","Glu-Phe","Glu-Trp","1-Oleoyl-sn-glycero-3-phosphocholine","Confertifoline","2-Palmitoyl-rac-glycerol","3-Hydroxydodecanoic acid","Soyasaponin Bb","Hyperoside"],["-","-","-","-","-","-","-","-","-","-"],["-","-","-","-","-","-","-","-","-","-"],["-","-","-","-","-","-","-","-","-","-"],["-","-","-","-","-","-","-","-","-","-"],["-","-","-","-","-","-","-","-","-","-"],[117.0556,245.09207,293.09872,334.13872,525.36328,233.15367,331.28346,215.1637,943.52609,463.08476],[118.063,348.0686,294.121572,333.132471,521.34814,234.1619799,330.27701,216.1725446,942.5188157,464.0955],["[M-H]-","[M+H-2H2O]+","[M-H]-","[M+H]+","[M+H]+","[M-H]-","[M+H]+","[M-H]-","[M+H]+","[M-H]-"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Index<\/th>\n      <th>Compounds<\/th>\n      <th>物质<\/th>\n      <th>Class.I<\/th>\n      <th>物质一级分类<\/th>\n      <th>Class.II<\/th>\n      <th>物质二级分类<\/th>\n      <th>Q1..Da.<\/th>\n      <th>Molecular.Weight..Da.<\/th>\n      <th>Ionization.model<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[8,9]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>mgs species: row names are Features, column names are Sample</p>
<p>metabolites_data: row names are Features, column names are Sample</p>
</div>
<div id="restructured-the-hierarchy-table" class="section level2 hasAnchor">
<h2><span class="header-section-number">3.4</span> restructured the hierarchy table<a href="example.html#restructured-the-hierarchy-table" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="example.html#cb64-1"></a><span class="co"># taxa hierarchy table</span></span>
<span id="cb64-2"><a href="example.html#cb64-2"></a>mgs_species &lt;-<span class="st"> </span>mgs_profile <span class="op">%&gt;%</span></span>
<span id="cb64-3"><a href="example.html#cb64-3"></a><span class="st">  </span>.[<span class="kw">sapply</span>(<span class="kw">str_split</span>(mgs_profile<span class="op">$</span>ID, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">|&quot;</span>), <span class="cf">function</span>(x) <span class="kw">length</span>(x) <span class="op">==</span><span class="st"> </span><span class="dv">7</span>), ] </span>
<span id="cb64-4"><a href="example.html#cb64-4"></a>taxa_table &lt;-<span class="st"> </span><span class="kw">str_split_fixed</span>(mgs_species<span class="op">$</span>ID,<span class="st">&quot;</span><span class="ch">\\</span><span class="st">|&quot;</span>, <span class="dv">7</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</span>
<span id="cb64-5"><a href="example.html#cb64-5"></a><span class="kw">colnames</span>(taxa_table) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>, <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb64-6"><a href="example.html#cb64-6"></a><span class="kw">rownames</span>(taxa_table) &lt;-<span class="st"> </span><span class="kw">str_replace</span>(taxa_table<span class="op">$</span>Species, <span class="st">&quot;s__&quot;</span>, <span class="st">&quot;&quot;</span>)</span>
<span id="cb64-7"><a href="example.html#cb64-7"></a></span>
<span id="cb64-8"><a href="example.html#cb64-8"></a><span class="co"># metabolic hierarchy table</span></span>
<span id="cb64-9"><a href="example.html#cb64-9"></a>metabolites_level &lt;-<span class="st"> </span>metabolites_level <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(Index, Class.I, Class.II)</span>
<span id="cb64-10"><a href="example.html#cb64-10"></a>metabolites_level &lt;-<span class="st"> </span>metabolites_level <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;Index&quot;</span>)</span></code></pre></div>
</div>
<div id="mgs-standardization" class="section level2 hasAnchor">
<h2><span class="header-section-number">3.5</span> mgs standardization<a href="example.html#mgs-standardization" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="check-data-format" class="section level3 hasAnchor">
<h3><span class="header-section-number">3.5.1</span> check data format<a href="example.html#check-data-format" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="example.html#cb65-1"></a><span class="kw">cat</span>(<span class="st">&quot;[mgs]&quot;</span>, <span class="kw">paste</span>(<span class="kw">c</span>(<span class="st">&quot;Feature: &quot;</span>, <span class="st">&quot;Sample: &quot;</span>), <span class="kw">dim</span>(mgs)))</span></code></pre></div>
<pre><code>## [mgs] Feature:  250 Sample:  32</code></pre>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="example.html#cb67-1"></a><span class="kw">cat</span>(<span class="st">&quot;[metabolites]&quot;</span>, <span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">&quot;Feature: &quot;</span>, <span class="st">&quot;Sample: &quot;</span>), <span class="kw">dim</span>(metabolites_data)))</span></code></pre></div>
<pre><code>## [metabolites] Feature: 811 Sample: 32</code></pre>
</div>
<div id="using-css-method" class="section level3 hasAnchor">
<h3><span class="header-section-number">3.5.2</span> using CSS method<a href="example.html#using-css-method" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="example.html#cb69-1"></a>data.metagenomeSeq &lt;-<span class="st"> </span>metagenomeSeq<span class="op">::</span><span class="kw">newMRexperiment</span>(mgs) <span class="co"># Variables as rows, samples as columns</span></span>
<span id="cb69-2"><a href="example.html#cb69-2"></a>p &lt;-<span class="st"> </span>metagenomeSeq<span class="op">::</span><span class="kw">cumNormStat</span>(data.metagenomeSeq)</span></code></pre></div>
<pre><code>## Default value being used.</code></pre>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="example.html#cb71-1"></a>data.cumnorm &lt;-<span class="st"> </span>metagenomeSeq<span class="op">::</span><span class="kw">cumNorm</span>(data.metagenomeSeq, <span class="dt">p=</span>p)</span>
<span id="cb71-2"><a href="example.html#cb71-2"></a>mgs &lt;-<span class="st"> </span>metagenomeSeq<span class="op">::</span><span class="kw">MRcounts</span>(data.cumnorm, <span class="dt">norm=</span><span class="ot">TRUE</span>, <span class="dt">log=</span><span class="ot">TRUE</span>) <span class="co"># log here is a +1 shifted log2</span></span>
<span id="cb71-3"><a href="example.html#cb71-3"></a></span>
<span id="cb71-4"><a href="example.html#cb71-4"></a><span class="kw">dim</span>(mgs)</span></code></pre></div>
<pre><code>## [1] 250  32</code></pre>
</div>
</div>
<div id="mgs-constructe-module" class="section level2 hasAnchor">
<h2><span class="header-section-number">3.6</span> mgs constructe module<a href="example.html#mgs-constructe-module" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="example.html#cb73-1"></a>mgs_t &lt;-<span class="st"> </span>mgs <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>() <span class="co"># Samples as rows, Features as columns</span></span></code></pre></div>
<div id="select-power" class="section level3 hasAnchor">
<h3><span class="header-section-number">3.6.1</span> select power<a href="example.html#select-power" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="example.html#cb74-1"></a>powers &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">seq</span>(<span class="dv">12</span>,<span class="dv">20</span>,<span class="dv">2</span>)) <span class="co"># default</span></span>
<span id="cb74-2"><a href="example.html#cb74-2"></a></span>
<span id="cb74-3"><a href="example.html#cb74-3"></a><span class="kw">suppressWarnings</span>(sft &lt;-<span class="st"> </span><span class="kw">pickSoftThreshold</span>(mgs_t, </span>
<span id="cb74-4"><a href="example.html#cb74-4"></a>                                          <span class="dt">powerVector =</span> powers, </span>
<span id="cb74-5"><a href="example.html#cb74-5"></a>                                          <span class="dt">verbose =</span> <span class="dv">1</span>, </span>
<span id="cb74-6"><a href="example.html#cb74-6"></a>                                          <span class="dt">networkType =</span> <span class="st">&quot;signed&quot;</span>,</span>
<span id="cb74-7"><a href="example.html#cb74-7"></a>                                          <span class="dt">corFn=</span> <span class="st">&quot;bicor&quot;</span>))</span></code></pre></div>
<pre><code>## pickSoftThreshold: will use block size 250.
##  pickSoftThreshold: calculating connectivity for given powers... ..0% ..100% 
##    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
## 1      1   0.6250 10.900          0.768  134.00   135.000 150.00
## 2      2   0.6400  4.310          0.814   75.60    75.700  93.20
## 3      3   0.5870  2.560          0.867   44.70    44.700  60.90
## 4      4   0.3110  1.210          0.715   27.80    27.900  41.70
## 5      5   0.2170  0.761          0.623   18.20    17.900  29.70
## 6      6   0.2000  0.561          0.699   12.50    12.100  21.80
## 7      7   0.1940  0.438          0.730    9.05     8.830  16.40
## 8      8   0.0156  0.114          0.753    6.80     6.510  13.60
## 9      9   0.0476 -0.223          0.620    5.30     4.830  11.80
## 10    10   0.1870 -0.526          0.470    4.26     3.720  10.50
## 11    12   0.5040 -0.857          0.661    2.99     2.310   8.89
## 12    14   0.6820 -0.965          0.666    2.27     1.470   7.93
## 13    16   0.7330 -1.080          0.659    1.84     1.070   7.33
## 14    18   0.8160 -1.020          0.804    1.57     0.777   6.94
## 15    20   0.8010 -0.986          0.808    1.38     0.587   6.67</code></pre>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="example.html#cb76-1"></a><span class="co"># Find the soft thresholding power beta to which co-expression similarity is raised to calculate adjacency.</span></span>
<span id="cb76-2"><a href="example.html#cb76-2"></a><span class="co"># based on the criterion of approximate scale-free topology.</span></span>
<span id="cb76-3"><a href="example.html#cb76-3"></a></span>
<span id="cb76-4"><a href="example.html#cb76-4"></a>idx &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">which</span>((<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.90</span>))</span>
<span id="cb76-5"><a href="example.html#cb76-5"></a><span class="cf">if</span>(<span class="kw">is.infinite</span>(idx)){</span>
<span id="cb76-6"><a href="example.html#cb76-6"></a>  idx &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">which</span>((<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.80</span>))</span>
<span id="cb76-7"><a href="example.html#cb76-7"></a>  <span class="cf">if</span>(<span class="op">!</span><span class="kw">is.infinite</span>(idx)){</span>
<span id="cb76-8"><a href="example.html#cb76-8"></a>    st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb76-9"><a href="example.html#cb76-9"></a>  } <span class="cf">else</span>{</span>
<span id="cb76-10"><a href="example.html#cb76-10"></a>    idx &lt;-<span class="st"> </span><span class="kw">which.max</span>(<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>])</span>
<span id="cb76-11"><a href="example.html#cb76-11"></a>    st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb76-12"><a href="example.html#cb76-12"></a>  }</span>
<span id="cb76-13"><a href="example.html#cb76-13"></a>} <span class="cf">else</span>{</span>
<span id="cb76-14"><a href="example.html#cb76-14"></a>  st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb76-15"><a href="example.html#cb76-15"></a>}</span>
<span id="cb76-16"><a href="example.html#cb76-16"></a></span>
<span id="cb76-17"><a href="example.html#cb76-17"></a></span>
<span id="cb76-18"><a href="example.html#cb76-18"></a><span class="co"># Plot Scale independence measure and Mean connectivity measure</span></span>
<span id="cb76-19"><a href="example.html#cb76-19"></a></span>
<span id="cb76-20"><a href="example.html#cb76-20"></a><span class="co"># Scale-free topology fit index as a function of the soft-thresholding power</span></span>
<span id="cb76-21"><a href="example.html#cb76-21"></a><span class="kw">data.frame</span>(<span class="dt">Indices =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb76-22"><a href="example.html#cb76-22"></a>           <span class="dt">sfApprox =</span> <span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb76-23"><a href="example.html#cb76-23"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb76-24"><a href="example.html#cb76-24"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="fl">0.9</span>, <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.6</span>) <span class="op">+</span><span class="st"> </span><span class="co"># corresponds to R^2 cut-off of 0.9</span></span>
<span id="cb76-25"><a href="example.html#cb76-25"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="fl">0.8</span>, <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.2</span>) <span class="op">+</span><span class="st"> </span><span class="co"># corresponds to R^2 cut-off of 0.8</span></span>
<span id="cb76-26"><a href="example.html#cb76-26"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> sfApprox), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb76-27"><a href="example.html#cb76-27"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> sfApprox, <span class="dt">label =</span> Indices), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb76-28"><a href="example.html#cb76-28"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Scale independence&quot;</span>) <span class="op">+</span></span>
<span id="cb76-29"><a href="example.html#cb76-29"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Soft Threshold (power)&quot;</span>) <span class="op">+</span></span>
<span id="cb76-30"><a href="example.html#cb76-30"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;SF Model Fit,signed R^2&quot;</span>) <span class="op">+</span></span>
<span id="cb76-31"><a href="example.html#cb76-31"></a><span class="st">  </span><span class="kw">xlim</span>(<span class="dv">1</span>,<span class="dv">20</span>) <span class="op">+</span></span>
<span id="cb76-32"><a href="example.html#cb76-32"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb76-33"><a href="example.html#cb76-33"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> st, <span class="dt">y =</span> <span class="fl">0.25</span>, <span class="dt">xend =</span> st, <span class="dt">yend =</span> sfApprox[idx]<span class="op">-</span><span class="fl">0.05</span>), </span>
<span id="cb76-34"><a href="example.html#cb76-34"></a>               <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.2</span>,<span class="st">&quot;cm&quot;</span>)), </span>
<span id="cb76-35"><a href="example.html#cb76-35"></a>               <span class="dt">size =</span> <span class="fl">0.5</span>)-&gt;<span class="st"> </span>scale_independence_plot </span>
<span id="cb76-36"><a href="example.html#cb76-36"></a>  </span>
<span id="cb76-37"><a href="example.html#cb76-37"></a> </span>
<span id="cb76-38"><a href="example.html#cb76-38"></a></span>
<span id="cb76-39"><a href="example.html#cb76-39"></a></span>
<span id="cb76-40"><a href="example.html#cb76-40"></a><span class="co"># Mean connectivity as a function of the soft-thresholding power</span></span>
<span id="cb76-41"><a href="example.html#cb76-41"></a></span>
<span id="cb76-42"><a href="example.html#cb76-42"></a><span class="kw">data.frame</span>(<span class="dt">Indices =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb76-43"><a href="example.html#cb76-43"></a>           <span class="dt">meanApprox =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">5</span>]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb76-44"><a href="example.html#cb76-44"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb76-45"><a href="example.html#cb76-45"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> meanApprox), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb76-46"><a href="example.html#cb76-46"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> meanApprox, <span class="dt">label =</span> Indices), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb76-47"><a href="example.html#cb76-47"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Soft Threshold (power)&quot;</span>) <span class="op">+</span></span>
<span id="cb76-48"><a href="example.html#cb76-48"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Mean Connectivity&quot;</span>) <span class="op">+</span></span>
<span id="cb76-49"><a href="example.html#cb76-49"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> st<span class="fl">-0.4</span>, </span>
<span id="cb76-50"><a href="example.html#cb76-50"></a>                   <span class="dt">y =</span> sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx], </span>
<span id="cb76-51"><a href="example.html#cb76-51"></a>                   <span class="dt">xend =</span> <span class="dv">0</span>, </span>
<span id="cb76-52"><a href="example.html#cb76-52"></a>                   <span class="dt">yend =</span> sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx]),</span>
<span id="cb76-53"><a href="example.html#cb76-53"></a>               <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.2</span>,<span class="st">&quot;cm&quot;</span>)), </span>
<span id="cb76-54"><a href="example.html#cb76-54"></a>               <span class="dt">size =</span> <span class="fl">0.4</span>) <span class="op">+</span></span>
<span id="cb76-55"><a href="example.html#cb76-55"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;Mean connectivity: &quot;</span>, </span>
<span id="cb76-56"><a href="example.html#cb76-56"></a>                 <span class="kw">round</span>(sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx],<span class="dv">2</span>))) -&gt;<span class="st"> </span>mean_connectivity_plot</span>
<span id="cb76-57"><a href="example.html#cb76-57"></a></span>
<span id="cb76-58"><a href="example.html#cb76-58"></a></span>
<span id="cb76-59"><a href="example.html#cb76-59"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(scale_independence_plot, mean_connectivity_plot, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">align =</span> <span class="st">&quot;h&quot;</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>) -&gt;<span class="st"> </span>si_mc_plot</span>
<span id="cb76-60"><a href="example.html#cb76-60"></a></span>
<span id="cb76-61"><a href="example.html#cb76-61"></a>si_mc_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-44-1.png" width="672" /></p>
<p>The number closest to the 0.8 line is 18, and the constructed adjacency matrix most closely fits with scale-free topology. Therefore power is 18.</p>
</div>
<div id="block-wise-network-construction-and-module-detection" class="section level3 hasAnchor">
<h3><span class="header-section-number">3.6.2</span> Block-wise network construction and module detection<a href="example.html#block-wise-network-construction-and-module-detection" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The function <code>blockwiseModules</code> will first pre cluster with fast crude clustering method to cluster OTUs into blocks not exceeding the maximum, blocks may therefore not be fully optimal in the end.</p>
<p>Change the parameters here to better reflect your own data.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="example.html#cb77-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">dir.exists</span>(<span class="st">&quot;./result/&quot;</span>)) {</span>
<span id="cb77-2"><a href="example.html#cb77-2"></a>  <span class="kw">dir.create</span>(<span class="st">&quot;./result/&quot;</span>)</span>
<span id="cb77-3"><a href="example.html#cb77-3"></a>}</span>
<span id="cb77-4"><a href="example.html#cb77-4"></a></span>
<span id="cb77-5"><a href="example.html#cb77-5"></a>Run_analysis &lt;-<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb77-6"><a href="example.html#cb77-6"></a><span class="cf">if</span>(Run_analysis){</span>
<span id="cb77-7"><a href="example.html#cb77-7"></a>  modules.mgs &lt;-<span class="st"> </span><span class="kw">blockwiseModules</span>(mgs_t,</span>
<span id="cb77-8"><a href="example.html#cb77-8"></a>                          <span class="dt">power =</span> st, </span>
<span id="cb77-9"><a href="example.html#cb77-9"></a>                          <span class="dt">networkType =</span> <span class="st">&quot;signed&quot;</span>, </span>
<span id="cb77-10"><a href="example.html#cb77-10"></a>                          <span class="dt">TOMType =</span> <span class="st">&quot;signed&quot;</span>,</span>
<span id="cb77-11"><a href="example.html#cb77-11"></a>                          <span class="dt">corType =</span> <span class="st">&#39;bicor&#39;</span>,</span>
<span id="cb77-12"><a href="example.html#cb77-12"></a>                          <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>,</span>
<span id="cb77-13"><a href="example.html#cb77-13"></a>                          <span class="dt">deepSplit =</span> <span class="dv">2</span>,</span>
<span id="cb77-14"><a href="example.html#cb77-14"></a>                          <span class="dt">pamStage =</span> <span class="ot">FALSE</span>, </span>
<span id="cb77-15"><a href="example.html#cb77-15"></a>                          <span class="dt">pamRespectsDendro =</span> <span class="ot">TRUE</span>,</span>
<span id="cb77-16"><a href="example.html#cb77-16"></a>                          <span class="dt">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb77-17"><a href="example.html#cb77-17"></a>                          <span class="dt">replaceMissingAdjacencies =</span> <span class="ot">TRUE</span>,</span>
<span id="cb77-18"><a href="example.html#cb77-18"></a>                          <span class="dt">minModuleSize =</span> <span class="dv">5</span>, <span class="co"># There are fewer otus than genes, and that many might not be connected</span></span>
<span id="cb77-19"><a href="example.html#cb77-19"></a>                          <span class="dt">numericLabels =</span> <span class="ot">TRUE</span>,</span>
<span id="cb77-20"><a href="example.html#cb77-20"></a>                          <span class="dt">saveTOMs =</span> <span class="ot">TRUE</span>,</span>
<span id="cb77-21"><a href="example.html#cb77-21"></a>                          <span class="dt">saveTOMFileBase =</span> <span class="st">&quot;./result/modules.mgs&quot;</span>, <span class="co">#&quot;/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/GVHD/bookdown_GVHD/result/modules.mgs&quot;,</span></span>
<span id="cb77-22"><a href="example.html#cb77-22"></a>                          <span class="dt">verbose =</span> <span class="dv">1</span>)</span>
<span id="cb77-23"><a href="example.html#cb77-23"></a>  </span>
<span id="cb77-24"><a href="example.html#cb77-24"></a>  <span class="kw">rownames</span>(modules.mgs<span class="op">$</span>MEs) &lt;-<span class="st"> </span><span class="kw">rownames</span>(mgs_t)</span>
<span id="cb77-25"><a href="example.html#cb77-25"></a>  <span class="kw">names</span>(modules.mgs<span class="op">$</span>colors) &lt;-<span class="st"> </span><span class="kw">colnames</span>(mgs_t)</span>
<span id="cb77-26"><a href="example.html#cb77-26"></a>  <span class="kw">names</span>(modules.mgs<span class="op">$</span>unmergedColors) &lt;-<span class="st"> </span><span class="kw">colnames</span>(mgs_t)</span>
<span id="cb77-27"><a href="example.html#cb77-27"></a>  </span>
<span id="cb77-28"><a href="example.html#cb77-28"></a>  hubs.mgs &lt;-<span class="st"> </span><span class="kw">chooseTopHubInEachModule</span>(mgs_t, modules.mgs<span class="op">$</span>colors)</span>
<span id="cb77-29"><a href="example.html#cb77-29"></a>}</span></code></pre></div>
<pre><code>##  Calculating module eigengenes block-wise from all genes</code></pre>
</div>
<div id="mgs-module-details" class="section level3 hasAnchor">
<h3><span class="header-section-number">3.6.3</span> mgs module details<a href="example.html#mgs-module-details" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="example.html#cb79-1"></a><span class="co"># Convert labels to colors for plotting</span></span>
<span id="cb79-2"><a href="example.html#cb79-2"></a>merged_colors &lt;-<span class="st"> </span><span class="kw">labels2colors</span>(modules.mgs<span class="op">$</span>colors)</span></code></pre></div>
<p>All samples are OK.<br />
All OTUs are OK.</p>
<p>There where 8 modules found.<br />
All module eigenOTUs are OK.</p>
<p>How many OTUs are there in each module?</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="example.html#cb80-1"></a><span class="kw">table</span>(modules.mgs<span class="op">$</span>colors) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb80-2"><a href="example.html#cb80-2"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb80-3"><a href="example.html#cb80-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">Module =</span> Var1, <span class="dt">Size =</span> Freq) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb80-4"><a href="example.html#cb80-4"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Module_color =</span> <span class="kw">labels2colors</span>(<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(Module)))) -&gt;<span class="st"> </span>module_size</span>
<span id="cb80-5"><a href="example.html#cb80-5"></a></span>
<span id="cb80-6"><a href="example.html#cb80-6"></a>module_size <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb80-7"><a href="example.html#cb80-7"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Module, <span class="dt">y =</span> Size, <span class="dt">fill =</span> Module)) <span class="op">+</span></span>
<span id="cb80-8"><a href="example.html#cb80-8"></a><span class="st">  </span><span class="kw">geom_col</span>(<span class="dt">color =</span>  <span class="st">&quot;#000000&quot;</span>) <span class="op">+</span></span>
<span id="cb80-9"><a href="example.html#cb80-9"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Number of OTUs in each module&quot;</span>) <span class="op">+</span></span>
<span id="cb80-10"><a href="example.html#cb80-10"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb80-11"><a href="example.html#cb80-11"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">setNames</span>(module_size<span class="op">$</span>Module_color,module_size<span class="op">$</span>Module)) <span class="op">+</span></span>
<span id="cb80-12"><a href="example.html#cb80-12"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label =</span> Size),<span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">hjust =</span> <span class="fl">-0.18</span>, <span class="dt">size =</span> <span class="fl">3.5</span>) <span class="op">+</span></span>
<span id="cb80-13"><a href="example.html#cb80-13"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="dv">0</span>, <span class="kw">max</span>(module_size<span class="op">$</span>Size)<span class="op">*</span><span class="fl">1.1</span>) <span class="op">+</span></span>
<span id="cb80-14"><a href="example.html#cb80-14"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">margin</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="st">&quot;pt&quot;</span>)) <span class="op">+</span></span>
<span id="cb80-15"><a href="example.html#cb80-15"></a><span class="st">  </span><span class="kw">coord_flip</span>()-&gt;<span class="st"> </span>module_size_barplot</span>
<span id="cb80-16"><a href="example.html#cb80-16"></a></span>
<span id="cb80-17"><a href="example.html#cb80-17"></a>module_size_barplot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-48-1.png" width="480" /></p>
<p>cluster relationships between the module</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="example.html#cb81-1"></a><span class="co"># labels2colors(modules.mgs$colors)</span></span>
<span id="cb81-2"><a href="example.html#cb81-2"></a></span>
<span id="cb81-3"><a href="example.html#cb81-3"></a><span class="kw">plotEigengeneNetworks</span>(modules.mgs<span class="op">$</span>MEs, <span class="st">&quot;Eigengene adjacency heatmap&quot;</span>,</span>
<span id="cb81-4"><a href="example.html#cb81-4"></a><span class="dt">marDendro =</span> <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">4</span>),</span>
<span id="cb81-5"><a href="example.html#cb81-5"></a><span class="dt">marHeatmap =</span> <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>), <span class="dt">plotDendrograms =</span> T,</span>
<span id="cb81-6"><a href="example.html#cb81-6"></a><span class="dt">xLabelsAngle =</span> <span class="dv">90</span>)</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-49-1.png" width="672" /></p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="example.html#cb82-1"></a><span class="kw">table</span>(modules.mgs<span class="op">$</span>colors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() -&gt;<span class="st"> </span>res</span>
<span id="cb82-2"><a href="example.html#cb82-2"></a>res<span class="op">$</span><span class="st">`</span><span class="dt">Module color</span><span class="st">`</span> &lt;-<span class="st"> </span>WGCNA<span class="op">::</span><span class="kw">labels2colors</span>(<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(res<span class="op">$</span>Var1)))</span>
<span id="cb82-3"><a href="example.html#cb82-3"></a>res &lt;-<span class="st"> </span>res[, <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>)]</span>
<span id="cb82-4"><a href="example.html#cb82-4"></a><span class="kw">colnames</span>(res) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Module&quot;</span>, <span class="st">&quot;Module color&quot;</span>, <span class="st">&quot;Number of OTUs&quot;</span>)</span>
<span id="cb82-5"><a href="example.html#cb82-5"></a>res <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">datatable</span>()</span></code></pre></div>
<div id="htmlwidget-913c7d5303b1c1225408" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-913c7d5303b1c1225408">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8"],["0","1","2","3","4","5","6","7"],["grey","turquoise","blue","brown","yellow","green","red","black"],[174,20,13,12,10,9,7,5]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Module<\/th>\n      <th>Module color<\/th>\n      <th>Number of OTUs<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Dendrogram and module colors</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="example.html#cb83-1"></a><span class="co"># Plot the dendrogram and the module colors underneath for each block</span></span>
<span id="cb83-2"><a href="example.html#cb83-2"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="kw">seq_along</span>(modules.mgs<span class="op">$</span>dendrograms)){</span>
<span id="cb83-3"><a href="example.html#cb83-3"></a>  <span class="kw">plotDendroAndColors</span>(modules.mgs<span class="op">$</span>dendrograms[[i]], merged_colors[modules.mgs<span class="op">$</span>blockGenes[[i]]],</span>
<span id="cb83-4"><a href="example.html#cb83-4"></a>                      <span class="st">&quot;Module colors&quot;</span>,</span>
<span id="cb83-5"><a href="example.html#cb83-5"></a>                      <span class="dt">dendroLabels =</span> <span class="ot">FALSE</span>, <span class="dt">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb83-6"><a href="example.html#cb83-6"></a>                      <span class="dt">addGuide =</span> <span class="ot">TRUE</span>, <span class="dt">guideHang =</span> <span class="fl">0.05</span>,</span>
<span id="cb83-7"><a href="example.html#cb83-7"></a>                      <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Cluster Dendrogram</span><span class="ch">\n</span><span class="st">&quot;</span>, </span>
<span id="cb83-8"><a href="example.html#cb83-8"></a>                                    <span class="st">&quot;for block &quot;</span>, </span>
<span id="cb83-9"><a href="example.html#cb83-9"></a>                                    i,<span class="st">&quot;: &quot;</span>,</span>
<span id="cb83-10"><a href="example.html#cb83-10"></a>                                    <span class="kw">length</span>(modules.mgs<span class="op">$</span>blockGenes[[i]]),</span>
<span id="cb83-11"><a href="example.html#cb83-11"></a>                                    <span class="st">&quot; OTUs&quot;</span>))</span>
<span id="cb83-12"><a href="example.html#cb83-12"></a>}</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-51-1.png" width="960" /></p>
<p>Module (Eigengene) correlation</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="example.html#cb84-1"></a>MEs &lt;-<span class="st"> </span>modules.mgs<span class="op">$</span>MEs</span>
<span id="cb84-2"><a href="example.html#cb84-2"></a></span>
<span id="cb84-3"><a href="example.html#cb84-3"></a><span class="co"># Module correlation to other modules</span></span>
<span id="cb84-4"><a href="example.html#cb84-4"></a>MEs_R &lt;-<span class="st"> </span><span class="kw">bicor</span>(MEs, MEs, <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## Warning in bicor(MEs, MEs, maxPOutliers = 0.05): bicor: zero MAD in variable
## &#39;x&#39;. Pearson correlation was used for individual columns with zero (or missing)
## MAD.</code></pre>
<pre><code>## Warning in bicor(MEs, MEs, maxPOutliers = 0.05): bicor: zero MAD in variable
## &#39;y&#39;. Pearson correlation was used for individual columns with zero (or missing)
## MAD.</code></pre>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="example.html#cb87-1"></a>idx.r &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rownames</span>(MEs_R) <span class="op">==</span><span class="st"> &quot;ME0&quot;</span>)</span>
<span id="cb87-2"><a href="example.html#cb87-2"></a>idx.c &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(MEs_R) <span class="op">==</span><span class="st"> &quot;ME0&quot;</span>)</span>
<span id="cb87-3"><a href="example.html#cb87-3"></a></span>
<span id="cb87-4"><a href="example.html#cb87-4"></a>MEs_R_noME0 &lt;-<span class="st"> </span>MEs_R[<span class="op">-</span>idx.r, <span class="op">-</span>idx.c]</span></code></pre></div>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="example.html#cb88-1"></a>MEs_R[<span class="kw">upper.tri</span>(MEs_R_noME0)] <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb88-2"><a href="example.html#cb88-2"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb88-3"><a href="example.html#cb88-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="st">&quot;correlation&quot;</span> =<span class="st"> &quot;.&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb88-4"><a href="example.html#cb88-4"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>correlation)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb88-5"><a href="example.html#cb88-5"></a><span class="st">  </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb88-6"><a href="example.html#cb88-6"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;mgs&quot;</span>,<span class="st">&quot;ME correlation density</span><span class="ch">\n</span><span class="st"> without &quot;</span>,<span class="st">&quot;mgs&quot;</span> ,<span class="st">&quot;ME0&quot;</span>)) -&gt;<span class="st"> </span>MEs_R_density</span>
<span id="cb88-7"><a href="example.html#cb88-7"></a></span>
<span id="cb88-8"><a href="example.html#cb88-8"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(MEs_R, <span class="dt">color =</span> <span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;Blue&quot;</span>, <span class="st">&quot;White&quot;</span>, <span class="st">&quot;Red&quot;</span>))(<span class="dv">100</span>),</span>
<span id="cb88-9"><a href="example.html#cb88-9"></a>                   <span class="dt">silent =</span> T, </span>
<span id="cb88-10"><a href="example.html#cb88-10"></a>                   <span class="dt">breaks =</span> <span class="kw">seq</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>,<span class="dt">length.out =</span> <span class="dv">101</span>),</span>
<span id="cb88-11"><a href="example.html#cb88-11"></a>                   <span class="dt">treeheight_row =</span> <span class="dv">5</span>, </span>
<span id="cb88-12"><a href="example.html#cb88-12"></a>                   <span class="dt">treeheight_col =</span> <span class="dv">5</span>,</span>
<span id="cb88-13"><a href="example.html#cb88-13"></a>                   <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;mgs&quot;</span>,<span class="st">&quot;ME correlation heatmap&quot;</span>),</span>
<span id="cb88-14"><a href="example.html#cb88-14"></a>                   <span class="dt">labels_row =</span> <span class="kw">paste0</span>(<span class="st">&quot;mgs&quot;</span>, <span class="kw">rownames</span>(MEs_R)),</span>
<span id="cb88-15"><a href="example.html#cb88-15"></a>                   <span class="dt">labels_col =</span> <span class="kw">paste0</span>(<span class="st">&quot;mgs&quot;</span>, <span class="kw">colnames</span>(MEs_R))) -&gt;<span class="st"> </span>MEs_R_Corr</span>
<span id="cb88-16"><a href="example.html#cb88-16"></a></span>
<span id="cb88-17"><a href="example.html#cb88-17"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(MEs_R_density, MEs_R_Corr<span class="op">$</span>gtable, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;D&quot;</span>, <span class="st">&quot;E&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.6</span>, <span class="dv">1</span>)) -&gt;<span class="st"> </span>density_eigen</span>
<span id="cb88-18"><a href="example.html#cb88-18"></a></span>
<span id="cb88-19"><a href="example.html#cb88-19"></a>density_eigen</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-53-1.png" width="960" /></p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="example.html#cb89-1"></a><span class="kw">all</span>(<span class="kw">rownames</span>(mgs_t) <span class="op">==</span><span class="st"> </span><span class="kw">rownames</span>(MEs))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="example.html#cb91-1"></a><span class="kw">dim</span>(mgs_t) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">&quot; samples&quot;</span>, <span class="st">&quot; OTUs&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;32 samples&quot; &quot;250 OTUs&quot;</code></pre>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="example.html#cb93-1"></a>kME &lt;-<span class="st"> </span><span class="kw">bicor</span>(mgs_t, MEs, <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span>
<span id="cb93-2"><a href="example.html#cb93-2"></a><span class="kw">dim</span>(kME) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">&quot; OTUs&quot;</span>, <span class="st">&quot; modules&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;250 OTUs&quot;  &quot;8 modules&quot;</code></pre>
<p>Show a plot of the intra modular correlation; How the OTUs within a module correlates to the module eigengene.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="example.html#cb95-1"></a>intra_cor &lt;-<span class="st"> </span><span class="kw">c</span>()</span>
<span id="cb95-2"><a href="example.html#cb95-2"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(mgs_t)) {</span>
<span id="cb95-3"><a href="example.html#cb95-3"></a>  m &lt;-<span class="st"> </span>modules.mgs<span class="op">$</span>colors[i]</span>
<span id="cb95-4"><a href="example.html#cb95-4"></a>  intra_cor[i] &lt;-<span class="st"> </span>kME[i, <span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>, m)]</span>
<span id="cb95-5"><a href="example.html#cb95-5"></a>  <span class="cf">if</span>(m <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>){</span>
<span id="cb95-6"><a href="example.html#cb95-6"></a>    intra_cor[i] &lt;-<span class="st"> </span>kME[i, <span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>, m)]</span>
<span id="cb95-7"><a href="example.html#cb95-7"></a>  } <span class="cf">else</span>{</span>
<span id="cb95-8"><a href="example.html#cb95-8"></a>    intra_cor[i] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb95-9"><a href="example.html#cb95-9"></a>  }</span>
<span id="cb95-10"><a href="example.html#cb95-10"></a>  </span>
<span id="cb95-11"><a href="example.html#cb95-11"></a>}</span>
<span id="cb95-12"><a href="example.html#cb95-12"></a></span>
<span id="cb95-13"><a href="example.html#cb95-13"></a>idx &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">is.na</span>(intra_cor))</span>
<span id="cb95-14"><a href="example.html#cb95-14"></a>intra_cor &lt;-<span class="st"> </span>intra_cor[<span class="op">-</span>idx]</span>
<span id="cb95-15"><a href="example.html#cb95-15"></a></span>
<span id="cb95-16"><a href="example.html#cb95-16"></a><span class="kw">plot</span>(<span class="kw">density</span>(intra_cor), <span class="dt">main =</span> <span class="st">&quot;Correlations with module-eigenOTU (within module correlation)</span><span class="ch">\n</span><span class="st">No ME0&quot;</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-55-1.png" width="672" /></p>
<p>Show the same thing, but for each module individually, and color by module color.</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="example.html#cb96-1"></a><span class="co"># Corr within modules</span></span>
<span id="cb96-2"><a href="example.html#cb96-2"></a>corr_within_module &lt;-<span class="st"> </span><span class="cf">function</span>(mgs_t, modules, <span class="dt">module_x =</span> <span class="dv">1</span>){</span>
<span id="cb96-3"><a href="example.html#cb96-3"></a>  idx.omics_data &lt;-<span class="st"> </span><span class="kw">which</span>(modules<span class="op">$</span>colors <span class="op">==</span><span class="st"> </span>module_x)</span>
<span id="cb96-4"><a href="example.html#cb96-4"></a>  idx.me &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(modules<span class="op">$</span>MEs) <span class="op">==</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>,module_x))</span>
<span id="cb96-5"><a href="example.html#cb96-5"></a>  kME_x &lt;-<span class="st"> </span><span class="kw">bicor</span>(mgs_t[,idx.omics_data], modules<span class="op">$</span>MEs[,idx.me], <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span>
<span id="cb96-6"><a href="example.html#cb96-6"></a>  kME_x</span>
<span id="cb96-7"><a href="example.html#cb96-7"></a>}</span>
<span id="cb96-8"><a href="example.html#cb96-8"></a></span>
<span id="cb96-9"><a href="example.html#cb96-9"></a>ggplot.list &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb96-10"><a href="example.html#cb96-10"></a></span>
<span id="cb96-11"><a href="example.html#cb96-11"></a><span class="cf">for</span>(m <span class="cf">in</span> <span class="kw">colnames</span>(modules.mgs<span class="op">$</span>MEs)){</span>
<span id="cb96-12"><a href="example.html#cb96-12"></a>  h &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">sub</span>(<span class="st">&quot;ME&quot;</span>,<span class="st">&quot;&quot;</span>, m))</span>
<span id="cb96-13"><a href="example.html#cb96-13"></a>  <span class="kw">data.frame</span>(<span class="dt">x =</span> <span class="kw">suppressWarnings</span>(<span class="kw">corr_within_module</span>(<span class="dt">mgs_t =</span> mgs_t, <span class="dt">modules =</span> modules.mgs, <span class="dt">module_x =</span> h))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb96-14"><a href="example.html#cb96-14"></a><span class="st">    </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb96-15"><a href="example.html#cb96-15"></a><span class="st">    </span><span class="kw">geom_density</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x), <span class="dt">fill =</span> <span class="kw">labels2colors</span>(h), <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb96-16"><a href="example.html#cb96-16"></a><span class="st">    </span><span class="kw">xlim</span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb96-17"><a href="example.html#cb96-17"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;OTU correlation&quot;</span>)<span class="op">+</span></span>
<span id="cb96-18"><a href="example.html#cb96-18"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;mgs&quot;</span>,m)) -&gt;<span class="st"> </span>da_plot</span>
<span id="cb96-19"><a href="example.html#cb96-19"></a>  </span>
<span id="cb96-20"><a href="example.html#cb96-20"></a>  ggplot.list[[m]] &lt;-<span class="st"> </span>da_plot</span>
<span id="cb96-21"><a href="example.html#cb96-21"></a>}</span>
<span id="cb96-22"><a href="example.html#cb96-22"></a></span>
<span id="cb96-23"><a href="example.html#cb96-23"></a>ggplot.list &lt;-<span class="st"> </span>ggplot.list[ggplot.list <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">names</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;ME&quot;</span>, <span class="st">&quot;&quot;</span>, .) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.numeric</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">order</span>()]</span></code></pre></div>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="example.html#cb97-1"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> ggplot.list, <span class="dt">ncol =</span> <span class="dv">2</span>) -&gt;<span class="st"> </span>density_all_plot</span>
<span id="cb97-2"><a href="example.html#cb97-2"></a>density_all_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-57-1.png" width="768" /></p>
<p>Combine to one plot</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="example.html#cb98-1"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(si_mc_plot , density_eigen, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>)) -&gt;<span class="st"> </span>part_<span class="dv">1</span></span>
<span id="cb98-2"><a href="example.html#cb98-2"></a></span>
<span id="cb98-3"><a href="example.html#cb98-3"></a></span>
<span id="cb98-4"><a href="example.html#cb98-4"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(part_<span class="dv">1</span>, module_size_barplot, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;C&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="fl">0.5</span>)) -&gt;<span class="st"> </span>part_<span class="dv">2</span></span>
<span id="cb98-5"><a href="example.html#cb98-5"></a></span>
<span id="cb98-6"><a href="example.html#cb98-6"></a></span>
<span id="cb98-7"><a href="example.html#cb98-7"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(part_<span class="dv">2</span>, density_all_plot, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;F&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>)</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-58-1.png" width="1152" /></p>
<p>Hub genes</p>
<p>For each module it is possible to pick a hub gene with the function <code>chooseTopHubInEachModule</code>.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="example.html#cb99-1"></a>hubs.mgs <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb99-2"><a href="example.html#cb99-2"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb99-3"><a href="example.html#cb99-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="st">&quot;OTU_name&quot;</span> =<span class="st"> &quot;.&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb99-4"><a href="example.html#cb99-4"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;Module&quot;</span>) -&gt;<span class="st"> </span>hubOTUs</span>
<span id="cb99-5"><a href="example.html#cb99-5"></a></span>
<span id="cb99-6"><a href="example.html#cb99-6"></a></span>
<span id="cb99-7"><a href="example.html#cb99-7"></a></span>
<span id="cb99-8"><a href="example.html#cb99-8"></a>dplyr<span class="op">::</span><span class="kw">left_join</span>(hubOTUs, </span>
<span id="cb99-9"><a href="example.html#cb99-9"></a>                 (taxa_table <span class="op">%&gt;%</span></span>
<span id="cb99-10"><a href="example.html#cb99-10"></a><span class="st">                    </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;OTU_name&quot;</span>)), </span>
<span id="cb99-11"><a href="example.html#cb99-11"></a>                 <span class="dt">by =</span> <span class="st">&quot;OTU_name&quot;</span>) -&gt;<span class="st"> </span>hubOTUs</span>
<span id="cb99-12"><a href="example.html#cb99-12"></a></span>
<span id="cb99-13"><a href="example.html#cb99-13"></a>hubOTUs <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">datatable</span>()</span></code></pre></div>
<div id="htmlwidget-04b026ecb877ad24b2f2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-04b026ecb877ad24b2f2">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8"],["0","1","2","3","4","5","6","7"],["Lachnospiraceae_bacterium_5_1_63FAA","Megamonas_rupellensis","Streptococcus_agalactiae","Lachnospiraceae_bacterium_5_1_57FAA","Actinomyces_massiliensis","Lactobacillus_plantarum","Erysipelotrichaceae_bacterium_6_1_45","Bacteroides_faecis"],["k__Bacteria","k__Bacteria","k__Bacteria","k__Bacteria","k__Bacteria","k__Bacteria","k__Bacteria","k__Bacteria"],["p__Firmicutes","p__Firmicutes","p__Firmicutes","p__Firmicutes","p__Actinobacteria","p__Firmicutes","p__Firmicutes","p__Bacteroidetes"],["c__Clostridia","c__Negativicutes","c__Bacilli","c__Clostridia","c__Actinobacteria","c__Bacilli","c__Erysipelotrichia","c__Bacteroidia"],["o__Clostridiales","o__Selenomonadales","o__Lactobacillales","o__Clostridiales","o__Actinomycetales","o__Lactobacillales","o__Erysipelotrichales","o__Bacteroidales"],["f__Lachnospiraceae","f__Veillonellaceae","f__Streptococcaceae","f__Lachnospiraceae","f__Actinomycetaceae","f__Lactobacillaceae","f__Erysipelotrichaceae","f__Bacteroidaceae"],["g__Lachnospiraceae_noname","g__Megamonas","g__Streptococcus","g__Lachnospiraceae_noname","g__Actinomyces","g__Lactobacillus","g__Erysipelotrichaceae_noname","g__Bacteroides"],["s__Lachnospiraceae_bacterium_5_1_63FAA","s__Megamonas_rupellensis","s__Streptococcus_agalactiae","s__Lachnospiraceae_bacterium_5_1_57FAA","s__Actinomyces_massiliensis","s__Lactobacillus_plantarum","s__Erysipelotrichaceae_bacterium_6_1_45","s__Bacteroides_faecis"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Module<\/th>\n      <th>OTU_name<\/th>\n      <th>Kingdom<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n      <th>Genus<\/th>\n      <th>Species<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="metabolites-constructe-module" class="section level2 hasAnchor">
<h2><span class="header-section-number">3.7</span> metabolites constructe module<a href="example.html#metabolites-constructe-module" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="example.html#cb100-1"></a>metabolites_t &lt;-<span class="st"> </span>metabolites_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>()</span></code></pre></div>
<div id="select-power-1" class="section level3 hasAnchor">
<h3><span class="header-section-number">3.7.1</span> select power<a href="example.html#select-power-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="example.html#cb101-1"></a>powers &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">seq</span>(<span class="dv">12</span>,<span class="dv">20</span>,<span class="dv">2</span>))</span>
<span id="cb101-2"><a href="example.html#cb101-2"></a></span>
<span id="cb101-3"><a href="example.html#cb101-3"></a><span class="kw">suppressWarnings</span>(sft &lt;-<span class="st"> </span><span class="kw">pickSoftThreshold</span>(metabolites_t, </span>
<span id="cb101-4"><a href="example.html#cb101-4"></a>                                          <span class="dt">powerVector =</span> powers, </span>
<span id="cb101-5"><a href="example.html#cb101-5"></a>                                          <span class="dt">verbose =</span> <span class="dv">1</span>, </span>
<span id="cb101-6"><a href="example.html#cb101-6"></a>                                          <span class="dt">networkType =</span> <span class="st">&quot;signed&quot;</span>,</span>
<span id="cb101-7"><a href="example.html#cb101-7"></a>                                          <span class="dt">corFn=</span> <span class="st">&quot;bicor&quot;</span>))</span></code></pre></div>
<pre><code>## pickSoftThreshold: will use block size 811.
##  pickSoftThreshold: calculating connectivity for given powers... ..0% ..100% 
##    Power SFT.R.sq   slope truncated.R.sq mean.k. median.k. max.k.
## 1      1  0.72200  9.9100          0.756  456.00    462.00  520.0
## 2      2  0.79300  4.6000          0.924  273.00    278.00  345.0
## 3      3  0.75800  2.8500          0.951  172.00    175.00  238.0
## 4      4  0.57200  1.7000          0.941  113.00    113.00  176.0
## 5      5  0.26700  0.7160          0.932   76.60     76.00  134.0
## 6      6  0.00268  0.0619          0.861   53.80     52.30  108.0
## 7      7  0.11500 -0.3880          0.920   38.90     36.70   89.5
## 8      8  0.30800 -0.7520          0.901   28.90     26.60   75.3
## 9      9  0.41300 -0.9160          0.918   22.00     19.40   64.3
## 10    10  0.53800 -1.2300          0.901   17.00     14.50   55.5
## 11    12  0.64600 -1.5400          0.935   10.80      8.60   42.5
## 12    14  0.71500 -1.6400          0.954    7.32      5.48   33.4
## 13    16  0.75500 -1.6800          0.960    5.23      3.52   26.8
## 14    18  0.77000 -1.7300          0.946    3.90      2.49   21.9
## 15    20  0.74400 -1.7800          0.912    3.03      1.74   18.1</code></pre>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="example.html#cb103-1"></a><span class="co"># Find the soft thresholding power beta to which co-expression similarity is raised to calculate adjacency.</span></span>
<span id="cb103-2"><a href="example.html#cb103-2"></a><span class="co"># based on the criterion of approximate scale-free topology.</span></span>
<span id="cb103-3"><a href="example.html#cb103-3"></a></span>
<span id="cb103-4"><a href="example.html#cb103-4"></a>idx &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">which</span>((<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.90</span>))</span>
<span id="cb103-5"><a href="example.html#cb103-5"></a><span class="cf">if</span>(<span class="kw">is.infinite</span>(idx)){</span>
<span id="cb103-6"><a href="example.html#cb103-6"></a>  idx &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">which</span>((<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.80</span>))</span>
<span id="cb103-7"><a href="example.html#cb103-7"></a>  <span class="cf">if</span>(<span class="op">!</span><span class="kw">is.infinite</span>(idx)){</span>
<span id="cb103-8"><a href="example.html#cb103-8"></a>    st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb103-9"><a href="example.html#cb103-9"></a>  } <span class="cf">else</span>{</span>
<span id="cb103-10"><a href="example.html#cb103-10"></a>    idx &lt;-<span class="st"> </span><span class="kw">which.max</span>(<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>])</span>
<span id="cb103-11"><a href="example.html#cb103-11"></a>    st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb103-12"><a href="example.html#cb103-12"></a>  }</span>
<span id="cb103-13"><a href="example.html#cb103-13"></a>} <span class="cf">else</span>{</span>
<span id="cb103-14"><a href="example.html#cb103-14"></a>  st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb103-15"><a href="example.html#cb103-15"></a>}</span>
<span id="cb103-16"><a href="example.html#cb103-16"></a></span>
<span id="cb103-17"><a href="example.html#cb103-17"></a></span>
<span id="cb103-18"><a href="example.html#cb103-18"></a><span class="co"># Plot Scale independence measure and Mean connectivity measure</span></span>
<span id="cb103-19"><a href="example.html#cb103-19"></a></span>
<span id="cb103-20"><a href="example.html#cb103-20"></a><span class="co"># Scale-free topology fit index as a function of the soft-thresholding power</span></span>
<span id="cb103-21"><a href="example.html#cb103-21"></a><span class="kw">data.frame</span>(<span class="dt">Indices =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb103-22"><a href="example.html#cb103-22"></a>           <span class="dt">sfApprox =</span> <span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb103-23"><a href="example.html#cb103-23"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb103-24"><a href="example.html#cb103-24"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="fl">0.9</span>, <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.6</span>) <span class="op">+</span><span class="st"> </span><span class="co"># corresponds to R^2 cut-off of 0.9</span></span>
<span id="cb103-25"><a href="example.html#cb103-25"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="fl">0.8</span>, <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.2</span>) <span class="op">+</span><span class="st"> </span><span class="co"># corresponds to R^2 cut-off of 0.8</span></span>
<span id="cb103-26"><a href="example.html#cb103-26"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> sfApprox), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb103-27"><a href="example.html#cb103-27"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> sfApprox, <span class="dt">label =</span> Indices), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb103-28"><a href="example.html#cb103-28"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Scale independence&quot;</span>) <span class="op">+</span></span>
<span id="cb103-29"><a href="example.html#cb103-29"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Soft Threshold (power)&quot;</span>) <span class="op">+</span></span>
<span id="cb103-30"><a href="example.html#cb103-30"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;SF Model Fit,signed R^2&quot;</span>) <span class="op">+</span></span>
<span id="cb103-31"><a href="example.html#cb103-31"></a><span class="st">  </span><span class="kw">xlim</span>(<span class="dv">1</span>,<span class="dv">20</span>) <span class="op">+</span></span>
<span id="cb103-32"><a href="example.html#cb103-32"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb103-33"><a href="example.html#cb103-33"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> st, <span class="dt">y =</span> <span class="fl">0.25</span>, <span class="dt">xend =</span> st, <span class="dt">yend =</span> sfApprox[idx]<span class="op">-</span><span class="fl">0.05</span>), </span>
<span id="cb103-34"><a href="example.html#cb103-34"></a>               <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.2</span>,<span class="st">&quot;cm&quot;</span>)), </span>
<span id="cb103-35"><a href="example.html#cb103-35"></a>               <span class="dt">size =</span> <span class="fl">0.5</span>)-&gt;<span class="st"> </span>scale_independence_plot </span>
<span id="cb103-36"><a href="example.html#cb103-36"></a>  </span>
<span id="cb103-37"><a href="example.html#cb103-37"></a> </span>
<span id="cb103-38"><a href="example.html#cb103-38"></a></span>
<span id="cb103-39"><a href="example.html#cb103-39"></a></span>
<span id="cb103-40"><a href="example.html#cb103-40"></a><span class="co"># Mean connectivity as a function of the soft-thresholding power</span></span>
<span id="cb103-41"><a href="example.html#cb103-41"></a></span>
<span id="cb103-42"><a href="example.html#cb103-42"></a><span class="kw">data.frame</span>(<span class="dt">Indices =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb103-43"><a href="example.html#cb103-43"></a>           <span class="dt">meanApprox =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">5</span>]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb103-44"><a href="example.html#cb103-44"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb103-45"><a href="example.html#cb103-45"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> meanApprox), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb103-46"><a href="example.html#cb103-46"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> meanApprox, <span class="dt">label =</span> Indices), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb103-47"><a href="example.html#cb103-47"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Soft Threshold (power)&quot;</span>) <span class="op">+</span></span>
<span id="cb103-48"><a href="example.html#cb103-48"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Mean Connectivity&quot;</span>) <span class="op">+</span></span>
<span id="cb103-49"><a href="example.html#cb103-49"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> st<span class="fl">-0.4</span>, </span>
<span id="cb103-50"><a href="example.html#cb103-50"></a>                   <span class="dt">y =</span> sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx], </span>
<span id="cb103-51"><a href="example.html#cb103-51"></a>                   <span class="dt">xend =</span> <span class="dv">0</span>, </span>
<span id="cb103-52"><a href="example.html#cb103-52"></a>                   <span class="dt">yend =</span> sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx]),</span>
<span id="cb103-53"><a href="example.html#cb103-53"></a>               <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.2</span>,<span class="st">&quot;cm&quot;</span>)), </span>
<span id="cb103-54"><a href="example.html#cb103-54"></a>               <span class="dt">size =</span> <span class="fl">0.4</span>) <span class="op">+</span></span>
<span id="cb103-55"><a href="example.html#cb103-55"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;Mean connectivity: &quot;</span>, </span>
<span id="cb103-56"><a href="example.html#cb103-56"></a>                 <span class="kw">round</span>(sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx],<span class="dv">2</span>))) -&gt;<span class="st"> </span>mean_connectivity_plot</span>
<span id="cb103-57"><a href="example.html#cb103-57"></a></span>
<span id="cb103-58"><a href="example.html#cb103-58"></a></span>
<span id="cb103-59"><a href="example.html#cb103-59"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(scale_independence_plot, mean_connectivity_plot, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">align =</span> <span class="st">&quot;h&quot;</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>) -&gt;<span class="st"> </span>si_mc_plot</span>
<span id="cb103-60"><a href="example.html#cb103-60"></a></span>
<span id="cb103-61"><a href="example.html#cb103-61"></a>si_mc_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-61-1.png" width="672" /></p>
<p>The power is 18</p>
</div>
<div id="block-wise-network-construction-and-module-detection-1" class="section level3 hasAnchor">
<h3><span class="header-section-number">3.7.2</span> Block-wise network construction and module detection<a href="example.html#block-wise-network-construction-and-module-detection-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The function <code>blockwiseModules</code> will first pre cluster with fast crude clustering method to cluster OTUs into blocks not exceeding the maximum, blocks may therefore not be fully optimal in the end.</p>
<p>Change the parameters here to better reflect your own data.</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="example.html#cb104-1"></a>Run_analysis &lt;-<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb104-2"><a href="example.html#cb104-2"></a><span class="cf">if</span>(Run_analysis){</span>
<span id="cb104-3"><a href="example.html#cb104-3"></a>  modules.metabolites &lt;-<span class="st"> </span><span class="kw">blockwiseModules</span>(metabolites_t,</span>
<span id="cb104-4"><a href="example.html#cb104-4"></a>                          <span class="dt">power =</span> st, </span>
<span id="cb104-5"><a href="example.html#cb104-5"></a>                          <span class="dt">networkType =</span> <span class="st">&quot;signed&quot;</span>, </span>
<span id="cb104-6"><a href="example.html#cb104-6"></a>                          <span class="dt">TOMType =</span> <span class="st">&quot;signed&quot;</span>,</span>
<span id="cb104-7"><a href="example.html#cb104-7"></a>                          <span class="dt">corType =</span> <span class="st">&#39;bicor&#39;</span>,</span>
<span id="cb104-8"><a href="example.html#cb104-8"></a>                          <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>,</span>
<span id="cb104-9"><a href="example.html#cb104-9"></a>                          <span class="dt">deepSplit =</span> <span class="dv">2</span>,</span>
<span id="cb104-10"><a href="example.html#cb104-10"></a>                          <span class="dt">pamStage =</span> <span class="ot">FALSE</span>, </span>
<span id="cb104-11"><a href="example.html#cb104-11"></a>                          <span class="dt">pamRespectsDendro =</span> <span class="ot">TRUE</span>,</span>
<span id="cb104-12"><a href="example.html#cb104-12"></a>                          <span class="dt">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb104-13"><a href="example.html#cb104-13"></a>                          <span class="dt">replaceMissingAdjacencies =</span> <span class="ot">TRUE</span>,</span>
<span id="cb104-14"><a href="example.html#cb104-14"></a>                          <span class="dt">minModuleSize =</span> <span class="dv">5</span>, <span class="co"># There are fewer otus than genes, and that many might not be connected</span></span>
<span id="cb104-15"><a href="example.html#cb104-15"></a>                          <span class="dt">numericLabels =</span> <span class="ot">TRUE</span>,</span>
<span id="cb104-16"><a href="example.html#cb104-16"></a>                          <span class="dt">saveTOMs =</span> <span class="ot">TRUE</span>,</span>
<span id="cb104-17"><a href="example.html#cb104-17"></a>                          <span class="dt">saveTOMFileBase =</span> <span class="st">&quot;./result/module.metabolites&quot;</span>, <span class="co">#&quot;/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/GVHD/bookdown_GVHD/result/module.metabolites&quot;,</span></span>
<span id="cb104-18"><a href="example.html#cb104-18"></a>                          <span class="dt">verbose =</span> <span class="dv">1</span>)</span>
<span id="cb104-19"><a href="example.html#cb104-19"></a>  </span>
<span id="cb104-20"><a href="example.html#cb104-20"></a>  <span class="kw">rownames</span>(modules.metabolites<span class="op">$</span>MEs) &lt;-<span class="st"> </span><span class="kw">rownames</span>(metabolites_t)</span>
<span id="cb104-21"><a href="example.html#cb104-21"></a>  <span class="kw">names</span>(modules.metabolites<span class="op">$</span>colors) &lt;-<span class="st"> </span><span class="kw">colnames</span>(metabolites_t)</span>
<span id="cb104-22"><a href="example.html#cb104-22"></a>  <span class="kw">names</span>(modules.metabolites<span class="op">$</span>unmergedColors) &lt;-<span class="st"> </span><span class="kw">colnames</span>(metabolites_t)</span>
<span id="cb104-23"><a href="example.html#cb104-23"></a>  </span>
<span id="cb104-24"><a href="example.html#cb104-24"></a>  hubs.metabolites &lt;-<span class="st"> </span><span class="kw">chooseTopHubInEachModule</span>(metabolites_t, modules.metabolites<span class="op">$</span>colors)</span>
<span id="cb104-25"><a href="example.html#cb104-25"></a>}</span></code></pre></div>
<pre><code>##  Calculating module eigengenes block-wise from all genes</code></pre>
</div>
<div id="metabolites-module-details" class="section level3 hasAnchor">
<h3><span class="header-section-number">3.7.3</span> metabolites module details<a href="example.html#metabolites-module-details" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="example.html#cb106-1"></a><span class="co"># Convert labels to colors for plotting</span></span>
<span id="cb106-2"><a href="example.html#cb106-2"></a>merged_colors &lt;-<span class="st"> </span><span class="kw">labels2colors</span>(modules.metabolites<span class="op">$</span>colors)</span></code></pre></div>
<p>All samples are OK.<br />
All OTUs are OK.</p>
<p>There where 27 modules found.<br />
All module eigenOTUs are OK.</p>
<p>How many Metabolites are there in each module?</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="example.html#cb107-1"></a><span class="kw">table</span>(modules.metabolites<span class="op">$</span>colors) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb107-2"><a href="example.html#cb107-2"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb107-3"><a href="example.html#cb107-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">Module =</span> Var1, <span class="dt">Size =</span> Freq) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb107-4"><a href="example.html#cb107-4"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Module_color =</span> <span class="kw">labels2colors</span>(<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(Module)))) -&gt;<span class="st"> </span>module_size</span>
<span id="cb107-5"><a href="example.html#cb107-5"></a></span>
<span id="cb107-6"><a href="example.html#cb107-6"></a>module_size <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb107-7"><a href="example.html#cb107-7"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Module, <span class="dt">y =</span> Size, <span class="dt">fill =</span> Module)) <span class="op">+</span></span>
<span id="cb107-8"><a href="example.html#cb107-8"></a><span class="st">  </span><span class="kw">geom_col</span>(<span class="dt">color =</span>  <span class="st">&quot;#000000&quot;</span>) <span class="op">+</span></span>
<span id="cb107-9"><a href="example.html#cb107-9"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Number of Metabolites in each module&quot;</span>) <span class="op">+</span></span>
<span id="cb107-10"><a href="example.html#cb107-10"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb107-11"><a href="example.html#cb107-11"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">setNames</span>(module_size<span class="op">$</span>Module_color,module_size<span class="op">$</span>Module)) <span class="op">+</span></span>
<span id="cb107-12"><a href="example.html#cb107-12"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label =</span> Size),<span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">hjust =</span> <span class="fl">-0.18</span>, <span class="dt">size =</span> <span class="fl">3.5</span>) <span class="op">+</span></span>
<span id="cb107-13"><a href="example.html#cb107-13"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="dv">0</span>, <span class="kw">max</span>(module_size<span class="op">$</span>Size)<span class="op">*</span><span class="fl">1.1</span>) <span class="op">+</span></span>
<span id="cb107-14"><a href="example.html#cb107-14"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">margin</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="st">&quot;pt&quot;</span>)) <span class="op">+</span></span>
<span id="cb107-15"><a href="example.html#cb107-15"></a><span class="st">  </span><span class="kw">coord_flip</span>()-&gt;<span class="st"> </span>module_size_barplot</span>
<span id="cb107-16"><a href="example.html#cb107-16"></a></span>
<span id="cb107-17"><a href="example.html#cb107-17"></a>module_size_barplot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-65-1.png" width="480" /></p>
<p>cluster relationships between the module</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="example.html#cb108-1"></a><span class="co"># labels2colors(modules.metabolites$colors)</span></span>
<span id="cb108-2"><a href="example.html#cb108-2"></a></span>
<span id="cb108-3"><a href="example.html#cb108-3"></a><span class="kw">plotEigengeneNetworks</span>(modules.metabolites<span class="op">$</span>MEs, <span class="st">&quot;Eigengene adjacency heatmap&quot;</span>,</span>
<span id="cb108-4"><a href="example.html#cb108-4"></a><span class="dt">marDendro =</span> <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">4</span>),</span>
<span id="cb108-5"><a href="example.html#cb108-5"></a><span class="dt">marHeatmap =</span> <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>), <span class="dt">plotDendrograms =</span> T,</span>
<span id="cb108-6"><a href="example.html#cb108-6"></a><span class="dt">xLabelsAngle =</span> <span class="dv">90</span>)</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-66-1.png" width="672" /></p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="example.html#cb109-1"></a><span class="kw">table</span>(modules.metabolites<span class="op">$</span>colors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() -&gt;<span class="st"> </span>res</span>
<span id="cb109-2"><a href="example.html#cb109-2"></a>res<span class="op">$</span><span class="st">`</span><span class="dt">Module color</span><span class="st">`</span> &lt;-<span class="st"> </span>WGCNA<span class="op">::</span><span class="kw">labels2colors</span>(<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(res<span class="op">$</span>Var1)))</span>
<span id="cb109-3"><a href="example.html#cb109-3"></a>res &lt;-<span class="st"> </span>res[, <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>)]</span>
<span id="cb109-4"><a href="example.html#cb109-4"></a><span class="kw">colnames</span>(res) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Module&quot;</span>, <span class="st">&quot;Module color&quot;</span>, <span class="st">&quot;Number of metabolic&quot;</span>)</span>
<span id="cb109-5"><a href="example.html#cb109-5"></a>res <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">datatable</span>()</span></code></pre></div>
<div id="htmlwidget-42180689214f46035eed" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-42180689214f46035eed">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27"],["0","1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26"],["grey","turquoise","blue","brown","yellow","green","red","black","pink","magenta","purple","greenyellow","tan","salmon","cyan","midnightblue","lightcyan","grey60","lightgreen","lightyellow","royalblue","darkred","darkgreen","darkturquoise","darkgrey","orange","darkorange"],[408,54,34,34,30,24,24,23,22,20,16,16,15,11,10,9,7,6,6,6,6,5,5,5,5,5,5]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Module<\/th>\n      <th>Module color<\/th>\n      <th>Number of metabolic<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Dendrogram and module colors</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="example.html#cb110-1"></a><span class="co"># Plot the dendrogram and the module colors underneath for each block</span></span>
<span id="cb110-2"><a href="example.html#cb110-2"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="kw">seq_along</span>(modules.metabolites<span class="op">$</span>dendrograms)){</span>
<span id="cb110-3"><a href="example.html#cb110-3"></a>  <span class="kw">plotDendroAndColors</span>(modules.metabolites<span class="op">$</span>dendrograms[[i]], merged_colors[modules.metabolites<span class="op">$</span>blockGenes[[i]]],</span>
<span id="cb110-4"><a href="example.html#cb110-4"></a>                      <span class="st">&quot;Module colors&quot;</span>,</span>
<span id="cb110-5"><a href="example.html#cb110-5"></a>                      <span class="dt">dendroLabels =</span> <span class="ot">FALSE</span>, <span class="dt">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb110-6"><a href="example.html#cb110-6"></a>                      <span class="dt">addGuide =</span> <span class="ot">TRUE</span>, <span class="dt">guideHang =</span> <span class="fl">0.05</span>,</span>
<span id="cb110-7"><a href="example.html#cb110-7"></a>                      <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Cluster Dendrogram</span><span class="ch">\n</span><span class="st">&quot;</span>, </span>
<span id="cb110-8"><a href="example.html#cb110-8"></a>                                    <span class="st">&quot;for block &quot;</span>, </span>
<span id="cb110-9"><a href="example.html#cb110-9"></a>                                    i,<span class="st">&quot;: &quot;</span>,</span>
<span id="cb110-10"><a href="example.html#cb110-10"></a>                                    <span class="kw">length</span>(modules.metabolites<span class="op">$</span>blockGenes[[i]]),</span>
<span id="cb110-11"><a href="example.html#cb110-11"></a>                                    <span class="st">&quot; metabolic&quot;</span>))</span>
<span id="cb110-12"><a href="example.html#cb110-12"></a>}</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-68-1.png" width="960" /></p>
<p>Module (Eigengene) correlation</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="example.html#cb111-1"></a>MEs &lt;-<span class="st"> </span>modules.metabolites<span class="op">$</span>MEs</span>
<span id="cb111-2"><a href="example.html#cb111-2"></a></span>
<span id="cb111-3"><a href="example.html#cb111-3"></a><span class="co"># Module correlation to other modules</span></span>
<span id="cb111-4"><a href="example.html#cb111-4"></a>MEs_R &lt;-<span class="st"> </span><span class="kw">bicor</span>(MEs, MEs, <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span>
<span id="cb111-5"><a href="example.html#cb111-5"></a></span>
<span id="cb111-6"><a href="example.html#cb111-6"></a>idx.r &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rownames</span>(MEs_R) <span class="op">==</span><span class="st"> &quot;ME0&quot;</span>)</span>
<span id="cb111-7"><a href="example.html#cb111-7"></a>idx.c &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(MEs_R) <span class="op">==</span><span class="st"> &quot;ME0&quot;</span>)</span>
<span id="cb111-8"><a href="example.html#cb111-8"></a></span>
<span id="cb111-9"><a href="example.html#cb111-9"></a>MEs_R_noME0 &lt;-<span class="st"> </span>MEs_R[<span class="op">-</span>idx.r, <span class="op">-</span>idx.c]</span></code></pre></div>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="example.html#cb112-1"></a>MEs_R[<span class="kw">upper.tri</span>(MEs_R_noME0)] <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb112-2"><a href="example.html#cb112-2"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb112-3"><a href="example.html#cb112-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="st">&quot;correlation&quot;</span> =<span class="st"> &quot;.&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb112-4"><a href="example.html#cb112-4"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>correlation)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb112-5"><a href="example.html#cb112-5"></a><span class="st">  </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb112-6"><a href="example.html#cb112-6"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;mbs&quot;</span>,<span class="st">&quot;ME correlation density</span><span class="ch">\n</span><span class="st"> without &quot;</span>,<span class="st">&quot;mbs&quot;</span> ,<span class="st">&quot;ME0&quot;</span>)) -&gt;<span class="st"> </span>MEs_R_density</span>
<span id="cb112-7"><a href="example.html#cb112-7"></a></span>
<span id="cb112-8"><a href="example.html#cb112-8"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(MEs_R, <span class="dt">color =</span> <span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;Blue&quot;</span>, <span class="st">&quot;White&quot;</span>, <span class="st">&quot;Red&quot;</span>))(<span class="dv">100</span>),</span>
<span id="cb112-9"><a href="example.html#cb112-9"></a>                   <span class="dt">silent =</span> T, </span>
<span id="cb112-10"><a href="example.html#cb112-10"></a>                   <span class="dt">breaks =</span> <span class="kw">seq</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>,<span class="dt">length.out =</span> <span class="dv">101</span>),</span>
<span id="cb112-11"><a href="example.html#cb112-11"></a>                   <span class="dt">treeheight_row =</span> <span class="dv">5</span>, </span>
<span id="cb112-12"><a href="example.html#cb112-12"></a>                   <span class="dt">treeheight_col =</span> <span class="dv">5</span>,</span>
<span id="cb112-13"><a href="example.html#cb112-13"></a>                   <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;mbs&quot;</span>,<span class="st">&quot;ME correlation heatmap&quot;</span>),</span>
<span id="cb112-14"><a href="example.html#cb112-14"></a>                   <span class="dt">labels_row =</span> <span class="kw">paste0</span>(<span class="st">&quot;mbs&quot;</span>, <span class="kw">rownames</span>(MEs_R)),</span>
<span id="cb112-15"><a href="example.html#cb112-15"></a>                   <span class="dt">labels_col =</span> <span class="kw">paste0</span>(<span class="st">&quot;mbs&quot;</span>, <span class="kw">colnames</span>(MEs_R))) -&gt;<span class="st"> </span>MEs_R_Corr</span>
<span id="cb112-16"><a href="example.html#cb112-16"></a></span>
<span id="cb112-17"><a href="example.html#cb112-17"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(MEs_R_density, MEs_R_Corr<span class="op">$</span>gtable, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;D&quot;</span>, <span class="st">&quot;E&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.6</span>, <span class="dv">1</span>)) -&gt;<span class="st"> </span>density_eigen</span>
<span id="cb112-18"><a href="example.html#cb112-18"></a></span>
<span id="cb112-19"><a href="example.html#cb112-19"></a>density_eigen</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-70-1.png" width="960" /></p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="example.html#cb113-1"></a><span class="kw">all</span>(<span class="kw">rownames</span>(metabolites_t) <span class="op">==</span><span class="st"> </span><span class="kw">rownames</span>(MEs))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="example.html#cb115-1"></a><span class="kw">dim</span>(metabolites_t) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">&quot; samples&quot;</span>, <span class="st">&quot; metabolic&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;32 samples&quot;    &quot;811 metabolic&quot;</code></pre>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="example.html#cb117-1"></a>kME &lt;-<span class="st"> </span><span class="kw">bicor</span>(metabolites_t, MEs, <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span>
<span id="cb117-2"><a href="example.html#cb117-2"></a><span class="kw">dim</span>(kME) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">&quot; metabolic&quot;</span>, <span class="st">&quot; modules&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;811 metabolic&quot; &quot;27 modules&quot;</code></pre>
<p>Show a plot of the intra modular correlation; How the OTUs within a module correlates to the module eigengene.</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="example.html#cb119-1"></a>intra_cor &lt;-<span class="st"> </span><span class="kw">c</span>()</span>
<span id="cb119-2"><a href="example.html#cb119-2"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(metabolites_t)) {</span>
<span id="cb119-3"><a href="example.html#cb119-3"></a>  m &lt;-<span class="st"> </span>modules.metabolites<span class="op">$</span>colors[i]</span>
<span id="cb119-4"><a href="example.html#cb119-4"></a>  intra_cor[i] &lt;-<span class="st"> </span>kME[i, <span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>, m)]</span>
<span id="cb119-5"><a href="example.html#cb119-5"></a>  <span class="cf">if</span>(m <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>){</span>
<span id="cb119-6"><a href="example.html#cb119-6"></a>    intra_cor[i] &lt;-<span class="st"> </span>kME[i, <span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>, m)]</span>
<span id="cb119-7"><a href="example.html#cb119-7"></a>  } <span class="cf">else</span>{</span>
<span id="cb119-8"><a href="example.html#cb119-8"></a>    intra_cor[i] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb119-9"><a href="example.html#cb119-9"></a>  }</span>
<span id="cb119-10"><a href="example.html#cb119-10"></a>  </span>
<span id="cb119-11"><a href="example.html#cb119-11"></a>}</span>
<span id="cb119-12"><a href="example.html#cb119-12"></a></span>
<span id="cb119-13"><a href="example.html#cb119-13"></a>idx &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">is.na</span>(intra_cor))</span>
<span id="cb119-14"><a href="example.html#cb119-14"></a>intra_cor &lt;-<span class="st"> </span>intra_cor[<span class="op">-</span>idx]</span>
<span id="cb119-15"><a href="example.html#cb119-15"></a></span>
<span id="cb119-16"><a href="example.html#cb119-16"></a><span class="kw">plot</span>(<span class="kw">density</span>(intra_cor), <span class="dt">main =</span> <span class="st">&quot;Correlations with module-eigenMBS (within module correlation)</span><span class="ch">\n</span><span class="st">No ME0&quot;</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-72-1.png" width="672" /></p>
<p>Show the same thing, but for each module individually, and color by module color.</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="example.html#cb120-1"></a><span class="co"># Corr within modules</span></span>
<span id="cb120-2"><a href="example.html#cb120-2"></a>corr_within_module &lt;-<span class="st"> </span><span class="cf">function</span>(metabolites_t, modules, <span class="dt">module_x =</span> <span class="dv">1</span>){</span>
<span id="cb120-3"><a href="example.html#cb120-3"></a>  idx.omics_data &lt;-<span class="st"> </span><span class="kw">which</span>(modules<span class="op">$</span>colors <span class="op">==</span><span class="st"> </span>module_x)</span>
<span id="cb120-4"><a href="example.html#cb120-4"></a>  idx.me &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(modules<span class="op">$</span>MEs) <span class="op">==</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>,module_x))</span>
<span id="cb120-5"><a href="example.html#cb120-5"></a>  kME_x &lt;-<span class="st"> </span><span class="kw">bicor</span>(metabolites_t[,idx.omics_data], modules<span class="op">$</span>MEs[,idx.me], <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span>
<span id="cb120-6"><a href="example.html#cb120-6"></a>  kME_x</span>
<span id="cb120-7"><a href="example.html#cb120-7"></a>}</span>
<span id="cb120-8"><a href="example.html#cb120-8"></a></span>
<span id="cb120-9"><a href="example.html#cb120-9"></a>ggplot.list &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb120-10"><a href="example.html#cb120-10"></a></span>
<span id="cb120-11"><a href="example.html#cb120-11"></a><span class="cf">for</span>(m <span class="cf">in</span> <span class="kw">colnames</span>(modules.metabolites<span class="op">$</span>MEs)){</span>
<span id="cb120-12"><a href="example.html#cb120-12"></a>  h &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">sub</span>(<span class="st">&quot;ME&quot;</span>,<span class="st">&quot;&quot;</span>, m))</span>
<span id="cb120-13"><a href="example.html#cb120-13"></a>  <span class="kw">data.frame</span>(<span class="dt">x =</span> <span class="kw">suppressWarnings</span>(<span class="kw">corr_within_module</span>(<span class="dt">metabolites_t =</span> metabolites_t, <span class="dt">modules =</span> modules.metabolites, <span class="dt">module_x =</span> h))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb120-14"><a href="example.html#cb120-14"></a><span class="st">    </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb120-15"><a href="example.html#cb120-15"></a><span class="st">    </span><span class="kw">geom_density</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x), <span class="dt">fill =</span> <span class="kw">labels2colors</span>(h), <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb120-16"><a href="example.html#cb120-16"></a><span class="st">    </span><span class="kw">xlim</span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb120-17"><a href="example.html#cb120-17"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;metabolic correlation&quot;</span>)<span class="op">+</span></span>
<span id="cb120-18"><a href="example.html#cb120-18"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;mbs&quot;</span>,m)) -&gt;<span class="st"> </span>da_plot</span>
<span id="cb120-19"><a href="example.html#cb120-19"></a>  </span>
<span id="cb120-20"><a href="example.html#cb120-20"></a>  ggplot.list[[m]] &lt;-<span class="st"> </span>da_plot</span>
<span id="cb120-21"><a href="example.html#cb120-21"></a>}</span>
<span id="cb120-22"><a href="example.html#cb120-22"></a></span>
<span id="cb120-23"><a href="example.html#cb120-23"></a>ggplot.list &lt;-<span class="st"> </span>ggplot.list[ggplot.list <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">names</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;ME&quot;</span>, <span class="st">&quot;&quot;</span>, .) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.numeric</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">order</span>()]</span></code></pre></div>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="example.html#cb121-1"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> ggplot.list, <span class="dt">ncol =</span> <span class="dv">5</span>) -&gt;<span class="st"> </span>density_all_plot <span class="co"># ncol 可以根据module的大小调整</span></span>
<span id="cb121-2"><a href="example.html#cb121-2"></a>density_all_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-74-1.png" width="768" /></p>
<p>Combine to one plot</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="example.html#cb122-1"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(si_mc_plot , density_eigen, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>)) -&gt;<span class="st"> </span>part_<span class="dv">1</span></span>
<span id="cb122-2"><a href="example.html#cb122-2"></a></span>
<span id="cb122-3"><a href="example.html#cb122-3"></a></span>
<span id="cb122-4"><a href="example.html#cb122-4"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(part_<span class="dv">1</span>, module_size_barplot, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;C&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="fl">0.5</span>)) -&gt;<span class="st"> </span>part_<span class="dv">2</span></span>
<span id="cb122-5"><a href="example.html#cb122-5"></a></span>
<span id="cb122-6"><a href="example.html#cb122-6"></a></span>
<span id="cb122-7"><a href="example.html#cb122-7"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(part_<span class="dv">2</span>, density_all_plot, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;F&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>)</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-75-1.png" width="1152" /></p>
<p>Hub metabolic</p>
<p>For each module it is possible to pick a hub metabolic with the function <code>chooseTopHubInEachModule</code>.</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="example.html#cb123-1"></a>hubs.metabolites <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb123-2"><a href="example.html#cb123-2"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb123-3"><a href="example.html#cb123-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="st">&quot;mbs_name&quot;</span> =<span class="st"> &quot;.&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb123-4"><a href="example.html#cb123-4"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;Module&quot;</span>) -&gt;<span class="st"> </span>hubMBS</span>
<span id="cb123-5"><a href="example.html#cb123-5"></a></span>
<span id="cb123-6"><a href="example.html#cb123-6"></a></span>
<span id="cb123-7"><a href="example.html#cb123-7"></a></span>
<span id="cb123-8"><a href="example.html#cb123-8"></a>dplyr<span class="op">::</span><span class="kw">left_join</span>(hubMBS, </span>
<span id="cb123-9"><a href="example.html#cb123-9"></a>                 (metabolites_level <span class="op">%&gt;%</span></span>
<span id="cb123-10"><a href="example.html#cb123-10"></a><span class="st">                    </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;mbs_name&quot;</span>)), </span>
<span id="cb123-11"><a href="example.html#cb123-11"></a>                 <span class="dt">by =</span> <span class="st">&quot;mbs_name&quot;</span>) -&gt;<span class="st"> </span>hubMBS</span>
<span id="cb123-12"><a href="example.html#cb123-12"></a></span>
<span id="cb123-13"><a href="example.html#cb123-13"></a>hubMBS <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">datatable</span>()</span></code></pre></div>
<div id="htmlwidget-f3a1f0b1d7d0c91020f3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f3a1f0b1d7d0c91020f3">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27"],["0","1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26"],["MEDP0170","MEDP0184","MEDN0059","MW0107773","MEDP1465","MW0055479","MW0122632","MEDP0434","MEDN0209","MW0055182","MEDP1201","MEDN0350","MW0012985","MEDN0210","MEDP0101","MEDP0061","MW0157921","MEDN0224","MEDN0498","MEDN0478","MEDP1441","MEDN0570","MEDP1273","MW0167779","MEDN1264","MEDN1287","MW0012612"],["Nucleotide And  Its metabolomics","Hormones and hormone related compunds","Amino acid  and Its metabolomics","-","Alcohol and amines","-","-","GP","Tryptamines,Cholines,Pigments","-","Others","FA","-","Carboxylic acids and derivatives","Benzene and substituted derivatives","Amino acid  and Its metabolomics","-","Carboxylic acids and derivatives","Carboxylic acids and derivatives","Organic acid And  Its derivatives","FA","Carboxylic acids and derivatives","Others","-","GP","GP","-"],["Nucleotide and  Its metabolomics","Hormones and hormone related compunds","Amino acid derivatives","-","Alcohols","-","-","LPC","Tryptamines","-","Others","Oxidized lipids","-","Sugar alcohols","Phenolic acids","Amino acid derivatives","-","Sugars","Phosphate sugars","Organic acid And  Its derivatives","CAR","Sugars","Others","-","LPE","LPE","-"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Module<\/th>\n      <th>mbs_name<\/th>\n      <th>Class.I<\/th>\n      <th>Class.II<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="metabolites-mgs-and-metadata-association" class="section level2 hasAnchor">
<h2><span class="header-section-number">3.8</span> metabolites, mgs and metadata association<a href="example.html#metabolites-mgs-and-metadata-association" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="example.html#cb124-1"></a>X_eigengenes &lt;-<span class="st"> </span>modules.metabolites<span class="op">$</span>MEs <span class="co"># X_eigengenes is metabolic</span></span>
<span id="cb124-2"><a href="example.html#cb124-2"></a>Y_eigengenes &lt;-<span class="st"> </span>modules.mgs<span class="op">$</span>MEs <span class="co"># Y_eigengenes is mgs</span></span></code></pre></div>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="example.html#cb125-1"></a><span class="co"># Create a dendrogram of the metabolites eigengenes to organise the final plots.</span></span>
<span id="cb125-2"><a href="example.html#cb125-2"></a>X_ME_dendro &lt;-<span class="st"> </span><span class="kw">hclust</span>(<span class="kw">as.dist</span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>WGCNA<span class="op">::</span><span class="kw">bicor</span>(X_eigengenes, <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)), <span class="dt">method =</span> <span class="st">&quot;ward.D2&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="example.html#cb126-1"></a>heatmap_colors &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;#18b29f&quot;</span>,<span class="st">&quot;#FFFFFF&quot;</span>,<span class="st">&quot;#ac6721&quot;</span>), <span class="dt">interpolate =</span> <span class="st">&quot;spline&quot;</span>, <span class="dt">space =</span> <span class="st">&quot;rgb&quot;</span>)(<span class="dv">51</span>)</span>
<span id="cb126-2"><a href="example.html#cb126-2"></a></span>
<span id="cb126-3"><a href="example.html#cb126-3"></a>annotation_col &lt;-<span class="st"> </span>metadata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb126-4"><a href="example.html#cb126-4"></a><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span></span>
<span id="cb126-5"><a href="example.html#cb126-5"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb126-6"><a href="example.html#cb126-6"></a><span class="st">  </span><span class="kw">select</span>(V1_outcome, GVHD_type, Donor)</span>
<span id="cb126-7"><a href="example.html#cb126-7"></a>  </span>
<span id="cb126-8"><a href="example.html#cb126-8"></a></span>
<span id="cb126-9"><a href="example.html#cb126-9"></a>annotation_colors &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb126-10"><a href="example.html#cb126-10"></a>  <span class="dt">V1_outcome =</span> <span class="kw">c</span>(<span class="st">`</span><span class="dt">CR</span><span class="st">`</span> =<span class="st"> &quot;#F08A46&quot;</span>, <span class="st">`</span><span class="dt">PR</span><span class="st">`</span> =<span class="st"> &quot;#8EB470&quot;</span>, <span class="st">`</span><span class="dt">NR</span><span class="st">`</span> =<span class="st"> &quot;#B7CFA4&quot;</span>),</span>
<span id="cb126-11"><a href="example.html#cb126-11"></a>  <span class="dt">GVHD_type =</span> <span class="kw">c</span>(<span class="st">`</span><span class="dt">aGVHD</span><span class="st">`</span> =<span class="st"> &quot;red&quot;</span>, <span class="st">`</span><span class="dt">cGVHD</span><span class="st">`</span> =<span class="st"> &quot;green&quot;</span>),</span>
<span id="cb126-12"><a href="example.html#cb126-12"></a>  <span class="dt">Donor =</span> <span class="kw">c</span>(<span class="st">`</span><span class="dt">DO1</span><span class="st">`</span> =<span class="st"> &quot;paleturquoise&quot;</span>, <span class="st">`</span><span class="dt">DO2</span><span class="st">`</span> =<span class="st"> &quot;palevioletred&quot;</span>)</span>
<span id="cb126-13"><a href="example.html#cb126-13"></a>)</span>
<span id="cb126-14"><a href="example.html#cb126-14"></a></span>
<span id="cb126-15"><a href="example.html#cb126-15"></a>X_eigengenes_to_plot &lt;-<span class="st"> </span></span>
<span id="cb126-16"><a href="example.html#cb126-16"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(annotation_col <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb126-17"><a href="example.html#cb126-17"></a><span class="st">               </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;sampleName&quot;</span>), </span>
<span id="cb126-18"><a href="example.html#cb126-18"></a>             X_eigengenes <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb126-19"><a href="example.html#cb126-19"></a><span class="st">               </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;sampleName&quot;</span>), </span>
<span id="cb126-20"><a href="example.html#cb126-20"></a>             <span class="dt">by =</span> <span class="st">&quot;sampleName&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb126-21"><a href="example.html#cb126-21"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(V1_outcome, GVHD_type, Donor) <span class="op">%&gt;%</span><span class="st">              </span><span class="co"># The order at which the columns should appear, given that there is no clustering.</span></span>
<span id="cb126-22"><a href="example.html#cb126-22"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(sampleName, <span class="kw">starts_with</span>(<span class="st">&quot;ME&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb126-23"><a href="example.html#cb126-23"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">column_to_rownames</span>(<span class="dt">var =</span> <span class="st">&quot;sampleName&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb126-24"><a href="example.html#cb126-24"></a><span class="st">  </span><span class="kw">t</span>()</span>
<span id="cb126-25"><a href="example.html#cb126-25"></a></span>
<span id="cb126-26"><a href="example.html#cb126-26"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(X_eigengenes_to_plot, </span>
<span id="cb126-27"><a href="example.html#cb126-27"></a>                   <span class="dt">cluster_cols =</span> <span class="ot">TRUE</span>,</span>
<span id="cb126-28"><a href="example.html#cb126-28"></a>                   <span class="dt">cluster_rows =</span> X_ME_dendro,</span>
<span id="cb126-29"><a href="example.html#cb126-29"></a>                   <span class="dt">treeheight_row =</span> <span class="dv">20</span>,</span>
<span id="cb126-30"><a href="example.html#cb126-30"></a>                   <span class="dt">cutree_rows =</span> <span class="dv">4</span>,</span>
<span id="cb126-31"><a href="example.html#cb126-31"></a>                   <span class="dt">cutree_cols =</span> <span class="dv">4</span>,</span>
<span id="cb126-32"><a href="example.html#cb126-32"></a>                   <span class="dt">color =</span> heatmap_colors,</span>
<span id="cb126-33"><a href="example.html#cb126-33"></a>                   <span class="dt">fontsize =</span> <span class="dv">10</span>,</span>
<span id="cb126-34"><a href="example.html#cb126-34"></a>                   <span class="dt">fontsize_col =</span> <span class="dv">6</span>,</span>
<span id="cb126-35"><a href="example.html#cb126-35"></a>                   <span class="dt">annotation_colors =</span> annotation_colors,</span>
<span id="cb126-36"><a href="example.html#cb126-36"></a>                   <span class="dt">annotation_col =</span> annotation_col, </span>
<span id="cb126-37"><a href="example.html#cb126-37"></a>                   <span class="dt">silent =</span> F,</span>
<span id="cb126-38"><a href="example.html#cb126-38"></a>                   <span class="dt">labels_row =</span> <span class="kw">paste0</span>(<span class="st">&quot;mb&quot;</span>, <span class="kw">rownames</span>(X_eigengenes_to_plot)),</span>
<span id="cb126-39"><a href="example.html#cb126-39"></a>                   <span class="dt">main =</span> <span class="kw">paste</span>(<span class="st">&quot;Metabolites Module &#39;expression&#39;</span><span class="ch">\n</span><span class="st">&quot;</span>)) -&gt;<span class="st"> </span>X_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-79-1.png" width="672" /></p>
<div id="correlate-modules-from-metabolic-and-metagenomics." class="section level4 hasAnchor">
<h4><span class="header-section-number">3.8.0.1</span> Correlate modules from metabolic and metagenomics.<a href="example.html#correlate-modules-from-metabolic-and-metagenomics." class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="example.html#cb127-1"></a>p.value_matr &lt;-<span class="st"> </span>corr.value_matr &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dt">ncol =</span> <span class="kw">ncol</span>(Y_eigengenes), </span>
<span id="cb127-2"><a href="example.html#cb127-2"></a>                                          <span class="dt">nrow =</span> <span class="kw">ncol</span>(X_eigengenes), </span>
<span id="cb127-3"><a href="example.html#cb127-3"></a>                                          <span class="dt">dimnames =</span> <span class="kw">list</span>(<span class="kw">colnames</span>(X_eigengenes), </span>
<span id="cb127-4"><a href="example.html#cb127-4"></a>                                                          <span class="kw">colnames</span>(Y_eigengenes)))</span>
<span id="cb127-5"><a href="example.html#cb127-5"></a></span>
<span id="cb127-6"><a href="example.html#cb127-6"></a></span>
<span id="cb127-7"><a href="example.html#cb127-7"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(X_eigengenes)){</span>
<span id="cb127-8"><a href="example.html#cb127-8"></a>  <span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(Y_eigengenes)){</span>
<span id="cb127-9"><a href="example.html#cb127-9"></a>    cor.res &lt;-<span class="st"> </span><span class="kw">cor.test</span>(X_eigengenes[,i], Y_eigengenes[,j])</span>
<span id="cb127-10"><a href="example.html#cb127-10"></a>    p.value_matr[i, j] &lt;-<span class="st"> </span>cor.res<span class="op">$</span>p.value</span>
<span id="cb127-11"><a href="example.html#cb127-11"></a>    corr.value_matr[i, j] &lt;-<span class="st"> </span>cor.res<span class="op">$</span>estimate</span>
<span id="cb127-12"><a href="example.html#cb127-12"></a>  }</span>
<span id="cb127-13"><a href="example.html#cb127-13"></a>}</span>
<span id="cb127-14"><a href="example.html#cb127-14"></a></span>
<span id="cb127-15"><a href="example.html#cb127-15"></a><span class="co"># Correct for number of tests</span></span>
<span id="cb127-16"><a href="example.html#cb127-16"></a>p.value_matr.adjust &lt;-<span class="st"> </span><span class="kw">p.adjust</span>(p.value_matr, <span class="dt">method =</span> <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb127-17"><a href="example.html#cb127-17"></a><span class="kw">dim</span>(p.value_matr.adjust) &lt;-<span class="st"> </span><span class="kw">dim</span>(p.value_matr)</span>
<span id="cb127-18"><a href="example.html#cb127-18"></a><span class="kw">dimnames</span>(p.value_matr.adjust) &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="kw">colnames</span>(X_eigengenes), <span class="kw">colnames</span>(Y_eigengenes))</span>
<span id="cb127-19"><a href="example.html#cb127-19"></a></span>
<span id="cb127-20"><a href="example.html#cb127-20"></a></span>
<span id="cb127-21"><a href="example.html#cb127-21"></a><span class="co"># Add significance level.  </span></span>
<span id="cb127-22"><a href="example.html#cb127-22"></a><span class="co"># One star means a p-value of less than 0.05; Two stars is less than 0.01, and three, is less than 0.001.</span></span>
<span id="cb127-23"><a href="example.html#cb127-23"></a></span>
<span id="cb127-24"><a href="example.html#cb127-24"></a>signif_matrix &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">&quot;&quot;</span>, <span class="kw">length</span>(p.value_matr))</span>
<span id="cb127-25"><a href="example.html#cb127-25"></a>three_star &lt;-<span class="st"> </span><span class="kw">which</span>( p.value_matr <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.001</span>)</span>
<span id="cb127-26"><a href="example.html#cb127-26"></a>signif_matrix[three_star] &lt;-<span class="st"> &quot;***&quot;</span></span>
<span id="cb127-27"><a href="example.html#cb127-27"></a>two_star &lt;-<span class="st"> </span><span class="kw">which</span>((p.value_matr <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.01</span>) <span class="op">&amp;</span><span class="st"> </span>(p.value_matr <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.001</span>))</span>
<span id="cb127-28"><a href="example.html#cb127-28"></a>signif_matrix[two_star] &lt;-<span class="st"> &quot;**&quot;</span></span>
<span id="cb127-29"><a href="example.html#cb127-29"></a>one_star &lt;-<span class="st"> </span><span class="kw">which</span>((p.value_matr <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">&amp;</span><span class="st"> </span>(p.value_matr <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.01</span>))</span>
<span id="cb127-30"><a href="example.html#cb127-30"></a>signif_matrix[one_star] &lt;-<span class="st"> &quot;*&quot;</span></span>
<span id="cb127-31"><a href="example.html#cb127-31"></a><span class="kw">dim</span>(signif_matrix) =<span class="st"> </span><span class="kw">dim</span>(p.value_matr) <span class="co"># Give textMatrix the correct dimensions </span></span>
<span id="cb127-32"><a href="example.html#cb127-32"></a></span>
<span id="cb127-33"><a href="example.html#cb127-33"></a></span>
<span id="cb127-34"><a href="example.html#cb127-34"></a><span class="co"># Collect all results into a list.</span></span>
<span id="cb127-35"><a href="example.html#cb127-35"></a>Y_corr_X &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">p_value =</span> p.value_matr, </span>
<span id="cb127-36"><a href="example.html#cb127-36"></a>                 <span class="dt">p_value_adj =</span> p.value_matr.adjust,</span>
<span id="cb127-37"><a href="example.html#cb127-37"></a>                 <span class="dt">signif_matrix =</span> signif_matrix,</span>
<span id="cb127-38"><a href="example.html#cb127-38"></a>                 <span class="dt">correlation =</span> corr.value_matr)</span>
<span id="cb127-39"><a href="example.html#cb127-39"></a><span class="kw">rm</span>(p.value_matr, p.value_matr.adjust, signif_matrix, corr.value_matr)</span></code></pre></div>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="example.html#cb128-1"></a>heatmap_colors &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">rev</span>(RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dt">n =</span> <span class="dv">6</span>, <span class="dt">name =</span><span class="st">&quot;RdBu&quot;</span>)))(<span class="dv">51</span>)</span>
<span id="cb128-2"><a href="example.html#cb128-2"></a></span>
<span id="cb128-3"><a href="example.html#cb128-3"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(Y_corr_X<span class="op">$</span>correlation, </span>
<span id="cb128-4"><a href="example.html#cb128-4"></a>                   <span class="dt">color =</span> heatmap_colors, </span>
<span id="cb128-5"><a href="example.html#cb128-5"></a>                   <span class="dt">treeheight_col =</span> <span class="dv">0</span>, </span>
<span id="cb128-6"><a href="example.html#cb128-6"></a>                   <span class="dt">treeheight_row =</span> <span class="dv">0</span>,  <span class="co"># will be shown on the transcriptomics ME heatmap</span></span>
<span id="cb128-7"><a href="example.html#cb128-7"></a>                   <span class="dt">cluster_rows =</span> X_ME_dendro,</span>
<span id="cb128-8"><a href="example.html#cb128-8"></a>                   <span class="dt">cutree_rows =</span> <span class="dv">4</span>,</span>
<span id="cb128-9"><a href="example.html#cb128-9"></a>                   <span class="dt">display_numbers =</span> Y_corr_X<span class="op">$</span>signif_matrix, </span>
<span id="cb128-10"><a href="example.html#cb128-10"></a>                   <span class="dt">fontsize_number =</span> <span class="dv">10</span>,</span>
<span id="cb128-11"><a href="example.html#cb128-11"></a>                   <span class="dt">breaks =</span> <span class="kw">seq</span>(<span class="dt">from =</span> <span class="dv">-1</span>, <span class="dt">to =</span> <span class="dv">1</span>, <span class="dt">length.out =</span> <span class="dv">51</span>), </span>
<span id="cb128-12"><a href="example.html#cb128-12"></a>                   <span class="dt">silent =</span> F,</span>
<span id="cb128-13"><a href="example.html#cb128-13"></a>                   <span class="dt">show_rownames =</span> F,</span>
<span id="cb128-14"><a href="example.html#cb128-14"></a>                   <span class="dt">labels_row =</span> <span class="kw">paste0</span>(<span class="st">&quot;mg&quot;</span>, <span class="kw">rownames</span>(Y_corr_X<span class="op">$</span>correlation)),</span>
<span id="cb128-15"><a href="example.html#cb128-15"></a>                   <span class="dt">labels_col =</span> <span class="kw">paste0</span>(<span class="st">&quot;mg&quot;</span>, <span class="kw">colnames</span>(Y_corr_X<span class="op">$</span>correlation)),</span>
<span id="cb128-16"><a href="example.html#cb128-16"></a>                   <span class="dt">main =</span> <span class="st">&quot;EigenOTUs&quot;</span>) -&gt;<span class="st"> </span>Y_corr_X_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-81-1.png" width="672" /></p>
</div>
<div id="combine-heatmaps" class="section level4 hasAnchor">
<h4><span class="header-section-number">3.8.0.2</span> Combine heatmaps<a href="example.html#combine-heatmaps" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="example.html#cb129-1"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(Y_corr_X_plot<span class="op">$</span>gtable,</span>
<span id="cb129-2"><a href="example.html#cb129-2"></a>                   X_plot<span class="op">$</span>gtable,</span>
<span id="cb129-3"><a href="example.html#cb129-3"></a>                   <span class="dt">ncol =</span> <span class="dv">2</span>, </span>
<span id="cb129-4"><a href="example.html#cb129-4"></a>                   <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="kw">dim</span>(Y_eigengenes)[<span class="dv">2</span>]<span class="op">/</span><span class="dv">3</span>, </span>
<span id="cb129-5"><a href="example.html#cb129-5"></a>                   <span class="kw">dim</span>(X_eigengenes)[<span class="dv">1</span>]<span class="op">/</span><span class="dv">9</span>),</span>
<span id="cb129-6"><a href="example.html#cb129-6"></a>                   <span class="dt">align =</span> <span class="st">&quot;h&quot;</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> ggplot2<span class="op">::</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">0</span>,<span class="fl">2.5</span>,<span class="dv">1</span>), <span class="st">&quot;cm&quot;</span>))</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-82-1.png" width="672" /></p>
</div>
</div>
<div id="visualization-network" class="section level2 hasAnchor">
<h2><span class="header-section-number">3.9</span> Visualization network<a href="example.html#visualization-network" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The strong correlation between mgME3 and mbME1 is the modules of interest, and we wondered if they contain those bacteria and metabolites, which ara published in the existing literature, or if there is a potential mining value.</p>
<p>So we visualize network in mgME3 and mbME1 (In the practical application，find the module you are interested in for visualization)</p>
<p>Visualize mgME3</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="example.html#cb130-1"></a><span class="co">#load(file = &quot;/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/GVHD/bookdown_GVHD/result/modules.mgs-block.1.RData&quot;)</span></span>
<span id="cb130-2"><a href="example.html#cb130-2"></a><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;./result/modules.mgs-block.1.RData&quot;</span>)</span>
<span id="cb130-3"><a href="example.html#cb130-3"></a></span>
<span id="cb130-4"><a href="example.html#cb130-4"></a><span class="co"># The TOM is saved as a dist object and needs to be converted to a matrix</span></span>
<span id="cb130-5"><a href="example.html#cb130-5"></a>TOM &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(TOM)</span>
<span id="cb130-6"><a href="example.html#cb130-6"></a><span class="co"># Add OTU names to the TOM matrix. It is symmetrical so rownames = colnames</span></span>
<span id="cb130-7"><a href="example.html#cb130-7"></a><span class="kw">rownames</span>(TOM) &lt;-<span class="st"> </span><span class="kw">colnames</span>(TOM) &lt;-<span class="st"> </span><span class="kw">names</span>(modules.mgs<span class="op">$</span>colors)</span></code></pre></div>
<p>Which taxonomic level should the graph be colored with in addition to modules?</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="example.html#cb131-1"></a>selected_taxa &lt;-<span class="st"> &quot;Genus&quot;</span></span></code></pre></div>
<p>Convert module labels and taxonomy to hex colors</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="example.html#cb132-1"></a>taxonomy_info &lt;-<span class="st"> </span></span>
<span id="cb132-2"><a href="example.html#cb132-2"></a><span class="st">  </span>taxa_table <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-3"><a href="example.html#cb132-3"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;OTU_name&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb132-4"><a href="example.html#cb132-4"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="st">&quot;OTU_name&quot;</span>, selected_taxa) <span class="op">%&gt;%</span></span>
<span id="cb132-5"><a href="example.html#cb132-5"></a><span class="st">  </span><span class="kw">mutate_all</span>(<span class="dt">.funs =</span> <span class="kw">list</span>(as.character)) </span>
<span id="cb132-6"><a href="example.html#cb132-6"></a></span>
<span id="cb132-7"><a href="example.html#cb132-7"></a>module_info &lt;-<span class="st"> </span></span>
<span id="cb132-8"><a href="example.html#cb132-8"></a><span class="st">  </span>modules.mgs<span class="op">$</span>colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-9"><a href="example.html#cb132-9"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb132-10"><a href="example.html#cb132-10"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;OTU_name&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-11"><a href="example.html#cb132-11"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">Module =</span> <span class="st">&quot;.&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb132-12"><a href="example.html#cb132-12"></a><span class="st">  </span><span class="kw">filter</span>(Module <span class="op">==</span><span class="st"> </span><span class="dv">3</span>)</span>
<span id="cb132-13"><a href="example.html#cb132-13"></a></span>
<span id="cb132-14"><a href="example.html#cb132-14"></a><span class="co"># selected module, here is ME3</span></span>
<span id="cb132-15"><a href="example.html#cb132-15"></a>TOM &lt;-<span class="st"> </span>TOM[module_info<span class="op">$</span>OTU_name, module_info<span class="op">$</span>OTU_name]</span>
<span id="cb132-16"><a href="example.html#cb132-16"></a></span>
<span id="cb132-17"><a href="example.html#cb132-17"></a>graph_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(module_info, taxonomy_info, <span class="dt">by =</span> <span class="st">&quot;OTU_name&quot;</span>)</span>
<span id="cb132-18"><a href="example.html#cb132-18"></a></span>
<span id="cb132-19"><a href="example.html#cb132-19"></a><span class="co"># Converts R-colors to hex colors</span></span>
<span id="cb132-20"><a href="example.html#cb132-20"></a>color2hex &lt;-<span class="st"> </span><span class="cf">function</span>(x){</span>
<span id="cb132-21"><a href="example.html#cb132-21"></a>  x &lt;-<span class="st"> </span><span class="kw">col2rgb</span>(x)</span>
<span id="cb132-22"><a href="example.html#cb132-22"></a>  <span class="kw">rgb</span>(x[<span class="dv">1</span>,], x[<span class="dv">2</span>,], x[<span class="dv">3</span>,], <span class="dt">maxColorValue =</span> <span class="dv">255</span>)</span>
<span id="cb132-23"><a href="example.html#cb132-23"></a>}</span>
<span id="cb132-24"><a href="example.html#cb132-24"></a></span>
<span id="cb132-25"><a href="example.html#cb132-25"></a><span class="co"># Add specific colors to the taxa</span></span>
<span id="cb132-26"><a href="example.html#cb132-26"></a>taxa_colors &lt;-<span class="st"> </span></span>
<span id="cb132-27"><a href="example.html#cb132-27"></a><span class="st">  </span>graph_info <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-28"><a href="example.html#cb132-28"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(selected_taxa) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-29"><a href="example.html#cb132-29"></a><span class="st">  </span><span class="kw">unique</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-30"><a href="example.html#cb132-30"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">tax_color =</span> <span class="kw">colorRampPalette</span>(RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&quot;Accent&quot;</span>))(<span class="kw">nrow</span>(.)))</span>
<span id="cb132-31"><a href="example.html#cb132-31"></a></span>
<span id="cb132-32"><a href="example.html#cb132-32"></a>graph_info_colors &lt;-<span class="st"> </span></span>
<span id="cb132-33"><a href="example.html#cb132-33"></a><span class="st">  </span><span class="kw">left_join</span>(graph_info, taxa_colors) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-34"><a href="example.html#cb132-34"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> WGCNA<span class="op">::</span><span class="kw">labels2colors</span>(Module)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-35"><a href="example.html#cb132-35"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> <span class="kw">color2hex</span>(module_color)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-36"><a href="example.html#cb132-36"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> <span class="kw">paste0</span>(module_color,<span class="dv">70</span>))</span></code></pre></div>
<pre><code>## Joining, by = &quot;Genus&quot;</code></pre>
<p>If all lines are too thick, reduce the strength (between 0 and 1).
The lower the number the weaker the lines.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="example.html#cb134-1"></a>strength_adjust =<span class="st"> </span><span class="dv">1</span></span></code></pre></div>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="example.html#cb135-1"></a>g &lt;-<span class="st"> </span><span class="kw">graph.adjacency</span>(TOM, <span class="dt">mode=</span><span class="st">&quot;undirected&quot;</span>, <span class="dt">weighted=</span> <span class="ot">TRUE</span>)</span>
<span id="cb135-2"><a href="example.html#cb135-2"></a></span>
<span id="cb135-3"><a href="example.html#cb135-3"></a><span class="co">#~https://stackoverflow.com/questions/28366329/how-to-scale-edge-colors-igraph</span></span>
<span id="cb135-4"><a href="example.html#cb135-4"></a>igraph<span class="op">::</span><span class="kw">delete.edges</span>(g, <span class="kw">which</span>(<span class="kw">E</span>(g)<span class="op">$</span>weight <span class="op">&lt;</span><span class="dv">1</span>))</span></code></pre></div>
<pre><code>## IGRAPH 7547a81 UNW- 12 0 -- 
## + attr: name (v/c), weight (e/n)
## + edges from 7547a81 (vertex names):</code></pre>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="example.html#cb137-1"></a><span class="kw">E</span>(g)<span class="op">$</span>width &lt;-<span class="st"> </span><span class="kw">E</span>(g)<span class="op">$</span>weight<span class="op">*</span>strength_adjust <span class="op">+</span><span class="st"> </span><span class="kw">min</span>(<span class="kw">E</span>(g)<span class="op">$</span>weight)</span>
<span id="cb137-2"><a href="example.html#cb137-2"></a><span class="kw">E</span>(g)<span class="op">$</span>color &lt;-<span class="st"> &quot;red&quot;</span></span>
<span id="cb137-3"><a href="example.html#cb137-3"></a></span>
<span id="cb137-4"><a href="example.html#cb137-4"></a><span class="kw">set.seed</span>(<span class="dv">231</span>) <span class="co"># Ensures the same layout given the same data.</span></span>
<span id="cb137-5"><a href="example.html#cb137-5"></a>l &lt;-<span class="st"> </span><span class="kw">layout_with_fr</span>(g, <span class="dt">weights =</span> <span class="kw">E</span>(g)<span class="op">$</span>weight)</span>
<span id="cb137-6"><a href="example.html#cb137-6"></a></span>
<span id="cb137-7"><a href="example.html#cb137-7"></a></span>
<span id="cb137-8"><a href="example.html#cb137-8"></a><span class="co"># Order graph_info_colors by the graph </span></span>
<span id="cb137-9"><a href="example.html#cb137-9"></a>graph_info_colors &lt;-<span class="st"> </span>graph_info_colors[<span class="kw">which</span>(graph_info_colors<span class="op">$</span>OTU_name <span class="op">%in%</span><span class="st"> </span><span class="kw">V</span>(g)<span class="op">$</span>name),]</span>
<span id="cb137-10"><a href="example.html#cb137-10"></a></span>
<span id="cb137-11"><a href="example.html#cb137-11"></a><span class="co"># Ensure that the names are in the same order</span></span>
<span id="cb137-12"><a href="example.html#cb137-12"></a><span class="cf">if</span>(<span class="kw">all</span>(<span class="kw">V</span>(g)<span class="op">$</span>name <span class="op">==</span><span class="st"> </span>graph_info_colors<span class="op">$</span>OTU_name)){<span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">OTU names match&quot;</span>)}</span></code></pre></div>
<pre><code>## 
## OTU names match</code></pre>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="example.html#cb139-1"></a><span class="co"># Add square shapes to hub OTUs</span></span>
<span id="cb139-2"><a href="example.html#cb139-2"></a><span class="kw">V</span>(g)<span class="op">$</span>shape &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">V</span>(g)<span class="op">$</span>name <span class="op">%in%</span><span class="st"> </span>hubs.mgs[<span class="op">-</span><span class="dv">1</span>], <span class="st">&quot;square&quot;</span>, <span class="st">&quot;circle&quot;</span>) <span class="co">#-1 means dont use module 0</span></span>
<span id="cb139-3"><a href="example.html#cb139-3"></a></span>
<span id="cb139-4"><a href="example.html#cb139-4"></a><span class="co"># OTUs in modules have larger nodes</span></span>
<span id="cb139-5"><a href="example.html#cb139-5"></a><span class="kw">V</span>(g)<span class="op">$</span>size &lt;-<span class="st"> </span><span class="kw">ifelse</span>(graph_info_colors<span class="op">$</span>Module <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">5</span>)</span>
<span id="cb139-6"><a href="example.html#cb139-6"></a></span>
<span id="cb139-7"><a href="example.html#cb139-7"></a><span class="co"># And larger text</span></span>
<span id="cb139-8"><a href="example.html#cb139-8"></a><span class="kw">V</span>(g)<span class="op">$</span>label.cex &lt;-<span class="st"> </span><span class="kw">ifelse</span>(graph_info_colors<span class="op">$</span>Module <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>, <span class="fl">0.8</span>, <span class="fl">0.4</span>)</span>
<span id="cb139-9"><a href="example.html#cb139-9"></a></span>
<span id="cb139-10"><a href="example.html#cb139-10"></a><span class="co"># Remove everything but the number to increase readability</span></span>
<span id="cb139-11"><a href="example.html#cb139-11"></a><span class="kw">V</span>(g)<span class="op">$</span>name =<span class="st">  </span><span class="kw">sub</span>(<span class="st">&quot;OTU_&quot;</span>, <span class="st">&quot;&quot;</span>, graph_info_colors<span class="op">$</span>OTU_name)</span></code></pre></div>
<p>Find distinct entires for the plot legends</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="example.html#cb140-1"></a>module_labels &lt;-<span class="st"> </span></span>
<span id="cb140-2"><a href="example.html#cb140-2"></a><span class="st">  </span>graph_info_colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb140-3"><a href="example.html#cb140-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(Module, module_color) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb140-4"><a href="example.html#cb140-4"></a><span class="st">  </span><span class="kw">distinct</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb140-5"><a href="example.html#cb140-5"></a><span class="st">  </span><span class="kw">arrange</span>(Module)</span>
<span id="cb140-6"><a href="example.html#cb140-6"></a></span>
<span id="cb140-7"><a href="example.html#cb140-7"></a>tax_labels &lt;-<span class="st"> </span></span>
<span id="cb140-8"><a href="example.html#cb140-8"></a><span class="st">  </span>graph_info_colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb140-9"><a href="example.html#cb140-9"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(selected_taxa, tax_color) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb140-10"><a href="example.html#cb140-10"></a><span class="st">  </span><span class="kw">distinct</span>()</span></code></pre></div>
<p>Plot the graphs, leftmost is colored by module, rightmost is colored by taxonomic classification</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="example.html#cb141-1"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb141-2"><a href="example.html#cb141-2"></a></span>
<span id="cb141-3"><a href="example.html#cb141-3"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb141-4"><a href="example.html#cb141-4"></a></span>
<span id="cb141-5"><a href="example.html#cb141-5"></a><span class="co"># plot(g, layout = l, vertex.color = graph_info_colors$module_color)</span></span>
<span id="cb141-6"><a href="example.html#cb141-6"></a><span class="co"># # legend(&quot;topleft&quot;, legend = paste0(&quot;mM&quot;, 0:(nrow(module_labels)-1)), fill=module_labels$module_color)</span></span>
<span id="cb141-7"><a href="example.html#cb141-7"></a><span class="co"># legend(&quot;topleft&quot;, legend = paste0(&quot;mgsM&quot;, 3), fill=module_labels$module_color)</span></span>
<span id="cb141-8"><a href="example.html#cb141-8"></a></span>
<span id="cb141-9"><a href="example.html#cb141-9"></a><span class="kw">plot</span>(g, <span class="dt">layout =</span> l, <span class="dt">vertex.color =</span> graph_info_colors<span class="op">$</span>tax_color)</span>
<span id="cb141-10"><a href="example.html#cb141-10"></a><span class="kw">legend</span>(<span class="st">&quot;topleft&quot;</span>, <span class="dt">legend =</span> tax_labels<span class="op">$</span>Genus, <span class="dt">fill=</span>tax_labels<span class="op">$</span>tax_color)</span>
<span id="cb141-11"><a href="example.html#cb141-11"></a></span>
<span id="cb141-12"><a href="example.html#cb141-12"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-89-1.png" width="1440" /></p>
<p>Visualize mbME1</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="example.html#cb142-1"></a><span class="co">#load(file = &quot;/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/GVHD/bookdown_GVHD/result/module.metabolites-block.1.RData&quot;)</span></span>
<span id="cb142-2"><a href="example.html#cb142-2"></a><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;./result/module.metabolites-block.1.RData&quot;</span>)</span>
<span id="cb142-3"><a href="example.html#cb142-3"></a><span class="co"># The TOM is saved as a dist object and needs to be converted to a matrix</span></span>
<span id="cb142-4"><a href="example.html#cb142-4"></a>TOM &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(TOM)</span>
<span id="cb142-5"><a href="example.html#cb142-5"></a><span class="co"># Add OTU names to the TOM matrix. It is symmetrical so rownames = colnames</span></span>
<span id="cb142-6"><a href="example.html#cb142-6"></a><span class="kw">rownames</span>(TOM) &lt;-<span class="st"> </span><span class="kw">colnames</span>(TOM) &lt;-<span class="st"> </span><span class="kw">names</span>(modules.metabolites<span class="op">$</span>colors)</span></code></pre></div>
<p>Which metabolic level should the graph be colored with in addition to modules?</p>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="example.html#cb143-1"></a>selected_taxa &lt;-<span class="st"> &quot;Class.II&quot;</span></span></code></pre></div>
<p>Convert module labels and taxonomy to hex colors</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="example.html#cb144-1"></a>taxonomy_info &lt;-<span class="st"> </span></span>
<span id="cb144-2"><a href="example.html#cb144-2"></a><span class="st">  </span>metabolites_level <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb144-3"><a href="example.html#cb144-3"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;mbs_name&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb144-4"><a href="example.html#cb144-4"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="st">&quot;mbs_name&quot;</span>, selected_taxa) <span class="op">%&gt;%</span></span>
<span id="cb144-5"><a href="example.html#cb144-5"></a><span class="st">  </span><span class="kw">mutate_all</span>(<span class="dt">.funs =</span> <span class="kw">list</span>(as.character)) </span>
<span id="cb144-6"><a href="example.html#cb144-6"></a></span>
<span id="cb144-7"><a href="example.html#cb144-7"></a>module_info &lt;-<span class="st"> </span></span>
<span id="cb144-8"><a href="example.html#cb144-8"></a><span class="st">  </span>modules.metabolites<span class="op">$</span>colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb144-9"><a href="example.html#cb144-9"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb144-10"><a href="example.html#cb144-10"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;mbs_name&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb144-11"><a href="example.html#cb144-11"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">Module =</span> <span class="st">&quot;.&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb144-12"><a href="example.html#cb144-12"></a><span class="st">  </span><span class="kw">filter</span>(Module <span class="op">==</span><span class="st"> </span><span class="dv">1</span>)</span>
<span id="cb144-13"><a href="example.html#cb144-13"></a></span>
<span id="cb144-14"><a href="example.html#cb144-14"></a><span class="co"># selected module, here is ME1</span></span>
<span id="cb144-15"><a href="example.html#cb144-15"></a>TOM &lt;-<span class="st"> </span>TOM[module_info<span class="op">$</span>mbs_name, module_info<span class="op">$</span>mbs_name]</span>
<span id="cb144-16"><a href="example.html#cb144-16"></a></span>
<span id="cb144-17"><a href="example.html#cb144-17"></a>graph_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(module_info, taxonomy_info, <span class="dt">by =</span> <span class="st">&quot;mbs_name&quot;</span>)</span>
<span id="cb144-18"><a href="example.html#cb144-18"></a></span>
<span id="cb144-19"><a href="example.html#cb144-19"></a><span class="co"># Converts R-colors to hex colors</span></span>
<span id="cb144-20"><a href="example.html#cb144-20"></a>color2hex &lt;-<span class="st"> </span><span class="cf">function</span>(x){</span>
<span id="cb144-21"><a href="example.html#cb144-21"></a>  x &lt;-<span class="st"> </span><span class="kw">col2rgb</span>(x)</span>
<span id="cb144-22"><a href="example.html#cb144-22"></a>  <span class="kw">rgb</span>(x[<span class="dv">1</span>,], x[<span class="dv">2</span>,], x[<span class="dv">3</span>,], <span class="dt">maxColorValue =</span> <span class="dv">255</span>)</span>
<span id="cb144-23"><a href="example.html#cb144-23"></a>}</span>
<span id="cb144-24"><a href="example.html#cb144-24"></a></span>
<span id="cb144-25"><a href="example.html#cb144-25"></a><span class="co"># Add specific colors to the taxa</span></span>
<span id="cb144-26"><a href="example.html#cb144-26"></a>taxa_colors &lt;-<span class="st"> </span></span>
<span id="cb144-27"><a href="example.html#cb144-27"></a><span class="st">  </span>graph_info <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb144-28"><a href="example.html#cb144-28"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(selected_taxa) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb144-29"><a href="example.html#cb144-29"></a><span class="st">  </span><span class="kw">unique</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb144-30"><a href="example.html#cb144-30"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">tax_color =</span> <span class="kw">colorRampPalette</span>(RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&quot;Accent&quot;</span>))(<span class="kw">nrow</span>(.)))</span>
<span id="cb144-31"><a href="example.html#cb144-31"></a></span>
<span id="cb144-32"><a href="example.html#cb144-32"></a>graph_info_colors &lt;-<span class="st"> </span></span>
<span id="cb144-33"><a href="example.html#cb144-33"></a><span class="st">  </span><span class="kw">left_join</span>(graph_info, taxa_colors) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb144-34"><a href="example.html#cb144-34"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> WGCNA<span class="op">::</span><span class="kw">labels2colors</span>(Module)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb144-35"><a href="example.html#cb144-35"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> <span class="kw">color2hex</span>(module_color)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb144-36"><a href="example.html#cb144-36"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> <span class="kw">paste0</span>(module_color,<span class="dv">70</span>))</span></code></pre></div>
<pre><code>## Joining, by = &quot;Class.II&quot;</code></pre>
<p>If all lines are too thick, reduce the strength (between 0 and 1).
The lower the number the weaker the lines.</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="example.html#cb146-1"></a>strength_adjust =<span class="st"> </span><span class="fl">0.2</span></span></code></pre></div>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="example.html#cb147-1"></a>g &lt;-<span class="st"> </span><span class="kw">graph.adjacency</span>(TOM, <span class="dt">mode=</span><span class="st">&quot;undirected&quot;</span>, <span class="dt">weighted=</span> <span class="ot">TRUE</span>)</span>
<span id="cb147-2"><a href="example.html#cb147-2"></a></span>
<span id="cb147-3"><a href="example.html#cb147-3"></a><span class="co">#~https://stackoverflow.com/questions/28366329/how-to-scale-edge-colors-igraph</span></span>
<span id="cb147-4"><a href="example.html#cb147-4"></a>igraph<span class="op">::</span><span class="kw">delete.edges</span>(g, <span class="kw">which</span>(<span class="kw">E</span>(g)<span class="op">$</span>weight <span class="op">&lt;</span><span class="dv">1</span>))</span></code></pre></div>
<pre><code>## IGRAPH 6d3dcb5 UNW- 54 0 -- 
## + attr: name (v/c), weight (e/n)
## + edges from 6d3dcb5 (vertex names):</code></pre>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="example.html#cb149-1"></a><span class="kw">E</span>(g)<span class="op">$</span>width &lt;-<span class="st"> </span><span class="kw">E</span>(g)<span class="op">$</span>weight<span class="op">*</span>strength_adjust <span class="op">+</span><span class="st"> </span><span class="kw">min</span>(<span class="kw">E</span>(g)<span class="op">$</span>weight)</span>
<span id="cb149-2"><a href="example.html#cb149-2"></a><span class="kw">E</span>(g)<span class="op">$</span>color &lt;-<span class="st"> &quot;red&quot;</span></span>
<span id="cb149-3"><a href="example.html#cb149-3"></a></span>
<span id="cb149-4"><a href="example.html#cb149-4"></a><span class="kw">set.seed</span>(<span class="dv">231</span>) <span class="co"># Ensures the same layout given the same data.</span></span>
<span id="cb149-5"><a href="example.html#cb149-5"></a>l &lt;-<span class="st"> </span><span class="kw">layout_with_fr</span>(g, <span class="dt">weights =</span> <span class="kw">E</span>(g)<span class="op">$</span>weight)</span>
<span id="cb149-6"><a href="example.html#cb149-6"></a></span>
<span id="cb149-7"><a href="example.html#cb149-7"></a></span>
<span id="cb149-8"><a href="example.html#cb149-8"></a><span class="co"># Order graph_info_colors by the graph </span></span>
<span id="cb149-9"><a href="example.html#cb149-9"></a>graph_info_colors &lt;-<span class="st"> </span>graph_info_colors[<span class="kw">which</span>(graph_info_colors<span class="op">$</span>mbs_name <span class="op">%in%</span><span class="st"> </span><span class="kw">V</span>(g)<span class="op">$</span>name),]</span>
<span id="cb149-10"><a href="example.html#cb149-10"></a></span>
<span id="cb149-11"><a href="example.html#cb149-11"></a><span class="co"># Ensure that the names are in the same order</span></span>
<span id="cb149-12"><a href="example.html#cb149-12"></a><span class="cf">if</span>(<span class="kw">all</span>(<span class="kw">V</span>(g)<span class="op">$</span>name <span class="op">==</span><span class="st"> </span>graph_info_colors<span class="op">$</span>mbs_name)){<span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">mbs names match&quot;</span>)}</span></code></pre></div>
<pre><code>## 
## mbs names match</code></pre>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="example.html#cb151-1"></a><span class="co"># Add square shapes to hub OTUs</span></span>
<span id="cb151-2"><a href="example.html#cb151-2"></a><span class="kw">V</span>(g)<span class="op">$</span>shape &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">V</span>(g)<span class="op">$</span>name <span class="op">%in%</span><span class="st"> </span>hubs.metabolites[<span class="op">-</span><span class="dv">1</span>], <span class="st">&quot;square&quot;</span>, <span class="st">&quot;circle&quot;</span>) <span class="co">#-1 means dont use module 0</span></span>
<span id="cb151-3"><a href="example.html#cb151-3"></a></span>
<span id="cb151-4"><a href="example.html#cb151-4"></a><span class="co"># OTUs in modules have larger nodes</span></span>
<span id="cb151-5"><a href="example.html#cb151-5"></a><span class="kw">V</span>(g)<span class="op">$</span>size &lt;-<span class="st"> </span><span class="kw">ifelse</span>(graph_info_colors<span class="op">$</span>Module <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">5</span>)</span>
<span id="cb151-6"><a href="example.html#cb151-6"></a></span>
<span id="cb151-7"><a href="example.html#cb151-7"></a><span class="co"># And larger text</span></span>
<span id="cb151-8"><a href="example.html#cb151-8"></a><span class="kw">V</span>(g)<span class="op">$</span>label.cex &lt;-<span class="st"> </span><span class="kw">ifelse</span>(graph_info_colors<span class="op">$</span>Module <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>, <span class="fl">0.8</span>, <span class="fl">0.4</span>)</span>
<span id="cb151-9"><a href="example.html#cb151-9"></a></span>
<span id="cb151-10"><a href="example.html#cb151-10"></a><span class="co"># Remove everything but the number to increase readability</span></span>
<span id="cb151-11"><a href="example.html#cb151-11"></a><span class="kw">V</span>(g)<span class="op">$</span>name =<span class="st">  </span><span class="kw">sub</span>(<span class="st">&quot;mbs_&quot;</span>, <span class="st">&quot;&quot;</span>, graph_info_colors<span class="op">$</span>mbs_name)</span></code></pre></div>
<p>Find distinct entires for the plot legends</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="example.html#cb152-1"></a>module_labels &lt;-<span class="st"> </span></span>
<span id="cb152-2"><a href="example.html#cb152-2"></a><span class="st">  </span>graph_info_colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb152-3"><a href="example.html#cb152-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(Module, module_color) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb152-4"><a href="example.html#cb152-4"></a><span class="st">  </span><span class="kw">distinct</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb152-5"><a href="example.html#cb152-5"></a><span class="st">  </span><span class="kw">arrange</span>(Module)</span>
<span id="cb152-6"><a href="example.html#cb152-6"></a></span>
<span id="cb152-7"><a href="example.html#cb152-7"></a>tax_labels &lt;-<span class="st"> </span></span>
<span id="cb152-8"><a href="example.html#cb152-8"></a><span class="st">  </span>graph_info_colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb152-9"><a href="example.html#cb152-9"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(selected_taxa, tax_color) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb152-10"><a href="example.html#cb152-10"></a><span class="st">  </span><span class="kw">distinct</span>()</span></code></pre></div>
<p>Plot the graphs, leftmost is colored by module, rightmost is colored by taxonomic classification</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="example.html#cb153-1"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb153-2"><a href="example.html#cb153-2"></a></span>
<span id="cb153-3"><a href="example.html#cb153-3"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb153-4"><a href="example.html#cb153-4"></a></span>
<span id="cb153-5"><a href="example.html#cb153-5"></a><span class="co"># plot(g, layout = l, vertex.color = graph_info_colors$module_color)</span></span>
<span id="cb153-6"><a href="example.html#cb153-6"></a><span class="co"># # legend(&quot;topleft&quot;, legend = paste0(&quot;mM&quot;, 0:(nrow(module_labels)-1)), fill=module_labels$module_color)</span></span>
<span id="cb153-7"><a href="example.html#cb153-7"></a><span class="co"># legend(&quot;topleft&quot;, legend = paste0(&quot;mbM&quot;, 1), fill=module_labels$module_color)</span></span>
<span id="cb153-8"><a href="example.html#cb153-8"></a></span>
<span id="cb153-9"><a href="example.html#cb153-9"></a><span class="kw">plot</span>(g, <span class="dt">layout =</span> l, <span class="dt">vertex.color =</span> graph_info_colors<span class="op">$</span>tax_color)</span>
<span id="cb153-10"><a href="example.html#cb153-10"></a><span class="kw">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="dt">legend =</span> tax_labels<span class="op">$</span>Class.II, <span class="dt">fill=</span>tax_labels<span class="op">$</span>tax_color)</span>
<span id="cb153-11"><a href="example.html#cb153-11"></a></span>
<span id="cb153-12"><a href="example.html#cb153-12"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-96-1.png" width="1440" /></p>
</div>
<div id="session-info" class="section level2 hasAnchor">
<h2><span class="header-section-number">3.10</span> Session Info<a href="example.html#session-info" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="example.html#cb154-1"></a>devtools<span class="op">::</span><span class="kw">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────
##  setting  value                       
##  version  R version 3.6.3 (2020-02-29)
##  os       Ubuntu 16.04.7 LTS          
##  system   x86_64, linux-gnu           
##  ui       RStudio                     
##  language (EN)                        
##  collate  en_IN.UTF-8                 
##  ctype    en_IN.UTF-8                 
##  tz       Asia/Hong_Kong              
##  date     2022-07-14                  
## 
## ─ Packages ───────────────────────────────────────────────────────────────────
##  package        * version  date       lib source                
##  acepack          1.4.1    2016-10-29 [1] CRAN (R 3.6.3)        
##  ade4             1.7-15   2020-02-13 [1] CRAN (R 3.6.3)        
##  AnnotationDbi    1.48.0   2019-10-29 [1] Bioconductor          
##  ape              5.4      2020-06-03 [1] CRAN (R 3.6.3)        
##  assertthat       0.2.1    2019-03-21 [1] CRAN (R 3.6.3)        
##  backports        1.1.8    2020-06-17 [1] CRAN (R 3.6.3)        
##  base64enc        0.1-3    2015-07-28 [1] CRAN (R 3.6.3)        
##  Biobase        * 2.46.0   2019-10-29 [1] Bioconductor          
##  BiocGenerics   * 0.32.0   2019-10-29 [1] Bioconductor          
##  biomformat       1.14.0   2019-10-29 [1] Bioconductor          
##  Biostrings       2.54.0   2019-10-29 [1] Bioconductor          
##  bit              1.1-15.2 2020-02-10 [1] CRAN (R 3.6.3)        
##  bit64            0.9-7    2017-05-08 [1] CRAN (R 3.6.3)        
##  bitops           1.0-6    2013-08-17 [1] CRAN (R 3.6.3)        
##  blob             1.2.1    2020-01-20 [1] CRAN (R 3.6.3)        
##  bookdown         0.27     2022-06-14 [1] CRAN (R 3.6.3)        
##  broom            0.8.0    2022-04-13 [2] CRAN (R 3.6.3)        
##  bslib            0.3.1    2021-10-06 [2] CRAN (R 3.6.3)        
##  callr            3.4.3    2020-03-28 [1] CRAN (R 3.6.3)        
##  caTools          1.18.0   2020-01-17 [1] CRAN (R 3.6.3)        
##  cellranger       1.1.0    2016-07-27 [1] CRAN (R 3.6.3)        
##  checkmate        2.0.0    2020-02-06 [1] CRAN (R 3.6.3)        
##  cli              3.0.1    2021-07-17 [1] CRAN (R 3.6.3)        
##  cluster          2.1.0    2019-06-19 [1] CRAN (R 3.6.3)        
##  codetools        0.2-16   2018-12-24 [2] CRAN (R 3.6.3)        
##  colorspace       2.0-2    2021-06-24 [1] CRAN (R 3.6.3)        
##  cowplot          1.0.0    2019-07-11 [1] CRAN (R 3.6.3)        
##  crayon           1.4.1    2021-02-08 [1] CRAN (R 3.6.3)        
##  crosstalk        1.1.1    2021-01-12 [1] CRAN (R 3.6.3)        
##  data.table       1.14.0   2021-02-21 [1] CRAN (R 3.6.3)        
##  DBI              1.1.0    2019-12-15 [1] CRAN (R 3.6.3)        
##  dbplyr           1.4.4    2020-05-27 [1] CRAN (R 3.6.3)        
##  desc             1.2.0    2018-05-01 [1] CRAN (R 3.6.3)        
##  devtools         2.3.0    2020-04-10 [1] CRAN (R 3.6.3)        
##  digest           0.6.27   2020-10-24 [1] CRAN (R 3.6.3)        
##  doParallel       1.0.17   2022-02-07 [2] CRAN (R 3.6.3)        
##  dplyr          * 1.0.7    2021-06-18 [1] CRAN (R 3.6.3)        
##  DT             * 0.14     2020-06-24 [1] CRAN (R 3.6.3)        
##  dynamicTreeCut * 1.63-1   2016-03-11 [1] CRAN (R 3.6.3)        
##  ellipsis         0.3.2    2021-04-29 [1] CRAN (R 3.6.3)        
##  evaluate         0.14     2019-05-28 [1] CRAN (R 3.6.3)        
##  fansi            0.5.0    2021-05-25 [1] CRAN (R 3.6.3)        
##  farver           2.1.0    2021-02-28 [1] CRAN (R 3.6.3)        
##  fastcluster    * 1.2.3    2021-05-24 [1] CRAN (R 3.6.3)        
##  fastmap          1.1.0    2021-01-25 [1] CRAN (R 3.6.3)        
##  fdrtool          1.2.16   2021-01-06 [1] CRAN (R 3.6.3)        
##  forcats        * 0.5.0    2020-03-01 [1] CRAN (R 3.6.3)        
##  foreach          1.5.0    2020-03-30 [1] CRAN (R 3.6.3)        
##  foreign          0.8-70.1 2018-07-13 [1] CRAN (R 3.5.2)        
##  Formula          1.2-3    2018-05-03 [1] CRAN (R 3.6.3)        
##  fs               1.4.2    2020-06-30 [1] CRAN (R 3.6.3)        
##  gdata            2.18.0   2017-06-06 [1] CRAN (R 3.6.3)        
##  generics         0.1.0    2020-10-31 [1] CRAN (R 3.6.3)        
##  ggplot2        * 3.3.5    2021-06-25 [1] CRAN (R 3.6.3)        
##  glmnet         * 4.0-2    2020-06-16 [1] CRAN (R 3.6.3)        
##  glue             1.4.2    2020-08-27 [1] CRAN (R 3.6.3)        
##  GO.db            3.10.0   2022-04-18 [1] Bioconductor          
##  gplots           3.0.4    2020-07-05 [1] CRAN (R 3.6.3)        
##  gridExtra        2.3      2017-09-09 [1] CRAN (R 3.6.3)        
##  gtable           0.3.0    2019-03-25 [1] CRAN (R 3.6.3)        
##  gtools           3.8.2    2020-03-31 [1] CRAN (R 3.6.3)        
##  haven            2.3.1    2020-06-01 [1] CRAN (R 3.6.3)        
##  Hmisc            4.4-0    2020-03-23 [1] CRAN (R 3.6.3)        
##  hms              0.5.3    2020-01-08 [1] CRAN (R 3.6.3)        
##  htmlTable        2.0.1    2020-07-05 [1] CRAN (R 3.6.3)        
##  htmltools        0.5.2    2021-08-25 [1] CRAN (R 3.6.3)        
##  htmlwidgets      1.5.3    2020-12-10 [1] CRAN (R 3.6.3)        
##  httr             1.4.2    2020-07-20 [1] CRAN (R 3.6.3)        
##  igraph         * 1.2.5    2020-03-19 [1] CRAN (R 3.6.3)        
##  IHW              1.14.0   2019-10-29 [1] Bioconductor          
##  impute           1.60.0   2019-10-29 [2] Bioconductor          
##  IRanges          2.20.2   2020-01-13 [1] Bioconductor          
##  iterators        1.0.12   2019-07-26 [1] CRAN (R 3.6.3)        
##  jpeg             0.1-8.1  2019-10-24 [1] CRAN (R 3.6.3)        
##  jquerylib        0.1.4    2021-04-26 [2] CRAN (R 3.6.3)        
##  jsonlite         1.7.2    2020-12-09 [1] CRAN (R 3.6.3)        
##  KernSmooth       2.23-17  2020-04-26 [1] CRAN (R 3.6.3)        
##  knitr            1.29     2020-06-23 [1] CRAN (R 3.6.3)        
##  labeling         0.4.2    2020-10-20 [1] CRAN (R 3.6.3)        
##  lattice          0.20-41  2020-04-02 [1] CRAN (R 3.6.3)        
##  latticeExtra     0.6-29   2019-12-19 [1] CRAN (R 3.6.3)        
##  lifecycle        1.0.0    2021-02-15 [1] CRAN (R 3.6.3)        
##  limma          * 3.42.2   2020-02-03 [2] Bioconductor          
##  locfit           1.5-9.4  2020-03-25 [1] CRAN (R 3.6.3)        
##  lpsymphony       1.14.0   2019-10-29 [1] Bioconductor (R 3.6.3)
##  lubridate        1.7.9    2020-06-08 [1] CRAN (R 3.6.3)        
##  magrittr         2.0.1    2020-11-17 [1] CRAN (R 3.6.3)        
##  MASS             7.3-51.6 2020-04-26 [1] CRAN (R 3.6.3)        
##  Matrix         * 1.2-18   2019-11-27 [1] CRAN (R 3.6.3)        
##  matrixStats      0.56.0   2020-03-13 [1] CRAN (R 3.6.3)        
##  memoise          1.1.0    2017-04-21 [1] CRAN (R 3.6.3)        
##  metagenomeSeq  * 1.28.2   2020-02-03 [1] Bioconductor          
##  mgcv             1.8-31   2019-11-09 [1] CRAN (R 3.6.3)        
##  modelr           0.1.8    2020-05-19 [1] CRAN (R 3.6.3)        
##  multtest         2.42.0   2019-10-29 [1] Bioconductor          
##  munsell          0.5.0    2018-06-12 [1] CRAN (R 3.6.3)        
##  nlme             3.1-148  2020-05-24 [1] CRAN (R 3.6.3)        
##  nnet             7.3-14   2020-04-26 [1] CRAN (R 3.6.3)        
##  permute          0.9-5    2019-03-12 [1] CRAN (R 3.6.3)        
##  pheatmap         1.0.12   2019-01-04 [1] CRAN (R 3.6.3)        
##  phyloseq       * 1.30.0   2019-10-29 [1] Bioconductor          
##  pillar           1.6.2    2021-07-29 [1] CRAN (R 3.6.3)        
##  pkgbuild         1.0.8    2020-05-07 [1] CRAN (R 3.6.3)        
##  pkgconfig        2.0.3    2019-09-22 [1] CRAN (R 3.6.3)        
##  pkgload          1.1.0    2020-05-29 [1] CRAN (R 3.6.3)        
##  plyr             1.8.6    2020-03-03 [1] CRAN (R 3.6.3)        
##  png              0.1-7    2013-12-03 [1] CRAN (R 3.6.3)        
##  preprocessCore   1.48.0   2019-10-29 [2] Bioconductor          
##  prettyunits      1.1.1    2020-01-24 [1] CRAN (R 3.6.3)        
##  processx         3.4.3    2020-07-05 [1] CRAN (R 3.6.3)        
##  propr            4.2.6    2019-12-16 [1] CRAN (R 3.6.3)        
##  ps               1.3.3    2020-05-08 [1] CRAN (R 3.6.3)        
##  purrr          * 0.3.4    2020-04-17 [1] CRAN (R 3.6.3)        
##  R6               2.5.0    2020-10-28 [1] CRAN (R 3.6.3)        
##  RColorBrewer   * 1.1-2    2014-12-07 [1] CRAN (R 3.6.3)        
##  Rcpp             1.0.7    2021-07-07 [1] CRAN (R 3.6.3)        
##  readr          * 1.3.1    2018-12-21 [1] CRAN (R 3.6.3)        
##  readxl           1.3.1    2019-03-13 [1] CRAN (R 3.6.3)        
##  remotes          2.1.1    2020-02-15 [1] CRAN (R 3.6.3)        
##  reprex           0.3.0    2019-05-16 [1] CRAN (R 3.6.3)        
##  reshape2         1.4.4    2020-04-09 [1] CRAN (R 3.6.3)        
##  rhdf5            2.30.1   2019-11-26 [1] Bioconductor          
##  Rhdf5lib         1.8.0    2019-10-29 [1] Bioconductor          
##  rJava            0.9-13   2020-07-06 [1] CRAN (R 3.6.3)        
##  rlang            0.4.11   2021-04-30 [1] CRAN (R 3.6.3)        
##  rmarkdown        2.14     2022-04-25 [1] CRAN (R 3.6.3)        
##  rpart            4.1-15   2019-04-12 [1] CRAN (R 3.6.3)        
##  rprojroot        2.0.2    2020-11-15 [1] CRAN (R 3.6.3)        
##  RSQLite          2.2.0    2020-01-07 [1] CRAN (R 3.6.3)        
##  rstudioapi       0.11     2020-02-07 [1] CRAN (R 3.6.3)        
##  rvest            0.3.5    2019-11-08 [1] CRAN (R 3.6.3)        
##  S4Vectors        0.24.4   2020-04-09 [1] Bioconductor          
##  sass             0.4.1    2022-03-23 [2] CRAN (R 3.6.3)        
##  scales           1.1.1    2020-05-11 [1] CRAN (R 3.6.3)        
##  sessioninfo      1.1.1    2018-11-05 [1] CRAN (R 3.6.3)        
##  shape            1.4.4    2018-02-07 [1] CRAN (R 3.6.3)        
##  slam             0.1-48   2020-12-03 [1] CRAN (R 3.6.3)        
##  stringi          1.4.6    2020-02-17 [1] CRAN (R 3.6.3)        
##  stringr        * 1.4.0    2019-02-10 [1] CRAN (R 3.6.3)        
##  survival         3.2-3    2020-06-13 [1] CRAN (R 3.6.3)        
##  testthat         2.3.2    2020-03-02 [1] CRAN (R 3.6.3)        
##  tibble         * 3.1.3    2021-07-23 [1] CRAN (R 3.6.3)        
##  tidyr          * 1.1.3    2021-03-03 [1] CRAN (R 3.6.3)        
##  tidyselect       1.1.1    2021-04-30 [1] CRAN (R 3.6.3)        
##  tidyverse      * 1.3.0    2019-11-21 [1] CRAN (R 3.6.3)        
##  usethis          1.6.1    2020-04-29 [1] CRAN (R 3.6.3)        
##  utf8             1.2.2    2021-07-24 [1] CRAN (R 3.6.3)        
##  vctrs            0.3.8    2021-04-29 [1] CRAN (R 3.6.3)        
##  vegan            2.5-6    2019-09-01 [1] CRAN (R 3.6.3)        
##  WGCNA          * 1.70-3   2021-02-28 [1] CRAN (R 3.6.3)        
##  withr            2.4.2    2021-04-18 [1] CRAN (R 3.6.3)        
##  Wrench           1.4.0    2019-10-29 [1] Bioconductor          
##  xfun             0.31     2022-05-10 [1] CRAN (R 3.6.3)        
##  xlsx           * 0.6.3    2020-02-28 [1] CRAN (R 3.6.3)        
##  xlsxjars         0.6.1    2014-08-22 [1] CRAN (R 3.6.3)        
##  xml2             1.3.2    2020-04-23 [1] CRAN (R 3.6.3)        
##  XVector          0.26.0   2019-10-29 [1] Bioconductor          
##  yaml             2.2.1    2020-02-01 [1] CRAN (R 3.6.3)        
##  zlibbioc         1.32.0   2019-10-29 [1] Bioconductor          
## 
## [1] /share/home/zhuzhengnong/R/x86_64-pc-linux-gnu-library/3.6
## [2] /opt/R-3.6.3/lib/R/library</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="利用wgcna构建网络.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="halla-tutorial.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/OMicsAnalysis/blob/main/02-GVHD_template.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMics analysis workflow.pdf", "OMics analysis workflow.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
