<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 HAllA | Omics analysis</title>
  <meta name="description" content="Tutorial for OMics analysis" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 HAllA | Omics analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Tutorial for OMics analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 HAllA | Omics analysis" />
  
  <meta name="twitter:description" content="Tutorial for OMics analysis" />
  

<meta name="author" content="Zhengnong Zhu, Yifan Zhao, Hua Zou, Bangzhuo Tong" />


<meta name="date" content="2022-09-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="wgcna.html"/>
<link rel="next" href="netcomi.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.23/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">OMics Data Analysis </a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Omics Analysis</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introducution-to-wgcna"><i class="fa fa-check"></i><b>1.1</b> Introducution to WGCNA</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#introducution-to-halla"><i class="fa fa-check"></i><b>1.2</b> Introducution to HAllA</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#introducution-to-netcomi"><i class="fa fa-check"></i><b>1.3</b> Introducution to NetCoMi</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="wgcna.html"><a href="wgcna.html"><i class="fa fa-check"></i><b>2</b> WGCNA</a><ul>
<li class="chapter" data-level="2.1" data-path="wgcna.html"><a href="wgcna.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a><ul>
<li class="chapter" data-level="2.1.1" data-path="wgcna.html"><a href="wgcna.html#introduction-to-the-7-steps-in-wgcna"><i class="fa fa-check"></i><b>2.1.1</b> Introduction to the 7 Steps in WGCNA</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="wgcna.html"><a href="wgcna.html#environment-setup"><i class="fa fa-check"></i><b>2.2</b> Environment Setup</a></li>
<li class="chapter" data-level="2.3" data-path="wgcna.html"><a href="wgcna.html#wgcna-for-16s-data"><i class="fa fa-check"></i><b>2.3</b> WGCNA for 16S data</a><ul>
<li class="chapter" data-level="2.3.1" data-path="wgcna.html"><a href="wgcna.html#data-loading-wgcna-16s"><i class="fa fa-check"></i><b>2.3.1</b> Data loading (WGCNA 16S)</a></li>
<li class="chapter" data-level="2.3.2" data-path="wgcna.html"><a href="wgcna.html#build-wgcna-network-wgcna-16s"><i class="fa fa-check"></i><b>2.3.2</b> Build WGCNA network (WGCNA 16S)</a></li>
<li class="chapter" data-level="2.3.3" data-path="wgcna.html"><a href="wgcna.html#block-wise-network-construction-and-module-detection"><i class="fa fa-check"></i><b>2.3.3</b> Block-wise network construction and module detection</a></li>
<li class="chapter" data-level="2.3.4" data-path="wgcna.html"><a href="wgcna.html#module-eigengene-correlation"><i class="fa fa-check"></i><b>2.3.4</b> Module (Eigengene) correlation</a></li>
<li class="chapter" data-level="2.3.5" data-path="wgcna.html"><a href="wgcna.html#find-hub-asv"><i class="fa fa-check"></i><b>2.3.5</b> Find Hub ASV</a></li>
<li class="chapter" data-level="2.3.6" data-path="wgcna.html"><a href="wgcna.html#visualize-network"><i class="fa fa-check"></i><b>2.3.6</b> Visualize Network</a></li>
<li class="chapter" data-level="2.3.7" data-path="wgcna.html"><a href="wgcna.html#clean-env-of-16s-data-analysis"><i class="fa fa-check"></i><b>2.3.7</b> Clean env of 16S data analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="wgcna.html"><a href="wgcna.html#wgcna-for-mgs-data"><i class="fa fa-check"></i><b>2.4</b> WGCNA for MGS data</a><ul>
<li class="chapter" data-level="2.4.1" data-path="wgcna.html"><a href="wgcna.html#data-loading-wgcna-mgs"><i class="fa fa-check"></i><b>2.4.1</b> Data loading (WGCNA MGS)</a></li>
<li class="chapter" data-level="2.4.2" data-path="wgcna.html"><a href="wgcna.html#data-normalization-wgcna-mgs"><i class="fa fa-check"></i><b>2.4.2</b> Data normalization (WGCNA MGS)</a></li>
<li class="chapter" data-level="2.4.3" data-path="wgcna.html"><a href="wgcna.html#build-wgcna-network-wgcna-mgs"><i class="fa fa-check"></i><b>2.4.3</b> Build WGCNA network (WGCNA MGS)</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="wgcna.html"><a href="wgcna.html#wgcna-for-metabolic-data"><i class="fa fa-check"></i><b>2.5</b> WGCNA for metabolic data</a><ul>
<li class="chapter" data-level="2.5.1" data-path="wgcna.html"><a href="wgcna.html#data-loading-wgcna-metabolites"><i class="fa fa-check"></i><b>2.5.1</b> Data loading (WGCNA metabolites)</a></li>
<li class="chapter" data-level="2.5.2" data-path="wgcna.html"><a href="wgcna.html#build-wgcna-network-wgcna-metabolites"><i class="fa fa-check"></i><b>2.5.2</b> Build WGCNA network (WGCNA metabolites)</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="wgcna.html"><a href="wgcna.html#metabolites-mgs-and-metadata-association"><i class="fa fa-check"></i><b>2.6</b> Metabolites, mgs and metadata association</a><ul>
<li class="chapter" data-level="2.6.1" data-path="wgcna.html"><a href="wgcna.html#use-heatmap-to-show-the-correlation-of-mgs-and-metabolites"><i class="fa fa-check"></i><b>2.6.1</b> Use heatmap to show the correlation of MGS and metabolites</a></li>
<li class="chapter" data-level="2.6.2" data-path="wgcna.html"><a href="wgcna.html#correlate-modules-from-metabolic-and-metagenomics"><i class="fa fa-check"></i><b>2.6.2</b> Correlate modules from metabolic and metagenomics</a></li>
<li class="chapter" data-level="2.6.3" data-path="wgcna.html"><a href="wgcna.html#visualization-network"><i class="fa fa-check"></i><b>2.6.3</b> Visualization network</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="wgcna.html"><a href="wgcna.html#session-info"><i class="fa fa-check"></i><b>2.7</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="halla.html"><a href="halla.html"><i class="fa fa-check"></i><b>3</b> HAllA</a><ul>
<li class="chapter" data-level="3.1" data-path="halla.html"><a href="halla.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a><ul>
<li class="chapter" data-level="3.1.1" data-path="halla.html"><a href="halla.html#available-pairwise-distance-metrics"><i class="fa fa-check"></i><b>3.1.1</b> Available pairwise distance metrics</a></li>
<li class="chapter" data-level="3.1.2" data-path="halla.html"><a href="halla.html#running-procedures"><i class="fa fa-check"></i><b>3.1.2</b> Running procedures</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="halla.html"><a href="halla.html#run-halla"><i class="fa fa-check"></i><b>3.2</b> Run HALLA</a></li>
<li class="chapter" data-level="3.3" data-path="halla.html"><a href="halla.html#result-of-halla"><i class="fa fa-check"></i><b>3.3</b> Result of HALLA</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="netcomi.html"><a href="netcomi.html"><i class="fa fa-check"></i><b>4</b> NetCoMi</a><ul>
<li class="chapter" data-level="4.1" data-path="netcomi.html"><a href="netcomi.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a><ul>
<li class="chapter" data-level="4.1.1" data-path="netcomi.html"><a href="netcomi.html#association-measures"><i class="fa fa-check"></i><b>4.1.1</b> Association measures</a></li>
<li class="chapter" data-level="4.1.2" data-path="netcomi.html"><a href="netcomi.html#dissimilarity-measures"><i class="fa fa-check"></i><b>4.1.2</b> Dissimilarity measures</a></li>
<li class="chapter" data-level="4.1.3" data-path="netcomi.html"><a href="netcomi.html#methods-for-zero-replacement"><i class="fa fa-check"></i><b>4.1.3</b> Methods for zero replacement</a></li>
<li class="chapter" data-level="4.1.4" data-path="netcomi.html"><a href="netcomi.html#normalization-methods"><i class="fa fa-check"></i><b>4.1.4</b> Normalization methods</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="netcomi.html"><a href="netcomi.html#environment-setup-1"><i class="fa fa-check"></i><b>4.2</b> Environment Setup</a></li>
<li class="chapter" data-level="4.3" data-path="netcomi.html"><a href="netcomi.html#loading-data"><i class="fa fa-check"></i><b>4.3</b> Loading data</a></li>
<li class="chapter" data-level="4.4" data-path="netcomi.html"><a href="netcomi.html#data-curation"><i class="fa fa-check"></i><b>4.4</b> Data curation</a></li>
<li class="chapter" data-level="4.5" data-path="netcomi.html"><a href="netcomi.html#associations-among-features"><i class="fa fa-check"></i><b>4.5</b> Associations Among Features</a><ul>
<li class="chapter" data-level="4.5.1" data-path="netcomi.html"><a href="netcomi.html#single-network-with-spring-as-association-measure"><i class="fa fa-check"></i><b>4.5.1</b> Single network with SPRING as association measure</a></li>
<li class="chapter" data-level="4.5.2" data-path="netcomi.html"><a href="netcomi.html#single-network-with-pearson-correlation-as-association-measure"><i class="fa fa-check"></i><b>4.5.2</b> Single network with Pearson correlation as association measure</a></li>
<li class="chapter" data-level="4.5.3" data-path="netcomi.html"><a href="netcomi.html#single-network-with-spieceasi-correlation-as-association-measure"><i class="fa fa-check"></i><b>4.5.3</b> Single network with SpiecEasi correlation as association measure</a></li>
<li class="chapter" data-level="4.5.4" data-path="netcomi.html"><a href="netcomi.html#single-network-with-wgcna-bicor-as-association-measure"><i class="fa fa-check"></i><b>4.5.4</b> Single network with WGCNA (bicor) as association measure</a></li>
<li class="chapter" data-level="4.5.5" data-path="netcomi.html"><a href="netcomi.html#single-network-with-sparcc-correlation-as-association-measure"><i class="fa fa-check"></i><b>4.5.5</b> Single network with sparcc correlation as association measure</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="netcomi.html"><a href="netcomi.html#dissimilarity-based-networks"><i class="fa fa-check"></i><b>4.6</b> Dissimilarity-based Networks</a><ul>
<li class="chapter" data-level="4.6.1" data-path="netcomi.html"><a href="netcomi.html#building-network-module-dissimilarity-based"><i class="fa fa-check"></i><b>4.6.1</b> Building network module (Dissimilarity-based)</a></li>
<li class="chapter" data-level="4.6.2" data-path="netcomi.html"><a href="netcomi.html#analyzing-the-constructed-network-dissimilarity-based"><i class="fa fa-check"></i><b>4.6.2</b> Analyzing the constructed network (Dissimilarity-based)</a></li>
<li class="chapter" data-level="4.6.3" data-path="netcomi.html"><a href="netcomi.html#visualizing-the-network-dissimilarity-based"><i class="fa fa-check"></i><b>4.6.3</b> Visualizing the network (Dissimilarity-based)</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="netcomi.html"><a href="netcomi.html#network-comparison"><i class="fa fa-check"></i><b>4.7</b> Network comparison</a><ul>
<li class="chapter" data-level="4.7.1" data-path="netcomi.html"><a href="netcomi.html#data-preparing"><i class="fa fa-check"></i><b>4.7.1</b> Data preparing</a></li>
<li class="chapter" data-level="4.7.2" data-path="netcomi.html"><a href="netcomi.html#building-network-model"><i class="fa fa-check"></i><b>4.7.2</b> Building network model</a></li>
<li class="chapter" data-level="4.7.3" data-path="netcomi.html"><a href="netcomi.html#network-analysis"><i class="fa fa-check"></i><b>4.7.3</b> Network analysis</a></li>
<li class="chapter" data-level="4.7.4" data-path="netcomi.html"><a href="netcomi.html#visualizing-the-network"><i class="fa fa-check"></i><b>4.7.4</b> Visualizing the network</a></li>
<li class="chapter" data-level="4.7.5" data-path="netcomi.html"><a href="netcomi.html#quantitative-network-comparison"><i class="fa fa-check"></i><b>4.7.5</b> Quantitative network comparison</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="netcomi.html"><a href="netcomi.html#session-info-1"><i class="fa fa-check"></i><b>4.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Omics analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="halla" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> HAllA</h1>
<p><em>For any questions about this chapter, please contact zhaoyifan (<a href="mailto:zhaoyifan@xbiome.com" class="email">zhaoyifan@xbiome.com</a>)</em></p>
<div id="introduction-1" class="section level2">
<h2><span class="header-section-number">3.1</span> Introduction</h2>
<p><a href="https://huttenhower.sph.harvard.edu/halla">HAllA</a> (Hierarchical All-against-All association) 是一个针对多维度、异质型数据集的多重关联分析工具。可用于连续型/分类型数值数据，且在同质型（homogeneous）数据集（所有数值皆相同类型，例如：RNA-seq 基因表現）与异质型（heterogeneous）数据集（不同单位或类型的数值，例如：病患临床指标）皆可高效分析。在目前微生物研究中，HAllA是一個常用于微生物组-代谢、微生物组-临床指标、微生物组-转录组（基因表現、miRNA、GO） 等关联分析的重要工具。</p>
<div class="figure">
<img src="figure/HAllA_workflow.png" alt="" />
<p class="caption">HAllA workflow</p>
</div>
<div id="available-pairwise-distance-metrics" class="section level3">
<h3><span class="header-section-number">3.1.1</span> Available pairwise distance metrics</h3>
<ul>
<li><p>spearman (default for continuous data)</p></li>
<li><p>pearson</p></li>
<li><p>mi (A set of properties of Mutual Information result from definition. default for mixed/categorical data)</p></li>
<li><p>nmi (Normalized Mutual Information)</p></li>
<li><p>xicor (Chatterjee correlation, uses rank differences to assess the degree to which one variable is a measurable function of another，适用于离散、非线性关系数据)</p></li>
<li><p>dcor (measures not necessarily linear dependency of two random variables with possibly different dimensions)</p></li>
</ul>
<div class="figure">
<img src="figure/HAllA_dataType.png" alt="" />
<p class="caption">HAllA correlation methods</p>
</div>
<hr />
</div>
<div id="running-procedures" class="section level3">
<h3><span class="header-section-number">3.1.2</span> Running procedures</h3>
<div id="step1" class="section level4">
<h4><span class="header-section-number">3.1.2.1</span> step1</h4>
<p>The pairwise similarity matrix between all features in X and Y is computed with a specified similarity measure, such as Spearman correlation and normalized mutual information (NMI). This step then generates the p-value and q-value tables.</p>
<p>Note that for handling heterogeneous data, all continuous features are first discretized into bins using a specified binning method.</p>
</div>
<div id="step2" class="section level4">
<h4><span class="header-section-number">3.1.2.2</span> step2</h4>
<p>Hierarchical clustering on the features in each dataset is performed using the converted similarity measure used in step 1. It produces a tree for each dataset.</p>
</div>
<div id="step3" class="section level4">
<h4><span class="header-section-number">3.1.2.3</span> step3</h4>
<p>Finding densely-associated blocks (pseudocode):</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode python"><code class="sourceCode python"><span id="cb153-1"><a href="halla.html#cb153-1"></a><span class="kw">def</span> find_densely_associated_blocks(x, y):</span>
<span id="cb153-2"><a href="halla.html#cb153-2"></a>    x_features <span class="op">=</span> <span class="bu">all</span> features <span class="kw">in</span> x</span>
<span id="cb153-3"><a href="halla.html#cb153-3"></a>    y_features <span class="op">=</span> <span class="bu">all</span> features <span class="kw">in</span> y</span>
<span id="cb153-4"><a href="halla.html#cb153-4"></a>    <span class="cf">if</span> is_densely_associated(x_features, y_features):</span>
<span id="cb153-5"><a href="halla.html#cb153-5"></a>        report block <span class="kw">and</span> terminate</span>
<span id="cb153-6"><a href="halla.html#cb153-6"></a>    <span class="cf">else</span>:</span>
<span id="cb153-7"><a href="halla.html#cb153-7"></a>        <span class="co"># bifurcate one according to Gini impurities of the splits</span></span>
<span id="cb153-8"><a href="halla.html#cb153-8"></a>        x_branches, y_branches <span class="op">=</span> bifurcate_one_of(x, y)</span>
<span id="cb153-9"><a href="halla.html#cb153-9"></a>        <span class="cf">if</span> both x <span class="kw">and</span> y are leaves:</span>
<span id="cb153-10"><a href="halla.html#cb153-10"></a>            terminate</span>
<span id="cb153-11"><a href="halla.html#cb153-11"></a>        <span class="cf">for</span> each x_branch <span class="kw">and</span> y_branch <span class="kw">in</span> x_branches <span class="kw">and</span> y_branches:</span>
<span id="cb153-12"><a href="halla.html#cb153-12"></a>            find_densely_associated_blocks(x_branch, y_branch)</span>
<span id="cb153-13"><a href="halla.html#cb153-13"></a>initial function call: find_densely_associated_blocks(X_root, Y_root)</span></code></pre></div>
<p>For example, given two datasets of X (features: X1, X2, X3, X4, X5) and Y (features: Y1, Y2, Y3, Y4) both hierarchically clustered in X tree and Y tree, the algorithm first evaluates the roots of both trees and checks if the block consisting of all features of X and Y are densely-associated (if %significance (%reject) &gt;= (1 - FNR)%).</p>
<p>If the block is not densely-associated, the algorithm would bifurcate one of the trees. It would pick one of:</p>
<p>·[X1 X2][X3 X4 X5] &gt;&lt; [Y1 Y2 Y3 Y4] or</p>
<p>·[X1 X2 X3 X4 X5] &gt;&lt; [Y1 Y2 Y3][Y4]</p>
<p>based on the Gini impurity of the splits (pick the split that produces a lower weighted Gini impurity),</p>
<p>Once it picks the split with the lower impurity (let’s say the first split), it will iteratively evaluate the branches:</p>
<p>·find densely-associated blocks in [X1 X2] vs [Y1 Y2 Y3 Y4], and</p>
<p>·find densely-associated blocks in [X3 X4 X5] vs [Y1 Y2 Y3 Y4]</p>
<p>and keep going until it terminates.</p>
<hr />
</div>
</div>
</div>
<div id="run-halla" class="section level2">
<h2><span class="header-section-number">3.2</span> Run HALLA</h2>
<p>Run HALLA on remote server</p>
<div class="sourceCode" id="cb154"><pre class="sourceCode markdown"><code class="sourceCode markdown"><span id="cb154-1"><a href="halla.html#cb154-1"></a></span>
<span id="cb154-2"><a href="halla.html#cb154-2"></a>conda activate /opt/conda/envs/Halla</span>
<span id="cb154-3"><a href="halla.html#cb154-3"></a></span>
<span id="cb154-4"><a href="halla.html#cb154-4"></a>halla -h</span>
<span id="cb154-5"><a href="halla.html#cb154-5"></a></span>
<span id="cb154-6"><a href="halla.html#cb154-6"></a>halla -x table_X -y table_Y -m correlation_methods -o output_file --fdr_alpha adjustP --fdr_method FDR_methods</span></code></pre></div>
</div>
<div id="result-of-halla" class="section level2">
<h2><span class="header-section-number">3.3</span> Result of HALLA</h2>
<p>HAllA result after WGCNA</p>
<div class="figure">
<img src="figure/GvHD_module.png" alt="" />
<p class="caption">Correlation with modules</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="wgcna.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="netcomi.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/OMicsAnalysis/blob/main/02-HALLA_network.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMics analysis workflow.pdf", "OMics analysis workflow.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
