<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 WGCNA | Omics analysis</title>
  <meta name="description" content="Tutorial for OMics analysis" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 WGCNA | Omics analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Tutorial for OMics analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 WGCNA | Omics analysis" />
  
  <meta name="twitter:description" content="Tutorial for OMics analysis" />
  

<meta name="author" content="Zhengnong Zhu, Yifan Zhao, Hua Zou, Bangzhuo Tong" />


<meta name="date" content="2022-09-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="halla.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.23/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">OMics Data Analysis </a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Omics Analysis</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introducution-to-wgcna"><i class="fa fa-check"></i><b>1.1</b> Introducution to WGCNA</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#introducution-to-halla"><i class="fa fa-check"></i><b>1.2</b> Introducution to HAllA</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#introducution-to-netcomi"><i class="fa fa-check"></i><b>1.3</b> Introducution to NetCoMi</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="wgcna.html"><a href="wgcna.html"><i class="fa fa-check"></i><b>2</b> WGCNA</a><ul>
<li class="chapter" data-level="2.1" data-path="wgcna.html"><a href="wgcna.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a><ul>
<li class="chapter" data-level="2.1.1" data-path="wgcna.html"><a href="wgcna.html#introduction-to-the-7-steps-in-wgcna"><i class="fa fa-check"></i><b>2.1.1</b> Introduction to the 7 Steps in WGCNA</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="wgcna.html"><a href="wgcna.html#environment-setup"><i class="fa fa-check"></i><b>2.2</b> Environment Setup</a></li>
<li class="chapter" data-level="2.3" data-path="wgcna.html"><a href="wgcna.html#wgcna-for-16s-data"><i class="fa fa-check"></i><b>2.3</b> WGCNA for 16S data</a><ul>
<li class="chapter" data-level="2.3.1" data-path="wgcna.html"><a href="wgcna.html#data-loading-wgcna-16s"><i class="fa fa-check"></i><b>2.3.1</b> Data loading (WGCNA 16S)</a></li>
<li class="chapter" data-level="2.3.2" data-path="wgcna.html"><a href="wgcna.html#build-wgcna-network-wgcna-16s"><i class="fa fa-check"></i><b>2.3.2</b> Build WGCNA network (WGCNA 16S)</a></li>
<li class="chapter" data-level="2.3.3" data-path="wgcna.html"><a href="wgcna.html#block-wise-network-construction-and-module-detection"><i class="fa fa-check"></i><b>2.3.3</b> Block-wise network construction and module detection</a></li>
<li class="chapter" data-level="2.3.4" data-path="wgcna.html"><a href="wgcna.html#module-eigengene-correlation"><i class="fa fa-check"></i><b>2.3.4</b> Module (Eigengene) correlation</a></li>
<li class="chapter" data-level="2.3.5" data-path="wgcna.html"><a href="wgcna.html#find-hub-asv"><i class="fa fa-check"></i><b>2.3.5</b> Find Hub ASV</a></li>
<li class="chapter" data-level="2.3.6" data-path="wgcna.html"><a href="wgcna.html#visualize-network"><i class="fa fa-check"></i><b>2.3.6</b> Visualize Network</a></li>
<li class="chapter" data-level="2.3.7" data-path="wgcna.html"><a href="wgcna.html#clean-env-of-16s-data-analysis"><i class="fa fa-check"></i><b>2.3.7</b> Clean env of 16S data analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="wgcna.html"><a href="wgcna.html#wgcna-for-mgs-data"><i class="fa fa-check"></i><b>2.4</b> WGCNA for MGS data</a><ul>
<li class="chapter" data-level="2.4.1" data-path="wgcna.html"><a href="wgcna.html#data-loading-wgcna-mgs"><i class="fa fa-check"></i><b>2.4.1</b> Data loading (WGCNA MGS)</a></li>
<li class="chapter" data-level="2.4.2" data-path="wgcna.html"><a href="wgcna.html#data-normalization-wgcna-mgs"><i class="fa fa-check"></i><b>2.4.2</b> Data normalization (WGCNA MGS)</a></li>
<li class="chapter" data-level="2.4.3" data-path="wgcna.html"><a href="wgcna.html#build-wgcna-network-wgcna-mgs"><i class="fa fa-check"></i><b>2.4.3</b> Build WGCNA network (WGCNA MGS)</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="wgcna.html"><a href="wgcna.html#wgcna-for-metabolic-data"><i class="fa fa-check"></i><b>2.5</b> WGCNA for metabolic data</a><ul>
<li class="chapter" data-level="2.5.1" data-path="wgcna.html"><a href="wgcna.html#data-loading-wgcna-metabolites"><i class="fa fa-check"></i><b>2.5.1</b> Data loading (WGCNA metabolites)</a></li>
<li class="chapter" data-level="2.5.2" data-path="wgcna.html"><a href="wgcna.html#build-wgcna-network-wgcna-metabolites"><i class="fa fa-check"></i><b>2.5.2</b> Build WGCNA network (WGCNA metabolites)</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="wgcna.html"><a href="wgcna.html#metabolites-mgs-and-metadata-association"><i class="fa fa-check"></i><b>2.6</b> Metabolites, mgs and metadata association</a><ul>
<li class="chapter" data-level="2.6.1" data-path="wgcna.html"><a href="wgcna.html#use-heatmap-to-show-the-correlation-of-mgs-and-metabolites"><i class="fa fa-check"></i><b>2.6.1</b> Use heatmap to show the correlation of MGS and metabolites</a></li>
<li class="chapter" data-level="2.6.2" data-path="wgcna.html"><a href="wgcna.html#correlate-modules-from-metabolic-and-metagenomics"><i class="fa fa-check"></i><b>2.6.2</b> Correlate modules from metabolic and metagenomics</a></li>
<li class="chapter" data-level="2.6.3" data-path="wgcna.html"><a href="wgcna.html#visualization-network"><i class="fa fa-check"></i><b>2.6.3</b> Visualization network</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="wgcna.html"><a href="wgcna.html#session-info"><i class="fa fa-check"></i><b>2.7</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="halla.html"><a href="halla.html"><i class="fa fa-check"></i><b>3</b> HAllA</a><ul>
<li class="chapter" data-level="3.1" data-path="halla.html"><a href="halla.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a><ul>
<li class="chapter" data-level="3.1.1" data-path="halla.html"><a href="halla.html#available-pairwise-distance-metrics"><i class="fa fa-check"></i><b>3.1.1</b> Available pairwise distance metrics</a></li>
<li class="chapter" data-level="3.1.2" data-path="halla.html"><a href="halla.html#running-procedures"><i class="fa fa-check"></i><b>3.1.2</b> Running procedures</a></li>
<li class="chapter" data-level="3.1.3" data-path="halla.html"><a href="halla.html#run-halla"><i class="fa fa-check"></i><b>3.1.3</b> Run HALLA</a></li>
<li class="chapter" data-level="3.1.4" data-path="halla.html"><a href="halla.html#result-of-halla"><i class="fa fa-check"></i><b>3.1.4</b> Result of HALLA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="netcomi.html"><a href="netcomi.html"><i class="fa fa-check"></i><b>4</b> NetCoMi</a><ul>
<li class="chapter" data-level="4.1" data-path="netcomi.html"><a href="netcomi.html#introduction-2"><i class="fa fa-check"></i><b>4.1</b> Introduction</a><ul>
<li class="chapter" data-level="4.1.1" data-path="netcomi.html"><a href="netcomi.html#association-measures"><i class="fa fa-check"></i><b>4.1.1</b> Association measures</a></li>
<li class="chapter" data-level="4.1.2" data-path="netcomi.html"><a href="netcomi.html#dissimilarity-measures"><i class="fa fa-check"></i><b>4.1.2</b> Dissimilarity measures</a></li>
<li class="chapter" data-level="4.1.3" data-path="netcomi.html"><a href="netcomi.html#methods-for-zero-replacement"><i class="fa fa-check"></i><b>4.1.3</b> Methods for zero replacement</a></li>
<li class="chapter" data-level="4.1.4" data-path="netcomi.html"><a href="netcomi.html#normalization-methods"><i class="fa fa-check"></i><b>4.1.4</b> Normalization methods</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="netcomi.html"><a href="netcomi.html#environment-setup-1"><i class="fa fa-check"></i><b>4.2</b> Environment Setup</a></li>
<li class="chapter" data-level="4.3" data-path="netcomi.html"><a href="netcomi.html#loading-data"><i class="fa fa-check"></i><b>4.3</b> Loading data</a></li>
<li class="chapter" data-level="4.4" data-path="netcomi.html"><a href="netcomi.html#data-curation"><i class="fa fa-check"></i><b>4.4</b> Data curation</a></li>
<li class="chapter" data-level="4.5" data-path="netcomi.html"><a href="netcomi.html#associations-among-features"><i class="fa fa-check"></i><b>4.5</b> Associations Among Features</a><ul>
<li class="chapter" data-level="4.5.1" data-path="netcomi.html"><a href="netcomi.html#single-network-with-spring-as-association-measure"><i class="fa fa-check"></i><b>4.5.1</b> Single network with SPRING as association measure</a></li>
<li class="chapter" data-level="4.5.2" data-path="netcomi.html"><a href="netcomi.html#single-network-with-pearson-correlation-as-association-measure"><i class="fa fa-check"></i><b>4.5.2</b> Single network with Pearson correlation as association measure</a></li>
<li class="chapter" data-level="4.5.3" data-path="netcomi.html"><a href="netcomi.html#single-network-with-spieceasi-correlation-as-association-measure"><i class="fa fa-check"></i><b>4.5.3</b> Single network with SpiecEasi correlation as association measure</a></li>
<li class="chapter" data-level="4.5.4" data-path="netcomi.html"><a href="netcomi.html#single-network-with-wgcna-bicor-as-association-measure"><i class="fa fa-check"></i><b>4.5.4</b> Single network with WGCNA (bicor) as association measure</a></li>
<li class="chapter" data-level="4.5.5" data-path="netcomi.html"><a href="netcomi.html#single-network-with-sparcc-correlation-as-association-measure"><i class="fa fa-check"></i><b>4.5.5</b> Single network with sparcc correlation as association measure</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="netcomi.html"><a href="netcomi.html#dissimilarity-based-networks"><i class="fa fa-check"></i><b>4.6</b> Dissimilarity-based Networks</a><ul>
<li class="chapter" data-level="4.6.1" data-path="netcomi.html"><a href="netcomi.html#building-network-module-dissimilarity-based"><i class="fa fa-check"></i><b>4.6.1</b> Building network module (Dissimilarity-based)</a></li>
<li class="chapter" data-level="4.6.2" data-path="netcomi.html"><a href="netcomi.html#analyzing-the-constructed-network-dissimilarity-based"><i class="fa fa-check"></i><b>4.6.2</b> Analyzing the constructed network (Dissimilarity-based)</a></li>
<li class="chapter" data-level="4.6.3" data-path="netcomi.html"><a href="netcomi.html#visualizing-the-network-dissimilarity-based"><i class="fa fa-check"></i><b>4.6.3</b> Visualizing the network (Dissimilarity-based)</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="netcomi.html"><a href="netcomi.html#network-comparison"><i class="fa fa-check"></i><b>4.7</b> Network comparison</a><ul>
<li class="chapter" data-level="4.7.1" data-path="netcomi.html"><a href="netcomi.html#data-preparing"><i class="fa fa-check"></i><b>4.7.1</b> Data preparing</a></li>
<li class="chapter" data-level="4.7.2" data-path="netcomi.html"><a href="netcomi.html#building-network-model"><i class="fa fa-check"></i><b>4.7.2</b> Building network model</a></li>
<li class="chapter" data-level="4.7.3" data-path="netcomi.html"><a href="netcomi.html#network-analysis"><i class="fa fa-check"></i><b>4.7.3</b> Network analysis</a></li>
<li class="chapter" data-level="4.7.4" data-path="netcomi.html"><a href="netcomi.html#visualizing-the-network"><i class="fa fa-check"></i><b>4.7.4</b> Visualizing the network</a></li>
<li class="chapter" data-level="4.7.5" data-path="netcomi.html"><a href="netcomi.html#quantitative-network-comparison"><i class="fa fa-check"></i><b>4.7.5</b> Quantitative network comparison</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="netcomi.html"><a href="netcomi.html#session-info-1"><i class="fa fa-check"></i><b>4.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Omics analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="wgcna" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> WGCNA</h1>
<p><em>For any questions about this chapter, please contact zhuzhengnong (<a href="mailto:zhuzhengnong@xbiome.com" class="email">zhuzhengnong@xbiome.com</a>)</em></p>
<div id="introduction" class="section level2">
<h2><span class="header-section-number">2.1</span> Introduction</h2>
<p>In this book, we will introduce an method of multi-omics association analysis, entitled WGCNA. We will introduce usage and principle of WGCNA. Moreover, we will utilize GVHD data to illustate pipelines of WGCNA.</p>
<p>Correlation networks are increasingly being used in bioinformatics applications. For example, weighted gene co-expression network analysis is a systems biology method for describing the correlation patterns among genes across microarray samples. Weighted correlation network analysis (WGCNA) can be used for finding clusters (modules) of highly correlated genes, for summarizing such clusters using the module eigengene or an intramodular hub gene, for relating modules to one another and to external sample traits (using eigengene network methodology), and for calculating module membership measures. Correlation networks facilitate network based gene screening methods that can be used to identify candidate biomarkers or therapeutic targets. These methods have been successfully applied in various biological contexts, e.g. cancer, mouse genetics, yeast genetics, and analysis of brain imaging data. While parts of the correlation network methodology have been described in separate publications, there is a need to provide a user-friendly, comprehensive, and consistent software implementation and an accompanying tutorial.</p>
<p>The WGCNA R software package is a comprehensive collection of R functions for performing various aspects of weighted correlation network analysis. The package includes functions for network construction, module detection, gene selection, calculations of topological properties, data simulation, visualization, and interfacing with external software. Along with the R package we also present R software tutorials. While the methods development was motivated by gene expression data, the underlying data mining approach can be applied to a variety of different settings. More details can be found on <a href="https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/">WGCNA website</a></p>
<div class="figure">
<img src="figure/WGCNA_workflow.png" alt="" />
<p class="caption">WGCNA workflow</p>
</div>
<hr />
<div id="introduction-to-the-7-steps-in-wgcna" class="section level3">
<h3><span class="header-section-number">2.1.1</span> Introduction to the 7 Steps in WGCNA</h3>
<p>Please read the follwing introduction to the steps of WGCNA. If you know WGCNA very well, please jump to <a href="wgcna.html#environment-setup">Environment Setup</a></p>
<hr />
<div id="step-one-the-definition-of-a-gene-co-expression-similarity" class="section level4">
<h4><span class="header-section-number">2.1.1.1</span> Step one: The Definition of a Gene Co-expression Similarity</h4>
<p>第一步首先确定需要构建网络模块使用那种 correlation 方法去计算节点之间的相关性，常见的方法为 Pearson correlation <span class="math inline">\(s_{ij} = |cor(i,j)|\)</span>，那么 similarity matrix <span class="math inline">\(S = [s_{ij}]\)</span>。当然，我们可以根据不同的数据使用不同的correlation算法。WGCNA默认的correlation算法为<a href="https://en.wikipedia.org/wiki/Biweight_midcorrelation">bicor</a></p>
<hr />
</div>
<div id="step-two-the-definition-of-a-family-of-adjacency-functions" class="section level4">
<h4><span class="header-section-number">2.1.1.2</span> Step two: The Definition of a Family of Adjacency Functions</h4>
<p>第二步是确定邻接矩阵。将similarity matrix 转化为 adjacency matrix，需要一个阈值。确定这个阈值有两种方法，hard threshold 和 soft threshold。</p>
<p>hard threshold 通常使用signum函数作为连接函数来实现：</p>
<p><span class="math display">\[
a_{ij} = signum(s_{ij}, \tau) \equiv
\begin{cases}
1 \quad if s_{ij} \geq \tau \\
0 \quad if s_{ij} \le \tau
\end{cases}
\]</span></p>
<p>Hard threshoding using the signum function leads to intuitive network concepts, but it may lead to a loss of information: if <span class="math inline">\(\tau\)</span> has been set to 0.8, there will be no connection between two nodes if their similarity equals 0.79.</p>
<p>To avoid the disadvantage of hard thresholding, we propose <strong>two types of ‘soft’ adjacency functions</strong>: the <strong>sigmoid function</strong></p>
<p><span class="math display">\[
a_{ij} = sigmoid(s_{ij}, \alpha, \tau_{0}) \equiv \frac{1}{1+e^{-\alpha(s_{ij}-\tau_{0})}}
\]</span>
the <strong>power adjacency function</strong></p>
<p><span class="math display">\[
a_{ij} = power(s_{ij}, \beta) \equiv |s_{ij}|^\beta
\]</span></p>
<div class="figure">
<img src="figure/adjacency_matrix_for_function.png" alt="" />
<p class="caption">根据不同adjacency function构建的不同adjacency matrix</p>
</div>
<p>a图为使用sigmoid function的adjacency matrix，b图为使用power function的adjacency matrix。从图中我们可以看出power function的adjacency matrix能够保留大部分强相关连的关系，弱关联的关系则会被消除，能够减少节点之间的连接噪音。在我们的WGCNA流程中我们默认选择的是power function方法去构建adjacency。</p>
<hr />
</div>
<div id="step-three-determining-the-parameters-of-the-adjacency-function" class="section level4">
<h4><span class="header-section-number">2.1.1.3</span> Step three: Determining the Parameters of the Adjacency Function</h4>
<p>第三步是确定adjacency function的参数，对我们的流程来说，就是确定 <span class="math inline">\(\beta\)</span> 参数。为了确定参数，WGCNA提出了无标度拓扑网络（The Scale-free Topology）的概念。无标度网络在pathways, metabolic等生物数据上已经得到了验证，因此我们也假设基因的表达，微生物的生长关系也是符合无标度拓扑网络的概念。因此，当选择的参数需要满足无标度拓扑网络的构建。</p>
<p>什么是无标度网络？</p>
<div class="figure">
<img src="figure/scale_free_network.png" alt="" />
<p class="caption">scale-free topology</p>
</div>
<p>无标度网络的形象描述就是，有一个hub node，大部分的节点与其相连接，而这些节点本身连接较少。</p>
<p>怎么判断合适的 <span class="math inline">\(\beta\)</span>?</p>
<p>Earlier, we have described that he linear regression model fitting index <span class="math inline">\(R^2\)</span> can be used to quantify how well a network satisfies a scale-free topology. There is a natural trade-off between maximizing scale-free topology model fit(<span class="math inline">\(R^2\)</span>) and maintaining a high mean number of connections: parameter values that lead to an <span class="math inline">\(R^2\)</span> value close to 1 may lead to networks with very few connections.</p>
<div class="figure">
<img src="figure/select_beta.png" alt="" />
<p class="caption">select beta</p>
</div>
<p>如上图所示，当<span class="math inline">\(\beta\)</span>为6时，<span class="math inline">\(R^2\)</span>接近与1，但是任然保留较高的connections</p>
<hr />
</div>
<div id="step-four-defining-a-measure-of-node-dissimilarity" class="section level4">
<h4><span class="header-section-number">2.1.1.4</span> Step four: Defining a Measure of Node Dissimilarity</h4>
<p>第四步确定节点之间的不相似性距离。通过构建node之间的dissimilarity，然后用cluster方法可以将node（gene or others）划分到不同的module，同时也可以观察一个module中的node的连接紧密程度。</p>
<p>The topology overlap matrix (TOM) <span class="math inline">\(\Omega = [w_{ij}]\)</span></p>
<p><span class="math display">\[
w_{ij} = \frac{l_{ij} + a_{ij}}{min\{k_{i}, k_{j}\} + 1 - a_{ij}}
\]</span>
<span class="math inline">\(l_{ij} = \sum_{u}a_{iu}a_{uj}\)</span>, 表示i和j共享连接的u之间的关系乘的和</p>
<p><span class="math inline">\(k_{i} = \sum_{j=1}^na_{ij}\)</span>,表示i的所有连接和</p>
<p>dissimilarity measure is defined by <span class="math inline">\(d_{ij}^w = 1 - w_{ij}\)</span></p>
<hr />
</div>
<div id="step-five-identifying-gene-modules" class="section level4">
<h4><span class="header-section-number">2.1.1.5</span> Step five: Identifying Gene Modules</h4>
<p>第5步划分module。根据<span class="math inline">\(d_{ij}^w\)</span>, 用cluster方法划分module</p>
<div class="figure">
<img src="figure/split_modules.png" alt="" />
<p class="caption">split modules</p>
</div>
<p>a height cutoff value is chosen in the dendrogram such that some of the resulting branches correspond to dark squares(modules) along the diagonal of the TOM plot.</p>
<hr />
</div>
<div id="step-six-relating-network-concepts-to-each-other" class="section level4">
<h4><span class="header-section-number">2.1.1.6</span> Step six: Relating Network Concepts to Each Other</h4>
<p>第六步定义模块中的属性。</p>
<ol style="list-style-type: decimal">
<li>比如hub node。Each module can be represented by a centroid, e.g., an intramodular hub gene. Singular value decomposition</li>
</ol>
<p><span class="math display">\[
X^{(q)} = [x_{ij}^{(q)}] = (x_{1}^{(q)} \quad x_{2}^{(q)} \quad x_{3}^{(q)} \dots x_{n}^{(q)})^T \\
X^{(q)} = U^{(q)}D^{(q)}(V^{(q)})^T \\
V^{(q)} = (v_{1}^{(q)} \quad v_{2}^{(q)} \quad v_{3}^{(q)} \dots v_{m}^{(q)})
\]</span>
Module Eigengene:
<span class="math display">\[
E^{(q)} = v_{1}^{(q)}
\]</span>
2. module之间的correlation，如果两个module的correlation很高，可以合并两个module为一个。</p>
<hr />
</div>
<div id="step-seven-relating-network-concepts-to-external-gene-or-sample-information" class="section level4">
<h4><span class="header-section-number">2.1.1.7</span> Step seven: Relating Network Concepts to External Gene or Sample Information</h4>
<p>第七步，将module与外部数据相关联，找到感兴趣的module。</p>
<p>A main purpose of many network analysis is to relate a connectivity measure to external gene information. For example, In our cancer microarray application, we show that for brown module genes intramodular connectivity is highly correlated with prognostic significance for cancer surival.</p>
<div class="figure">
<img src="figure/relating_to_external_information.png" alt="" />
<p class="caption">relating to external information</p>
</div>
<p>如上图所示，临床数据（weight）与brown module最相关，表示这个module中的gene or node对weight有显著性的影响。</p>
<hr />
<p><strong>Users can choose different chapters to read according to their own data type: <a href="wgcna.html#wgcna-for-16s-data">16S</a>, <a href="wgcna.html#wgcna-for-mgs-data">MGS</a> and <a href="wgcna.html#wgcna-for-metabolic-data">metabolic data</a>.</strong> You can also [integrate the modules from different data type into a network.</p>
</div>
</div>
</div>
<div id="environment-setup" class="section level2">
<h2><span class="header-section-number">2.2</span> Environment Setup</h2>
<p>Seek help from Zhuzhengnong(<a href="mailto:zhuzhengnong@xbiome.com" class="email">zhuzhengnong@xbiome.com</a>) on any issues about the installation of R packages/dependencies in this tutorial.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode markdown"><code class="sourceCode markdown"><span id="cb1-1"><a href="wgcna.html#cb1-1"></a>library(tidyverse)</span>
<span id="cb1-2"><a href="wgcna.html#cb1-2"></a>library(xlsx)</span>
<span id="cb1-3"><a href="wgcna.html#cb1-3"></a>library(WGCNA)</span>
<span id="cb1-4"><a href="wgcna.html#cb1-4"></a>library(igraph)</span>
<span id="cb1-5"><a href="wgcna.html#cb1-5"></a>library(phyloseq)</span>
<span id="cb1-6"><a href="wgcna.html#cb1-6"></a>library(DT)</span>
<span id="cb1-7"><a href="wgcna.html#cb1-7"></a></span>
<span id="cb1-8"><a href="wgcna.html#cb1-8"></a>options(stringsAsFactors = FALSE)</span></code></pre></div>
</div>
<div id="wgcna-for-16s-data" class="section level2">
<h2><span class="header-section-number">2.3</span> WGCNA for 16S data</h2>
<p><strong>Data used in this sub-chapter belongs to SLE project, including 45 samples with 1364 ASVs.</strong></p>
<div id="data-loading-wgcna-16s" class="section level3">
<h3><span class="header-section-number">2.3.1</span> Data loading (WGCNA 16S)</h3>
<div id="read-counts-table-from-rds" class="section level4">
<h4><span class="header-section-number">2.3.1.1</span> Read counts table from rds</h4>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="wgcna.html#cb2-1"></a>seq_tab &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/SLE/data/dada2_res.rds&quot;</span>)<span class="op">$</span>seq_tab</span>
<span id="cb2-2"><a href="wgcna.html#cb2-2"></a><span class="co"># 将ASV替换为ASV1，ASV2，...</span></span>
<span id="cb2-3"><a href="wgcna.html#cb2-3"></a>ASV_index &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">ASV=</span><span class="kw">colnames</span>(seq_tab))</span>
<span id="cb2-4"><a href="wgcna.html#cb2-4"></a>ASV_index<span class="op">$</span>index &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ASV&quot;</span>,<span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">dim</span>(ASV_index)[<span class="dv">1</span>]))</span>
<span id="cb2-5"><a href="wgcna.html#cb2-5"></a><span class="kw">colnames</span>(seq_tab) &lt;-<span class="st"> </span>ASV_index<span class="op">$</span>index</span>
<span id="cb2-6"><a href="wgcna.html#cb2-6"></a></span>
<span id="cb2-7"><a href="wgcna.html#cb2-7"></a><span class="kw">datatable</span>(seq_tab[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>])</span></code></pre></div>
<div id="htmlwidget-6d04450b19ed71fe40cd" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6d04450b19ed71fe40cd">{"x":{"filter":"none","vertical":false,"data":[["6029","6030","6031","6032","6033","6034","6035","6036","6037","6038"],[5494,1247,4890,14616,2884,3700,0,3548,8483,2911],[20133,7103,286,2915,2184,4228,0,1231,19620,0],[4049,2794,1464,1552,1839,1129,11885,0,5632,259],[165,6983,1078,15950,1172,1173,255,12138,874,11685],[1077,0,48,2278,94,5383,13767,0,4453,680],[4372,2111,7518,0,16579,6182,0,4868,0,0],[0,0,6,0,0,0,0,0,6232,0],[0,489,129,0,199,0,0,815,0,2859],[0,757,3941,0,4056,0,0,1266,0,0],[5695,1042,2813,0,530,2091,0,469,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>ASV1<\/th>\n      <th>ASV2<\/th>\n      <th>ASV3<\/th>\n      <th>ASV4<\/th>\n      <th>ASV5<\/th>\n      <th>ASV6<\/th>\n      <th>ASV7<\/th>\n      <th>ASV8<\/th>\n      <th>ASV9<\/th>\n      <th>ASV10<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="read-taxa-table-from-rds" class="section level4">
<h4><span class="header-section-number">2.3.1.2</span> Read taxa table from rds</h4>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="wgcna.html#cb3-1"></a>taxa_tb &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&quot;/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/SLE/data/dada2_res.rds&quot;</span>)<span class="op">$</span>tax_tab</span>
<span id="cb3-2"><a href="wgcna.html#cb3-2"></a>taxa_tb &lt;-<span class="st"> </span><span class="kw">merge</span>(ASV_index, taxa_tb, <span class="dt">by.x =</span> <span class="st">&quot;ASV&quot;</span>, <span class="dt">by.y =</span> <span class="dv">0</span>)</span>
<span id="cb3-3"><a href="wgcna.html#cb3-3"></a>taxa_tb &lt;-<span class="st"> </span>taxa_tb[,<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span>)] <span class="co"># remove ASV column</span></span>
<span id="cb3-4"><a href="wgcna.html#cb3-4"></a>taxa_tb &lt;-<span class="st"> </span>taxa_tb <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;index&quot;</span>)</span>
<span id="cb3-5"><a href="wgcna.html#cb3-5"></a><span class="kw">datatable</span>(taxa_tb[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, ])</span></code></pre></div>
<div id="htmlwidget-ffd785b26fae2e4fcede" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ffd785b26fae2e4fcede">{"x":{"filter":"none","vertical":false,"data":[["ASV269","ASV552","ASV368","ASV624","ASV26"],["Bacteria","Bacteria","Bacteria","Bacteria","Bacteria"],["Firmicutes","Firmicutes","Firmicutes","Firmicutes","Firmicutes"],["Clostridia","Clostridia","Clostridia","Clostridia","Clostridia"],["Clostridiales","Clostridiales","Clostridiales","Clostridiales","Clostridiales"],["Peptococcaceae","Ruminococcaceae","Ruminococcaceae","Ruminococcaceae","Ruminococcaceae"],["Peptococcus","Subdoligranulum","Subdoligranulum","Subdoligranulum","Subdoligranulum"],[null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Kingdom<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n      <th>Genus<\/th>\n      <th>Species<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="remove-low-count-number-asvs-count-number-500" class="section level4">
<h4><span class="header-section-number">2.3.1.3</span> Remove low count-number ASVs (count number &lt; 500)</h4>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="wgcna.html#cb4-1"></a><span class="kw">dim</span>(seq_tab)</span></code></pre></div>
<pre><code>## [1]   45 1364</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="wgcna.html#cb6-1"></a>seq_tab_filter &lt;-<span class="st"> </span>seq_tab <span class="op">%&gt;%</span><span class="st"> </span>.[, <span class="kw">apply</span>(.<span class="op">&gt;</span><span class="dv">500</span>, <span class="dv">2</span>, any)]</span>
<span id="cb6-2"><a href="wgcna.html#cb6-2"></a><span class="kw">dim</span>(seq_tab_filter)</span></code></pre></div>
<pre><code>## [1]  45 231</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="wgcna.html#cb8-1"></a>taxa_tb &lt;-<span class="st"> </span>taxa_tb[<span class="kw">colnames</span>(seq_tab_filter),]</span>
<span id="cb8-2"><a href="wgcna.html#cb8-2"></a><span class="kw">dim</span>(taxa_tb)</span></code></pre></div>
<pre><code>## [1] 231   7</code></pre>
</div>
</div>
<div id="build-wgcna-network-wgcna-16s" class="section level3">
<h3><span class="header-section-number">2.3.2</span> Build WGCNA network (WGCNA 16S)</h3>
<div id="set-parameters-wgcna-16s" class="section level4">
<h4><span class="header-section-number">2.3.2.1</span> Set parameters (WGCNA 16S)</h4>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="wgcna.html#cb10-1"></a><span class="co"># Set seed as a precaution for reproducibility as some methods are non-deterministic.</span></span>
<span id="cb10-2"><a href="wgcna.html#cb10-2"></a><span class="kw">set.seed</span>(<span class="dv">13118</span>)</span>
<span id="cb10-3"><a href="wgcna.html#cb10-3"></a>Run_analysis &lt;-<span class="st"> </span><span class="ot">TRUE</span> <span class="co"># if FALSE it tries to load data instead of running</span></span>
<span id="cb10-4"><a href="wgcna.html#cb10-4"></a></span>
<span id="cb10-5"><a href="wgcna.html#cb10-5"></a>save_plots &lt;-<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb10-6"><a href="wgcna.html#cb10-6"></a>plot_labeling_size =<span class="st"> </span><span class="dv">15</span></span>
<span id="cb10-7"><a href="wgcna.html#cb10-7"></a>prefix &lt;-<span class="st"> &quot;m&quot;</span></span>
<span id="cb10-8"><a href="wgcna.html#cb10-8"></a></span>
<span id="cb10-9"><a href="wgcna.html#cb10-9"></a>save_TOM &lt;-<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb10-10"><a href="wgcna.html#cb10-10"></a>pam_stage &lt;-<span class="st"> </span><span class="ot">FALSE</span>    <span class="co"># Partitioning around medoids, tries to put more OTUs into modules where it is not directly clear from the dendrogram cutting.</span></span>
<span id="cb10-11"><a href="wgcna.html#cb10-11"></a>set_verbose =<span class="st"> </span><span class="dv">1</span> <span class="co"># How much detail from the WGCNA functions? Higher means more detail.</span></span>
<span id="cb10-12"><a href="wgcna.html#cb10-12"></a></span>
<span id="cb10-13"><a href="wgcna.html#cb10-13"></a>parameter_sets &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">set_1 =</span> <span class="kw">list</span>(<span class="dt">applied_norm =</span> <span class="st">&quot;TSS&quot;</span>, <span class="dt">applied_transf =</span> <span class="st">&quot;CLR&quot;</span>, <span class="dt">assoc_measure =</span> <span class="st">&quot;bicor&quot;</span>),</span>
<span id="cb10-14"><a href="wgcna.html#cb10-14"></a>                       <span class="dt">set_2 =</span> <span class="kw">list</span>(<span class="dt">applied_norm =</span> <span class="st">&quot;CSS&quot;</span>, <span class="dt">applied_transf =</span> <span class="st">&quot;log2&quot;</span>, <span class="dt">assoc_measure =</span> <span class="st">&quot;bicor&quot;</span>))</span>
<span id="cb10-15"><a href="wgcna.html#cb10-15"></a></span>
<span id="cb10-16"><a href="wgcna.html#cb10-16"></a>chosen_parameter_set &lt;-<span class="st"> </span>parameter_sets<span class="op">$</span>set_<span class="dv">2</span></span>
<span id="cb10-17"><a href="wgcna.html#cb10-17"></a></span>
<span id="cb10-18"><a href="wgcna.html#cb10-18"></a>pcCorrection &lt;-<span class="st"> </span>F</span>
<span id="cb10-19"><a href="wgcna.html#cb10-19"></a><span class="cf">if</span>(pcCorrection){</span>
<span id="cb10-20"><a href="wgcna.html#cb10-20"></a>  estimate_n.pc =<span class="st"> </span>F</span>
<span id="cb10-21"><a href="wgcna.html#cb10-21"></a>  <span class="cf">if</span>(<span class="op">!</span>estimate_n.pc){</span>
<span id="cb10-22"><a href="wgcna.html#cb10-22"></a>    number_of_pcs_to_remove =<span class="st"> </span><span class="dv">0</span> <span class="co"># Does not matter when pcCorrection is FALSE</span></span>
<span id="cb10-23"><a href="wgcna.html#cb10-23"></a>  }</span>
<span id="cb10-24"><a href="wgcna.html#cb10-24"></a>}</span>
<span id="cb10-25"><a href="wgcna.html#cb10-25"></a></span>
<span id="cb10-26"><a href="wgcna.html#cb10-26"></a>save_name &lt;-<span class="st"> &quot;16S_network_module&quot;</span></span></code></pre></div>
</div>
<div id="data-normalization-wgcna-16s" class="section level4">
<h4><span class="header-section-number">2.3.2.2</span> Data normalization (WGCNA 16S)</h4>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="wgcna.html#cb11-1"></a><span class="cf">if</span>(chosen_parameter_set<span class="op">$</span>applied_norm <span class="op">==</span><span class="st"> &quot;TSS&quot;</span>){</span>
<span id="cb11-2"><a href="wgcna.html#cb11-2"></a>  <span class="co"># Normalize</span></span>
<span id="cb11-3"><a href="wgcna.html#cb11-3"></a>  seq_tab_filter &lt;-<span class="st"> </span><span class="kw">apply</span>(seq_tab_filter, <span class="dt">MARGIN =</span> <span class="dv">2</span>, <span class="cf">function</span>(x){x<span class="op">/</span><span class="kw">sum</span>(x) <span class="op">*</span><span class="st"> </span><span class="dv">100</span>})</span>
<span id="cb11-4"><a href="wgcna.html#cb11-4"></a>  seq_tab_filter &lt;-<span class="st"> </span>propr<span class="op">:::</span><span class="kw">proprCLR</span>(<span class="kw">t</span>(seq_tab_filter) <span class="op">+</span><span class="st"> </span><span class="dv">1</span>) <span class="co"># t(), samples as rows, variables as columns</span></span>
<span id="cb11-5"><a href="wgcna.html#cb11-5"></a>  seq_tab_filter &lt;-<span class="st"> </span><span class="kw">t</span>(seq_tab_filter)</span>
<span id="cb11-6"><a href="wgcna.html#cb11-6"></a>}</span>
<span id="cb11-7"><a href="wgcna.html#cb11-7"></a></span>
<span id="cb11-8"><a href="wgcna.html#cb11-8"></a><span class="cf">if</span>(chosen_parameter_set<span class="op">$</span>applied_norm <span class="op">==</span><span class="st"> &quot;CSS&quot;</span>){</span>
<span id="cb11-9"><a href="wgcna.html#cb11-9"></a>  <span class="co"># Normalization and transformation in one</span></span>
<span id="cb11-10"><a href="wgcna.html#cb11-10"></a>  data.metagenomeSeq &lt;-<span class="st"> </span>metagenomeSeq<span class="op">::</span><span class="kw">newMRexperiment</span>(seq_tab_filter) <span class="co"># Variables as rows, samples as columns</span></span>
<span id="cb11-11"><a href="wgcna.html#cb11-11"></a>  p &lt;-<span class="st"> </span>metagenomeSeq<span class="op">::</span><span class="kw">cumNormStat</span>(data.metagenomeSeq)</span>
<span id="cb11-12"><a href="wgcna.html#cb11-12"></a>  data.cumnorm &lt;-<span class="st"> </span>metagenomeSeq<span class="op">::</span><span class="kw">cumNorm</span>(data.metagenomeSeq, <span class="dt">p=</span>p)</span>
<span id="cb11-13"><a href="wgcna.html#cb11-13"></a>  seq_tab_filter &lt;-<span class="st"> </span>metagenomeSeq<span class="op">::</span><span class="kw">MRcounts</span>(data.cumnorm, <span class="dt">norm=</span><span class="ot">TRUE</span>, <span class="dt">log=</span><span class="ot">TRUE</span>) <span class="co"># log here is a +1 shifted log2</span></span>
<span id="cb11-14"><a href="wgcna.html#cb11-14"></a>}</span></code></pre></div>
<pre><code>## Default value being used.</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="wgcna.html#cb13-1"></a>seq_tab_filter &lt;-<span class="st"> </span>seq_tab_filter</span>
<span id="cb13-2"><a href="wgcna.html#cb13-2"></a></span>
<span id="cb13-3"><a href="wgcna.html#cb13-3"></a><span class="kw">dim</span>(seq_tab_filter) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">paste</span>(<span class="kw">c</span>(<span class="st">&quot;Samples&quot;</span>, <span class="st">&quot;OTUs&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;45 Samples&quot; &quot;231 OTUs&quot;</code></pre>
</div>
<div id="select-the-best-power-parameter-for-soft-threshold" class="section level4">
<h4><span class="header-section-number">2.3.2.3</span> Select the best power parameter for Soft threshold</h4>
<p><strong>Before running this chunk and the the chunks onwards, please make sure that your data has been <a href="#remove-low-count-number-asvs-count-number--500">filterd</a> and <a href="wgcna.html#data-normalization-wgcna-16s">normalized</a>. Also make sure that samples are on row and features are on column in your input profile table</strong></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="wgcna.html#cb15-1"></a>powers &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">seq</span>(<span class="dv">12</span>,<span class="dv">20</span>,<span class="dv">2</span>))</span>
<span id="cb15-2"><a href="wgcna.html#cb15-2"></a></span>
<span id="cb15-3"><a href="wgcna.html#cb15-3"></a><span class="kw">suppressWarnings</span>(sft &lt;-<span class="st"> </span><span class="kw">pickSoftThreshold</span>(seq_tab_filter, </span>
<span id="cb15-4"><a href="wgcna.html#cb15-4"></a>                                          <span class="dt">powerVector =</span> powers, </span>
<span id="cb15-5"><a href="wgcna.html#cb15-5"></a>                                          <span class="dt">verbose =</span> set_verbose, </span>
<span id="cb15-6"><a href="wgcna.html#cb15-6"></a>                                          <span class="dt">networkType =</span> <span class="st">&quot;signed&quot;</span>,</span>
<span id="cb15-7"><a href="wgcna.html#cb15-7"></a>                                          <span class="dt">corFn=</span> chosen_parameter_set<span class="op">$</span>assoc_measure))</span></code></pre></div>
<pre><code>## pickSoftThreshold: will use block size 231.
##  pickSoftThreshold: calculating connectivity for given powers... ..0% ..100% 
##    Power SFT.R.sq slope truncated.R.sq mean.k. median.k. max.k.
## 1      1   0.6140 11.20         0.9500 118.000  119.0000 129.00
## 2      2   0.0977  1.59         0.8770  63.000   62.8000  76.10
## 3      3   0.1530 -1.20         0.7990  34.800   34.2000  46.90
## 4      4   0.2970 -1.44         0.8540  19.900   19.2000  30.00
## 5      5   0.2820 -1.17         0.8240  11.900   11.3000  19.70
## 6      6   0.3040 -1.02         0.9330   7.370    7.0000  13.70
## 7      7   0.3360 -1.00         0.9280   4.760    4.4200   9.77
## 8      8   0.4560 -1.09         0.8500   3.210    2.8400   7.18
## 9      9   0.5630 -1.07         0.7710   2.250    1.9100   5.49
## 10    10   0.7380 -1.33         0.8710   1.640    1.3400   4.85
## 11    12   0.1900 -3.14        -0.0251   0.977    0.6970   4.12
## 12    14   0.1970 -2.85        -0.0237   0.663    0.3710   3.73
## 13    16   0.2850 -4.10         0.1900   0.499    0.2150   3.50
## 14    18   0.2450 -4.12         0.0383   0.407    0.1220   3.36
## 15    20   0.2480 -3.81         0.0410   0.351    0.0795   3.26</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="wgcna.html#cb17-1"></a><span class="co"># Find the soft thresholding power beta to which co-expression similarity is raised to calculate adjacency.</span></span>
<span id="cb17-2"><a href="wgcna.html#cb17-2"></a><span class="co"># based on the criterion of approximate scale-free topology.</span></span>
<span id="cb17-3"><a href="wgcna.html#cb17-3"></a></span>
<span id="cb17-4"><a href="wgcna.html#cb17-4"></a>idx &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">which</span>((<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.90</span>))</span>
<span id="cb17-5"><a href="wgcna.html#cb17-5"></a><span class="cf">if</span>(<span class="kw">is.infinite</span>(idx)){</span>
<span id="cb17-6"><a href="wgcna.html#cb17-6"></a>  idx &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">which</span>((<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.80</span>))</span>
<span id="cb17-7"><a href="wgcna.html#cb17-7"></a>  <span class="cf">if</span>(<span class="op">!</span><span class="kw">is.infinite</span>(idx)){</span>
<span id="cb17-8"><a href="wgcna.html#cb17-8"></a>    st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb17-9"><a href="wgcna.html#cb17-9"></a>  } <span class="cf">else</span>{</span>
<span id="cb17-10"><a href="wgcna.html#cb17-10"></a>    idx &lt;-<span class="st"> </span><span class="kw">which.max</span>(<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>])</span>
<span id="cb17-11"><a href="wgcna.html#cb17-11"></a>    st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb17-12"><a href="wgcna.html#cb17-12"></a>  }</span>
<span id="cb17-13"><a href="wgcna.html#cb17-13"></a>} <span class="cf">else</span>{</span>
<span id="cb17-14"><a href="wgcna.html#cb17-14"></a>  st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb17-15"><a href="wgcna.html#cb17-15"></a>}</span>
<span id="cb17-16"><a href="wgcna.html#cb17-16"></a></span>
<span id="cb17-17"><a href="wgcna.html#cb17-17"></a></span>
<span id="cb17-18"><a href="wgcna.html#cb17-18"></a><span class="co"># Plot Scale independence measure and Mean connectivity measure</span></span>
<span id="cb17-19"><a href="wgcna.html#cb17-19"></a></span>
<span id="cb17-20"><a href="wgcna.html#cb17-20"></a><span class="co"># Scale-free topology fit index as a function of the soft-thresholding power</span></span>
<span id="cb17-21"><a href="wgcna.html#cb17-21"></a><span class="kw">data.frame</span>(<span class="dt">Indices =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb17-22"><a href="wgcna.html#cb17-22"></a>           <span class="dt">sfApprox =</span> <span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb17-23"><a href="wgcna.html#cb17-23"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb17-24"><a href="wgcna.html#cb17-24"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="fl">0.9</span>, <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.6</span>) <span class="op">+</span><span class="st"> </span><span class="co"># corresponds to R^2 cut-off of 0.9</span></span>
<span id="cb17-25"><a href="wgcna.html#cb17-25"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="fl">0.8</span>, <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.2</span>) <span class="op">+</span><span class="st"> </span><span class="co"># corresponds to R^2 cut-off of 0.8</span></span>
<span id="cb17-26"><a href="wgcna.html#cb17-26"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> sfApprox), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb17-27"><a href="wgcna.html#cb17-27"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> sfApprox, <span class="dt">label =</span> Indices), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb17-28"><a href="wgcna.html#cb17-28"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Scale independence&quot;</span>) <span class="op">+</span></span>
<span id="cb17-29"><a href="wgcna.html#cb17-29"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Soft Threshold (power)&quot;</span>) <span class="op">+</span></span>
<span id="cb17-30"><a href="wgcna.html#cb17-30"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;SF Model Fit,signed R^2&quot;</span>) <span class="op">+</span></span>
<span id="cb17-31"><a href="wgcna.html#cb17-31"></a><span class="st">  </span><span class="kw">xlim</span>(<span class="dv">1</span>,<span class="dv">20</span>) <span class="op">+</span></span>
<span id="cb17-32"><a href="wgcna.html#cb17-32"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb17-33"><a href="wgcna.html#cb17-33"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> st, <span class="dt">y =</span> <span class="fl">0.25</span>, <span class="dt">xend =</span> st, <span class="dt">yend =</span> sfApprox[idx]<span class="op">-</span><span class="fl">0.05</span>), </span>
<span id="cb17-34"><a href="wgcna.html#cb17-34"></a>               <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.2</span>,<span class="st">&quot;cm&quot;</span>)), </span>
<span id="cb17-35"><a href="wgcna.html#cb17-35"></a>               <span class="dt">size =</span> <span class="fl">0.5</span>)-&gt;<span class="st"> </span>scale_independence_plot </span>
<span id="cb17-36"><a href="wgcna.html#cb17-36"></a>  </span>
<span id="cb17-37"><a href="wgcna.html#cb17-37"></a> </span>
<span id="cb17-38"><a href="wgcna.html#cb17-38"></a></span>
<span id="cb17-39"><a href="wgcna.html#cb17-39"></a></span>
<span id="cb17-40"><a href="wgcna.html#cb17-40"></a><span class="co"># Mean connectivity as a function of the soft-thresholding power</span></span>
<span id="cb17-41"><a href="wgcna.html#cb17-41"></a></span>
<span id="cb17-42"><a href="wgcna.html#cb17-42"></a><span class="kw">data.frame</span>(<span class="dt">Indices =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb17-43"><a href="wgcna.html#cb17-43"></a>           <span class="dt">meanApprox =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">5</span>]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb17-44"><a href="wgcna.html#cb17-44"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb17-45"><a href="wgcna.html#cb17-45"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> meanApprox), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb17-46"><a href="wgcna.html#cb17-46"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> meanApprox, <span class="dt">label =</span> Indices), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb17-47"><a href="wgcna.html#cb17-47"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Soft Threshold (power)&quot;</span>) <span class="op">+</span></span>
<span id="cb17-48"><a href="wgcna.html#cb17-48"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Mean Connectivity&quot;</span>) <span class="op">+</span></span>
<span id="cb17-49"><a href="wgcna.html#cb17-49"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> st<span class="fl">-0.4</span>, </span>
<span id="cb17-50"><a href="wgcna.html#cb17-50"></a>                   <span class="dt">y =</span> sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx], </span>
<span id="cb17-51"><a href="wgcna.html#cb17-51"></a>                   <span class="dt">xend =</span> <span class="dv">0</span>, </span>
<span id="cb17-52"><a href="wgcna.html#cb17-52"></a>                   <span class="dt">yend =</span> sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx]),</span>
<span id="cb17-53"><a href="wgcna.html#cb17-53"></a>               <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.2</span>,<span class="st">&quot;cm&quot;</span>)), </span>
<span id="cb17-54"><a href="wgcna.html#cb17-54"></a>               <span class="dt">size =</span> <span class="fl">0.4</span>) <span class="op">+</span></span>
<span id="cb17-55"><a href="wgcna.html#cb17-55"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;Mean connectivity: &quot;</span>, </span>
<span id="cb17-56"><a href="wgcna.html#cb17-56"></a>                 <span class="kw">round</span>(sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx],<span class="dv">2</span>))) -&gt;<span class="st"> </span>mean_connectivity_plot</span>
<span id="cb17-57"><a href="wgcna.html#cb17-57"></a></span>
<span id="cb17-58"><a href="wgcna.html#cb17-58"></a></span>
<span id="cb17-59"><a href="wgcna.html#cb17-59"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(scale_independence_plot, mean_connectivity_plot, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">align =</span> <span class="st">&quot;h&quot;</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>), <span class="dt">label_size =</span> plot_labeling_size) -&gt;<span class="st"> </span>si_mc_plot</span>
<span id="cb17-60"><a href="wgcna.html#cb17-60"></a></span>
<span id="cb17-61"><a href="wgcna.html#cb17-61"></a>si_mc_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
</div>
</div>
<div id="block-wise-network-construction-and-module-detection" class="section level3">
<h3><span class="header-section-number">2.3.3</span> Block-wise network construction and module detection</h3>
<p>The function <code>blockwiseModules</code> will first pre cluster with fast crude clustering method to cluster OTUs into blocks not exceeding the maximum, blocks may therefore not be fully optimal in the end.</p>
<p>Adjust the parameters to fit your own data.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="wgcna.html#cb18-1"></a>Run_analysis &lt;-<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb18-2"><a href="wgcna.html#cb18-2"></a><span class="cf">if</span>(Run_analysis){</span>
<span id="cb18-3"><a href="wgcna.html#cb18-3"></a>  modules.otu &lt;-<span class="st"> </span><span class="kw">blockwiseModules</span>(seq_tab_filter,</span>
<span id="cb18-4"><a href="wgcna.html#cb18-4"></a>                          <span class="dt">power =</span> st, </span>
<span id="cb18-5"><a href="wgcna.html#cb18-5"></a>                          <span class="dt">networkType =</span> <span class="st">&quot;signed&quot;</span>, </span>
<span id="cb18-6"><a href="wgcna.html#cb18-6"></a>                          <span class="dt">TOMType =</span> <span class="st">&quot;signed&quot;</span>,</span>
<span id="cb18-7"><a href="wgcna.html#cb18-7"></a>                          <span class="dt">corType =</span> chosen_parameter_set<span class="op">$</span>assoc_measure,</span>
<span id="cb18-8"><a href="wgcna.html#cb18-8"></a>                          <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>,</span>
<span id="cb18-9"><a href="wgcna.html#cb18-9"></a>                          <span class="dt">deepSplit =</span> <span class="dv">2</span>,</span>
<span id="cb18-10"><a href="wgcna.html#cb18-10"></a>                          <span class="dt">pamStage =</span> pam_stage, </span>
<span id="cb18-11"><a href="wgcna.html#cb18-11"></a>                          <span class="dt">pamRespectsDendro =</span> <span class="ot">TRUE</span>,</span>
<span id="cb18-12"><a href="wgcna.html#cb18-12"></a>                          <span class="dt">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb18-13"><a href="wgcna.html#cb18-13"></a>                          <span class="dt">replaceMissingAdjacencies =</span> <span class="ot">TRUE</span>,</span>
<span id="cb18-14"><a href="wgcna.html#cb18-14"></a>                          <span class="dt">minModuleSize =</span> <span class="dv">5</span>, <span class="co"># There are fewer otus than genes, and that many might not be connected</span></span>
<span id="cb18-15"><a href="wgcna.html#cb18-15"></a>                          <span class="dt">numericLabels =</span> <span class="ot">TRUE</span>,</span>
<span id="cb18-16"><a href="wgcna.html#cb18-16"></a>                          <span class="dt">saveTOMs =</span> save_TOM,</span>
<span id="cb18-17"><a href="wgcna.html#cb18-17"></a>                          <span class="dt">saveTOMFileBase =</span> <span class="kw">paste0</span>(<span class="st">&quot;/share/projects/Analytics/analytics/MultiOmics/tools/Bookdown/result/&quot;</span>, save_name),</span>
<span id="cb18-18"><a href="wgcna.html#cb18-18"></a>                          <span class="dt">verbose =</span> set_verbose)</span>
<span id="cb18-19"><a href="wgcna.html#cb18-19"></a>  </span>
<span id="cb18-20"><a href="wgcna.html#cb18-20"></a>  <span class="kw">rownames</span>(modules.otu<span class="op">$</span>MEs) &lt;-<span class="st"> </span><span class="kw">rownames</span>(seq_tab_filter)</span>
<span id="cb18-21"><a href="wgcna.html#cb18-21"></a>  <span class="kw">names</span>(modules.otu<span class="op">$</span>colors) &lt;-<span class="st"> </span><span class="kw">colnames</span>(seq_tab_filter)</span>
<span id="cb18-22"><a href="wgcna.html#cb18-22"></a>  <span class="kw">names</span>(modules.otu<span class="op">$</span>unmergedColors) &lt;-<span class="st"> </span><span class="kw">colnames</span>(seq_tab_filter)</span>
<span id="cb18-23"><a href="wgcna.html#cb18-23"></a>  </span>
<span id="cb18-24"><a href="wgcna.html#cb18-24"></a>  hubs &lt;-<span class="st"> </span><span class="kw">chooseTopHubInEachModule</span>(seq_tab_filter, modules.otu<span class="op">$</span>colors)</span>
<span id="cb18-25"><a href="wgcna.html#cb18-25"></a>}</span></code></pre></div>
<pre><code>##  Calculating module eigengenes block-wise from all genes</code></pre>
<div id="visualizing-module-characteristics" class="section level4">
<h4><span class="header-section-number">2.3.3.1</span> Visualizing module characteristics</h4>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="wgcna.html#cb20-1"></a><span class="co"># Convert labels to colors for plotting</span></span>
<span id="cb20-2"><a href="wgcna.html#cb20-2"></a>merged_colors &lt;-<span class="st"> </span><span class="kw">labels2colors</span>(modules.otu<span class="op">$</span>colors)</span>
<span id="cb20-3"><a href="wgcna.html#cb20-3"></a></span>
<span id="cb20-4"><a href="wgcna.html#cb20-4"></a>n_modules &lt;-<span class="st"> </span><span class="kw">unique</span>(merged_colors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">length</span>()</span>
<span id="cb20-5"><a href="wgcna.html#cb20-5"></a></span>
<span id="cb20-6"><a href="wgcna.html#cb20-6"></a>samples_good &lt;-<span class="st"> </span><span class="kw">sum</span>(modules.otu<span class="op">$</span>goodSamples) <span class="op">==</span><span class="st"> </span><span class="kw">length</span>(modules.otu<span class="op">$</span>goodSamples)</span>
<span id="cb20-7"><a href="wgcna.html#cb20-7"></a>OTUs_good &lt;-<span class="st"> </span><span class="kw">sum</span>(modules.otu<span class="op">$</span>goodGenes) <span class="op">==</span><span class="st"> </span><span class="kw">length</span>(modules.otu<span class="op">$</span>goodGenes)</span>
<span id="cb20-8"><a href="wgcna.html#cb20-8"></a></span>
<span id="cb20-9"><a href="wgcna.html#cb20-9"></a>ME_good &lt;-<span class="st"> </span><span class="kw">sum</span>(modules.otu<span class="op">$</span>MEsOK) <span class="op">==</span><span class="st"> </span><span class="kw">length</span>(modules.otu<span class="op">$</span>MEsOK)</span></code></pre></div>
</div>
<div id="display-otus-in-each-module" class="section level4">
<h4><span class="header-section-number">2.3.3.2</span> Display OTUs in each module</h4>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="wgcna.html#cb21-1"></a><span class="kw">table</span>(modules.otu<span class="op">$</span>colors) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb21-2"><a href="wgcna.html#cb21-2"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb21-3"><a href="wgcna.html#cb21-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">Module =</span> Var1, <span class="dt">Size =</span> Freq) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb21-4"><a href="wgcna.html#cb21-4"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Module_color =</span> <span class="kw">labels2colors</span>(<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(Module)))) -&gt;<span class="st"> </span>module_size</span>
<span id="cb21-5"><a href="wgcna.html#cb21-5"></a></span>
<span id="cb21-6"><a href="wgcna.html#cb21-6"></a>module_size <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb21-7"><a href="wgcna.html#cb21-7"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Module, <span class="dt">y =</span> Size, <span class="dt">fill =</span> Module)) <span class="op">+</span></span>
<span id="cb21-8"><a href="wgcna.html#cb21-8"></a><span class="st">  </span><span class="kw">geom_col</span>(<span class="dt">color =</span>  <span class="st">&quot;#000000&quot;</span>) <span class="op">+</span></span>
<span id="cb21-9"><a href="wgcna.html#cb21-9"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Number of OTUs in each module&quot;</span>) <span class="op">+</span></span>
<span id="cb21-10"><a href="wgcna.html#cb21-10"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb21-11"><a href="wgcna.html#cb21-11"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">setNames</span>(module_size<span class="op">$</span>Module_color,module_size<span class="op">$</span>Module)) <span class="op">+</span></span>
<span id="cb21-12"><a href="wgcna.html#cb21-12"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label =</span> Size),<span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">hjust =</span> <span class="fl">-0.18</span>, <span class="dt">size =</span> <span class="fl">3.5</span>) <span class="op">+</span></span>
<span id="cb21-13"><a href="wgcna.html#cb21-13"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="dv">0</span>, <span class="kw">max</span>(module_size<span class="op">$</span>Size)<span class="op">*</span><span class="fl">1.1</span>) <span class="op">+</span></span>
<span id="cb21-14"><a href="wgcna.html#cb21-14"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">margin</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="st">&quot;pt&quot;</span>)) <span class="op">+</span></span>
<span id="cb21-15"><a href="wgcna.html#cb21-15"></a><span class="st">  </span><span class="kw">coord_flip</span>()-&gt;<span class="st"> </span>module_size_barplot</span>
<span id="cb21-16"><a href="wgcna.html#cb21-16"></a></span>
<span id="cb21-17"><a href="wgcna.html#cb21-17"></a>module_size_barplot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/How%20many%20OTUs%20are%20there%20in%20each%20module-1.png" width="480" /></p>
</div>
<div id="extract-modules-in-the-network" class="section level4">
<h4><span class="header-section-number">2.3.3.3</span> Extract modules in the network</h4>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="wgcna.html#cb22-1"></a><span class="kw">plotEigengeneNetworks</span>(modules.otu<span class="op">$</span>MEs, <span class="st">&quot;Eigengene adjacency heatmap&quot;</span>,</span>
<span id="cb22-2"><a href="wgcna.html#cb22-2"></a><span class="dt">marDendro =</span> <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">4</span>),</span>
<span id="cb22-3"><a href="wgcna.html#cb22-3"></a><span class="dt">marHeatmap =</span> <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>), <span class="dt">plotDendrograms =</span> T,</span>
<span id="cb22-4"><a href="wgcna.html#cb22-4"></a><span class="dt">xLabelsAngle =</span> <span class="dv">90</span>)</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Extract%20modules%20in%20the%20network-1.png" width="672" /></p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="wgcna.html#cb23-1"></a><span class="co">## Show module info</span></span>
<span id="cb23-2"><a href="wgcna.html#cb23-2"></a><span class="kw">table</span>(modules.otu<span class="op">$</span>colors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() -&gt;<span class="st"> </span>res</span>
<span id="cb23-3"><a href="wgcna.html#cb23-3"></a>res<span class="op">$</span><span class="st">`</span><span class="dt">Module color</span><span class="st">`</span> &lt;-<span class="st"> </span>WGCNA<span class="op">::</span><span class="kw">labels2colors</span>(<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(res<span class="op">$</span>Var1)))</span>
<span id="cb23-4"><a href="wgcna.html#cb23-4"></a>res &lt;-<span class="st"> </span>res[, <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>)]</span>
<span id="cb23-5"><a href="wgcna.html#cb23-5"></a><span class="kw">colnames</span>(res) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Module&quot;</span>, <span class="st">&quot;Module color&quot;</span>, <span class="st">&quot;Number of OTUs&quot;</span>)</span>
<span id="cb23-6"><a href="wgcna.html#cb23-6"></a><span class="kw">print</span>(res)</span></code></pre></div>
<pre><code>##   Module Module color Number of OTUs
## 1      0         grey            141
## 2      1    turquoise             58
## 3      2         blue             10
## 4      3        brown              8
## 5      4       yellow              8
## 6      5        green              6</code></pre>
</div>
<div id="dendrogram-of-modules" class="section level4">
<h4><span class="header-section-number">2.3.3.4</span> Dendrogram of modules</h4>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="wgcna.html#cb25-1"></a><span class="co"># Plot the dendrogram and the module colors underneath for each block</span></span>
<span id="cb25-2"><a href="wgcna.html#cb25-2"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="kw">seq_along</span>(modules.otu<span class="op">$</span>dendrograms)){</span>
<span id="cb25-3"><a href="wgcna.html#cb25-3"></a>  <span class="kw">plotDendroAndColors</span>(modules.otu<span class="op">$</span>dendrograms[[i]], merged_colors[modules.otu<span class="op">$</span>blockGenes[[i]]],</span>
<span id="cb25-4"><a href="wgcna.html#cb25-4"></a>                      <span class="st">&quot;Module colors&quot;</span>,</span>
<span id="cb25-5"><a href="wgcna.html#cb25-5"></a>                      <span class="dt">dendroLabels =</span> <span class="ot">FALSE</span>, <span class="dt">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb25-6"><a href="wgcna.html#cb25-6"></a>                      <span class="dt">addGuide =</span> <span class="ot">TRUE</span>, <span class="dt">guideHang =</span> <span class="fl">0.05</span>,</span>
<span id="cb25-7"><a href="wgcna.html#cb25-7"></a>                      <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Cluster Dendrogram</span><span class="ch">\n</span><span class="st">&quot;</span>, </span>
<span id="cb25-8"><a href="wgcna.html#cb25-8"></a>                                    <span class="st">&quot;for block &quot;</span>, </span>
<span id="cb25-9"><a href="wgcna.html#cb25-9"></a>                                    i,<span class="st">&quot;: &quot;</span>,</span>
<span id="cb25-10"><a href="wgcna.html#cb25-10"></a>                                    <span class="kw">length</span>(modules.otu<span class="op">$</span>blockGenes[[i]]),</span>
<span id="cb25-11"><a href="wgcna.html#cb25-11"></a>                                    <span class="st">&quot; OTUs&quot;</span>))</span>
<span id="cb25-12"><a href="wgcna.html#cb25-12"></a>}</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Dendrogram%20of%20modules-1.png" width="960" /></p>
</div>
</div>
<div id="module-eigengene-correlation" class="section level3">
<h3><span class="header-section-number">2.3.4</span> Module (Eigengene) correlation</h3>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="wgcna.html#cb26-1"></a>MEs &lt;-<span class="st"> </span>modules.otu<span class="op">$</span>MEs</span>
<span id="cb26-2"><a href="wgcna.html#cb26-2"></a></span>
<span id="cb26-3"><a href="wgcna.html#cb26-3"></a><span class="co"># Module correlation to other modules</span></span>
<span id="cb26-4"><a href="wgcna.html#cb26-4"></a>MEs_R &lt;-<span class="st"> </span><span class="kw">bicor</span>(MEs, MEs, <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## Warning in bicor(MEs, MEs, maxPOutliers = 0.05): bicor: zero MAD in variable &#39;x&#39;. Pearson correlation was used for individual columns with zero (or missing) MAD.</code></pre>
<pre><code>## Warning in bicor(MEs, MEs, maxPOutliers = 0.05): bicor: zero MAD in variable &#39;y&#39;. Pearson correlation was used for individual columns with zero (or missing) MAD.</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="wgcna.html#cb29-1"></a>idx.r &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rownames</span>(MEs_R) <span class="op">==</span><span class="st"> &quot;ME0&quot;</span>)</span>
<span id="cb29-2"><a href="wgcna.html#cb29-2"></a>idx.c &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(MEs_R) <span class="op">==</span><span class="st"> &quot;ME0&quot;</span>)</span>
<span id="cb29-3"><a href="wgcna.html#cb29-3"></a></span>
<span id="cb29-4"><a href="wgcna.html#cb29-4"></a>MEs_R_noME0 &lt;-<span class="st"> </span>MEs_R[<span class="op">-</span>idx.r, <span class="op">-</span>idx.c]</span>
<span id="cb29-5"><a href="wgcna.html#cb29-5"></a></span>
<span id="cb29-6"><a href="wgcna.html#cb29-6"></a>MEs_R[<span class="kw">upper.tri</span>(MEs_R_noME0)] <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb29-7"><a href="wgcna.html#cb29-7"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb29-8"><a href="wgcna.html#cb29-8"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="st">&quot;correlation&quot;</span> =<span class="st"> &quot;.&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb29-9"><a href="wgcna.html#cb29-9"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>correlation)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb29-10"><a href="wgcna.html#cb29-10"></a><span class="st">  </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb29-11"><a href="wgcna.html#cb29-11"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(prefix,<span class="st">&quot;ME correlation density</span><span class="ch">\n</span><span class="st"> without &quot;</span>,prefix ,<span class="st">&quot;ME0&quot;</span>)) -&gt;<span class="st"> </span>MEs_R_density</span>
<span id="cb29-12"><a href="wgcna.html#cb29-12"></a></span>
<span id="cb29-13"><a href="wgcna.html#cb29-13"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(MEs_R, <span class="dt">color =</span> <span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;Blue&quot;</span>, <span class="st">&quot;White&quot;</span>, <span class="st">&quot;Red&quot;</span>))(<span class="dv">100</span>),</span>
<span id="cb29-14"><a href="wgcna.html#cb29-14"></a>                   <span class="dt">silent =</span> T, </span>
<span id="cb29-15"><a href="wgcna.html#cb29-15"></a>                   <span class="dt">breaks =</span> <span class="kw">seq</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>,<span class="dt">length.out =</span> <span class="dv">101</span>),</span>
<span id="cb29-16"><a href="wgcna.html#cb29-16"></a>                   <span class="dt">treeheight_row =</span> <span class="dv">5</span>, </span>
<span id="cb29-17"><a href="wgcna.html#cb29-17"></a>                   <span class="dt">treeheight_col =</span> <span class="dv">5</span>,</span>
<span id="cb29-18"><a href="wgcna.html#cb29-18"></a>                   <span class="dt">main =</span> <span class="kw">paste0</span>(prefix,<span class="st">&quot;ME correlation heatmap&quot;</span>),</span>
<span id="cb29-19"><a href="wgcna.html#cb29-19"></a>                   <span class="dt">labels_row =</span> <span class="kw">paste0</span>(prefix, <span class="kw">rownames</span>(MEs_R)),</span>
<span id="cb29-20"><a href="wgcna.html#cb29-20"></a>                   <span class="dt">labels_col =</span> <span class="kw">paste0</span>(prefix, <span class="kw">colnames</span>(MEs_R))) -&gt;<span class="st"> </span>MEs_R_Corr</span>
<span id="cb29-21"><a href="wgcna.html#cb29-21"></a></span>
<span id="cb29-22"><a href="wgcna.html#cb29-22"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(MEs_R_density, MEs_R_Corr<span class="op">$</span>gtable, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;D&quot;</span>, <span class="st">&quot;E&quot;</span>), <span class="dt">label_size =</span> plot_labeling_size, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.6</span>, <span class="dv">1</span>)) -&gt;<span class="st"> </span>density_eigen</span>
<span id="cb29-23"><a href="wgcna.html#cb29-23"></a></span>
<span id="cb29-24"><a href="wgcna.html#cb29-24"></a>density_eigen</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Module%20(Eigengene)%20correlation-1.png" width="960" /></p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="wgcna.html#cb30-1"></a><span class="co">## Display module info</span></span>
<span id="cb30-2"><a href="wgcna.html#cb30-2"></a><span class="kw">all</span>(<span class="kw">rownames</span>(seq_tab_filter) <span class="op">==</span><span class="st"> </span><span class="kw">rownames</span>(MEs))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="wgcna.html#cb32-1"></a><span class="kw">dim</span>(seq_tab_filter) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">&quot; samples&quot;</span>, <span class="st">&quot; OTUs&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;45 samples&quot; &quot;231 OTUs&quot;</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="wgcna.html#cb34-1"></a>kME &lt;-<span class="st"> </span><span class="kw">bicor</span>(seq_tab_filter, MEs, <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## Warning in bicor(seq_tab_filter, MEs, maxPOutliers = 0.05): bicor: zero MAD in variable &#39;x&#39;. Pearson correlation was used for individual columns with zero (or missing)
## MAD.</code></pre>
<pre><code>## Warning in bicor(seq_tab_filter, MEs, maxPOutliers = 0.05): bicor: zero MAD in variable &#39;y&#39;. Pearson correlation was used for individual columns with zero (or missing)
## MAD.</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="wgcna.html#cb37-1"></a><span class="kw">dim</span>(kME) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">&quot; OTUs&quot;</span>, <span class="st">&quot; modules&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;231 OTUs&quot;  &quot;6 modules&quot;</code></pre>
<div id="plot-intramodular-correlation" class="section level4">
<h4><span class="header-section-number">2.3.4.1</span> Plot intramodular correlation</h4>
<p>How the OTUs within a module correlates to the module eigengene.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="wgcna.html#cb39-1"></a>intra_cor &lt;-<span class="st"> </span><span class="kw">c</span>()</span>
<span id="cb39-2"><a href="wgcna.html#cb39-2"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(seq_tab_filter)) {</span>
<span id="cb39-3"><a href="wgcna.html#cb39-3"></a>  m &lt;-<span class="st"> </span>modules.otu<span class="op">$</span>colors[i]</span>
<span id="cb39-4"><a href="wgcna.html#cb39-4"></a>  intra_cor[i] &lt;-<span class="st"> </span>kME[i, <span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>, m)]</span>
<span id="cb39-5"><a href="wgcna.html#cb39-5"></a>  <span class="cf">if</span>(m <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>){</span>
<span id="cb39-6"><a href="wgcna.html#cb39-6"></a>    intra_cor[i] &lt;-<span class="st"> </span>kME[i, <span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>, m)]</span>
<span id="cb39-7"><a href="wgcna.html#cb39-7"></a>  } <span class="cf">else</span>{</span>
<span id="cb39-8"><a href="wgcna.html#cb39-8"></a>    intra_cor[i] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb39-9"><a href="wgcna.html#cb39-9"></a>  }</span>
<span id="cb39-10"><a href="wgcna.html#cb39-10"></a>  </span>
<span id="cb39-11"><a href="wgcna.html#cb39-11"></a>}</span>
<span id="cb39-12"><a href="wgcna.html#cb39-12"></a></span>
<span id="cb39-13"><a href="wgcna.html#cb39-13"></a>idx &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">is.na</span>(intra_cor))</span>
<span id="cb39-14"><a href="wgcna.html#cb39-14"></a>intra_cor &lt;-<span class="st"> </span>intra_cor[<span class="op">-</span>idx]</span>
<span id="cb39-15"><a href="wgcna.html#cb39-15"></a></span>
<span id="cb39-16"><a href="wgcna.html#cb39-16"></a><span class="kw">plot</span>(<span class="kw">density</span>(intra_cor), <span class="dt">main =</span> <span class="st">&quot;Correlations with module-eigenOTU (within module correlation)</span><span class="ch">\n</span><span class="st">No ME0&quot;</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/intramodular%20correlation-1.png" width="672" /></p>
</div>
<div id="plot-intramodular-correlation-for-each-module" class="section level4">
<h4><span class="header-section-number">2.3.4.2</span> Plot intramodular correlation for each module</h4>
<p><strong>Plot intramodular correlation for each module individually, and color by module color.</strong></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="wgcna.html#cb40-1"></a><span class="co"># Corr within modules</span></span>
<span id="cb40-2"><a href="wgcna.html#cb40-2"></a>corr_within_module &lt;-<span class="st"> </span><span class="cf">function</span>(seq_tab_filter, modules, <span class="dt">module_x =</span> <span class="dv">1</span>){</span>
<span id="cb40-3"><a href="wgcna.html#cb40-3"></a>  idx.omics_data &lt;-<span class="st"> </span><span class="kw">which</span>(modules<span class="op">$</span>colors <span class="op">==</span><span class="st"> </span>module_x)</span>
<span id="cb40-4"><a href="wgcna.html#cb40-4"></a>  idx.me &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(modules<span class="op">$</span>MEs) <span class="op">==</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>,module_x))</span>
<span id="cb40-5"><a href="wgcna.html#cb40-5"></a>  kME_x &lt;-<span class="st"> </span><span class="kw">bicor</span>(seq_tab_filter[,idx.omics_data], modules<span class="op">$</span>MEs[,idx.me], <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span>
<span id="cb40-6"><a href="wgcna.html#cb40-6"></a>  kME_x</span>
<span id="cb40-7"><a href="wgcna.html#cb40-7"></a>}</span>
<span id="cb40-8"><a href="wgcna.html#cb40-8"></a></span>
<span id="cb40-9"><a href="wgcna.html#cb40-9"></a>ggplot.list &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb40-10"><a href="wgcna.html#cb40-10"></a></span>
<span id="cb40-11"><a href="wgcna.html#cb40-11"></a><span class="cf">for</span>(m <span class="cf">in</span> <span class="kw">colnames</span>(modules.otu<span class="op">$</span>MEs)){</span>
<span id="cb40-12"><a href="wgcna.html#cb40-12"></a>  h &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">sub</span>(<span class="st">&quot;ME&quot;</span>,<span class="st">&quot;&quot;</span>, m))</span>
<span id="cb40-13"><a href="wgcna.html#cb40-13"></a>  <span class="kw">data.frame</span>(<span class="dt">x =</span> <span class="kw">suppressWarnings</span>(<span class="kw">corr_within_module</span>(<span class="dt">seq_tab_filter =</span> seq_tab_filter, <span class="dt">modules =</span> modules.otu, <span class="dt">module_x =</span> h))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb40-14"><a href="wgcna.html#cb40-14"></a><span class="st">    </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb40-15"><a href="wgcna.html#cb40-15"></a><span class="st">    </span><span class="kw">geom_density</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x), <span class="dt">fill =</span> <span class="kw">labels2colors</span>(h), <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb40-16"><a href="wgcna.html#cb40-16"></a><span class="st">    </span><span class="kw">xlim</span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb40-17"><a href="wgcna.html#cb40-17"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;OTU correlation&quot;</span>)<span class="op">+</span></span>
<span id="cb40-18"><a href="wgcna.html#cb40-18"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(prefix,m)) -&gt;<span class="st"> </span>da_plot</span>
<span id="cb40-19"><a href="wgcna.html#cb40-19"></a>  </span>
<span id="cb40-20"><a href="wgcna.html#cb40-20"></a>  ggplot.list[[m]] &lt;-<span class="st"> </span>da_plot</span>
<span id="cb40-21"><a href="wgcna.html#cb40-21"></a>}</span>
<span id="cb40-22"><a href="wgcna.html#cb40-22"></a></span>
<span id="cb40-23"><a href="wgcna.html#cb40-23"></a>ggplot.list &lt;-<span class="st"> </span>ggplot.list[ggplot.list <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">names</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;ME&quot;</span>, <span class="st">&quot;&quot;</span>, .) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.numeric</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">order</span>()]</span>
<span id="cb40-24"><a href="wgcna.html#cb40-24"></a></span>
<span id="cb40-25"><a href="wgcna.html#cb40-25"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> ggplot.list, <span class="dt">ncol =</span> <span class="dv">2</span>) -&gt;<span class="st"> </span>density_all_plot</span>
<span id="cb40-26"><a href="wgcna.html#cb40-26"></a>density_all_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Plot%20intramodular%20correlation%20for%20each%20module-1.png" width="1152" /></p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="wgcna.html#cb41-1"></a><span class="co">## Combine to one plot</span></span>
<span id="cb41-2"><a href="wgcna.html#cb41-2"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(si_mc_plot , density_eigen, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>)) -&gt;<span class="st"> </span>part_<span class="dv">1</span></span>
<span id="cb41-3"><a href="wgcna.html#cb41-3"></a></span>
<span id="cb41-4"><a href="wgcna.html#cb41-4"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(part_<span class="dv">1</span>, module_size_barplot, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;C&quot;</span>), <span class="dt">label_size =</span> plot_labeling_size, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="fl">0.5</span>)) -&gt;<span class="st"> </span>part_<span class="dv">2</span></span>
<span id="cb41-5"><a href="wgcna.html#cb41-5"></a></span>
<span id="cb41-6"><a href="wgcna.html#cb41-6"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(part_<span class="dv">2</span>, density_all_plot, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;F&quot;</span>), <span class="dt">label_size =</span> plot_labeling_size)</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Plot%20intramodular%20correlation%20for%20each%20module-2.png" width="1152" /></p>
</div>
</div>
<div id="find-hub-asv" class="section level3">
<h3><span class="header-section-number">2.3.5</span> Find Hub ASV</h3>
<p><strong>For each module it is possible to pick a hub ASV with the function <code>chooseTopHubInEachModule</code>.</strong></p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="wgcna.html#cb42-1"></a>hubs <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb42-2"><a href="wgcna.html#cb42-2"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb42-3"><a href="wgcna.html#cb42-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="st">&quot;OTU_name&quot;</span> =<span class="st"> &quot;.&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb42-4"><a href="wgcna.html#cb42-4"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;Module&quot;</span>) -&gt;<span class="st"> </span>hubOTUs</span>
<span id="cb42-5"><a href="wgcna.html#cb42-5"></a></span>
<span id="cb42-6"><a href="wgcna.html#cb42-6"></a></span>
<span id="cb42-7"><a href="wgcna.html#cb42-7"></a></span>
<span id="cb42-8"><a href="wgcna.html#cb42-8"></a>dplyr<span class="op">::</span><span class="kw">left_join</span>(hubOTUs, </span>
<span id="cb42-9"><a href="wgcna.html#cb42-9"></a>                 (taxa_tb <span class="op">%&gt;%</span></span>
<span id="cb42-10"><a href="wgcna.html#cb42-10"></a><span class="st">                    </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;OTU_name&quot;</span>)), </span>
<span id="cb42-11"><a href="wgcna.html#cb42-11"></a>                 <span class="dt">by =</span> <span class="st">&quot;OTU_name&quot;</span>) -&gt;<span class="st"> </span>hubOTUs</span>
<span id="cb42-12"><a href="wgcna.html#cb42-12"></a></span>
<span id="cb42-13"><a href="wgcna.html#cb42-13"></a>hubOTUs</span></code></pre></div>
<pre><code>##   Module OTU_name  Kingdom        Phylum            Class              Order                Family        Genus     Species
## 1      0    ASV75 Bacteria Bacteroidetes      Bacteroidia      Bacteroidales        Bacteroidaceae  Bacteroides        &lt;NA&gt;
## 2      1    ASV20 Bacteria    Firmicutes       Clostridia      Clostridiales       Lachnospiraceae        Dorea longicatena
## 3      2   ASV248 Bacteria    Firmicutes       Clostridia      Clostridiales       Lachnospiraceae      Blautia        &lt;NA&gt;
## 4      3   ASV222 Bacteria    Firmicutes       Clostridia      Clostridiales Peptostreptococcaceae         &lt;NA&gt;        &lt;NA&gt;
## 5      4   ASV237 Bacteria Bacteroidetes      Bacteroidia      Bacteroidales        Bacteroidaceae  Bacteroides        &lt;NA&gt;
## 6      5   ASV214 Bacteria    Firmicutes Erysipelotrichia Erysipelotrichales   Erysipelotrichaceae Holdemanella        &lt;NA&gt;</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="wgcna.html#cb44-1"></a>res &lt;-<span class="st"> </span>hubOTUs[, <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">7</span>)]</span>
<span id="cb44-2"><a href="wgcna.html#cb44-2"></a><span class="kw">colnames</span>(res) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Module&quot;</span>, <span class="st">&quot;OTU&quot;</span>, <span class="st">&quot;Phylum&quot;</span>,<span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb44-3"><a href="wgcna.html#cb44-3"></a><span class="kw">print</span>(res)</span></code></pre></div>
<pre><code>##   Module    OTU        Phylum                 Genus
## 1      0  ASV75 Bacteroidetes        Bacteroidaceae
## 2      1  ASV20    Firmicutes       Lachnospiraceae
## 3      2 ASV248    Firmicutes       Lachnospiraceae
## 4      3 ASV222    Firmicutes Peptostreptococcaceae
## 5      4 ASV237 Bacteroidetes        Bacteroidaceae
## 6      5 ASV214    Firmicutes   Erysipelotrichaceae</code></pre>
</div>
<div id="visualize-network" class="section level3">
<h3><span class="header-section-number">2.3.6</span> Visualize Network</h3>
<div id="construct-network-with-igragh" class="section level4">
<h4><span class="header-section-number">2.3.6.1</span> Construct network with igragh</h4>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="wgcna.html#cb46-1"></a><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;/share/projects/Analytics/analytics/MultiOmics/tools/Bookdown/result/16S_network_module-block.1.RData&quot;</span>)</span>
<span id="cb46-2"><a href="wgcna.html#cb46-2"></a><span class="co"># The TOM is saved as a dist object and needs to be converted to a matrix</span></span>
<span id="cb46-3"><a href="wgcna.html#cb46-3"></a>TOM &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(TOM)</span>
<span id="cb46-4"><a href="wgcna.html#cb46-4"></a><span class="co"># Add OTU names to the TOM matrix. It is symmetrical so rownames = colnames</span></span>
<span id="cb46-5"><a href="wgcna.html#cb46-5"></a><span class="kw">rownames</span>(TOM) &lt;-<span class="st"> </span><span class="kw">colnames</span>(TOM) &lt;-<span class="st"> </span><span class="kw">names</span>(modules.otu<span class="op">$</span>colors)</span>
<span id="cb46-6"><a href="wgcna.html#cb46-6"></a></span>
<span id="cb46-7"><a href="wgcna.html#cb46-7"></a><span class="co">## Choose taxonomic level should the graph be colored with in addition to modules?</span></span>
<span id="cb46-8"><a href="wgcna.html#cb46-8"></a>selected_taxa &lt;-<span class="st"> &quot;Phylum&quot;</span></span>
<span id="cb46-9"><a href="wgcna.html#cb46-9"></a></span>
<span id="cb46-10"><a href="wgcna.html#cb46-10"></a><span class="co">## Convert module labels and taxonomy to hex colors</span></span>
<span id="cb46-11"><a href="wgcna.html#cb46-11"></a>taxonomy_info &lt;-<span class="st"> </span></span>
<span id="cb46-12"><a href="wgcna.html#cb46-12"></a><span class="st">  </span>taxa_tb <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb46-13"><a href="wgcna.html#cb46-13"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;OTU_name&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb46-14"><a href="wgcna.html#cb46-14"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="st">&quot;OTU_name&quot;</span>, selected_taxa) <span class="op">%&gt;%</span></span>
<span id="cb46-15"><a href="wgcna.html#cb46-15"></a><span class="st">  </span><span class="kw">mutate_all</span>(<span class="dt">.funs =</span> <span class="kw">list</span>(as.character)) </span></code></pre></div>
<pre><code>## Note: Using an external vector in selections is ambiguous.
## ℹ Use `all_of(selected_taxa)` instead of `selected_taxa` to silence this message.
## ℹ See &lt;https://tidyselect.r-lib.org/reference/faq-external-vector.html&gt;.
## This message is displayed once per session.</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="wgcna.html#cb48-1"></a>module_info &lt;-<span class="st"> </span></span>
<span id="cb48-2"><a href="wgcna.html#cb48-2"></a><span class="st">  </span>modules.otu<span class="op">$</span>colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb48-3"><a href="wgcna.html#cb48-3"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb48-4"><a href="wgcna.html#cb48-4"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;OTU_name&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb48-5"><a href="wgcna.html#cb48-5"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">Module =</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb48-6"><a href="wgcna.html#cb48-6"></a></span>
<span id="cb48-7"><a href="wgcna.html#cb48-7"></a><span class="co">## 选择指定模块</span></span>
<span id="cb48-8"><a href="wgcna.html#cb48-8"></a>TOM &lt;-<span class="st"> </span>TOM[module_info<span class="op">$</span>OTU_name, module_info<span class="op">$</span>OTU_name]</span>
<span id="cb48-9"><a href="wgcna.html#cb48-9"></a></span>
<span id="cb48-10"><a href="wgcna.html#cb48-10"></a>graph_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(module_info, taxonomy_info, <span class="dt">by =</span> <span class="st">&quot;OTU_name&quot;</span>)</span>
<span id="cb48-11"><a href="wgcna.html#cb48-11"></a></span>
<span id="cb48-12"><a href="wgcna.html#cb48-12"></a><span class="co">## Converts R-colors to hex colors</span></span>
<span id="cb48-13"><a href="wgcna.html#cb48-13"></a>color2hex &lt;-<span class="st"> </span><span class="cf">function</span>(x){</span>
<span id="cb48-14"><a href="wgcna.html#cb48-14"></a>  x &lt;-<span class="st"> </span><span class="kw">col2rgb</span>(x)</span>
<span id="cb48-15"><a href="wgcna.html#cb48-15"></a>  <span class="kw">rgb</span>(x[<span class="dv">1</span>,], x[<span class="dv">2</span>,], x[<span class="dv">3</span>,], <span class="dt">maxColorValue =</span> <span class="dv">255</span>)</span>
<span id="cb48-16"><a href="wgcna.html#cb48-16"></a>}</span>
<span id="cb48-17"><a href="wgcna.html#cb48-17"></a></span>
<span id="cb48-18"><a href="wgcna.html#cb48-18"></a><span class="co">## Add specific colors to the taxa</span></span>
<span id="cb48-19"><a href="wgcna.html#cb48-19"></a>taxa_colors &lt;-<span class="st"> </span></span>
<span id="cb48-20"><a href="wgcna.html#cb48-20"></a><span class="st">  </span>graph_info <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb48-21"><a href="wgcna.html#cb48-21"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(selected_taxa) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb48-22"><a href="wgcna.html#cb48-22"></a><span class="st">  </span><span class="kw">unique</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb48-23"><a href="wgcna.html#cb48-23"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">tax_color =</span> <span class="kw">colorRampPalette</span>(RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&quot;Accent&quot;</span>))(<span class="kw">nrow</span>(.)))</span>
<span id="cb48-24"><a href="wgcna.html#cb48-24"></a></span>
<span id="cb48-25"><a href="wgcna.html#cb48-25"></a>graph_info_colors &lt;-<span class="st"> </span></span>
<span id="cb48-26"><a href="wgcna.html#cb48-26"></a><span class="st">  </span><span class="kw">left_join</span>(graph_info, taxa_colors) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb48-27"><a href="wgcna.html#cb48-27"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> WGCNA<span class="op">::</span><span class="kw">labels2colors</span>(Module)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb48-28"><a href="wgcna.html#cb48-28"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> <span class="kw">color2hex</span>(module_color)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb48-29"><a href="wgcna.html#cb48-29"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> <span class="kw">paste0</span>(module_color,<span class="dv">70</span>))</span></code></pre></div>
<pre><code>## Joining, by = &quot;Phylum&quot;</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="wgcna.html#cb50-1"></a><span class="co">## If all lines are too thick, reduce the strength (between 0 and 1).</span></span>
<span id="cb50-2"><a href="wgcna.html#cb50-2"></a><span class="co">## The lower the number the weaker the lines.</span></span>
<span id="cb50-3"><a href="wgcna.html#cb50-3"></a>strength_adjust =<span class="st"> </span><span class="dv">1</span></span>
<span id="cb50-4"><a href="wgcna.html#cb50-4"></a></span>
<span id="cb50-5"><a href="wgcna.html#cb50-5"></a></span>
<span id="cb50-6"><a href="wgcna.html#cb50-6"></a>g &lt;-<span class="st"> </span><span class="kw">graph.adjacency</span>(TOM, <span class="dt">mode=</span><span class="st">&quot;undirected&quot;</span>, <span class="dt">weighted=</span> <span class="ot">TRUE</span>)</span>
<span id="cb50-7"><a href="wgcna.html#cb50-7"></a></span>
<span id="cb50-8"><a href="wgcna.html#cb50-8"></a><span class="co">#~https://stackoverflow.com/questions/28366329/how-to-scale-edge-colors-igraph</span></span>
<span id="cb50-9"><a href="wgcna.html#cb50-9"></a>igraph<span class="op">::</span><span class="kw">delete.edges</span>(g, <span class="kw">which</span>(<span class="kw">E</span>(g)<span class="op">$</span>weight <span class="op">&lt;</span><span class="dv">1</span>))</span></code></pre></div>
<pre><code>## IGRAPH d753a0d UNW- 231 0 -- 
## + attr: name (v/c), weight (e/n)
## + edges from d753a0d (vertex names):</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="wgcna.html#cb52-1"></a><span class="kw">E</span>(g)<span class="op">$</span>width &lt;-<span class="st"> </span><span class="kw">E</span>(g)<span class="op">$</span>weight<span class="op">*</span>strength_adjust <span class="op">+</span><span class="st"> </span><span class="kw">min</span>(<span class="kw">E</span>(g)<span class="op">$</span>weight)</span>
<span id="cb52-2"><a href="wgcna.html#cb52-2"></a></span>
<span id="cb52-3"><a href="wgcna.html#cb52-3"></a><span class="kw">set.seed</span>(<span class="dv">231</span>) <span class="co"># Ensures the same layout given the same data.</span></span>
<span id="cb52-4"><a href="wgcna.html#cb52-4"></a>l &lt;-<span class="st"> </span><span class="kw">layout_with_fr</span>(g, <span class="dt">weights =</span> <span class="kw">E</span>(g)<span class="op">$</span>weight)</span>
<span id="cb52-5"><a href="wgcna.html#cb52-5"></a></span>
<span id="cb52-6"><a href="wgcna.html#cb52-6"></a></span>
<span id="cb52-7"><a href="wgcna.html#cb52-7"></a><span class="co"># Order graph_info_colors by the graph </span></span>
<span id="cb52-8"><a href="wgcna.html#cb52-8"></a>graph_info_colors &lt;-<span class="st"> </span>graph_info_colors[<span class="kw">which</span>(graph_info_colors<span class="op">$</span>OTU_name <span class="op">%in%</span><span class="st"> </span><span class="kw">V</span>(g)<span class="op">$</span>name),]</span>
<span id="cb52-9"><a href="wgcna.html#cb52-9"></a></span>
<span id="cb52-10"><a href="wgcna.html#cb52-10"></a><span class="co"># Ensure that the names are in the same order</span></span>
<span id="cb52-11"><a href="wgcna.html#cb52-11"></a><span class="cf">if</span>(<span class="kw">all</span>(<span class="kw">V</span>(g)<span class="op">$</span>name <span class="op">==</span><span class="st"> </span>graph_info_colors<span class="op">$</span>OTU_name)){<span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">OTU names match&quot;</span>)}</span></code></pre></div>
<pre><code>## 
## OTU names match</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="wgcna.html#cb54-1"></a><span class="co"># Add square shapes to hub OTUs</span></span>
<span id="cb54-2"><a href="wgcna.html#cb54-2"></a><span class="kw">V</span>(g)<span class="op">$</span>shape &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">V</span>(g)<span class="op">$</span>name <span class="op">%in%</span><span class="st"> </span>hubs[<span class="op">-</span><span class="dv">1</span>], <span class="st">&quot;square&quot;</span>, <span class="st">&quot;circle&quot;</span>) <span class="co">#-1 means dont use module 0</span></span>
<span id="cb54-3"><a href="wgcna.html#cb54-3"></a></span>
<span id="cb54-4"><a href="wgcna.html#cb54-4"></a><span class="co"># OTUs in modules have larger nodes</span></span>
<span id="cb54-5"><a href="wgcna.html#cb54-5"></a><span class="kw">V</span>(g)<span class="op">$</span>size &lt;-<span class="st"> </span><span class="kw">ifelse</span>(graph_info_colors<span class="op">$</span>Module <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">5</span>)</span>
<span id="cb54-6"><a href="wgcna.html#cb54-6"></a></span>
<span id="cb54-7"><a href="wgcna.html#cb54-7"></a><span class="co"># And larger text</span></span>
<span id="cb54-8"><a href="wgcna.html#cb54-8"></a><span class="kw">V</span>(g)<span class="op">$</span>label.cex &lt;-<span class="st"> </span><span class="kw">ifelse</span>(graph_info_colors<span class="op">$</span>Module <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>, <span class="fl">0.8</span>, <span class="fl">0.4</span>)</span>
<span id="cb54-9"><a href="wgcna.html#cb54-9"></a></span>
<span id="cb54-10"><a href="wgcna.html#cb54-10"></a><span class="co"># Remove everything but the number to increase readability</span></span>
<span id="cb54-11"><a href="wgcna.html#cb54-11"></a><span class="kw">V</span>(g)<span class="op">$</span>name =<span class="st">  </span><span class="kw">sub</span>(<span class="st">&quot;OTU_&quot;</span>, <span class="st">&quot;&quot;</span>, graph_info_colors<span class="op">$</span>OTU_name)</span></code></pre></div>
</div>
<div id="plot-network" class="section level4">
<h4><span class="header-section-number">2.3.6.2</span> Plot network</h4>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="wgcna.html#cb55-1"></a><span class="co">## Find distinct entires for the plot legends</span></span>
<span id="cb55-2"><a href="wgcna.html#cb55-2"></a>module_labels &lt;-<span class="st"> </span></span>
<span id="cb55-3"><a href="wgcna.html#cb55-3"></a><span class="st">  </span>graph_info_colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb55-4"><a href="wgcna.html#cb55-4"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(Module, module_color) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb55-5"><a href="wgcna.html#cb55-5"></a><span class="st">  </span><span class="kw">distinct</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb55-6"><a href="wgcna.html#cb55-6"></a><span class="st">  </span><span class="kw">arrange</span>(Module)</span>
<span id="cb55-7"><a href="wgcna.html#cb55-7"></a></span>
<span id="cb55-8"><a href="wgcna.html#cb55-8"></a>tax_labels &lt;-<span class="st"> </span></span>
<span id="cb55-9"><a href="wgcna.html#cb55-9"></a><span class="st">  </span>graph_info_colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb55-10"><a href="wgcna.html#cb55-10"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(selected_taxa, tax_color) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb55-11"><a href="wgcna.html#cb55-11"></a><span class="st">  </span><span class="kw">distinct</span>()</span>
<span id="cb55-12"><a href="wgcna.html#cb55-12"></a></span>
<span id="cb55-13"><a href="wgcna.html#cb55-13"></a><span class="co">## Plot the graphs, leftmost is colored by module, rightmost is colored by taxonomic classification</span></span>
<span id="cb55-14"><a href="wgcna.html#cb55-14"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb55-15"><a href="wgcna.html#cb55-15"></a></span>
<span id="cb55-16"><a href="wgcna.html#cb55-16"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb55-17"><a href="wgcna.html#cb55-17"></a></span>
<span id="cb55-18"><a href="wgcna.html#cb55-18"></a><span class="kw">plot</span>(g, <span class="dt">layout =</span> l, <span class="dt">vertex.color =</span> graph_info_colors<span class="op">$</span>module_color)</span>
<span id="cb55-19"><a href="wgcna.html#cb55-19"></a><span class="co"># legend(&quot;topleft&quot;, legend = paste0(&quot;mM&quot;, 0:(nrow(module_labels)-1)), fill=module_labels$module_color)</span></span>
<span id="cb55-20"><a href="wgcna.html#cb55-20"></a><span class="kw">legend</span>(<span class="st">&quot;topleft&quot;</span>, <span class="dt">legend =</span> <span class="kw">paste0</span>(<span class="st">&quot;mM&quot;</span>, <span class="dv">0</span><span class="op">:</span>(<span class="kw">nrow</span>(module_labels)<span class="op">-</span><span class="dv">1</span>)), <span class="dt">fill=</span>module_labels<span class="op">$</span>module_color)</span>
<span id="cb55-21"><a href="wgcna.html#cb55-21"></a></span>
<span id="cb55-22"><a href="wgcna.html#cb55-22"></a><span class="kw">plot</span>(g, <span class="dt">layout =</span> l, <span class="dt">vertex.color =</span> graph_info_colors<span class="op">$</span>tax_color)</span>
<span id="cb55-23"><a href="wgcna.html#cb55-23"></a><span class="kw">legend</span>(<span class="st">&quot;topleft&quot;</span>, <span class="dt">legend =</span> tax_labels <span class="op">%&gt;%</span><span class="st"> </span>.[,<span class="dv">1</span>], <span class="dt">fill=</span>tax_labels<span class="op">$</span>tax_color)</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Plot%20network-1.png" width="1440" /></p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="wgcna.html#cb56-1"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
</div>
</div>
<div id="clean-env-of-16s-data-analysis" class="section level3">
<h3><span class="header-section-number">2.3.7</span> Clean env of 16S data analysis</h3>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="wgcna.html#cb57-1"></a><span class="kw">rm</span>(<span class="dt">list=</span><span class="kw">ls</span>())</span></code></pre></div>
</div>
</div>
<div id="wgcna-for-mgs-data" class="section level2">
<h2><span class="header-section-number">2.4</span> WGCNA for MGS data</h2>
<p><strong>MGS data from GvHD project (32 samples with 250 species features) was used as demo data in this sub-chapter.</strong></p>
<div id="data-loading-wgcna-mgs" class="section level3">
<h3><span class="header-section-number">2.4.1</span> Data loading (WGCNA MGS)</h3>
<div id="read-demo-input-files-wgcna-mgs" class="section level4">
<h4><span class="header-section-number">2.4.1.1</span> Read demo input files (WGCNA MGS)</h4>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="wgcna.html#cb58-1"></a>mgs_profile &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;/share/projects/Analytics/analytics/MultiOmics/tools/HAllA/GvHD/metaphlan2_merged.tsv&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>) <span class="co"># read metagenomic hirerachy table，only need ID column</span></span>
<span id="cb58-2"><a href="wgcna.html#cb58-2"></a></span>
<span id="cb58-3"><a href="wgcna.html#cb58-3"></a>mgs &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;/share/projects/Analytics/analytics/MultiOmics/tools/HAllA/GvHD/metagenomics.txt&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>) <span class="co"># read metagenomic species level data</span></span>
<span id="cb58-4"><a href="wgcna.html#cb58-4"></a></span>
<span id="cb58-5"><a href="wgcna.html#cb58-5"></a>metadata &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;/share/projects/Analytics/analytics/MultiOmics/tools/HAllA/GvHD/metadata.txt&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>) <span class="co"># read metadata</span></span>
<span id="cb58-6"><a href="wgcna.html#cb58-6"></a></span>
<span id="cb58-7"><a href="wgcna.html#cb58-7"></a><span class="kw">dim</span>(mgs_profile)</span></code></pre></div>
<pre><code>## [1] 1028   69</code></pre>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="wgcna.html#cb60-1"></a><span class="kw">dim</span>(mgs)</span></code></pre></div>
<pre><code>## [1] 250  32</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="wgcna.html#cb62-1"></a><span class="kw">dim</span>(metadata)</span></code></pre></div>
<pre><code>## [1] 15 32</code></pre>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="wgcna.html#cb64-1"></a><span class="kw">cat</span>(<span class="st">&quot;[mgs]&quot;</span>, <span class="kw">paste</span>(<span class="kw">c</span>(<span class="st">&quot;Feature: &quot;</span>, <span class="st">&quot;Sample: &quot;</span>), <span class="kw">dim</span>(mgs)))</span></code></pre></div>
<pre><code>## [mgs] Feature:  250 Sample:  32</code></pre>
</div>
<div id="construct-the-hierarchy-table-for-taxa-data" class="section level4">
<h4><span class="header-section-number">2.4.1.2</span> Construct the hierarchy table for taxa data</h4>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="wgcna.html#cb66-1"></a><span class="co"># taxa hierarchy table</span></span>
<span id="cb66-2"><a href="wgcna.html#cb66-2"></a>mgs_species &lt;-<span class="st"> </span>mgs_profile <span class="op">%&gt;%</span></span>
<span id="cb66-3"><a href="wgcna.html#cb66-3"></a><span class="st">  </span>.[<span class="kw">sapply</span>(<span class="kw">str_split</span>(mgs_profile<span class="op">$</span>ID, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">|&quot;</span>), <span class="cf">function</span>(x) <span class="kw">length</span>(x) <span class="op">==</span><span class="st"> </span><span class="dv">7</span>), ] </span>
<span id="cb66-4"><a href="wgcna.html#cb66-4"></a>taxa_table &lt;-<span class="st"> </span><span class="kw">str_split_fixed</span>(mgs_species<span class="op">$</span>ID,<span class="st">&quot;</span><span class="ch">\\</span><span class="st">|&quot;</span>, <span class="dv">7</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</span>
<span id="cb66-5"><a href="wgcna.html#cb66-5"></a><span class="kw">colnames</span>(taxa_table) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>, <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb66-6"><a href="wgcna.html#cb66-6"></a><span class="kw">rownames</span>(taxa_table) &lt;-<span class="st"> </span><span class="kw">str_replace</span>(taxa_table<span class="op">$</span>Species, <span class="st">&quot;s__&quot;</span>, <span class="st">&quot;&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="data-normalization-wgcna-mgs" class="section level3">
<h3><span class="header-section-number">2.4.2</span> Data normalization (WGCNA MGS)</h3>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="wgcna.html#cb67-1"></a>data.metagenomeSeq &lt;-<span class="st"> </span>metagenomeSeq<span class="op">::</span><span class="kw">newMRexperiment</span>(mgs) <span class="co"># Variables as rows, samples as columns</span></span>
<span id="cb67-2"><a href="wgcna.html#cb67-2"></a>p &lt;-<span class="st"> </span>metagenomeSeq<span class="op">::</span><span class="kw">cumNormStat</span>(data.metagenomeSeq)</span></code></pre></div>
<pre><code>## Default value being used.</code></pre>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="wgcna.html#cb69-1"></a>data.cumnorm &lt;-<span class="st"> </span>metagenomeSeq<span class="op">::</span><span class="kw">cumNorm</span>(data.metagenomeSeq, <span class="dt">p=</span>p)</span>
<span id="cb69-2"><a href="wgcna.html#cb69-2"></a>mgs &lt;-<span class="st"> </span>metagenomeSeq<span class="op">::</span><span class="kw">MRcounts</span>(data.cumnorm, <span class="dt">norm=</span><span class="ot">TRUE</span>, <span class="dt">log=</span><span class="ot">TRUE</span>) <span class="co"># log here is a +1 shifted log2</span></span>
<span id="cb69-3"><a href="wgcna.html#cb69-3"></a></span>
<span id="cb69-4"><a href="wgcna.html#cb69-4"></a><span class="kw">dim</span>(mgs)</span></code></pre></div>
<pre><code>## [1] 250  32</code></pre>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="wgcna.html#cb71-1"></a>mgs_t &lt;-<span class="st"> </span>mgs <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>() <span class="co"># Samples as rows, Features as columns</span></span></code></pre></div>
</div>
<div id="build-wgcna-network-wgcna-mgs" class="section level3">
<h3><span class="header-section-number">2.4.3</span> Build WGCNA network (WGCNA MGS)</h3>
<div id="select-power-wgcna-mgs" class="section level4">
<h4><span class="header-section-number">2.4.3.1</span> select power (WGCNA MGS)</h4>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="wgcna.html#cb72-1"></a>powers &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">seq</span>(<span class="dv">12</span>,<span class="dv">20</span>,<span class="dv">2</span>)) <span class="co"># default</span></span>
<span id="cb72-2"><a href="wgcna.html#cb72-2"></a></span>
<span id="cb72-3"><a href="wgcna.html#cb72-3"></a><span class="kw">suppressWarnings</span>(sft &lt;-<span class="st"> </span><span class="kw">pickSoftThreshold</span>(mgs_t, </span>
<span id="cb72-4"><a href="wgcna.html#cb72-4"></a>                                          <span class="dt">powerVector =</span> powers, </span>
<span id="cb72-5"><a href="wgcna.html#cb72-5"></a>                                          <span class="dt">verbose =</span> <span class="dv">1</span>, </span>
<span id="cb72-6"><a href="wgcna.html#cb72-6"></a>                                          <span class="dt">networkType =</span> <span class="st">&quot;signed&quot;</span>,</span>
<span id="cb72-7"><a href="wgcna.html#cb72-7"></a>                                          <span class="dt">corFn=</span> <span class="st">&quot;bicor&quot;</span>))</span></code></pre></div>
<pre><code>## pickSoftThreshold: will use block size 250.
##  pickSoftThreshold: calculating connectivity for given powers... ..0% ..100% 
##    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
## 1      1   0.6250 10.900          0.768  134.00   135.000 150.00
## 2      2   0.6400  4.310          0.814   75.60    75.700  93.20
## 3      3   0.5870  2.560          0.867   44.70    44.700  60.90
## 4      4   0.3110  1.210          0.715   27.80    27.900  41.70
## 5      5   0.2170  0.761          0.623   18.20    17.900  29.70
## 6      6   0.2000  0.561          0.699   12.50    12.100  21.80
## 7      7   0.1940  0.438          0.730    9.05     8.830  16.40
## 8      8   0.0156  0.114          0.753    6.80     6.510  13.60
## 9      9   0.0476 -0.223          0.620    5.30     4.830  11.80
## 10    10   0.1870 -0.526          0.470    4.26     3.720  10.50
## 11    12   0.5040 -0.857          0.661    2.99     2.310   8.89
## 12    14   0.6820 -0.965          0.666    2.27     1.470   7.93
## 13    16   0.7330 -1.080          0.659    1.84     1.070   7.33
## 14    18   0.8160 -1.020          0.804    1.57     0.777   6.94
## 15    20   0.8010 -0.986          0.808    1.38     0.587   6.67</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="wgcna.html#cb74-1"></a><span class="co"># Find the soft thresholding power beta to which co-expression similarity is raised to calculate adjacency.</span></span>
<span id="cb74-2"><a href="wgcna.html#cb74-2"></a><span class="co"># based on the criterion of approximate scale-free topology.</span></span>
<span id="cb74-3"><a href="wgcna.html#cb74-3"></a></span>
<span id="cb74-4"><a href="wgcna.html#cb74-4"></a>idx &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">which</span>((<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.90</span>))</span>
<span id="cb74-5"><a href="wgcna.html#cb74-5"></a><span class="cf">if</span>(<span class="kw">is.infinite</span>(idx)){</span>
<span id="cb74-6"><a href="wgcna.html#cb74-6"></a>  idx &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">which</span>((<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.80</span>))</span>
<span id="cb74-7"><a href="wgcna.html#cb74-7"></a>  <span class="cf">if</span>(<span class="op">!</span><span class="kw">is.infinite</span>(idx)){</span>
<span id="cb74-8"><a href="wgcna.html#cb74-8"></a>    st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb74-9"><a href="wgcna.html#cb74-9"></a>  } <span class="cf">else</span>{</span>
<span id="cb74-10"><a href="wgcna.html#cb74-10"></a>    idx &lt;-<span class="st"> </span><span class="kw">which.max</span>(<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>])</span>
<span id="cb74-11"><a href="wgcna.html#cb74-11"></a>    st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb74-12"><a href="wgcna.html#cb74-12"></a>  }</span>
<span id="cb74-13"><a href="wgcna.html#cb74-13"></a>} <span class="cf">else</span>{</span>
<span id="cb74-14"><a href="wgcna.html#cb74-14"></a>  st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb74-15"><a href="wgcna.html#cb74-15"></a>}</span>
<span id="cb74-16"><a href="wgcna.html#cb74-16"></a></span>
<span id="cb74-17"><a href="wgcna.html#cb74-17"></a></span>
<span id="cb74-18"><a href="wgcna.html#cb74-18"></a><span class="co"># Plot Scale independence measure and Mean connectivity measure</span></span>
<span id="cb74-19"><a href="wgcna.html#cb74-19"></a></span>
<span id="cb74-20"><a href="wgcna.html#cb74-20"></a><span class="co"># Scale-free topology fit index as a function of the soft-thresholding power</span></span>
<span id="cb74-21"><a href="wgcna.html#cb74-21"></a><span class="kw">data.frame</span>(<span class="dt">Indices =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb74-22"><a href="wgcna.html#cb74-22"></a>           <span class="dt">sfApprox =</span> <span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb74-23"><a href="wgcna.html#cb74-23"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb74-24"><a href="wgcna.html#cb74-24"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="fl">0.9</span>, <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.6</span>) <span class="op">+</span><span class="st"> </span><span class="co"># corresponds to R^2 cut-off of 0.9</span></span>
<span id="cb74-25"><a href="wgcna.html#cb74-25"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="fl">0.8</span>, <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.2</span>) <span class="op">+</span><span class="st"> </span><span class="co"># corresponds to R^2 cut-off of 0.8</span></span>
<span id="cb74-26"><a href="wgcna.html#cb74-26"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> sfApprox), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb74-27"><a href="wgcna.html#cb74-27"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> sfApprox, <span class="dt">label =</span> Indices), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb74-28"><a href="wgcna.html#cb74-28"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Scale independence&quot;</span>) <span class="op">+</span></span>
<span id="cb74-29"><a href="wgcna.html#cb74-29"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Soft Threshold (power)&quot;</span>) <span class="op">+</span></span>
<span id="cb74-30"><a href="wgcna.html#cb74-30"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;SF Model Fit,signed R^2&quot;</span>) <span class="op">+</span></span>
<span id="cb74-31"><a href="wgcna.html#cb74-31"></a><span class="st">  </span><span class="kw">xlim</span>(<span class="dv">1</span>,<span class="dv">20</span>) <span class="op">+</span></span>
<span id="cb74-32"><a href="wgcna.html#cb74-32"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb74-33"><a href="wgcna.html#cb74-33"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> st, <span class="dt">y =</span> <span class="fl">0.25</span>, <span class="dt">xend =</span> st, <span class="dt">yend =</span> sfApprox[idx]<span class="op">-</span><span class="fl">0.05</span>), </span>
<span id="cb74-34"><a href="wgcna.html#cb74-34"></a>               <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.2</span>,<span class="st">&quot;cm&quot;</span>)), </span>
<span id="cb74-35"><a href="wgcna.html#cb74-35"></a>               <span class="dt">size =</span> <span class="fl">0.5</span>)-&gt;<span class="st"> </span>scale_independence_plot </span>
<span id="cb74-36"><a href="wgcna.html#cb74-36"></a>  </span>
<span id="cb74-37"><a href="wgcna.html#cb74-37"></a> </span>
<span id="cb74-38"><a href="wgcna.html#cb74-38"></a></span>
<span id="cb74-39"><a href="wgcna.html#cb74-39"></a></span>
<span id="cb74-40"><a href="wgcna.html#cb74-40"></a><span class="co"># Mean connectivity as a function of the soft-thresholding power</span></span>
<span id="cb74-41"><a href="wgcna.html#cb74-41"></a></span>
<span id="cb74-42"><a href="wgcna.html#cb74-42"></a><span class="kw">data.frame</span>(<span class="dt">Indices =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb74-43"><a href="wgcna.html#cb74-43"></a>           <span class="dt">meanApprox =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">5</span>]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb74-44"><a href="wgcna.html#cb74-44"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb74-45"><a href="wgcna.html#cb74-45"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> meanApprox), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb74-46"><a href="wgcna.html#cb74-46"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> meanApprox, <span class="dt">label =</span> Indices), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb74-47"><a href="wgcna.html#cb74-47"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Soft Threshold (power)&quot;</span>) <span class="op">+</span></span>
<span id="cb74-48"><a href="wgcna.html#cb74-48"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Mean Connectivity&quot;</span>) <span class="op">+</span></span>
<span id="cb74-49"><a href="wgcna.html#cb74-49"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> st<span class="fl">-0.4</span>, </span>
<span id="cb74-50"><a href="wgcna.html#cb74-50"></a>                   <span class="dt">y =</span> sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx], </span>
<span id="cb74-51"><a href="wgcna.html#cb74-51"></a>                   <span class="dt">xend =</span> <span class="dv">0</span>, </span>
<span id="cb74-52"><a href="wgcna.html#cb74-52"></a>                   <span class="dt">yend =</span> sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx]),</span>
<span id="cb74-53"><a href="wgcna.html#cb74-53"></a>               <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.2</span>,<span class="st">&quot;cm&quot;</span>)), </span>
<span id="cb74-54"><a href="wgcna.html#cb74-54"></a>               <span class="dt">size =</span> <span class="fl">0.4</span>) <span class="op">+</span></span>
<span id="cb74-55"><a href="wgcna.html#cb74-55"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;Mean connectivity: &quot;</span>, </span>
<span id="cb74-56"><a href="wgcna.html#cb74-56"></a>                 <span class="kw">round</span>(sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx],<span class="dv">2</span>))) -&gt;<span class="st"> </span>mean_connectivity_plot</span>
<span id="cb74-57"><a href="wgcna.html#cb74-57"></a></span>
<span id="cb74-58"><a href="wgcna.html#cb74-58"></a></span>
<span id="cb74-59"><a href="wgcna.html#cb74-59"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(scale_independence_plot, mean_connectivity_plot, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">align =</span> <span class="st">&quot;h&quot;</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>) -&gt;<span class="st"> </span>si_mc_plot</span>
<span id="cb74-60"><a href="wgcna.html#cb74-60"></a></span>
<span id="cb74-61"><a href="wgcna.html#cb74-61"></a>si_mc_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<p>The number closest to the 0.8 line is 18, and the constructed adjacency matrix most closely fits with scale-free topology. Therefore power is 18.</p>
</div>
<div id="block-wise-network-construction-and-module-detection-wgcna-mgs" class="section level4">
<h4><span class="header-section-number">2.4.3.2</span> Block-wise network construction and module detection (WGCNA MGS)</h4>
<p>The function <code>blockwiseModules</code> will first pre cluster with fast crude clustering method to cluster OTUs into blocks not exceeding the maximum, blocks may therefore not be fully optimal in the end.</p>
<p>Change the parameters here to better reflect your own data.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="wgcna.html#cb75-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">dir.exists</span>(<span class="st">&quot;./result/&quot;</span>)) {</span>
<span id="cb75-2"><a href="wgcna.html#cb75-2"></a>  <span class="kw">dir.create</span>(<span class="st">&quot;./result/&quot;</span>)</span>
<span id="cb75-3"><a href="wgcna.html#cb75-3"></a>}</span>
<span id="cb75-4"><a href="wgcna.html#cb75-4"></a></span>
<span id="cb75-5"><a href="wgcna.html#cb75-5"></a>Run_analysis &lt;-<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb75-6"><a href="wgcna.html#cb75-6"></a><span class="cf">if</span>(Run_analysis){</span>
<span id="cb75-7"><a href="wgcna.html#cb75-7"></a>  modules.mgs &lt;-<span class="st"> </span><span class="kw">blockwiseModules</span>(mgs_t,</span>
<span id="cb75-8"><a href="wgcna.html#cb75-8"></a>                          <span class="dt">power =</span> st, </span>
<span id="cb75-9"><a href="wgcna.html#cb75-9"></a>                          <span class="dt">networkType =</span> <span class="st">&quot;signed&quot;</span>, </span>
<span id="cb75-10"><a href="wgcna.html#cb75-10"></a>                          <span class="dt">TOMType =</span> <span class="st">&quot;signed&quot;</span>,</span>
<span id="cb75-11"><a href="wgcna.html#cb75-11"></a>                          <span class="dt">corType =</span> <span class="st">&#39;bicor&#39;</span>,</span>
<span id="cb75-12"><a href="wgcna.html#cb75-12"></a>                          <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>,</span>
<span id="cb75-13"><a href="wgcna.html#cb75-13"></a>                          <span class="dt">deepSplit =</span> <span class="dv">2</span>,</span>
<span id="cb75-14"><a href="wgcna.html#cb75-14"></a>                          <span class="dt">pamStage =</span> <span class="ot">FALSE</span>, </span>
<span id="cb75-15"><a href="wgcna.html#cb75-15"></a>                          <span class="dt">pamRespectsDendro =</span> <span class="ot">TRUE</span>,</span>
<span id="cb75-16"><a href="wgcna.html#cb75-16"></a>                          <span class="dt">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb75-17"><a href="wgcna.html#cb75-17"></a>                          <span class="dt">replaceMissingAdjacencies =</span> <span class="ot">TRUE</span>,</span>
<span id="cb75-18"><a href="wgcna.html#cb75-18"></a>                          <span class="dt">minModuleSize =</span> <span class="dv">5</span>, <span class="co"># There are fewer otus than genes, and that many might not be connected</span></span>
<span id="cb75-19"><a href="wgcna.html#cb75-19"></a>                          <span class="dt">numericLabels =</span> <span class="ot">TRUE</span>,</span>
<span id="cb75-20"><a href="wgcna.html#cb75-20"></a>                          <span class="dt">saveTOMs =</span> <span class="ot">TRUE</span>,</span>
<span id="cb75-21"><a href="wgcna.html#cb75-21"></a>                          <span class="dt">saveTOMFileBase =</span> <span class="st">&quot;./result/modules.mgs&quot;</span>, <span class="co">#&quot;/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/GVHD/bookdown_GVHD/result/modules.mgs&quot;,</span></span>
<span id="cb75-22"><a href="wgcna.html#cb75-22"></a>                          <span class="dt">verbose =</span> <span class="dv">1</span>)</span>
<span id="cb75-23"><a href="wgcna.html#cb75-23"></a>  </span>
<span id="cb75-24"><a href="wgcna.html#cb75-24"></a>  <span class="kw">rownames</span>(modules.mgs<span class="op">$</span>MEs) &lt;-<span class="st"> </span><span class="kw">rownames</span>(mgs_t)</span>
<span id="cb75-25"><a href="wgcna.html#cb75-25"></a>  <span class="kw">names</span>(modules.mgs<span class="op">$</span>colors) &lt;-<span class="st"> </span><span class="kw">colnames</span>(mgs_t)</span>
<span id="cb75-26"><a href="wgcna.html#cb75-26"></a>  <span class="kw">names</span>(modules.mgs<span class="op">$</span>unmergedColors) &lt;-<span class="st"> </span><span class="kw">colnames</span>(mgs_t)</span>
<span id="cb75-27"><a href="wgcna.html#cb75-27"></a>  </span>
<span id="cb75-28"><a href="wgcna.html#cb75-28"></a>  hubs.mgs &lt;-<span class="st"> </span><span class="kw">chooseTopHubInEachModule</span>(mgs_t, modules.mgs<span class="op">$</span>colors)</span>
<span id="cb75-29"><a href="wgcna.html#cb75-29"></a>}</span></code></pre></div>
<pre><code>##  Calculating module eigengenes block-wise from all genes</code></pre>
</div>
<div id="mgs-module-details-wgcna-mgs" class="section level4">
<h4><span class="header-section-number">2.4.3.3</span> mgs module details (WGCNA MGS)</h4>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="wgcna.html#cb77-1"></a><span class="co">## Convert labels to colors for plotting</span></span>
<span id="cb77-2"><a href="wgcna.html#cb77-2"></a>merged_colors &lt;-<span class="st"> </span><span class="kw">labels2colors</span>(modules.mgs<span class="op">$</span>colors)</span>
<span id="cb77-3"><a href="wgcna.html#cb77-3"></a></span>
<span id="cb77-4"><a href="wgcna.html#cb77-4"></a>n_modules &lt;-<span class="st"> </span><span class="kw">unique</span>(merged_colors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">length</span>()</span>
<span id="cb77-5"><a href="wgcna.html#cb77-5"></a></span>
<span id="cb77-6"><a href="wgcna.html#cb77-6"></a>samples_good &lt;-<span class="st"> </span><span class="kw">sum</span>(modules.mgs<span class="op">$</span>goodSamples) <span class="op">==</span><span class="st"> </span><span class="kw">length</span>(modules.mgs<span class="op">$</span>goodSamples)</span>
<span id="cb77-7"><a href="wgcna.html#cb77-7"></a>OTUs_good &lt;-<span class="st"> </span><span class="kw">sum</span>(modules.mgs<span class="op">$</span>goodGenes) <span class="op">==</span><span class="st"> </span><span class="kw">length</span>(modules.mgs<span class="op">$</span>goodGenes)</span>
<span id="cb77-8"><a href="wgcna.html#cb77-8"></a></span>
<span id="cb77-9"><a href="wgcna.html#cb77-9"></a>ME_good &lt;-<span class="st"> </span><span class="kw">sum</span>(modules.mgs<span class="op">$</span>MEsOK) <span class="op">==</span><span class="st"> </span><span class="kw">length</span>(modules.mgs<span class="op">$</span>MEsOK)</span></code></pre></div>
<p>All samples are OK.<br />
All OTUs are OK.</p>
<p>There where 8 modules found.<br />
All module eigenOTUs are OK.</p>
</div>
<div id="plot-mgs-module-details-wgcna-mgs" class="section level4">
<h4><span class="header-section-number">2.4.3.4</span> plot mgs module details (WGCNA MGS)</h4>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="wgcna.html#cb78-1"></a><span class="kw">table</span>(modules.mgs<span class="op">$</span>colors) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb78-2"><a href="wgcna.html#cb78-2"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb78-3"><a href="wgcna.html#cb78-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">Module =</span> Var1, <span class="dt">Size =</span> Freq) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb78-4"><a href="wgcna.html#cb78-4"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Module_color =</span> <span class="kw">labels2colors</span>(<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(Module)))) -&gt;<span class="st"> </span>module_size</span>
<span id="cb78-5"><a href="wgcna.html#cb78-5"></a></span>
<span id="cb78-6"><a href="wgcna.html#cb78-6"></a>module_size <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb78-7"><a href="wgcna.html#cb78-7"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Module, <span class="dt">y =</span> Size, <span class="dt">fill =</span> Module)) <span class="op">+</span></span>
<span id="cb78-8"><a href="wgcna.html#cb78-8"></a><span class="st">  </span><span class="kw">geom_col</span>(<span class="dt">color =</span>  <span class="st">&quot;#000000&quot;</span>) <span class="op">+</span></span>
<span id="cb78-9"><a href="wgcna.html#cb78-9"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Number of OTUs in each module&quot;</span>) <span class="op">+</span></span>
<span id="cb78-10"><a href="wgcna.html#cb78-10"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb78-11"><a href="wgcna.html#cb78-11"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">setNames</span>(module_size<span class="op">$</span>Module_color,module_size<span class="op">$</span>Module)) <span class="op">+</span></span>
<span id="cb78-12"><a href="wgcna.html#cb78-12"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label =</span> Size),<span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">hjust =</span> <span class="fl">-0.18</span>, <span class="dt">size =</span> <span class="fl">3.5</span>) <span class="op">+</span></span>
<span id="cb78-13"><a href="wgcna.html#cb78-13"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="dv">0</span>, <span class="kw">max</span>(module_size<span class="op">$</span>Size)<span class="op">*</span><span class="fl">1.1</span>) <span class="op">+</span></span>
<span id="cb78-14"><a href="wgcna.html#cb78-14"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">margin</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="st">&quot;pt&quot;</span>)) <span class="op">+</span></span>
<span id="cb78-15"><a href="wgcna.html#cb78-15"></a><span class="st">  </span><span class="kw">coord_flip</span>()-&gt;<span class="st"> </span>module_size_barplot</span>
<span id="cb78-16"><a href="wgcna.html#cb78-16"></a></span>
<span id="cb78-17"><a href="wgcna.html#cb78-17"></a>module_size_barplot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-3-1.png" width="480" /></p>
</div>
<div id="cluster-modules-wgcna-mgs" class="section level4">
<h4><span class="header-section-number">2.4.3.5</span> Cluster modules (WGCNA MGS)</h4>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="wgcna.html#cb79-1"></a><span class="kw">plotEigengeneNetworks</span>(modules.mgs<span class="op">$</span>MEs, <span class="st">&quot;Eigengene adjacency heatmap&quot;</span>,</span>
<span id="cb79-2"><a href="wgcna.html#cb79-2"></a><span class="dt">marDendro =</span> <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">4</span>),</span>
<span id="cb79-3"><a href="wgcna.html#cb79-3"></a><span class="dt">marHeatmap =</span> <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>), <span class="dt">plotDendrograms =</span> T,</span>
<span id="cb79-4"><a href="wgcna.html#cb79-4"></a><span class="dt">xLabelsAngle =</span> <span class="dv">90</span>)</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Cluster%20modules-1.png" width="672" /></p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="wgcna.html#cb80-1"></a><span class="kw">table</span>(modules.mgs<span class="op">$</span>colors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() -&gt;<span class="st"> </span>res</span>
<span id="cb80-2"><a href="wgcna.html#cb80-2"></a>res<span class="op">$</span><span class="st">`</span><span class="dt">Module color</span><span class="st">`</span> &lt;-<span class="st"> </span>WGCNA<span class="op">::</span><span class="kw">labels2colors</span>(<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(res<span class="op">$</span>Var1)))</span>
<span id="cb80-3"><a href="wgcna.html#cb80-3"></a>res &lt;-<span class="st"> </span>res[, <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>)]</span>
<span id="cb80-4"><a href="wgcna.html#cb80-4"></a><span class="kw">colnames</span>(res) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Module&quot;</span>, <span class="st">&quot;Module color&quot;</span>, <span class="st">&quot;Number of OTUs&quot;</span>)</span>
<span id="cb80-5"><a href="wgcna.html#cb80-5"></a><span class="kw">print</span>(res)</span></code></pre></div>
<pre><code>##   Module Module color Number of OTUs
## 1      0         grey            174
## 2      1    turquoise             20
## 3      2         blue             13
## 4      3        brown             12
## 5      4       yellow             10
## 6      5        green              9
## 7      6          red              7
## 8      7        black              5</code></pre>
</div>
<div id="plot-dendrogram-of-module-clusters-wgcna-mgs" class="section level4">
<h4><span class="header-section-number">2.4.3.6</span> Plot dendrogram of module clusters (WGCNA MGS)</h4>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="wgcna.html#cb82-1"></a><span class="co"># Plot the dendrogram and the module colors underneath for each block</span></span>
<span id="cb82-2"><a href="wgcna.html#cb82-2"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="kw">seq_along</span>(modules.mgs<span class="op">$</span>dendrograms)){</span>
<span id="cb82-3"><a href="wgcna.html#cb82-3"></a>  <span class="kw">plotDendroAndColors</span>(modules.mgs<span class="op">$</span>dendrograms[[i]], merged_colors[modules.mgs<span class="op">$</span>blockGenes[[i]]],</span>
<span id="cb82-4"><a href="wgcna.html#cb82-4"></a>                      <span class="st">&quot;Module colors&quot;</span>,</span>
<span id="cb82-5"><a href="wgcna.html#cb82-5"></a>                      <span class="dt">dendroLabels =</span> <span class="ot">FALSE</span>, <span class="dt">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb82-6"><a href="wgcna.html#cb82-6"></a>                      <span class="dt">addGuide =</span> <span class="ot">TRUE</span>, <span class="dt">guideHang =</span> <span class="fl">0.05</span>,</span>
<span id="cb82-7"><a href="wgcna.html#cb82-7"></a>                      <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Cluster Dendrogram</span><span class="ch">\n</span><span class="st">&quot;</span>, </span>
<span id="cb82-8"><a href="wgcna.html#cb82-8"></a>                                    <span class="st">&quot;for block &quot;</span>, </span>
<span id="cb82-9"><a href="wgcna.html#cb82-9"></a>                                    i,<span class="st">&quot;: &quot;</span>,</span>
<span id="cb82-10"><a href="wgcna.html#cb82-10"></a>                                    <span class="kw">length</span>(modules.mgs<span class="op">$</span>blockGenes[[i]]),</span>
<span id="cb82-11"><a href="wgcna.html#cb82-11"></a>                                    <span class="st">&quot; OTUs&quot;</span>))</span>
<span id="cb82-12"><a href="wgcna.html#cb82-12"></a>}</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Plot%20dendrogram%20of%20module%20clusters-1.png" width="960" /></p>
</div>
<div id="module-eigengene-correlation-wgcna-mgs" class="section level4">
<h4><span class="header-section-number">2.4.3.7</span> Module (Eigengene) correlation (WGCNA MGS)</h4>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="wgcna.html#cb83-1"></a>MEs &lt;-<span class="st"> </span>modules.mgs<span class="op">$</span>MEs</span>
<span id="cb83-2"><a href="wgcna.html#cb83-2"></a></span>
<span id="cb83-3"><a href="wgcna.html#cb83-3"></a><span class="co"># Module correlation to other modules</span></span>
<span id="cb83-4"><a href="wgcna.html#cb83-4"></a>MEs_R &lt;-<span class="st"> </span><span class="kw">bicor</span>(MEs, MEs, <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## Warning in bicor(MEs, MEs, maxPOutliers = 0.05): bicor: zero MAD in variable &#39;x&#39;. Pearson correlation was used for individual columns with zero (or missing) MAD.</code></pre>
<pre><code>## Warning in bicor(MEs, MEs, maxPOutliers = 0.05): bicor: zero MAD in variable &#39;y&#39;. Pearson correlation was used for individual columns with zero (or missing) MAD.</code></pre>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="wgcna.html#cb86-1"></a>idx.r &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rownames</span>(MEs_R) <span class="op">==</span><span class="st"> &quot;ME0&quot;</span>)</span>
<span id="cb86-2"><a href="wgcna.html#cb86-2"></a>idx.c &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(MEs_R) <span class="op">==</span><span class="st"> &quot;ME0&quot;</span>)</span>
<span id="cb86-3"><a href="wgcna.html#cb86-3"></a></span>
<span id="cb86-4"><a href="wgcna.html#cb86-4"></a>MEs_R_noME0 &lt;-<span class="st"> </span>MEs_R[<span class="op">-</span>idx.r, <span class="op">-</span>idx.c]</span>
<span id="cb86-5"><a href="wgcna.html#cb86-5"></a></span>
<span id="cb86-6"><a href="wgcna.html#cb86-6"></a>MEs_R[<span class="kw">upper.tri</span>(MEs_R_noME0)] <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb86-7"><a href="wgcna.html#cb86-7"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb86-8"><a href="wgcna.html#cb86-8"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="st">&quot;correlation&quot;</span> =<span class="st"> &quot;.&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb86-9"><a href="wgcna.html#cb86-9"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>correlation)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb86-10"><a href="wgcna.html#cb86-10"></a><span class="st">  </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb86-11"><a href="wgcna.html#cb86-11"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;mgs&quot;</span>,<span class="st">&quot;ME correlation density</span><span class="ch">\n</span><span class="st"> without &quot;</span>,<span class="st">&quot;mgs&quot;</span> ,<span class="st">&quot;ME0&quot;</span>)) -&gt;<span class="st"> </span>MEs_R_density</span>
<span id="cb86-12"><a href="wgcna.html#cb86-12"></a></span>
<span id="cb86-13"><a href="wgcna.html#cb86-13"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(MEs_R, <span class="dt">color =</span> <span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;Blue&quot;</span>, <span class="st">&quot;White&quot;</span>, <span class="st">&quot;Red&quot;</span>))(<span class="dv">100</span>),</span>
<span id="cb86-14"><a href="wgcna.html#cb86-14"></a>                   <span class="dt">silent =</span> T, </span>
<span id="cb86-15"><a href="wgcna.html#cb86-15"></a>                   <span class="dt">breaks =</span> <span class="kw">seq</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>,<span class="dt">length.out =</span> <span class="dv">101</span>),</span>
<span id="cb86-16"><a href="wgcna.html#cb86-16"></a>                   <span class="dt">treeheight_row =</span> <span class="dv">5</span>, </span>
<span id="cb86-17"><a href="wgcna.html#cb86-17"></a>                   <span class="dt">treeheight_col =</span> <span class="dv">5</span>,</span>
<span id="cb86-18"><a href="wgcna.html#cb86-18"></a>                   <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;mgs&quot;</span>,<span class="st">&quot;ME correlation heatmap&quot;</span>),</span>
<span id="cb86-19"><a href="wgcna.html#cb86-19"></a>                   <span class="dt">labels_row =</span> <span class="kw">paste0</span>(<span class="st">&quot;mgs&quot;</span>, <span class="kw">rownames</span>(MEs_R)),</span>
<span id="cb86-20"><a href="wgcna.html#cb86-20"></a>                   <span class="dt">labels_col =</span> <span class="kw">paste0</span>(<span class="st">&quot;mgs&quot;</span>, <span class="kw">colnames</span>(MEs_R))) -&gt;<span class="st"> </span>MEs_R_Corr</span>
<span id="cb86-21"><a href="wgcna.html#cb86-21"></a></span>
<span id="cb86-22"><a href="wgcna.html#cb86-22"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(MEs_R_density, MEs_R_Corr<span class="op">$</span>gtable, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;D&quot;</span>, <span class="st">&quot;E&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.6</span>, <span class="dv">1</span>)) -&gt;<span class="st"> </span>density_eigen</span>
<span id="cb86-23"><a href="wgcna.html#cb86-23"></a></span>
<span id="cb86-24"><a href="wgcna.html#cb86-24"></a>density_eigen</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Module%20(Eigengene)%20correlation%20(WGCNA%20MGS)-1.png" width="960" /></p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="wgcna.html#cb87-1"></a><span class="kw">all</span>(<span class="kw">rownames</span>(mgs_t) <span class="op">==</span><span class="st"> </span><span class="kw">rownames</span>(MEs))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="wgcna.html#cb89-1"></a><span class="kw">dim</span>(mgs_t) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">&quot; samples&quot;</span>, <span class="st">&quot; OTUs&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;32 samples&quot; &quot;250 OTUs&quot;</code></pre>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="wgcna.html#cb91-1"></a>kME &lt;-<span class="st"> </span><span class="kw">bicor</span>(mgs_t, MEs, <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## Warning in bicor(mgs_t, MEs, maxPOutliers = 0.05): bicor: zero MAD in variable &#39;x&#39;. Pearson correlation was used for individual columns with zero (or missing) MAD.</code></pre>
<pre><code>## Warning in bicor(mgs_t, MEs, maxPOutliers = 0.05): bicor: zero MAD in variable &#39;y&#39;. Pearson correlation was used for individual columns with zero (or missing) MAD.</code></pre>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="wgcna.html#cb94-1"></a><span class="kw">dim</span>(kME) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">&quot; OTUs&quot;</span>, <span class="st">&quot; modules&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;250 OTUs&quot;  &quot;8 modules&quot;</code></pre>
</div>
<div id="show-plots-of-the-intra-modular-correlation" class="section level4">
<h4><span class="header-section-number">2.4.3.8</span> Show plots of the intra modular correlation</h4>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="wgcna.html#cb96-1"></a><span class="co">## How the OTUs within a module correlates to the module eigengene</span></span>
<span id="cb96-2"><a href="wgcna.html#cb96-2"></a>intra_cor &lt;-<span class="st"> </span><span class="kw">c</span>()</span>
<span id="cb96-3"><a href="wgcna.html#cb96-3"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(mgs_t)) {</span>
<span id="cb96-4"><a href="wgcna.html#cb96-4"></a>  m &lt;-<span class="st"> </span>modules.mgs<span class="op">$</span>colors[i]</span>
<span id="cb96-5"><a href="wgcna.html#cb96-5"></a>  intra_cor[i] &lt;-<span class="st"> </span>kME[i, <span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>, m)]</span>
<span id="cb96-6"><a href="wgcna.html#cb96-6"></a>  <span class="cf">if</span>(m <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>){</span>
<span id="cb96-7"><a href="wgcna.html#cb96-7"></a>    intra_cor[i] &lt;-<span class="st"> </span>kME[i, <span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>, m)]</span>
<span id="cb96-8"><a href="wgcna.html#cb96-8"></a>  } <span class="cf">else</span>{</span>
<span id="cb96-9"><a href="wgcna.html#cb96-9"></a>    intra_cor[i] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb96-10"><a href="wgcna.html#cb96-10"></a>  }</span>
<span id="cb96-11"><a href="wgcna.html#cb96-11"></a>  </span>
<span id="cb96-12"><a href="wgcna.html#cb96-12"></a>}</span>
<span id="cb96-13"><a href="wgcna.html#cb96-13"></a></span>
<span id="cb96-14"><a href="wgcna.html#cb96-14"></a>idx &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">is.na</span>(intra_cor))</span>
<span id="cb96-15"><a href="wgcna.html#cb96-15"></a>intra_cor &lt;-<span class="st"> </span>intra_cor[<span class="op">-</span>idx]</span>
<span id="cb96-16"><a href="wgcna.html#cb96-16"></a></span>
<span id="cb96-17"><a href="wgcna.html#cb96-17"></a><span class="kw">plot</span>(<span class="kw">density</span>(intra_cor), <span class="dt">main =</span> <span class="st">&quot;Correlations with module-eigenOTU (within module correlation)</span><span class="ch">\n</span><span class="st">No ME0&quot;</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Show%20a%20plot%20of%20the%20intra%20modular%20correlation-1.png" width="768" /></p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="wgcna.html#cb97-1"></a><span class="co">## Show the same thing, but for each module individually, and color by module color.</span></span>
<span id="cb97-2"><a href="wgcna.html#cb97-2"></a></span>
<span id="cb97-3"><a href="wgcna.html#cb97-3"></a><span class="co"># Corr within modules</span></span>
<span id="cb97-4"><a href="wgcna.html#cb97-4"></a>corr_within_module &lt;-<span class="st"> </span><span class="cf">function</span>(mgs_t, modules, <span class="dt">module_x =</span> <span class="dv">1</span>){</span>
<span id="cb97-5"><a href="wgcna.html#cb97-5"></a>  idx.omics_data &lt;-<span class="st"> </span><span class="kw">which</span>(modules<span class="op">$</span>colors <span class="op">==</span><span class="st"> </span>module_x)</span>
<span id="cb97-6"><a href="wgcna.html#cb97-6"></a>  idx.me &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(modules<span class="op">$</span>MEs) <span class="op">==</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>,module_x))</span>
<span id="cb97-7"><a href="wgcna.html#cb97-7"></a>  kME_x &lt;-<span class="st"> </span><span class="kw">bicor</span>(mgs_t[,idx.omics_data], modules<span class="op">$</span>MEs[,idx.me], <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span>
<span id="cb97-8"><a href="wgcna.html#cb97-8"></a>  kME_x</span>
<span id="cb97-9"><a href="wgcna.html#cb97-9"></a>}</span>
<span id="cb97-10"><a href="wgcna.html#cb97-10"></a></span>
<span id="cb97-11"><a href="wgcna.html#cb97-11"></a>ggplot.list &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb97-12"><a href="wgcna.html#cb97-12"></a></span>
<span id="cb97-13"><a href="wgcna.html#cb97-13"></a><span class="cf">for</span>(m <span class="cf">in</span> <span class="kw">colnames</span>(modules.mgs<span class="op">$</span>MEs)){</span>
<span id="cb97-14"><a href="wgcna.html#cb97-14"></a>  h &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">sub</span>(<span class="st">&quot;ME&quot;</span>,<span class="st">&quot;&quot;</span>, m))</span>
<span id="cb97-15"><a href="wgcna.html#cb97-15"></a>  <span class="kw">data.frame</span>(<span class="dt">x =</span> <span class="kw">suppressWarnings</span>(<span class="kw">corr_within_module</span>(<span class="dt">mgs_t =</span> mgs_t, <span class="dt">modules =</span> modules.mgs, <span class="dt">module_x =</span> h))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb97-16"><a href="wgcna.html#cb97-16"></a><span class="st">    </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb97-17"><a href="wgcna.html#cb97-17"></a><span class="st">    </span><span class="kw">geom_density</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x), <span class="dt">fill =</span> <span class="kw">labels2colors</span>(h), <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb97-18"><a href="wgcna.html#cb97-18"></a><span class="st">    </span><span class="kw">xlim</span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb97-19"><a href="wgcna.html#cb97-19"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;OTU correlation&quot;</span>)<span class="op">+</span></span>
<span id="cb97-20"><a href="wgcna.html#cb97-20"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;mgs&quot;</span>,m)) -&gt;<span class="st"> </span>da_plot</span>
<span id="cb97-21"><a href="wgcna.html#cb97-21"></a>  </span>
<span id="cb97-22"><a href="wgcna.html#cb97-22"></a>  ggplot.list[[m]] &lt;-<span class="st"> </span>da_plot</span>
<span id="cb97-23"><a href="wgcna.html#cb97-23"></a>}</span>
<span id="cb97-24"><a href="wgcna.html#cb97-24"></a></span>
<span id="cb97-25"><a href="wgcna.html#cb97-25"></a>ggplot.list &lt;-<span class="st"> </span>ggplot.list[ggplot.list <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">names</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;ME&quot;</span>, <span class="st">&quot;&quot;</span>, .) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.numeric</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">order</span>()]</span>
<span id="cb97-26"><a href="wgcna.html#cb97-26"></a></span>
<span id="cb97-27"><a href="wgcna.html#cb97-27"></a></span>
<span id="cb97-28"><a href="wgcna.html#cb97-28"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> ggplot.list, <span class="dt">ncol =</span> <span class="dv">2</span>) -&gt;<span class="st"> </span>density_all_plot</span>
<span id="cb97-29"><a href="wgcna.html#cb97-29"></a>density_all_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Show%20a%20plot%20of%20the%20intra%20modular%20correlation-2.png" width="768" /></p>
<p>Combine to one plot</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="wgcna.html#cb98-1"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(si_mc_plot , density_eigen, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>)) -&gt;<span class="st"> </span>part_<span class="dv">1</span></span>
<span id="cb98-2"><a href="wgcna.html#cb98-2"></a></span>
<span id="cb98-3"><a href="wgcna.html#cb98-3"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(part_<span class="dv">1</span>, module_size_barplot, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;C&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="fl">0.5</span>)) -&gt;<span class="st"> </span>part_<span class="dv">2</span></span>
<span id="cb98-4"><a href="wgcna.html#cb98-4"></a></span>
<span id="cb98-5"><a href="wgcna.html#cb98-5"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(part_<span class="dv">2</span>, density_all_plot, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;F&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>)</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Combine%20to%20one%20plot-1.png" width="1152" /></p>
</div>
<div id="hub-genes-wgcna-mgs" class="section level4">
<h4><span class="header-section-number">2.4.3.9</span> Hub genes (WGCNA MGS)</h4>
<p>For each module it is possible to pick a hub gene with the function <code>chooseTopHubInEachModule</code>.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="wgcna.html#cb99-1"></a>hubs.mgs <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb99-2"><a href="wgcna.html#cb99-2"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb99-3"><a href="wgcna.html#cb99-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="st">&quot;OTU_name&quot;</span> =<span class="st"> &quot;.&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb99-4"><a href="wgcna.html#cb99-4"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;Module&quot;</span>) -&gt;<span class="st"> </span>hubOTUs</span>
<span id="cb99-5"><a href="wgcna.html#cb99-5"></a></span>
<span id="cb99-6"><a href="wgcna.html#cb99-6"></a></span>
<span id="cb99-7"><a href="wgcna.html#cb99-7"></a></span>
<span id="cb99-8"><a href="wgcna.html#cb99-8"></a>dplyr<span class="op">::</span><span class="kw">left_join</span>(hubOTUs, </span>
<span id="cb99-9"><a href="wgcna.html#cb99-9"></a>                 (taxa_table <span class="op">%&gt;%</span></span>
<span id="cb99-10"><a href="wgcna.html#cb99-10"></a><span class="st">                    </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;OTU_name&quot;</span>)), </span>
<span id="cb99-11"><a href="wgcna.html#cb99-11"></a>                 <span class="dt">by =</span> <span class="st">&quot;OTU_name&quot;</span>) -&gt;<span class="st"> </span>hubOTUs</span>
<span id="cb99-12"><a href="wgcna.html#cb99-12"></a></span>
<span id="cb99-13"><a href="wgcna.html#cb99-13"></a>hubOTUs <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">datatable</span>()</span></code></pre></div>
<div id="htmlwidget-f52419000ffd95477b56" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-f52419000ffd95477b56">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8"],["0","1","2","3","4","5","6","7"],["Lachnospiraceae_bacterium_5_1_63FAA","Megamonas_rupellensis","Streptococcus_agalactiae","Lachnospiraceae_bacterium_5_1_57FAA","Actinomyces_massiliensis","Lactobacillus_plantarum","Erysipelotrichaceae_bacterium_6_1_45","Bacteroides_faecis"],["k__Bacteria","k__Bacteria","k__Bacteria","k__Bacteria","k__Bacteria","k__Bacteria","k__Bacteria","k__Bacteria"],["p__Firmicutes","p__Firmicutes","p__Firmicutes","p__Firmicutes","p__Actinobacteria","p__Firmicutes","p__Firmicutes","p__Bacteroidetes"],["c__Clostridia","c__Negativicutes","c__Bacilli","c__Clostridia","c__Actinobacteria","c__Bacilli","c__Erysipelotrichia","c__Bacteroidia"],["o__Clostridiales","o__Selenomonadales","o__Lactobacillales","o__Clostridiales","o__Actinomycetales","o__Lactobacillales","o__Erysipelotrichales","o__Bacteroidales"],["f__Lachnospiraceae","f__Veillonellaceae","f__Streptococcaceae","f__Lachnospiraceae","f__Actinomycetaceae","f__Lactobacillaceae","f__Erysipelotrichaceae","f__Bacteroidaceae"],["g__Lachnospiraceae_noname","g__Megamonas","g__Streptococcus","g__Lachnospiraceae_noname","g__Actinomyces","g__Lactobacillus","g__Erysipelotrichaceae_noname","g__Bacteroides"],["s__Lachnospiraceae_bacterium_5_1_63FAA","s__Megamonas_rupellensis","s__Streptococcus_agalactiae","s__Lachnospiraceae_bacterium_5_1_57FAA","s__Actinomyces_massiliensis","s__Lactobacillus_plantarum","s__Erysipelotrichaceae_bacterium_6_1_45","s__Bacteroides_faecis"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Module<\/th>\n      <th>OTU_name<\/th>\n      <th>Kingdom<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n      <th>Genus<\/th>\n      <th>Species<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<div id="wgcna-for-metabolic-data" class="section level2">
<h2><span class="header-section-number">2.5</span> WGCNA for metabolic data</h2>
<p><strong>Metabolite data from GvHD project (32 samples with 811 metabolite features) was used as demo data in this sub-chapter.</strong></p>
<div id="data-loading-wgcna-metabolites" class="section level3">
<h3><span class="header-section-number">2.5.1</span> Data loading (WGCNA metabolites)</h3>
<div id="read-demo-input-files-wgcna-metabolites" class="section level4">
<h4><span class="header-section-number">2.5.1.1</span> Read demo input files (WGCNA metabolites)</h4>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="wgcna.html#cb100-1"></a>metabolites_level &lt;-<span class="st"> </span><span class="kw">read.xlsx</span>(<span class="st">&quot;/share/projects/Analytics/analytics/MultiOmics/tools/HAllA/GvHD/ALL_sample_data.xlsx&quot;</span>, <span class="dt">sheetIndex =</span> <span class="dv">1</span>) <span class="co"># read metabolic hirerachy table</span></span>
<span id="cb100-2"><a href="wgcna.html#cb100-2"></a></span>
<span id="cb100-3"><a href="wgcna.html#cb100-3"></a>metabolites_data &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;/share/projects/Analytics/analytics/MultiOmics/tools/HAllA/GvHD/metabolic.txt&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>) <span class="co"># read metabolic data</span></span>
<span id="cb100-4"><a href="wgcna.html#cb100-4"></a></span>
<span id="cb100-5"><a href="wgcna.html#cb100-5"></a><span class="kw">dim</span>(metabolites_level)</span></code></pre></div>
<pre><code>## [1] 1235   78</code></pre>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="wgcna.html#cb102-1"></a><span class="kw">dim</span>(metabolites_data)</span></code></pre></div>
<pre><code>## [1] 811  32</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="wgcna.html#cb104-1"></a><span class="kw">cat</span>(<span class="st">&quot;[metabolites]&quot;</span>, <span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">&quot;Feature: &quot;</span>, <span class="st">&quot;Sample: &quot;</span>), <span class="kw">dim</span>(metabolites_data)))</span></code></pre></div>
<pre><code>## [metabolites] Feature: 811 Sample: 32</code></pre>
</div>
<div id="construct-metabolic-hierarchy-table-wgcna-metabolites" class="section level4">
<h4><span class="header-section-number">2.5.1.2</span> Construct metabolic hierarchy table (WGCNA metabolites)</h4>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="wgcna.html#cb106-1"></a>metabolites_level &lt;-<span class="st"> </span>metabolites_level <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(Index, Class.I, Class.II)</span>
<span id="cb106-2"><a href="wgcna.html#cb106-2"></a>metabolites_level &lt;-<span class="st"> </span>metabolites_level <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;Index&quot;</span>)</span>
<span id="cb106-3"><a href="wgcna.html#cb106-3"></a></span>
<span id="cb106-4"><a href="wgcna.html#cb106-4"></a>metabolites_t &lt;-<span class="st"> </span>metabolites_data <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>()</span></code></pre></div>
</div>
</div>
<div id="build-wgcna-network-wgcna-metabolites" class="section level3">
<h3><span class="header-section-number">2.5.2</span> Build WGCNA network (WGCNA metabolites)</h3>
<div id="select-power-wgcna-metabolites" class="section level4">
<h4><span class="header-section-number">2.5.2.1</span> Select power (WGCNA metabolites)</h4>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="wgcna.html#cb107-1"></a>powers &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="kw">seq</span>(<span class="dv">12</span>,<span class="dv">20</span>,<span class="dv">2</span>))</span>
<span id="cb107-2"><a href="wgcna.html#cb107-2"></a></span>
<span id="cb107-3"><a href="wgcna.html#cb107-3"></a><span class="kw">suppressWarnings</span>(sft &lt;-<span class="st"> </span><span class="kw">pickSoftThreshold</span>(metabolites_t, </span>
<span id="cb107-4"><a href="wgcna.html#cb107-4"></a>                                          <span class="dt">powerVector =</span> powers, </span>
<span id="cb107-5"><a href="wgcna.html#cb107-5"></a>                                          <span class="dt">verbose =</span> <span class="dv">1</span>, </span>
<span id="cb107-6"><a href="wgcna.html#cb107-6"></a>                                          <span class="dt">networkType =</span> <span class="st">&quot;signed&quot;</span>,</span>
<span id="cb107-7"><a href="wgcna.html#cb107-7"></a>                                          <span class="dt">corFn=</span> <span class="st">&quot;bicor&quot;</span>))</span></code></pre></div>
<pre><code>## pickSoftThreshold: will use block size 811.
##  pickSoftThreshold: calculating connectivity for given powers... ..0% ..100% 
##    Power SFT.R.sq   slope truncated.R.sq mean.k. median.k. max.k.
## 1      1  0.72200  9.9100          0.756  456.00    462.00  520.0
## 2      2  0.79300  4.6000          0.924  273.00    278.00  345.0
## 3      3  0.75800  2.8500          0.951  172.00    175.00  238.0
## 4      4  0.57200  1.7000          0.941  113.00    113.00  176.0
## 5      5  0.26700  0.7160          0.932   76.60     76.00  134.0
## 6      6  0.00268  0.0619          0.861   53.80     52.30  108.0
## 7      7  0.11500 -0.3880          0.920   38.90     36.70   89.5
## 8      8  0.30800 -0.7520          0.901   28.90     26.60   75.3
## 9      9  0.41300 -0.9160          0.918   22.00     19.40   64.3
## 10    10  0.53800 -1.2300          0.901   17.00     14.50   55.5
## 11    12  0.64600 -1.5400          0.935   10.80      8.60   42.5
## 12    14  0.71500 -1.6400          0.954    7.32      5.48   33.4
## 13    16  0.75500 -1.6800          0.960    5.23      3.52   26.8
## 14    18  0.77000 -1.7300          0.946    3.90      2.49   21.9
## 15    20  0.74400 -1.7800          0.912    3.03      1.74   18.1</code></pre>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="wgcna.html#cb109-1"></a><span class="co"># Find the soft thresholding power beta to which co-expression similarity is raised to calculate adjacency.</span></span>
<span id="cb109-2"><a href="wgcna.html#cb109-2"></a><span class="co"># based on the criterion of approximate scale-free topology.</span></span>
<span id="cb109-3"><a href="wgcna.html#cb109-3"></a></span>
<span id="cb109-4"><a href="wgcna.html#cb109-4"></a>idx &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">which</span>((<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.90</span>))</span>
<span id="cb109-5"><a href="wgcna.html#cb109-5"></a><span class="cf">if</span>(<span class="kw">is.infinite</span>(idx)){</span>
<span id="cb109-6"><a href="wgcna.html#cb109-6"></a>  idx &lt;-<span class="st"> </span><span class="kw">min</span>(<span class="kw">which</span>((<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.80</span>))</span>
<span id="cb109-7"><a href="wgcna.html#cb109-7"></a>  <span class="cf">if</span>(<span class="op">!</span><span class="kw">is.infinite</span>(idx)){</span>
<span id="cb109-8"><a href="wgcna.html#cb109-8"></a>    st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb109-9"><a href="wgcna.html#cb109-9"></a>  } <span class="cf">else</span>{</span>
<span id="cb109-10"><a href="wgcna.html#cb109-10"></a>    idx &lt;-<span class="st"> </span><span class="kw">which.max</span>(<span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>])</span>
<span id="cb109-11"><a href="wgcna.html#cb109-11"></a>    st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb109-12"><a href="wgcna.html#cb109-12"></a>  }</span>
<span id="cb109-13"><a href="wgcna.html#cb109-13"></a>} <span class="cf">else</span>{</span>
<span id="cb109-14"><a href="wgcna.html#cb109-14"></a>  st &lt;-<span class="st"> </span>sft<span class="op">$</span>fitIndices[idx,<span class="dv">1</span>]</span>
<span id="cb109-15"><a href="wgcna.html#cb109-15"></a>}</span>
<span id="cb109-16"><a href="wgcna.html#cb109-16"></a></span>
<span id="cb109-17"><a href="wgcna.html#cb109-17"></a></span>
<span id="cb109-18"><a href="wgcna.html#cb109-18"></a><span class="co"># Plot Scale independence measure and Mean connectivity measure</span></span>
<span id="cb109-19"><a href="wgcna.html#cb109-19"></a></span>
<span id="cb109-20"><a href="wgcna.html#cb109-20"></a><span class="co"># Scale-free topology fit index as a function of the soft-thresholding power</span></span>
<span id="cb109-21"><a href="wgcna.html#cb109-21"></a><span class="kw">data.frame</span>(<span class="dt">Indices =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb109-22"><a href="wgcna.html#cb109-22"></a>           <span class="dt">sfApprox =</span> <span class="op">-</span><span class="kw">sign</span>(sft<span class="op">$</span>fitIndices[,<span class="dv">3</span>])<span class="op">*</span>sft<span class="op">$</span>fitIndices[,<span class="dv">2</span>]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb109-23"><a href="wgcna.html#cb109-23"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb109-24"><a href="wgcna.html#cb109-24"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="fl">0.9</span>, <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.6</span>) <span class="op">+</span><span class="st"> </span><span class="co"># corresponds to R^2 cut-off of 0.9</span></span>
<span id="cb109-25"><a href="wgcna.html#cb109-25"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="fl">0.8</span>, <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.2</span>) <span class="op">+</span><span class="st"> </span><span class="co"># corresponds to R^2 cut-off of 0.8</span></span>
<span id="cb109-26"><a href="wgcna.html#cb109-26"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> sfApprox), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb109-27"><a href="wgcna.html#cb109-27"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> sfApprox, <span class="dt">label =</span> Indices), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb109-28"><a href="wgcna.html#cb109-28"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Scale independence&quot;</span>) <span class="op">+</span></span>
<span id="cb109-29"><a href="wgcna.html#cb109-29"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Soft Threshold (power)&quot;</span>) <span class="op">+</span></span>
<span id="cb109-30"><a href="wgcna.html#cb109-30"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;SF Model Fit,signed R^2&quot;</span>) <span class="op">+</span></span>
<span id="cb109-31"><a href="wgcna.html#cb109-31"></a><span class="st">  </span><span class="kw">xlim</span>(<span class="dv">1</span>,<span class="dv">20</span>) <span class="op">+</span></span>
<span id="cb109-32"><a href="wgcna.html#cb109-32"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb109-33"><a href="wgcna.html#cb109-33"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> st, <span class="dt">y =</span> <span class="fl">0.25</span>, <span class="dt">xend =</span> st, <span class="dt">yend =</span> sfApprox[idx]<span class="op">-</span><span class="fl">0.05</span>), </span>
<span id="cb109-34"><a href="wgcna.html#cb109-34"></a>               <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.2</span>,<span class="st">&quot;cm&quot;</span>)), </span>
<span id="cb109-35"><a href="wgcna.html#cb109-35"></a>               <span class="dt">size =</span> <span class="fl">0.5</span>)-&gt;<span class="st"> </span>scale_independence_plot </span>
<span id="cb109-36"><a href="wgcna.html#cb109-36"></a>  </span>
<span id="cb109-37"><a href="wgcna.html#cb109-37"></a> </span>
<span id="cb109-38"><a href="wgcna.html#cb109-38"></a></span>
<span id="cb109-39"><a href="wgcna.html#cb109-39"></a></span>
<span id="cb109-40"><a href="wgcna.html#cb109-40"></a><span class="co"># Mean connectivity as a function of the soft-thresholding power</span></span>
<span id="cb109-41"><a href="wgcna.html#cb109-41"></a></span>
<span id="cb109-42"><a href="wgcna.html#cb109-42"></a><span class="kw">data.frame</span>(<span class="dt">Indices =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">1</span>],</span>
<span id="cb109-43"><a href="wgcna.html#cb109-43"></a>           <span class="dt">meanApprox =</span> sft<span class="op">$</span>fitIndices[,<span class="dv">5</span>]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb109-44"><a href="wgcna.html#cb109-44"></a><span class="st">  </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb109-45"><a href="wgcna.html#cb109-45"></a><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> meanApprox), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">size =</span> <span class="fl">2.5</span>) <span class="op">+</span></span>
<span id="cb109-46"><a href="wgcna.html#cb109-46"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="dt">mapping =</span> <span class="kw">aes</span>(<span class="dt">x =</span> Indices, <span class="dt">y =</span> meanApprox, <span class="dt">label =</span> Indices), <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">size =</span> <span class="dv">4</span>) <span class="op">+</span></span>
<span id="cb109-47"><a href="wgcna.html#cb109-47"></a><span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Soft Threshold (power)&quot;</span>) <span class="op">+</span></span>
<span id="cb109-48"><a href="wgcna.html#cb109-48"></a><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Mean Connectivity&quot;</span>) <span class="op">+</span></span>
<span id="cb109-49"><a href="wgcna.html#cb109-49"></a><span class="st">  </span><span class="kw">geom_segment</span>(<span class="kw">aes</span>(<span class="dt">x =</span> st<span class="fl">-0.4</span>, </span>
<span id="cb109-50"><a href="wgcna.html#cb109-50"></a>                   <span class="dt">y =</span> sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx], </span>
<span id="cb109-51"><a href="wgcna.html#cb109-51"></a>                   <span class="dt">xend =</span> <span class="dv">0</span>, </span>
<span id="cb109-52"><a href="wgcna.html#cb109-52"></a>                   <span class="dt">yend =</span> sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx]),</span>
<span id="cb109-53"><a href="wgcna.html#cb109-53"></a>               <span class="dt">arrow =</span> <span class="kw">arrow</span>(<span class="dt">length =</span> <span class="kw">unit</span>(<span class="fl">0.2</span>,<span class="st">&quot;cm&quot;</span>)), </span>
<span id="cb109-54"><a href="wgcna.html#cb109-54"></a>               <span class="dt">size =</span> <span class="fl">0.4</span>) <span class="op">+</span></span>
<span id="cb109-55"><a href="wgcna.html#cb109-55"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;Mean connectivity: &quot;</span>, </span>
<span id="cb109-56"><a href="wgcna.html#cb109-56"></a>                 <span class="kw">round</span>(sft<span class="op">$</span>fitIndices<span class="op">$</span>mean.k.[idx],<span class="dv">2</span>))) -&gt;<span class="st"> </span>mean_connectivity_plot</span>
<span id="cb109-57"><a href="wgcna.html#cb109-57"></a></span>
<span id="cb109-58"><a href="wgcna.html#cb109-58"></a></span>
<span id="cb109-59"><a href="wgcna.html#cb109-59"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(scale_independence_plot, mean_connectivity_plot, <span class="dt">ncol =</span> <span class="dv">2</span>, <span class="dt">align =</span> <span class="st">&quot;h&quot;</span>, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>) -&gt;<span class="st"> </span>si_mc_plot</span>
<span id="cb109-60"><a href="wgcna.html#cb109-60"></a></span>
<span id="cb109-61"><a href="wgcna.html#cb109-61"></a>si_mc_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Select%20power%20(WGCNA%20metabolites)-1.png" width="672" /></p>
<p>The power is 18</p>
</div>
<div id="block-wise-network-construction-and-module-detection-wgcna-metabolites" class="section level4">
<h4><span class="header-section-number">2.5.2.2</span> Block-wise network construction and module detection (WGCNA metabolites)</h4>
<p>The function <code>blockwiseModules</code> will first pre cluster with fast crude clustering method to cluster OTUs into blocks not exceeding the maximum, blocks may therefore not be fully optimal in the end.</p>
<p>Change the parameters here to better reflect your own data.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="wgcna.html#cb110-1"></a>Run_analysis &lt;-<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb110-2"><a href="wgcna.html#cb110-2"></a><span class="cf">if</span>(Run_analysis){</span>
<span id="cb110-3"><a href="wgcna.html#cb110-3"></a>  modules.metabolites &lt;-<span class="st"> </span><span class="kw">blockwiseModules</span>(metabolites_t,</span>
<span id="cb110-4"><a href="wgcna.html#cb110-4"></a>                          <span class="dt">power =</span> st, </span>
<span id="cb110-5"><a href="wgcna.html#cb110-5"></a>                          <span class="dt">networkType =</span> <span class="st">&quot;signed&quot;</span>, </span>
<span id="cb110-6"><a href="wgcna.html#cb110-6"></a>                          <span class="dt">TOMType =</span> <span class="st">&quot;signed&quot;</span>,</span>
<span id="cb110-7"><a href="wgcna.html#cb110-7"></a>                          <span class="dt">corType =</span> <span class="st">&#39;bicor&#39;</span>,</span>
<span id="cb110-8"><a href="wgcna.html#cb110-8"></a>                          <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>,</span>
<span id="cb110-9"><a href="wgcna.html#cb110-9"></a>                          <span class="dt">deepSplit =</span> <span class="dv">2</span>,</span>
<span id="cb110-10"><a href="wgcna.html#cb110-10"></a>                          <span class="dt">pamStage =</span> <span class="ot">FALSE</span>, </span>
<span id="cb110-11"><a href="wgcna.html#cb110-11"></a>                          <span class="dt">pamRespectsDendro =</span> <span class="ot">TRUE</span>,</span>
<span id="cb110-12"><a href="wgcna.html#cb110-12"></a>                          <span class="dt">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb110-13"><a href="wgcna.html#cb110-13"></a>                          <span class="dt">replaceMissingAdjacencies =</span> <span class="ot">TRUE</span>,</span>
<span id="cb110-14"><a href="wgcna.html#cb110-14"></a>                          <span class="dt">minModuleSize =</span> <span class="dv">5</span>, <span class="co"># There are fewer otus than genes, and that many might not be connected</span></span>
<span id="cb110-15"><a href="wgcna.html#cb110-15"></a>                          <span class="dt">numericLabels =</span> <span class="ot">TRUE</span>,</span>
<span id="cb110-16"><a href="wgcna.html#cb110-16"></a>                          <span class="dt">saveTOMs =</span> <span class="ot">TRUE</span>,</span>
<span id="cb110-17"><a href="wgcna.html#cb110-17"></a>                          <span class="dt">saveTOMFileBase =</span> <span class="st">&quot;./result/module.metabolites&quot;</span>, <span class="co">#&quot;/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/GVHD/bookdown_GVHD/result/module.metabolites&quot;,</span></span>
<span id="cb110-18"><a href="wgcna.html#cb110-18"></a>                          <span class="dt">verbose =</span> <span class="dv">1</span>)</span>
<span id="cb110-19"><a href="wgcna.html#cb110-19"></a>  </span>
<span id="cb110-20"><a href="wgcna.html#cb110-20"></a>  <span class="kw">rownames</span>(modules.metabolites<span class="op">$</span>MEs) &lt;-<span class="st"> </span><span class="kw">rownames</span>(metabolites_t)</span>
<span id="cb110-21"><a href="wgcna.html#cb110-21"></a>  <span class="kw">names</span>(modules.metabolites<span class="op">$</span>colors) &lt;-<span class="st"> </span><span class="kw">colnames</span>(metabolites_t)</span>
<span id="cb110-22"><a href="wgcna.html#cb110-22"></a>  <span class="kw">names</span>(modules.metabolites<span class="op">$</span>unmergedColors) &lt;-<span class="st"> </span><span class="kw">colnames</span>(metabolites_t)</span>
<span id="cb110-23"><a href="wgcna.html#cb110-23"></a>  </span>
<span id="cb110-24"><a href="wgcna.html#cb110-24"></a>  hubs.metabolites &lt;-<span class="st"> </span><span class="kw">chooseTopHubInEachModule</span>(metabolites_t, modules.metabolites<span class="op">$</span>colors)</span>
<span id="cb110-25"><a href="wgcna.html#cb110-25"></a>}</span></code></pre></div>
<pre><code>##  Calculating module eigengenes block-wise from all genes</code></pre>
</div>
<div id="metabolites-module-details-wgcna-metabolites" class="section level4">
<h4><span class="header-section-number">2.5.2.3</span> metabolites module details (WGCNA metabolites)</h4>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="wgcna.html#cb112-1"></a><span class="co"># Convert labels to colors for plotting</span></span>
<span id="cb112-2"><a href="wgcna.html#cb112-2"></a>merged_colors &lt;-<span class="st"> </span><span class="kw">labels2colors</span>(modules.metabolites<span class="op">$</span>colors)</span>
<span id="cb112-3"><a href="wgcna.html#cb112-3"></a></span>
<span id="cb112-4"><a href="wgcna.html#cb112-4"></a>n_modules &lt;-<span class="st"> </span><span class="kw">unique</span>(merged_colors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">length</span>()</span>
<span id="cb112-5"><a href="wgcna.html#cb112-5"></a></span>
<span id="cb112-6"><a href="wgcna.html#cb112-6"></a>samples_good &lt;-<span class="st"> </span><span class="kw">sum</span>(modules.metabolites<span class="op">$</span>goodSamples) <span class="op">==</span><span class="st"> </span><span class="kw">length</span>(modules.metabolites<span class="op">$</span>goodSamples)</span>
<span id="cb112-7"><a href="wgcna.html#cb112-7"></a>OTUs_good &lt;-<span class="st"> </span><span class="kw">sum</span>(modules.metabolites<span class="op">$</span>goodGenes) <span class="op">==</span><span class="st"> </span><span class="kw">length</span>(modules.metabolites<span class="op">$</span>goodGenes)</span>
<span id="cb112-8"><a href="wgcna.html#cb112-8"></a></span>
<span id="cb112-9"><a href="wgcna.html#cb112-9"></a>ME_good &lt;-<span class="st"> </span><span class="kw">sum</span>(modules.metabolites<span class="op">$</span>MEsOK) <span class="op">==</span><span class="st"> </span><span class="kw">length</span>(modules.metabolites<span class="op">$</span>MEsOK)</span></code></pre></div>
<p>All samples are OK.<br />
All OTUs are OK.</p>
<p>There where 27 modules found.<br />
All module eigenOTUs are OK.</p>
<p>How many Metabolites are there in each module?</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="wgcna.html#cb113-1"></a><span class="kw">table</span>(modules.metabolites<span class="op">$</span>colors) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb113-2"><a href="wgcna.html#cb113-2"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb113-3"><a href="wgcna.html#cb113-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">Module =</span> Var1, <span class="dt">Size =</span> Freq) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb113-4"><a href="wgcna.html#cb113-4"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">Module_color =</span> <span class="kw">labels2colors</span>(<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(Module)))) -&gt;<span class="st"> </span>module_size</span>
<span id="cb113-5"><a href="wgcna.html#cb113-5"></a></span>
<span id="cb113-6"><a href="wgcna.html#cb113-6"></a>module_size <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb113-7"><a href="wgcna.html#cb113-7"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> Module, <span class="dt">y =</span> Size, <span class="dt">fill =</span> Module)) <span class="op">+</span></span>
<span id="cb113-8"><a href="wgcna.html#cb113-8"></a><span class="st">  </span><span class="kw">geom_col</span>(<span class="dt">color =</span>  <span class="st">&quot;#000000&quot;</span>) <span class="op">+</span></span>
<span id="cb113-9"><a href="wgcna.html#cb113-9"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="st">&quot;Number of Metabolites in each module&quot;</span>) <span class="op">+</span></span>
<span id="cb113-10"><a href="wgcna.html#cb113-10"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb113-11"><a href="wgcna.html#cb113-11"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="dt">values =</span> <span class="kw">setNames</span>(module_size<span class="op">$</span>Module_color,module_size<span class="op">$</span>Module)) <span class="op">+</span></span>
<span id="cb113-12"><a href="wgcna.html#cb113-12"></a><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label =</span> Size),<span class="dt">vjust =</span> <span class="fl">0.5</span>, <span class="dt">hjust =</span> <span class="fl">-0.18</span>, <span class="dt">size =</span> <span class="fl">3.5</span>) <span class="op">+</span></span>
<span id="cb113-13"><a href="wgcna.html#cb113-13"></a><span class="st">  </span><span class="kw">ylim</span>(<span class="dv">0</span>, <span class="kw">max</span>(module_size<span class="op">$</span>Size)<span class="op">*</span><span class="fl">1.1</span>) <span class="op">+</span></span>
<span id="cb113-14"><a href="wgcna.html#cb113-14"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> <span class="kw">margin</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="st">&quot;pt&quot;</span>)) <span class="op">+</span></span>
<span id="cb113-15"><a href="wgcna.html#cb113-15"></a><span class="st">  </span><span class="kw">coord_flip</span>()-&gt;<span class="st"> </span>module_size_barplot</span>
<span id="cb113-16"><a href="wgcna.html#cb113-16"></a></span>
<span id="cb113-17"><a href="wgcna.html#cb113-17"></a>module_size_barplot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/How%20many%20Metabolites%20are%20there%20in%20each%20module?-1.png" width="480" /></p>
<p>cluster relationships between the module</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="wgcna.html#cb114-1"></a><span class="co"># labels2colors(modules.metabolites$colors)</span></span>
<span id="cb114-2"><a href="wgcna.html#cb114-2"></a></span>
<span id="cb114-3"><a href="wgcna.html#cb114-3"></a><span class="kw">plotEigengeneNetworks</span>(modules.metabolites<span class="op">$</span>MEs, <span class="st">&quot;Eigengene adjacency heatmap&quot;</span>,</span>
<span id="cb114-4"><a href="wgcna.html#cb114-4"></a><span class="dt">marDendro =</span> <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">2</span>,<span class="dv">4</span>),</span>
<span id="cb114-5"><a href="wgcna.html#cb114-5"></a><span class="dt">marHeatmap =</span> <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>), <span class="dt">plotDendrograms =</span> T,</span>
<span id="cb114-6"><a href="wgcna.html#cb114-6"></a><span class="dt">xLabelsAngle =</span> <span class="dv">90</span>)</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/cluster%20relationships%20between%20the%20module-1.png" width="672" /></p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="wgcna.html#cb115-1"></a><span class="kw">table</span>(modules.metabolites<span class="op">$</span>colors) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() -&gt;<span class="st"> </span>res</span>
<span id="cb115-2"><a href="wgcna.html#cb115-2"></a>res<span class="op">$</span><span class="st">`</span><span class="dt">Module color</span><span class="st">`</span> &lt;-<span class="st"> </span>WGCNA<span class="op">::</span><span class="kw">labels2colors</span>(<span class="kw">as.numeric</span>(<span class="kw">as.character</span>(res<span class="op">$</span>Var1)))</span>
<span id="cb115-3"><a href="wgcna.html#cb115-3"></a>res &lt;-<span class="st"> </span>res[, <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">2</span>)]</span>
<span id="cb115-4"><a href="wgcna.html#cb115-4"></a><span class="kw">colnames</span>(res) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Module&quot;</span>, <span class="st">&quot;Module color&quot;</span>, <span class="st">&quot;Number of metabolic&quot;</span>)</span>
<span id="cb115-5"><a href="wgcna.html#cb115-5"></a><span class="kw">print</span>(res <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">datatable</span>())</span></code></pre></div>
<p>Dendrogram and module colors</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="wgcna.html#cb116-1"></a><span class="co"># Plot the dendrogram and the module colors underneath for each block</span></span>
<span id="cb116-2"><a href="wgcna.html#cb116-2"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="kw">seq_along</span>(modules.metabolites<span class="op">$</span>dendrograms)){</span>
<span id="cb116-3"><a href="wgcna.html#cb116-3"></a>  <span class="kw">plotDendroAndColors</span>(modules.metabolites<span class="op">$</span>dendrograms[[i]], merged_colors[modules.metabolites<span class="op">$</span>blockGenes[[i]]],</span>
<span id="cb116-4"><a href="wgcna.html#cb116-4"></a>                      <span class="st">&quot;Module colors&quot;</span>,</span>
<span id="cb116-5"><a href="wgcna.html#cb116-5"></a>                      <span class="dt">dendroLabels =</span> <span class="ot">FALSE</span>, <span class="dt">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb116-6"><a href="wgcna.html#cb116-6"></a>                      <span class="dt">addGuide =</span> <span class="ot">TRUE</span>, <span class="dt">guideHang =</span> <span class="fl">0.05</span>,</span>
<span id="cb116-7"><a href="wgcna.html#cb116-7"></a>                      <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;Cluster Dendrogram</span><span class="ch">\n</span><span class="st">&quot;</span>, </span>
<span id="cb116-8"><a href="wgcna.html#cb116-8"></a>                                    <span class="st">&quot;for block &quot;</span>, </span>
<span id="cb116-9"><a href="wgcna.html#cb116-9"></a>                                    i,<span class="st">&quot;: &quot;</span>,</span>
<span id="cb116-10"><a href="wgcna.html#cb116-10"></a>                                    <span class="kw">length</span>(modules.metabolites<span class="op">$</span>blockGenes[[i]]),</span>
<span id="cb116-11"><a href="wgcna.html#cb116-11"></a>                                    <span class="st">&quot; metabolic&quot;</span>))</span>
<span id="cb116-12"><a href="wgcna.html#cb116-12"></a>}</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-5-1.png" width="960" /></p>
<p>Module (Eigengene) correlation (metabolites)</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="wgcna.html#cb117-1"></a>MEs &lt;-<span class="st"> </span>modules.metabolites<span class="op">$</span>MEs</span>
<span id="cb117-2"><a href="wgcna.html#cb117-2"></a></span>
<span id="cb117-3"><a href="wgcna.html#cb117-3"></a><span class="co"># Module correlation to other modules</span></span>
<span id="cb117-4"><a href="wgcna.html#cb117-4"></a>MEs_R &lt;-<span class="st"> </span><span class="kw">bicor</span>(MEs, MEs, <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span>
<span id="cb117-5"><a href="wgcna.html#cb117-5"></a></span>
<span id="cb117-6"><a href="wgcna.html#cb117-6"></a>idx.r &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">rownames</span>(MEs_R) <span class="op">==</span><span class="st"> &quot;ME0&quot;</span>)</span>
<span id="cb117-7"><a href="wgcna.html#cb117-7"></a>idx.c &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(MEs_R) <span class="op">==</span><span class="st"> &quot;ME0&quot;</span>)</span>
<span id="cb117-8"><a href="wgcna.html#cb117-8"></a></span>
<span id="cb117-9"><a href="wgcna.html#cb117-9"></a>MEs_R_noME0 &lt;-<span class="st"> </span>MEs_R[<span class="op">-</span>idx.r, <span class="op">-</span>idx.c]</span>
<span id="cb117-10"><a href="wgcna.html#cb117-10"></a></span>
<span id="cb117-11"><a href="wgcna.html#cb117-11"></a></span>
<span id="cb117-12"><a href="wgcna.html#cb117-12"></a>MEs_R[<span class="kw">upper.tri</span>(MEs_R_noME0)] <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb117-13"><a href="wgcna.html#cb117-13"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb117-14"><a href="wgcna.html#cb117-14"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="st">&quot;correlation&quot;</span> =<span class="st"> &quot;.&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb117-15"><a href="wgcna.html#cb117-15"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x=</span>correlation)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb117-16"><a href="wgcna.html#cb117-16"></a><span class="st">  </span><span class="kw">geom_density</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb117-17"><a href="wgcna.html#cb117-17"></a><span class="st">  </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;mbs&quot;</span>,<span class="st">&quot;ME correlation density</span><span class="ch">\n</span><span class="st"> without &quot;</span>,<span class="st">&quot;mbs&quot;</span> ,<span class="st">&quot;ME0&quot;</span>)) -&gt;<span class="st"> </span>MEs_R_density</span>
<span id="cb117-18"><a href="wgcna.html#cb117-18"></a></span>
<span id="cb117-19"><a href="wgcna.html#cb117-19"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(MEs_R, <span class="dt">color =</span> <span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;Blue&quot;</span>, <span class="st">&quot;White&quot;</span>, <span class="st">&quot;Red&quot;</span>))(<span class="dv">100</span>),</span>
<span id="cb117-20"><a href="wgcna.html#cb117-20"></a>                   <span class="dt">silent =</span> T, </span>
<span id="cb117-21"><a href="wgcna.html#cb117-21"></a>                   <span class="dt">breaks =</span> <span class="kw">seq</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>,<span class="dt">length.out =</span> <span class="dv">101</span>),</span>
<span id="cb117-22"><a href="wgcna.html#cb117-22"></a>                   <span class="dt">treeheight_row =</span> <span class="dv">5</span>, </span>
<span id="cb117-23"><a href="wgcna.html#cb117-23"></a>                   <span class="dt">treeheight_col =</span> <span class="dv">5</span>,</span>
<span id="cb117-24"><a href="wgcna.html#cb117-24"></a>                   <span class="dt">main =</span> <span class="kw">paste0</span>(<span class="st">&quot;mbs&quot;</span>,<span class="st">&quot;ME correlation heatmap&quot;</span>),</span>
<span id="cb117-25"><a href="wgcna.html#cb117-25"></a>                   <span class="dt">labels_row =</span> <span class="kw">paste0</span>(<span class="st">&quot;mbs&quot;</span>, <span class="kw">rownames</span>(MEs_R)),</span>
<span id="cb117-26"><a href="wgcna.html#cb117-26"></a>                   <span class="dt">labels_col =</span> <span class="kw">paste0</span>(<span class="st">&quot;mbs&quot;</span>, <span class="kw">colnames</span>(MEs_R))) -&gt;<span class="st"> </span>MEs_R_Corr</span>
<span id="cb117-27"><a href="wgcna.html#cb117-27"></a></span>
<span id="cb117-28"><a href="wgcna.html#cb117-28"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(MEs_R_density, MEs_R_Corr<span class="op">$</span>gtable, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;D&quot;</span>, <span class="st">&quot;E&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="fl">0.6</span>, <span class="dv">1</span>)) -&gt;<span class="st"> </span>density_eigen</span>
<span id="cb117-29"><a href="wgcna.html#cb117-29"></a></span>
<span id="cb117-30"><a href="wgcna.html#cb117-30"></a>density_eigen</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Module%20(Eigengene)%20correlation%20(metabolites)-1.png" width="960" /></p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="wgcna.html#cb118-1"></a><span class="kw">all</span>(<span class="kw">rownames</span>(metabolites_t) <span class="op">==</span><span class="st"> </span><span class="kw">rownames</span>(MEs))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="wgcna.html#cb120-1"></a><span class="kw">dim</span>(metabolites_t) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">&quot; samples&quot;</span>, <span class="st">&quot; metabolic&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;32 samples&quot;    &quot;811 metabolic&quot;</code></pre>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="wgcna.html#cb122-1"></a>kME &lt;-<span class="st"> </span><span class="kw">bicor</span>(metabolites_t, MEs, <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span>
<span id="cb122-2"><a href="wgcna.html#cb122-2"></a><span class="kw">dim</span>(kME) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">paste0</span>(<span class="kw">c</span>(<span class="st">&quot; metabolic&quot;</span>, <span class="st">&quot; modules&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;811 metabolic&quot; &quot;27 modules&quot;</code></pre>
<p>Show a plot of the intra modular correlation (metabolites)
How the OTUs within a module correlates to the module eigengene.</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="wgcna.html#cb124-1"></a>intra_cor &lt;-<span class="st"> </span><span class="kw">c</span>()</span>
<span id="cb124-2"><a href="wgcna.html#cb124-2"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(metabolites_t)) {</span>
<span id="cb124-3"><a href="wgcna.html#cb124-3"></a>  m &lt;-<span class="st"> </span>modules.metabolites<span class="op">$</span>colors[i]</span>
<span id="cb124-4"><a href="wgcna.html#cb124-4"></a>  intra_cor[i] &lt;-<span class="st"> </span>kME[i, <span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>, m)]</span>
<span id="cb124-5"><a href="wgcna.html#cb124-5"></a>  <span class="cf">if</span>(m <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>){</span>
<span id="cb124-6"><a href="wgcna.html#cb124-6"></a>    intra_cor[i] &lt;-<span class="st"> </span>kME[i, <span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>, m)]</span>
<span id="cb124-7"><a href="wgcna.html#cb124-7"></a>  } <span class="cf">else</span>{</span>
<span id="cb124-8"><a href="wgcna.html#cb124-8"></a>    intra_cor[i] &lt;-<span class="st"> </span><span class="ot">NA</span></span>
<span id="cb124-9"><a href="wgcna.html#cb124-9"></a>  }</span>
<span id="cb124-10"><a href="wgcna.html#cb124-10"></a>  </span>
<span id="cb124-11"><a href="wgcna.html#cb124-11"></a>}</span>
<span id="cb124-12"><a href="wgcna.html#cb124-12"></a></span>
<span id="cb124-13"><a href="wgcna.html#cb124-13"></a>idx &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">is.na</span>(intra_cor))</span>
<span id="cb124-14"><a href="wgcna.html#cb124-14"></a>intra_cor &lt;-<span class="st"> </span>intra_cor[<span class="op">-</span>idx]</span>
<span id="cb124-15"><a href="wgcna.html#cb124-15"></a></span>
<span id="cb124-16"><a href="wgcna.html#cb124-16"></a><span class="kw">plot</span>(<span class="kw">density</span>(intra_cor), <span class="dt">main =</span> <span class="st">&quot;Correlations with module-eigenMBS (within module correlation)</span><span class="ch">\n</span><span class="st">No ME0&quot;</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Show%20a%20plot%20of%20the%20intra%20modular%20correlation%20(metabolites)-1.png" width="672" /></p>
<p>Show the same thing, but for each module individually, and color by module color.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="wgcna.html#cb125-1"></a><span class="co"># Corr within modules</span></span>
<span id="cb125-2"><a href="wgcna.html#cb125-2"></a>corr_within_module &lt;-<span class="st"> </span><span class="cf">function</span>(metabolites_t, modules, <span class="dt">module_x =</span> <span class="dv">1</span>){</span>
<span id="cb125-3"><a href="wgcna.html#cb125-3"></a>  idx.omics_data &lt;-<span class="st"> </span><span class="kw">which</span>(modules<span class="op">$</span>colors <span class="op">==</span><span class="st"> </span>module_x)</span>
<span id="cb125-4"><a href="wgcna.html#cb125-4"></a>  idx.me &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colnames</span>(modules<span class="op">$</span>MEs) <span class="op">==</span><span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;ME&quot;</span>,module_x))</span>
<span id="cb125-5"><a href="wgcna.html#cb125-5"></a>  kME_x &lt;-<span class="st"> </span><span class="kw">bicor</span>(metabolites_t[,idx.omics_data], modules<span class="op">$</span>MEs[,idx.me], <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)</span>
<span id="cb125-6"><a href="wgcna.html#cb125-6"></a>  kME_x</span>
<span id="cb125-7"><a href="wgcna.html#cb125-7"></a>}</span>
<span id="cb125-8"><a href="wgcna.html#cb125-8"></a></span>
<span id="cb125-9"><a href="wgcna.html#cb125-9"></a>ggplot.list &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb125-10"><a href="wgcna.html#cb125-10"></a></span>
<span id="cb125-11"><a href="wgcna.html#cb125-11"></a><span class="cf">for</span>(m <span class="cf">in</span> <span class="kw">colnames</span>(modules.metabolites<span class="op">$</span>MEs)){</span>
<span id="cb125-12"><a href="wgcna.html#cb125-12"></a>  h &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">sub</span>(<span class="st">&quot;ME&quot;</span>,<span class="st">&quot;&quot;</span>, m))</span>
<span id="cb125-13"><a href="wgcna.html#cb125-13"></a>  <span class="kw">data.frame</span>(<span class="dt">x =</span> <span class="kw">suppressWarnings</span>(<span class="kw">corr_within_module</span>(<span class="dt">metabolites_t =</span> metabolites_t, <span class="dt">modules =</span> modules.metabolites, <span class="dt">module_x =</span> h))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb125-14"><a href="wgcna.html#cb125-14"></a><span class="st">    </span><span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb125-15"><a href="wgcna.html#cb125-15"></a><span class="st">    </span><span class="kw">geom_density</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x), <span class="dt">fill =</span> <span class="kw">labels2colors</span>(h), <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>, <span class="dt">alpha =</span> <span class="fl">0.5</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb125-16"><a href="wgcna.html#cb125-16"></a><span class="st">    </span><span class="kw">xlim</span>(<span class="op">-</span><span class="dv">1</span>, <span class="dv">1</span>) <span class="op">+</span></span>
<span id="cb125-17"><a href="wgcna.html#cb125-17"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;metabolic correlation&quot;</span>)<span class="op">+</span></span>
<span id="cb125-18"><a href="wgcna.html#cb125-18"></a><span class="st">    </span><span class="kw">ggtitle</span>(<span class="kw">paste0</span>(<span class="st">&quot;mbs&quot;</span>,m)) -&gt;<span class="st"> </span>da_plot</span>
<span id="cb125-19"><a href="wgcna.html#cb125-19"></a>  </span>
<span id="cb125-20"><a href="wgcna.html#cb125-20"></a>  ggplot.list[[m]] &lt;-<span class="st"> </span>da_plot</span>
<span id="cb125-21"><a href="wgcna.html#cb125-21"></a>}</span>
<span id="cb125-22"><a href="wgcna.html#cb125-22"></a></span>
<span id="cb125-23"><a href="wgcna.html#cb125-23"></a>ggplot.list &lt;-<span class="st"> </span>ggplot.list[ggplot.list <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">names</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;ME&quot;</span>, <span class="st">&quot;&quot;</span>, .) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.numeric</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">order</span>()]</span>
<span id="cb125-24"><a href="wgcna.html#cb125-24"></a></span>
<span id="cb125-25"><a href="wgcna.html#cb125-25"></a></span>
<span id="cb125-26"><a href="wgcna.html#cb125-26"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(<span class="dt">plotlist =</span> ggplot.list, <span class="dt">ncol =</span> <span class="dv">5</span>) -&gt;<span class="st"> </span>density_all_plot <span class="co"># ncol 可以根据module的大小调整</span></span>
<span id="cb125-27"><a href="wgcna.html#cb125-27"></a>density_all_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Show%20the%20same%20thing,%20but%20for%20each%20module%20individually,%20and%20color%20by%20module%20color.-1.png" width="672" /></p>
<p>Combine to one plot (metabolites)</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="wgcna.html#cb126-1"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(si_mc_plot , density_eigen, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>)) -&gt;<span class="st"> </span>part_<span class="dv">1</span></span>
<span id="cb126-2"><a href="wgcna.html#cb126-2"></a></span>
<span id="cb126-3"><a href="wgcna.html#cb126-3"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(part_<span class="dv">1</span>, module_size_barplot, <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;C&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>, <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="fl">0.5</span>)) -&gt;<span class="st"> </span>part_<span class="dv">2</span></span>
<span id="cb126-4"><a href="wgcna.html#cb126-4"></a></span>
<span id="cb126-5"><a href="wgcna.html#cb126-5"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(part_<span class="dv">2</span>, density_all_plot, <span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">rel_heights =</span> <span class="kw">c</span>(<span class="fl">0.8</span>,<span class="dv">1</span>), <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;F&quot;</span>), <span class="dt">label_size =</span> <span class="dv">15</span>)</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Combine%20to%20one%20plot%20(metabolites)-1.png" width="1152" /></p>
</div>
<div id="hub-metabolites-wgcna-metabolites" class="section level4">
<h4><span class="header-section-number">2.5.2.4</span> Hub metabolites (WGCNA metabolites)</h4>
<p>For each module it is possible to pick a hub metabolites with the function <code>chooseTopHubInEachModule</code>.</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="wgcna.html#cb127-1"></a>hubs.metabolites <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb127-2"><a href="wgcna.html#cb127-2"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb127-3"><a href="wgcna.html#cb127-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="st">&quot;mbs_name&quot;</span> =<span class="st"> &quot;.&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb127-4"><a href="wgcna.html#cb127-4"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;Module&quot;</span>) -&gt;<span class="st"> </span>hubMBS</span>
<span id="cb127-5"><a href="wgcna.html#cb127-5"></a></span>
<span id="cb127-6"><a href="wgcna.html#cb127-6"></a></span>
<span id="cb127-7"><a href="wgcna.html#cb127-7"></a></span>
<span id="cb127-8"><a href="wgcna.html#cb127-8"></a>dplyr<span class="op">::</span><span class="kw">left_join</span>(hubMBS, </span>
<span id="cb127-9"><a href="wgcna.html#cb127-9"></a>                 (metabolites_level <span class="op">%&gt;%</span></span>
<span id="cb127-10"><a href="wgcna.html#cb127-10"></a><span class="st">                    </span>tibble<span class="op">::</span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;mbs_name&quot;</span>)), </span>
<span id="cb127-11"><a href="wgcna.html#cb127-11"></a>                 <span class="dt">by =</span> <span class="st">&quot;mbs_name&quot;</span>) -&gt;<span class="st"> </span>hubMBS</span>
<span id="cb127-12"><a href="wgcna.html#cb127-12"></a></span>
<span id="cb127-13"><a href="wgcna.html#cb127-13"></a>hubMBS <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">datatable</span>()</span></code></pre></div>
<div id="htmlwidget-15e402f405d941c85e16" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-15e402f405d941c85e16">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27"],["0","1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26"],["MEDP0170","MEDP0184","MEDN0059","MW0107773","MEDP1465","MW0055479","MW0122632","MEDP0434","MEDN0209","MW0055182","MEDP1201","MEDN0350","MW0012985","MEDN0210","MEDP0101","MEDP0061","MW0157921","MEDN0224","MEDN0498","MEDN0478","MEDP1441","MEDN0570","MEDP1273","MW0167779","MEDN1264","MEDN1287","MW0012612"],["Nucleotide And  Its metabolomics","Hormones and hormone related compunds","Amino acid  and Its metabolomics","-","Alcohol and amines","-","-","GP","Tryptamines,Cholines,Pigments","-","Others","FA","-","Carboxylic acids and derivatives","Benzene and substituted derivatives","Amino acid  and Its metabolomics","-","Carboxylic acids and derivatives","Carboxylic acids and derivatives","Organic acid And  Its derivatives","FA","Carboxylic acids and derivatives","Others","-","GP","GP","-"],["Nucleotide and  Its metabolomics","Hormones and hormone related compunds","Amino acid derivatives","-","Alcohols","-","-","LPC","Tryptamines","-","Others","Oxidized lipids","-","Sugar alcohols","Phenolic acids","Amino acid derivatives","-","Sugars","Phosphate sugars","Organic acid And  Its derivatives","CAR","Sugars","Others","-","LPE","LPE","-"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Module<\/th>\n      <th>mbs_name<\/th>\n      <th>Class.I<\/th>\n      <th>Class.II<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<div id="metabolites-mgs-and-metadata-association" class="section level2">
<h2><span class="header-section-number">2.6</span> Metabolites, mgs and metadata association</h2>
<p>In this subchapter, you will learn to associate the modules extracted from different types of data. Here we associated the modules extracted from <a href="wgcna.html#wgcna-for-mgs-data">MGS</a> and <a href="wgcna.html#wgcna-for-metabolic-data">Metabolites</a> with <strong>cor.test</strong> funciton from <strong>stats</strong> R package.</p>
<div id="use-heatmap-to-show-the-correlation-of-mgs-and-metabolites" class="section level3">
<h3><span class="header-section-number">2.6.1</span> Use heatmap to show the correlation of MGS and metabolites</h3>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="wgcna.html#cb128-1"></a>X_eigengenes &lt;-<span class="st"> </span>modules.metabolites<span class="op">$</span>MEs <span class="co"># X_eigengenes is metabolic</span></span>
<span id="cb128-2"><a href="wgcna.html#cb128-2"></a>Y_eigengenes &lt;-<span class="st"> </span>modules.mgs<span class="op">$</span>MEs <span class="co"># Y_eigengenes is mgs</span></span>
<span id="cb128-3"><a href="wgcna.html#cb128-3"></a></span>
<span id="cb128-4"><a href="wgcna.html#cb128-4"></a><span class="co"># Create a dendrogram of the metabolites eigengenes to organise the final plots.</span></span>
<span id="cb128-5"><a href="wgcna.html#cb128-5"></a>X_ME_dendro &lt;-<span class="st"> </span><span class="kw">hclust</span>(<span class="kw">as.dist</span>(<span class="dv">1</span> <span class="op">-</span><span class="st"> </span>WGCNA<span class="op">::</span><span class="kw">bicor</span>(X_eigengenes, <span class="dt">maxPOutliers =</span> <span class="fl">0.05</span>)), <span class="dt">method =</span> <span class="st">&quot;ward.D2&quot;</span>)</span>
<span id="cb128-6"><a href="wgcna.html#cb128-6"></a></span>
<span id="cb128-7"><a href="wgcna.html#cb128-7"></a>heatmap_colors &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">c</span>(<span class="st">&quot;#18b29f&quot;</span>,<span class="st">&quot;#FFFFFF&quot;</span>,<span class="st">&quot;#ac6721&quot;</span>), <span class="dt">interpolate =</span> <span class="st">&quot;spline&quot;</span>, <span class="dt">space =</span> <span class="st">&quot;rgb&quot;</span>)(<span class="dv">51</span>)</span>
<span id="cb128-8"><a href="wgcna.html#cb128-8"></a></span>
<span id="cb128-9"><a href="wgcna.html#cb128-9"></a>annotation_col &lt;-<span class="st"> </span>metadata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb128-10"><a href="wgcna.html#cb128-10"></a><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span></span>
<span id="cb128-11"><a href="wgcna.html#cb128-11"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb128-12"><a href="wgcna.html#cb128-12"></a><span class="st">  </span><span class="kw">select</span>(V1_outcome, GVHD_type, Donor)</span>
<span id="cb128-13"><a href="wgcna.html#cb128-13"></a>  </span>
<span id="cb128-14"><a href="wgcna.html#cb128-14"></a></span>
<span id="cb128-15"><a href="wgcna.html#cb128-15"></a>annotation_colors &lt;-<span class="st"> </span><span class="kw">list</span>(</span>
<span id="cb128-16"><a href="wgcna.html#cb128-16"></a>  <span class="dt">V1_outcome =</span> <span class="kw">c</span>(<span class="st">`</span><span class="dt">CR</span><span class="st">`</span> =<span class="st"> &quot;#F08A46&quot;</span>, <span class="st">`</span><span class="dt">PR</span><span class="st">`</span> =<span class="st"> &quot;#8EB470&quot;</span>, <span class="st">`</span><span class="dt">NR</span><span class="st">`</span> =<span class="st"> &quot;#B7CFA4&quot;</span>),</span>
<span id="cb128-17"><a href="wgcna.html#cb128-17"></a>  <span class="dt">GVHD_type =</span> <span class="kw">c</span>(<span class="st">`</span><span class="dt">aGVHD</span><span class="st">`</span> =<span class="st"> &quot;red&quot;</span>, <span class="st">`</span><span class="dt">cGVHD</span><span class="st">`</span> =<span class="st"> &quot;green&quot;</span>),</span>
<span id="cb128-18"><a href="wgcna.html#cb128-18"></a>  <span class="dt">Donor =</span> <span class="kw">c</span>(<span class="st">`</span><span class="dt">DO1</span><span class="st">`</span> =<span class="st"> &quot;paleturquoise&quot;</span>, <span class="st">`</span><span class="dt">DO2</span><span class="st">`</span> =<span class="st"> &quot;palevioletred&quot;</span>)</span>
<span id="cb128-19"><a href="wgcna.html#cb128-19"></a>)</span>
<span id="cb128-20"><a href="wgcna.html#cb128-20"></a></span>
<span id="cb128-21"><a href="wgcna.html#cb128-21"></a>X_eigengenes_to_plot &lt;-<span class="st"> </span></span>
<span id="cb128-22"><a href="wgcna.html#cb128-22"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">inner_join</span>(annotation_col <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb128-23"><a href="wgcna.html#cb128-23"></a><span class="st">               </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;sampleName&quot;</span>), </span>
<span id="cb128-24"><a href="wgcna.html#cb128-24"></a>             X_eigengenes <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb128-25"><a href="wgcna.html#cb128-25"></a><span class="st">               </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;sampleName&quot;</span>), </span>
<span id="cb128-26"><a href="wgcna.html#cb128-26"></a>             <span class="dt">by =</span> <span class="st">&quot;sampleName&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb128-27"><a href="wgcna.html#cb128-27"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(V1_outcome, GVHD_type, Donor) <span class="op">%&gt;%</span><span class="st">              </span><span class="co"># The order at which the columns should appear, given that there is no clustering.</span></span>
<span id="cb128-28"><a href="wgcna.html#cb128-28"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(sampleName, <span class="kw">starts_with</span>(<span class="st">&quot;ME&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb128-29"><a href="wgcna.html#cb128-29"></a><span class="st">  </span>tibble<span class="op">::</span><span class="kw">column_to_rownames</span>(<span class="dt">var =</span> <span class="st">&quot;sampleName&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb128-30"><a href="wgcna.html#cb128-30"></a><span class="st">  </span><span class="kw">t</span>()</span>
<span id="cb128-31"><a href="wgcna.html#cb128-31"></a></span>
<span id="cb128-32"><a href="wgcna.html#cb128-32"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(X_eigengenes_to_plot, </span>
<span id="cb128-33"><a href="wgcna.html#cb128-33"></a>                   <span class="dt">cluster_cols =</span> <span class="ot">TRUE</span>,</span>
<span id="cb128-34"><a href="wgcna.html#cb128-34"></a>                   <span class="dt">cluster_rows =</span> X_ME_dendro,</span>
<span id="cb128-35"><a href="wgcna.html#cb128-35"></a>                   <span class="dt">treeheight_row =</span> <span class="dv">20</span>,</span>
<span id="cb128-36"><a href="wgcna.html#cb128-36"></a>                   <span class="dt">cutree_rows =</span> <span class="dv">4</span>,</span>
<span id="cb128-37"><a href="wgcna.html#cb128-37"></a>                   <span class="dt">cutree_cols =</span> <span class="dv">4</span>,</span>
<span id="cb128-38"><a href="wgcna.html#cb128-38"></a>                   <span class="dt">color =</span> heatmap_colors,</span>
<span id="cb128-39"><a href="wgcna.html#cb128-39"></a>                   <span class="dt">fontsize =</span> <span class="dv">10</span>,</span>
<span id="cb128-40"><a href="wgcna.html#cb128-40"></a>                   <span class="dt">fontsize_col =</span> <span class="dv">6</span>,</span>
<span id="cb128-41"><a href="wgcna.html#cb128-41"></a>                   <span class="dt">annotation_colors =</span> annotation_colors,</span>
<span id="cb128-42"><a href="wgcna.html#cb128-42"></a>                   <span class="dt">annotation_col =</span> annotation_col, </span>
<span id="cb128-43"><a href="wgcna.html#cb128-43"></a>                   <span class="dt">silent =</span> F,</span>
<span id="cb128-44"><a href="wgcna.html#cb128-44"></a>                   <span class="dt">labels_row =</span> <span class="kw">paste0</span>(<span class="st">&quot;mb&quot;</span>, <span class="kw">rownames</span>(X_eigengenes_to_plot)),</span>
<span id="cb128-45"><a href="wgcna.html#cb128-45"></a>                   <span class="dt">main =</span> <span class="kw">paste</span>(<span class="st">&quot;Metabolites Module &#39;expression&#39;</span><span class="ch">\n</span><span class="st">&quot;</span>)) -&gt;<span class="st"> </span>X_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Use%20heatmap%20to%20show%20the%20correlation%20of%20MGS%20and%20metabolites-1.png" width="672" /></p>
</div>
<div id="correlate-modules-from-metabolic-and-metagenomics" class="section level3">
<h3><span class="header-section-number">2.6.2</span> Correlate modules from metabolic and metagenomics</h3>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="wgcna.html#cb129-1"></a>p.value_matr &lt;-<span class="st"> </span>corr.value_matr &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dt">ncol =</span> <span class="kw">ncol</span>(Y_eigengenes), </span>
<span id="cb129-2"><a href="wgcna.html#cb129-2"></a>                                          <span class="dt">nrow =</span> <span class="kw">ncol</span>(X_eigengenes), </span>
<span id="cb129-3"><a href="wgcna.html#cb129-3"></a>                                          <span class="dt">dimnames =</span> <span class="kw">list</span>(<span class="kw">colnames</span>(X_eigengenes), </span>
<span id="cb129-4"><a href="wgcna.html#cb129-4"></a>                                                          <span class="kw">colnames</span>(Y_eigengenes)))</span>
<span id="cb129-5"><a href="wgcna.html#cb129-5"></a></span>
<span id="cb129-6"><a href="wgcna.html#cb129-6"></a></span>
<span id="cb129-7"><a href="wgcna.html#cb129-7"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(X_eigengenes)){</span>
<span id="cb129-8"><a href="wgcna.html#cb129-8"></a>  <span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(Y_eigengenes)){</span>
<span id="cb129-9"><a href="wgcna.html#cb129-9"></a>    cor.res &lt;-<span class="st"> </span><span class="kw">cor.test</span>(X_eigengenes[,i], Y_eigengenes[,j])</span>
<span id="cb129-10"><a href="wgcna.html#cb129-10"></a>    p.value_matr[i, j] &lt;-<span class="st"> </span>cor.res<span class="op">$</span>p.value</span>
<span id="cb129-11"><a href="wgcna.html#cb129-11"></a>    corr.value_matr[i, j] &lt;-<span class="st"> </span>cor.res<span class="op">$</span>estimate</span>
<span id="cb129-12"><a href="wgcna.html#cb129-12"></a>  }</span>
<span id="cb129-13"><a href="wgcna.html#cb129-13"></a>}</span>
<span id="cb129-14"><a href="wgcna.html#cb129-14"></a></span>
<span id="cb129-15"><a href="wgcna.html#cb129-15"></a><span class="co"># Correct for number of tests</span></span>
<span id="cb129-16"><a href="wgcna.html#cb129-16"></a>p.value_matr.adjust &lt;-<span class="st"> </span><span class="kw">p.adjust</span>(p.value_matr, <span class="dt">method =</span> <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb129-17"><a href="wgcna.html#cb129-17"></a><span class="kw">dim</span>(p.value_matr.adjust) &lt;-<span class="st"> </span><span class="kw">dim</span>(p.value_matr)</span>
<span id="cb129-18"><a href="wgcna.html#cb129-18"></a><span class="kw">dimnames</span>(p.value_matr.adjust) &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="kw">colnames</span>(X_eigengenes), <span class="kw">colnames</span>(Y_eigengenes))</span>
<span id="cb129-19"><a href="wgcna.html#cb129-19"></a></span>
<span id="cb129-20"><a href="wgcna.html#cb129-20"></a></span>
<span id="cb129-21"><a href="wgcna.html#cb129-21"></a><span class="co"># Add significance level.  </span></span>
<span id="cb129-22"><a href="wgcna.html#cb129-22"></a><span class="co"># One star means a p-value of less than 0.05; Two stars is less than 0.01, and three, is less than 0.001.</span></span>
<span id="cb129-23"><a href="wgcna.html#cb129-23"></a></span>
<span id="cb129-24"><a href="wgcna.html#cb129-24"></a>signif_matrix &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="st">&quot;&quot;</span>, <span class="kw">length</span>(p.value_matr))</span>
<span id="cb129-25"><a href="wgcna.html#cb129-25"></a>three_star &lt;-<span class="st"> </span><span class="kw">which</span>( p.value_matr <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.001</span>)</span>
<span id="cb129-26"><a href="wgcna.html#cb129-26"></a>signif_matrix[three_star] &lt;-<span class="st"> &quot;***&quot;</span></span>
<span id="cb129-27"><a href="wgcna.html#cb129-27"></a>two_star &lt;-<span class="st"> </span><span class="kw">which</span>((p.value_matr <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.01</span>) <span class="op">&amp;</span><span class="st"> </span>(p.value_matr <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.001</span>))</span>
<span id="cb129-28"><a href="wgcna.html#cb129-28"></a>signif_matrix[two_star] &lt;-<span class="st"> &quot;**&quot;</span></span>
<span id="cb129-29"><a href="wgcna.html#cb129-29"></a>one_star &lt;-<span class="st"> </span><span class="kw">which</span>((p.value_matr <span class="op">&lt;=</span><span class="st"> </span><span class="fl">0.05</span>) <span class="op">&amp;</span><span class="st"> </span>(p.value_matr <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.01</span>))</span>
<span id="cb129-30"><a href="wgcna.html#cb129-30"></a>signif_matrix[one_star] &lt;-<span class="st"> &quot;*&quot;</span></span>
<span id="cb129-31"><a href="wgcna.html#cb129-31"></a><span class="kw">dim</span>(signif_matrix) =<span class="st"> </span><span class="kw">dim</span>(p.value_matr) <span class="co"># Give textMatrix the correct dimensions </span></span>
<span id="cb129-32"><a href="wgcna.html#cb129-32"></a></span>
<span id="cb129-33"><a href="wgcna.html#cb129-33"></a></span>
<span id="cb129-34"><a href="wgcna.html#cb129-34"></a><span class="co"># Collect all results into a list.</span></span>
<span id="cb129-35"><a href="wgcna.html#cb129-35"></a>Y_corr_X &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">p_value =</span> p.value_matr, </span>
<span id="cb129-36"><a href="wgcna.html#cb129-36"></a>                 <span class="dt">p_value_adj =</span> p.value_matr.adjust,</span>
<span id="cb129-37"><a href="wgcna.html#cb129-37"></a>                 <span class="dt">signif_matrix =</span> signif_matrix,</span>
<span id="cb129-38"><a href="wgcna.html#cb129-38"></a>                 <span class="dt">correlation =</span> corr.value_matr)</span>
<span id="cb129-39"><a href="wgcna.html#cb129-39"></a><span class="kw">rm</span>(p.value_matr, p.value_matr.adjust, signif_matrix, corr.value_matr)</span>
<span id="cb129-40"><a href="wgcna.html#cb129-40"></a></span>
<span id="cb129-41"><a href="wgcna.html#cb129-41"></a></span>
<span id="cb129-42"><a href="wgcna.html#cb129-42"></a>heatmap_colors &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">rev</span>(RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dt">n =</span> <span class="dv">6</span>, <span class="dt">name =</span><span class="st">&quot;RdBu&quot;</span>)))(<span class="dv">51</span>)</span>
<span id="cb129-43"><a href="wgcna.html#cb129-43"></a></span>
<span id="cb129-44"><a href="wgcna.html#cb129-44"></a>pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(Y_corr_X<span class="op">$</span>correlation, </span>
<span id="cb129-45"><a href="wgcna.html#cb129-45"></a>                   <span class="dt">color =</span> heatmap_colors, </span>
<span id="cb129-46"><a href="wgcna.html#cb129-46"></a>                   <span class="dt">treeheight_col =</span> <span class="dv">0</span>, </span>
<span id="cb129-47"><a href="wgcna.html#cb129-47"></a>                   <span class="dt">treeheight_row =</span> <span class="dv">0</span>,  <span class="co"># will be shown on the transcriptomics ME heatmap</span></span>
<span id="cb129-48"><a href="wgcna.html#cb129-48"></a>                   <span class="dt">cluster_rows =</span> X_ME_dendro,</span>
<span id="cb129-49"><a href="wgcna.html#cb129-49"></a>                   <span class="dt">cutree_rows =</span> <span class="dv">4</span>,</span>
<span id="cb129-50"><a href="wgcna.html#cb129-50"></a>                   <span class="dt">display_numbers =</span> Y_corr_X<span class="op">$</span>signif_matrix, </span>
<span id="cb129-51"><a href="wgcna.html#cb129-51"></a>                   <span class="dt">fontsize_number =</span> <span class="dv">10</span>,</span>
<span id="cb129-52"><a href="wgcna.html#cb129-52"></a>                   <span class="dt">breaks =</span> <span class="kw">seq</span>(<span class="dt">from =</span> <span class="dv">-1</span>, <span class="dt">to =</span> <span class="dv">1</span>, <span class="dt">length.out =</span> <span class="dv">51</span>), </span>
<span id="cb129-53"><a href="wgcna.html#cb129-53"></a>                   <span class="dt">silent =</span> F,</span>
<span id="cb129-54"><a href="wgcna.html#cb129-54"></a>                   <span class="dt">show_rownames =</span> F,</span>
<span id="cb129-55"><a href="wgcna.html#cb129-55"></a>                   <span class="dt">labels_row =</span> <span class="kw">paste0</span>(<span class="st">&quot;mg&quot;</span>, <span class="kw">rownames</span>(Y_corr_X<span class="op">$</span>correlation)),</span>
<span id="cb129-56"><a href="wgcna.html#cb129-56"></a>                   <span class="dt">labels_col =</span> <span class="kw">paste0</span>(<span class="st">&quot;mg&quot;</span>, <span class="kw">colnames</span>(Y_corr_X<span class="op">$</span>correlation)),</span>
<span id="cb129-57"><a href="wgcna.html#cb129-57"></a>                   <span class="dt">main =</span> <span class="st">&quot;EigenOTUs&quot;</span>) -&gt;<span class="st"> </span>Y_corr_X_plot</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Correlate%20modules%20from%20metabolic%20and%20metagenomics-1.png" width="672" /></p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="wgcna.html#cb130-1"></a><span class="co">## Combine heatmaps</span></span>
<span id="cb130-2"><a href="wgcna.html#cb130-2"></a></span>
<span id="cb130-3"><a href="wgcna.html#cb130-3"></a>cowplot<span class="op">::</span><span class="kw">plot_grid</span>(Y_corr_X_plot<span class="op">$</span>gtable,</span>
<span id="cb130-4"><a href="wgcna.html#cb130-4"></a>                   X_plot<span class="op">$</span>gtable,</span>
<span id="cb130-5"><a href="wgcna.html#cb130-5"></a>                   <span class="dt">ncol =</span> <span class="dv">2</span>, </span>
<span id="cb130-6"><a href="wgcna.html#cb130-6"></a>                   <span class="dt">rel_widths =</span> <span class="kw">c</span>(<span class="kw">dim</span>(Y_eigengenes)[<span class="dv">2</span>]<span class="op">/</span><span class="dv">3</span>, </span>
<span id="cb130-7"><a href="wgcna.html#cb130-7"></a>                   <span class="kw">dim</span>(X_eigengenes)[<span class="dv">1</span>]<span class="op">/</span><span class="dv">9</span>),</span>
<span id="cb130-8"><a href="wgcna.html#cb130-8"></a>                   <span class="dt">align =</span> <span class="st">&quot;h&quot;</span>) <span class="op">+</span><span class="st"> </span>ggplot2<span class="op">::</span><span class="kw">theme</span>(<span class="dt">plot.margin =</span> ggplot2<span class="op">::</span><span class="kw">unit</span>(<span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">0</span>,<span class="fl">2.5</span>,<span class="dv">1</span>), <span class="st">&quot;cm&quot;</span>))</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/Correlate%20modules%20from%20metabolic%20and%20metagenomics-2.png" width="672" /></p>
</div>
<div id="visualization-network" class="section level3">
<h3><span class="header-section-number">2.6.3</span> Visualization network</h3>
<p>The strong correlation between mgME3 and mbME1 is the modules of interest, and we wondered if they contain those bacteria and metabolites, which ara published in the existing literature, or if there is a potential mining value.</p>
<p>So we visualize network in mgME3 and mbME1 (In the practical application，find the module you are interested in for visualization)</p>
<p>Visualize mgME3</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="wgcna.html#cb131-1"></a><span class="co">#load(file = &quot;/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/GVHD/bookdown_GVHD/result/modules.mgs-block.1.RData&quot;)</span></span>
<span id="cb131-2"><a href="wgcna.html#cb131-2"></a><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;./result/modules.mgs-block.1.RData&quot;</span>)</span>
<span id="cb131-3"><a href="wgcna.html#cb131-3"></a></span>
<span id="cb131-4"><a href="wgcna.html#cb131-4"></a><span class="co"># The TOM is saved as a dist object and needs to be converted to a matrix</span></span>
<span id="cb131-5"><a href="wgcna.html#cb131-5"></a>TOM &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(TOM)</span>
<span id="cb131-6"><a href="wgcna.html#cb131-6"></a><span class="co"># Add OTU names to the TOM matrix. It is symmetrical so rownames = colnames</span></span>
<span id="cb131-7"><a href="wgcna.html#cb131-7"></a><span class="kw">rownames</span>(TOM) &lt;-<span class="st"> </span><span class="kw">colnames</span>(TOM) &lt;-<span class="st"> </span><span class="kw">names</span>(modules.mgs<span class="op">$</span>colors)</span>
<span id="cb131-8"><a href="wgcna.html#cb131-8"></a></span>
<span id="cb131-9"><a href="wgcna.html#cb131-9"></a><span class="co">## Which taxonomic level should the graph be colored with in addition to modules?</span></span>
<span id="cb131-10"><a href="wgcna.html#cb131-10"></a>selected_taxa &lt;-<span class="st"> &quot;Genus&quot;</span></span></code></pre></div>
<p>Convert module labels and taxonomy to hex colors (MGS)</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="wgcna.html#cb132-1"></a>taxonomy_info &lt;-<span class="st"> </span></span>
<span id="cb132-2"><a href="wgcna.html#cb132-2"></a><span class="st">  </span>taxa_table <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-3"><a href="wgcna.html#cb132-3"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;OTU_name&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb132-4"><a href="wgcna.html#cb132-4"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="st">&quot;OTU_name&quot;</span>, selected_taxa) <span class="op">%&gt;%</span></span>
<span id="cb132-5"><a href="wgcna.html#cb132-5"></a><span class="st">  </span><span class="kw">mutate_all</span>(<span class="dt">.funs =</span> <span class="kw">list</span>(as.character)) </span>
<span id="cb132-6"><a href="wgcna.html#cb132-6"></a></span>
<span id="cb132-7"><a href="wgcna.html#cb132-7"></a>module_info &lt;-<span class="st"> </span></span>
<span id="cb132-8"><a href="wgcna.html#cb132-8"></a><span class="st">  </span>modules.mgs<span class="op">$</span>colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-9"><a href="wgcna.html#cb132-9"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb132-10"><a href="wgcna.html#cb132-10"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;OTU_name&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-11"><a href="wgcna.html#cb132-11"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">Module =</span> <span class="st">&quot;.&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb132-12"><a href="wgcna.html#cb132-12"></a><span class="st">  </span><span class="kw">filter</span>(Module <span class="op">==</span><span class="st"> </span><span class="dv">3</span>)</span>
<span id="cb132-13"><a href="wgcna.html#cb132-13"></a></span>
<span id="cb132-14"><a href="wgcna.html#cb132-14"></a><span class="co"># selected module, here is ME3</span></span>
<span id="cb132-15"><a href="wgcna.html#cb132-15"></a>TOM &lt;-<span class="st"> </span>TOM[module_info<span class="op">$</span>OTU_name, module_info<span class="op">$</span>OTU_name]</span>
<span id="cb132-16"><a href="wgcna.html#cb132-16"></a></span>
<span id="cb132-17"><a href="wgcna.html#cb132-17"></a>graph_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(module_info, taxonomy_info, <span class="dt">by =</span> <span class="st">&quot;OTU_name&quot;</span>)</span>
<span id="cb132-18"><a href="wgcna.html#cb132-18"></a></span>
<span id="cb132-19"><a href="wgcna.html#cb132-19"></a><span class="co"># Converts R-colors to hex colors</span></span>
<span id="cb132-20"><a href="wgcna.html#cb132-20"></a>color2hex &lt;-<span class="st"> </span><span class="cf">function</span>(x){</span>
<span id="cb132-21"><a href="wgcna.html#cb132-21"></a>  x &lt;-<span class="st"> </span><span class="kw">col2rgb</span>(x)</span>
<span id="cb132-22"><a href="wgcna.html#cb132-22"></a>  <span class="kw">rgb</span>(x[<span class="dv">1</span>,], x[<span class="dv">2</span>,], x[<span class="dv">3</span>,], <span class="dt">maxColorValue =</span> <span class="dv">255</span>)</span>
<span id="cb132-23"><a href="wgcna.html#cb132-23"></a>}</span>
<span id="cb132-24"><a href="wgcna.html#cb132-24"></a></span>
<span id="cb132-25"><a href="wgcna.html#cb132-25"></a><span class="co"># Add specific colors to the taxa</span></span>
<span id="cb132-26"><a href="wgcna.html#cb132-26"></a>taxa_colors &lt;-<span class="st"> </span></span>
<span id="cb132-27"><a href="wgcna.html#cb132-27"></a><span class="st">  </span>graph_info <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-28"><a href="wgcna.html#cb132-28"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(selected_taxa) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-29"><a href="wgcna.html#cb132-29"></a><span class="st">  </span><span class="kw">unique</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-30"><a href="wgcna.html#cb132-30"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">tax_color =</span> <span class="kw">colorRampPalette</span>(RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&quot;Accent&quot;</span>))(<span class="kw">nrow</span>(.)))</span>
<span id="cb132-31"><a href="wgcna.html#cb132-31"></a></span>
<span id="cb132-32"><a href="wgcna.html#cb132-32"></a>graph_info_colors &lt;-<span class="st"> </span></span>
<span id="cb132-33"><a href="wgcna.html#cb132-33"></a><span class="st">  </span><span class="kw">left_join</span>(graph_info, taxa_colors) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-34"><a href="wgcna.html#cb132-34"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> WGCNA<span class="op">::</span><span class="kw">labels2colors</span>(Module)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-35"><a href="wgcna.html#cb132-35"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> <span class="kw">color2hex</span>(module_color)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb132-36"><a href="wgcna.html#cb132-36"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> <span class="kw">paste0</span>(module_color,<span class="dv">70</span>))</span></code></pre></div>
<pre><code>## Joining, by = &quot;Genus&quot;</code></pre>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="wgcna.html#cb134-1"></a><span class="co">## If all lines are too thick, reduce the strength (between 0 and 1).</span></span>
<span id="cb134-2"><a href="wgcna.html#cb134-2"></a><span class="co">## The lower the number the weaker the lines.</span></span>
<span id="cb134-3"><a href="wgcna.html#cb134-3"></a></span>
<span id="cb134-4"><a href="wgcna.html#cb134-4"></a>strength_adjust =<span class="st"> </span><span class="dv">1</span></span></code></pre></div>
<p>Construct taxa network with igraph</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="wgcna.html#cb135-1"></a>g &lt;-<span class="st"> </span><span class="kw">graph.adjacency</span>(TOM, <span class="dt">mode=</span><span class="st">&quot;undirected&quot;</span>, <span class="dt">weighted=</span> <span class="ot">TRUE</span>)</span>
<span id="cb135-2"><a href="wgcna.html#cb135-2"></a></span>
<span id="cb135-3"><a href="wgcna.html#cb135-3"></a><span class="co">#~https://stackoverflow.com/questions/28366329/how-to-scale-edge-colors-igraph</span></span>
<span id="cb135-4"><a href="wgcna.html#cb135-4"></a>igraph<span class="op">::</span><span class="kw">delete.edges</span>(g, <span class="kw">which</span>(<span class="kw">E</span>(g)<span class="op">$</span>weight <span class="op">&lt;</span><span class="dv">1</span>))</span></code></pre></div>
<pre><code>## IGRAPH ba692c3 UNW- 12 0 -- 
## + attr: name (v/c), weight (e/n)
## + edges from ba692c3 (vertex names):</code></pre>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="wgcna.html#cb137-1"></a><span class="kw">E</span>(g)<span class="op">$</span>width &lt;-<span class="st"> </span><span class="kw">E</span>(g)<span class="op">$</span>weight<span class="op">*</span>strength_adjust <span class="op">+</span><span class="st"> </span><span class="kw">min</span>(<span class="kw">E</span>(g)<span class="op">$</span>weight)</span>
<span id="cb137-2"><a href="wgcna.html#cb137-2"></a><span class="kw">E</span>(g)<span class="op">$</span>color &lt;-<span class="st"> &quot;red&quot;</span></span>
<span id="cb137-3"><a href="wgcna.html#cb137-3"></a></span>
<span id="cb137-4"><a href="wgcna.html#cb137-4"></a><span class="kw">set.seed</span>(<span class="dv">231</span>) <span class="co"># Ensures the same layout given the same data.</span></span>
<span id="cb137-5"><a href="wgcna.html#cb137-5"></a>l &lt;-<span class="st"> </span><span class="kw">layout_with_fr</span>(g, <span class="dt">weights =</span> <span class="kw">E</span>(g)<span class="op">$</span>weight)</span>
<span id="cb137-6"><a href="wgcna.html#cb137-6"></a></span>
<span id="cb137-7"><a href="wgcna.html#cb137-7"></a></span>
<span id="cb137-8"><a href="wgcna.html#cb137-8"></a><span class="co"># Order graph_info_colors by the graph </span></span>
<span id="cb137-9"><a href="wgcna.html#cb137-9"></a>graph_info_colors &lt;-<span class="st"> </span>graph_info_colors[<span class="kw">which</span>(graph_info_colors<span class="op">$</span>OTU_name <span class="op">%in%</span><span class="st"> </span><span class="kw">V</span>(g)<span class="op">$</span>name),]</span>
<span id="cb137-10"><a href="wgcna.html#cb137-10"></a></span>
<span id="cb137-11"><a href="wgcna.html#cb137-11"></a><span class="co"># Ensure that the names are in the same order</span></span>
<span id="cb137-12"><a href="wgcna.html#cb137-12"></a><span class="cf">if</span>(<span class="kw">all</span>(<span class="kw">V</span>(g)<span class="op">$</span>name <span class="op">==</span><span class="st"> </span>graph_info_colors<span class="op">$</span>OTU_name)){<span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">OTU names match&quot;</span>)}</span></code></pre></div>
<pre><code>## 
## OTU names match</code></pre>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="wgcna.html#cb139-1"></a><span class="co"># Add square shapes to hub OTUs</span></span>
<span id="cb139-2"><a href="wgcna.html#cb139-2"></a><span class="kw">V</span>(g)<span class="op">$</span>shape &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">V</span>(g)<span class="op">$</span>name <span class="op">%in%</span><span class="st"> </span>hubs.mgs[<span class="op">-</span><span class="dv">1</span>], <span class="st">&quot;square&quot;</span>, <span class="st">&quot;circle&quot;</span>) <span class="co">#-1 means dont use module 0</span></span>
<span id="cb139-3"><a href="wgcna.html#cb139-3"></a></span>
<span id="cb139-4"><a href="wgcna.html#cb139-4"></a><span class="co"># OTUs in modules have larger nodes</span></span>
<span id="cb139-5"><a href="wgcna.html#cb139-5"></a><span class="kw">V</span>(g)<span class="op">$</span>size &lt;-<span class="st"> </span><span class="kw">ifelse</span>(graph_info_colors<span class="op">$</span>Module <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">5</span>)</span>
<span id="cb139-6"><a href="wgcna.html#cb139-6"></a></span>
<span id="cb139-7"><a href="wgcna.html#cb139-7"></a><span class="co"># And larger text</span></span>
<span id="cb139-8"><a href="wgcna.html#cb139-8"></a><span class="kw">V</span>(g)<span class="op">$</span>label.cex &lt;-<span class="st"> </span><span class="kw">ifelse</span>(graph_info_colors<span class="op">$</span>Module <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>, <span class="fl">0.8</span>, <span class="fl">0.4</span>)</span>
<span id="cb139-9"><a href="wgcna.html#cb139-9"></a></span>
<span id="cb139-10"><a href="wgcna.html#cb139-10"></a><span class="co"># Remove everything but the number to increase readability</span></span>
<span id="cb139-11"><a href="wgcna.html#cb139-11"></a><span class="kw">V</span>(g)<span class="op">$</span>name =<span class="st">  </span><span class="kw">sub</span>(<span class="st">&quot;OTU_&quot;</span>, <span class="st">&quot;&quot;</span>, graph_info_colors<span class="op">$</span>OTU_name)</span></code></pre></div>
<p>Find distinct entires for the plot legends</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="wgcna.html#cb140-1"></a>module_labels &lt;-<span class="st"> </span></span>
<span id="cb140-2"><a href="wgcna.html#cb140-2"></a><span class="st">  </span>graph_info_colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb140-3"><a href="wgcna.html#cb140-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(Module, module_color) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb140-4"><a href="wgcna.html#cb140-4"></a><span class="st">  </span><span class="kw">distinct</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb140-5"><a href="wgcna.html#cb140-5"></a><span class="st">  </span><span class="kw">arrange</span>(Module)</span>
<span id="cb140-6"><a href="wgcna.html#cb140-6"></a></span>
<span id="cb140-7"><a href="wgcna.html#cb140-7"></a>tax_labels &lt;-<span class="st"> </span></span>
<span id="cb140-8"><a href="wgcna.html#cb140-8"></a><span class="st">  </span>graph_info_colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb140-9"><a href="wgcna.html#cb140-9"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(selected_taxa, tax_color) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb140-10"><a href="wgcna.html#cb140-10"></a><span class="st">  </span><span class="kw">distinct</span>()</span></code></pre></div>
<p>Plot the graphs, leftmost is colored by module, rightmost is colored by taxonomic classification</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="wgcna.html#cb141-1"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb141-2"><a href="wgcna.html#cb141-2"></a></span>
<span id="cb141-3"><a href="wgcna.html#cb141-3"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb141-4"><a href="wgcna.html#cb141-4"></a></span>
<span id="cb141-5"><a href="wgcna.html#cb141-5"></a><span class="co"># plot(g, layout = l, vertex.color = graph_info_colors$module_color)</span></span>
<span id="cb141-6"><a href="wgcna.html#cb141-6"></a><span class="co"># # legend(&quot;topleft&quot;, legend = paste0(&quot;mM&quot;, 0:(nrow(module_labels)-1)), fill=module_labels$module_color)</span></span>
<span id="cb141-7"><a href="wgcna.html#cb141-7"></a><span class="co"># legend(&quot;topleft&quot;, legend = paste0(&quot;mgsM&quot;, 3), fill=module_labels$module_color)</span></span>
<span id="cb141-8"><a href="wgcna.html#cb141-8"></a></span>
<span id="cb141-9"><a href="wgcna.html#cb141-9"></a><span class="kw">plot</span>(g, <span class="dt">layout =</span> l, <span class="dt">vertex.color =</span> graph_info_colors<span class="op">$</span>tax_color)</span>
<span id="cb141-10"><a href="wgcna.html#cb141-10"></a><span class="kw">legend</span>(<span class="st">&quot;topleft&quot;</span>, <span class="dt">legend =</span> tax_labels<span class="op">$</span>Genus, <span class="dt">fill=</span>tax_labels<span class="op">$</span>tax_color)</span>
<span id="cb141-11"><a href="wgcna.html#cb141-11"></a></span>
<span id="cb141-12"><a href="wgcna.html#cb141-12"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-6-1.png" width="1440" /></p>
<p>Visualize mbME1</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="wgcna.html#cb142-1"></a><span class="co">#load(file = &quot;/share/projects/Analytics/analytics/MultiOmics/tools/WGCNA/test/GVHD/bookdown_GVHD/result/module.metabolites-block.1.RData&quot;)</span></span>
<span id="cb142-2"><a href="wgcna.html#cb142-2"></a><span class="kw">load</span>(<span class="dt">file =</span> <span class="st">&quot;./result/module.metabolites-block.1.RData&quot;</span>)</span>
<span id="cb142-3"><a href="wgcna.html#cb142-3"></a><span class="co"># The TOM is saved as a dist object and needs to be converted to a matrix</span></span>
<span id="cb142-4"><a href="wgcna.html#cb142-4"></a>TOM &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(TOM)</span>
<span id="cb142-5"><a href="wgcna.html#cb142-5"></a><span class="co"># Add OTU names to the TOM matrix. It is symmetrical so rownames = colnames</span></span>
<span id="cb142-6"><a href="wgcna.html#cb142-6"></a><span class="kw">rownames</span>(TOM) &lt;-<span class="st"> </span><span class="kw">colnames</span>(TOM) &lt;-<span class="st"> </span><span class="kw">names</span>(modules.metabolites<span class="op">$</span>colors)</span>
<span id="cb142-7"><a href="wgcna.html#cb142-7"></a></span>
<span id="cb142-8"><a href="wgcna.html#cb142-8"></a><span class="co">## Which metabolic level should the graph be colored with in addition to modules?</span></span>
<span id="cb142-9"><a href="wgcna.html#cb142-9"></a>selected_taxa &lt;-<span class="st"> &quot;Class.II&quot;</span></span></code></pre></div>
<p>Convert module labels and taxonomy to hex colors</p>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="wgcna.html#cb143-1"></a>taxonomy_info &lt;-<span class="st"> </span></span>
<span id="cb143-2"><a href="wgcna.html#cb143-2"></a><span class="st">  </span>metabolites_level <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb143-3"><a href="wgcna.html#cb143-3"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;mbs_name&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb143-4"><a href="wgcna.html#cb143-4"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="st">&quot;mbs_name&quot;</span>, selected_taxa) <span class="op">%&gt;%</span></span>
<span id="cb143-5"><a href="wgcna.html#cb143-5"></a><span class="st">  </span><span class="kw">mutate_all</span>(<span class="dt">.funs =</span> <span class="kw">list</span>(as.character)) </span>
<span id="cb143-6"><a href="wgcna.html#cb143-6"></a></span>
<span id="cb143-7"><a href="wgcna.html#cb143-7"></a>module_info &lt;-<span class="st"> </span></span>
<span id="cb143-8"><a href="wgcna.html#cb143-8"></a><span class="st">  </span>modules.metabolites<span class="op">$</span>colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb143-9"><a href="wgcna.html#cb143-9"></a><span class="st">  </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb143-10"><a href="wgcna.html#cb143-10"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;mbs_name&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb143-11"><a href="wgcna.html#cb143-11"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">rename</span>(<span class="dt">Module =</span> <span class="st">&quot;.&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb143-12"><a href="wgcna.html#cb143-12"></a><span class="st">  </span><span class="kw">filter</span>(Module <span class="op">==</span><span class="st"> </span><span class="dv">1</span>)</span>
<span id="cb143-13"><a href="wgcna.html#cb143-13"></a></span>
<span id="cb143-14"><a href="wgcna.html#cb143-14"></a><span class="co"># selected module, here is ME1</span></span>
<span id="cb143-15"><a href="wgcna.html#cb143-15"></a>TOM &lt;-<span class="st"> </span>TOM[module_info<span class="op">$</span>mbs_name, module_info<span class="op">$</span>mbs_name]</span>
<span id="cb143-16"><a href="wgcna.html#cb143-16"></a></span>
<span id="cb143-17"><a href="wgcna.html#cb143-17"></a>graph_info &lt;-<span class="st"> </span><span class="kw">left_join</span>(module_info, taxonomy_info, <span class="dt">by =</span> <span class="st">&quot;mbs_name&quot;</span>)</span>
<span id="cb143-18"><a href="wgcna.html#cb143-18"></a></span>
<span id="cb143-19"><a href="wgcna.html#cb143-19"></a><span class="co"># Converts R-colors to hex colors</span></span>
<span id="cb143-20"><a href="wgcna.html#cb143-20"></a>color2hex &lt;-<span class="st"> </span><span class="cf">function</span>(x){</span>
<span id="cb143-21"><a href="wgcna.html#cb143-21"></a>  x &lt;-<span class="st"> </span><span class="kw">col2rgb</span>(x)</span>
<span id="cb143-22"><a href="wgcna.html#cb143-22"></a>  <span class="kw">rgb</span>(x[<span class="dv">1</span>,], x[<span class="dv">2</span>,], x[<span class="dv">3</span>,], <span class="dt">maxColorValue =</span> <span class="dv">255</span>)</span>
<span id="cb143-23"><a href="wgcna.html#cb143-23"></a>}</span>
<span id="cb143-24"><a href="wgcna.html#cb143-24"></a></span>
<span id="cb143-25"><a href="wgcna.html#cb143-25"></a><span class="co"># Add specific colors to the taxa</span></span>
<span id="cb143-26"><a href="wgcna.html#cb143-26"></a>taxa_colors &lt;-<span class="st"> </span></span>
<span id="cb143-27"><a href="wgcna.html#cb143-27"></a><span class="st">  </span>graph_info <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb143-28"><a href="wgcna.html#cb143-28"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(selected_taxa) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb143-29"><a href="wgcna.html#cb143-29"></a><span class="st">  </span><span class="kw">unique</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb143-30"><a href="wgcna.html#cb143-30"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">tax_color =</span> <span class="kw">colorRampPalette</span>(RColorBrewer<span class="op">::</span><span class="kw">brewer.pal</span>(<span class="dv">8</span>, <span class="st">&quot;Accent&quot;</span>))(<span class="kw">nrow</span>(.)))</span>
<span id="cb143-31"><a href="wgcna.html#cb143-31"></a></span>
<span id="cb143-32"><a href="wgcna.html#cb143-32"></a>graph_info_colors &lt;-<span class="st"> </span></span>
<span id="cb143-33"><a href="wgcna.html#cb143-33"></a><span class="st">  </span><span class="kw">left_join</span>(graph_info, taxa_colors) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb143-34"><a href="wgcna.html#cb143-34"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> WGCNA<span class="op">::</span><span class="kw">labels2colors</span>(Module)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb143-35"><a href="wgcna.html#cb143-35"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> <span class="kw">color2hex</span>(module_color)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb143-36"><a href="wgcna.html#cb143-36"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">module_color =</span> <span class="kw">paste0</span>(module_color,<span class="dv">70</span>))</span></code></pre></div>
<pre><code>## Joining, by = &quot;Class.II&quot;</code></pre>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="wgcna.html#cb145-1"></a><span class="co">## If all lines are too thick, reduce the strength (between 0 and 1).</span></span>
<span id="cb145-2"><a href="wgcna.html#cb145-2"></a><span class="co">## The lower the number the weaker the lines.</span></span>
<span id="cb145-3"><a href="wgcna.html#cb145-3"></a></span>
<span id="cb145-4"><a href="wgcna.html#cb145-4"></a>strength_adjust =<span class="st"> </span><span class="fl">0.2</span></span></code></pre></div>
<p>Construct metabolites network with igraph</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="wgcna.html#cb146-1"></a>g &lt;-<span class="st"> </span><span class="kw">graph.adjacency</span>(TOM, <span class="dt">mode=</span><span class="st">&quot;undirected&quot;</span>, <span class="dt">weighted=</span> <span class="ot">TRUE</span>)</span>
<span id="cb146-2"><a href="wgcna.html#cb146-2"></a></span>
<span id="cb146-3"><a href="wgcna.html#cb146-3"></a><span class="co">#~https://stackoverflow.com/questions/28366329/how-to-scale-edge-colors-igraph</span></span>
<span id="cb146-4"><a href="wgcna.html#cb146-4"></a>igraph<span class="op">::</span><span class="kw">delete.edges</span>(g, <span class="kw">which</span>(<span class="kw">E</span>(g)<span class="op">$</span>weight <span class="op">&lt;</span><span class="dv">1</span>))</span></code></pre></div>
<pre><code>## IGRAPH f4aba4d UNW- 54 0 -- 
## + attr: name (v/c), weight (e/n)
## + edges from f4aba4d (vertex names):</code></pre>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="wgcna.html#cb148-1"></a><span class="kw">E</span>(g)<span class="op">$</span>width &lt;-<span class="st"> </span><span class="kw">E</span>(g)<span class="op">$</span>weight<span class="op">*</span>strength_adjust <span class="op">+</span><span class="st"> </span><span class="kw">min</span>(<span class="kw">E</span>(g)<span class="op">$</span>weight)</span>
<span id="cb148-2"><a href="wgcna.html#cb148-2"></a><span class="kw">E</span>(g)<span class="op">$</span>color &lt;-<span class="st"> &quot;red&quot;</span></span>
<span id="cb148-3"><a href="wgcna.html#cb148-3"></a></span>
<span id="cb148-4"><a href="wgcna.html#cb148-4"></a><span class="kw">set.seed</span>(<span class="dv">231</span>) <span class="co"># Ensures the same layout given the same data.</span></span>
<span id="cb148-5"><a href="wgcna.html#cb148-5"></a>l &lt;-<span class="st"> </span><span class="kw">layout_with_fr</span>(g, <span class="dt">weights =</span> <span class="kw">E</span>(g)<span class="op">$</span>weight)</span>
<span id="cb148-6"><a href="wgcna.html#cb148-6"></a></span>
<span id="cb148-7"><a href="wgcna.html#cb148-7"></a></span>
<span id="cb148-8"><a href="wgcna.html#cb148-8"></a><span class="co"># Order graph_info_colors by the graph </span></span>
<span id="cb148-9"><a href="wgcna.html#cb148-9"></a>graph_info_colors &lt;-<span class="st"> </span>graph_info_colors[<span class="kw">which</span>(graph_info_colors<span class="op">$</span>mbs_name <span class="op">%in%</span><span class="st"> </span><span class="kw">V</span>(g)<span class="op">$</span>name),]</span>
<span id="cb148-10"><a href="wgcna.html#cb148-10"></a></span>
<span id="cb148-11"><a href="wgcna.html#cb148-11"></a><span class="co"># Ensure that the names are in the same order</span></span>
<span id="cb148-12"><a href="wgcna.html#cb148-12"></a><span class="cf">if</span>(<span class="kw">all</span>(<span class="kw">V</span>(g)<span class="op">$</span>name <span class="op">==</span><span class="st"> </span>graph_info_colors<span class="op">$</span>mbs_name)){<span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">mbs names match&quot;</span>)}</span></code></pre></div>
<pre><code>## 
## mbs names match</code></pre>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="wgcna.html#cb150-1"></a><span class="co"># Add square shapes to hub OTUs</span></span>
<span id="cb150-2"><a href="wgcna.html#cb150-2"></a><span class="kw">V</span>(g)<span class="op">$</span>shape &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">V</span>(g)<span class="op">$</span>name <span class="op">%in%</span><span class="st"> </span>hubs.metabolites[<span class="op">-</span><span class="dv">1</span>], <span class="st">&quot;square&quot;</span>, <span class="st">&quot;circle&quot;</span>) <span class="co">#-1 means dont use module 0</span></span>
<span id="cb150-3"><a href="wgcna.html#cb150-3"></a></span>
<span id="cb150-4"><a href="wgcna.html#cb150-4"></a><span class="co"># OTUs in modules have larger nodes</span></span>
<span id="cb150-5"><a href="wgcna.html#cb150-5"></a><span class="kw">V</span>(g)<span class="op">$</span>size &lt;-<span class="st"> </span><span class="kw">ifelse</span>(graph_info_colors<span class="op">$</span>Module <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>, <span class="dv">10</span>, <span class="dv">5</span>)</span>
<span id="cb150-6"><a href="wgcna.html#cb150-6"></a></span>
<span id="cb150-7"><a href="wgcna.html#cb150-7"></a><span class="co"># And larger text</span></span>
<span id="cb150-8"><a href="wgcna.html#cb150-8"></a><span class="kw">V</span>(g)<span class="op">$</span>label.cex &lt;-<span class="st"> </span><span class="kw">ifelse</span>(graph_info_colors<span class="op">$</span>Module <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>, <span class="fl">0.8</span>, <span class="fl">0.4</span>)</span>
<span id="cb150-9"><a href="wgcna.html#cb150-9"></a></span>
<span id="cb150-10"><a href="wgcna.html#cb150-10"></a><span class="co"># Remove everything but the number to increase readability</span></span>
<span id="cb150-11"><a href="wgcna.html#cb150-11"></a><span class="kw">V</span>(g)<span class="op">$</span>name =<span class="st">  </span><span class="kw">sub</span>(<span class="st">&quot;mbs_&quot;</span>, <span class="st">&quot;&quot;</span>, graph_info_colors<span class="op">$</span>mbs_name)</span></code></pre></div>
<p>Find distinct entires for the plot legends metabolites</p>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="wgcna.html#cb151-1"></a>module_labels &lt;-<span class="st"> </span></span>
<span id="cb151-2"><a href="wgcna.html#cb151-2"></a><span class="st">  </span>graph_info_colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb151-3"><a href="wgcna.html#cb151-3"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(Module, module_color) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb151-4"><a href="wgcna.html#cb151-4"></a><span class="st">  </span><span class="kw">distinct</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb151-5"><a href="wgcna.html#cb151-5"></a><span class="st">  </span><span class="kw">arrange</span>(Module)</span>
<span id="cb151-6"><a href="wgcna.html#cb151-6"></a></span>
<span id="cb151-7"><a href="wgcna.html#cb151-7"></a>tax_labels &lt;-<span class="st"> </span></span>
<span id="cb151-8"><a href="wgcna.html#cb151-8"></a><span class="st">  </span>graph_info_colors <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb151-9"><a href="wgcna.html#cb151-9"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(selected_taxa, tax_color) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb151-10"><a href="wgcna.html#cb151-10"></a><span class="st">  </span><span class="kw">distinct</span>()</span></code></pre></div>
<p>Plot the graphs, leftmost is colored by module, rightmost is colored by taxonomic classification</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="wgcna.html#cb152-1"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb152-2"><a href="wgcna.html#cb152-2"></a></span>
<span id="cb152-3"><a href="wgcna.html#cb152-3"></a><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb152-4"><a href="wgcna.html#cb152-4"></a></span>
<span id="cb152-5"><a href="wgcna.html#cb152-5"></a><span class="co"># plot(g, layout = l, vertex.color = graph_info_colors$module_color)</span></span>
<span id="cb152-6"><a href="wgcna.html#cb152-6"></a><span class="co"># # legend(&quot;topleft&quot;, legend = paste0(&quot;mM&quot;, 0:(nrow(module_labels)-1)), fill=module_labels$module_color)</span></span>
<span id="cb152-7"><a href="wgcna.html#cb152-7"></a><span class="co"># legend(&quot;topleft&quot;, legend = paste0(&quot;mbM&quot;, 1), fill=module_labels$module_color)</span></span>
<span id="cb152-8"><a href="wgcna.html#cb152-8"></a></span>
<span id="cb152-9"><a href="wgcna.html#cb152-9"></a><span class="kw">plot</span>(g, <span class="dt">layout =</span> l, <span class="dt">vertex.color =</span> graph_info_colors<span class="op">$</span>tax_color)</span>
<span id="cb152-10"><a href="wgcna.html#cb152-10"></a><span class="kw">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="dt">legend =</span> tax_labels<span class="op">$</span>Class.II, <span class="dt">fill=</span>tax_labels<span class="op">$</span>tax_color)</span>
<span id="cb152-11"><a href="wgcna.html#cb152-11"></a></span>
<span id="cb152-12"><a href="wgcna.html#cb152-12"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p><img src="OMics-analysis-workflow_files/figure-html/unnamed-chunk-7-1.png" width="1440" /></p>
</div>
</div>
<div id="session-info" class="section level2">
<h2><span class="header-section-number">2.7</span> Session Info</h2>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="wgcna.html#cb153-1"></a>devtools<span class="op">::</span><span class="kw">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 3.6.3 (2020-02-29)
##  os       Ubuntu 16.04.7 LTS
##  system   x86_64, linux-gnu
##  ui       RStudio
##  language (EN)
##  collate  en_IN.UTF-8
##  ctype    en_IN.UTF-8
##  tz       Asia/Hong_Kong
##  date     2022-09-15
##  rstudio  1.1.419 (server)
##  pandoc   2.7.3 @ /usr/bin/ (via rmarkdown)
## 
## ─ Packages ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  ! package        * version date (UTC) lib source
##    ade4             1.7-17  2021-06-17 [1] CRAN (R 3.6.3)
##    AnnotationDbi    1.58.0  2022-04-26 [1] Bioconductor
##    ape              5.5     2021-04-25 [1] CRAN (R 3.6.3)
##    assertthat       0.2.1   2019-03-21 [2] CRAN (R 3.6.3)
##    backports        1.4.1   2021-12-13 [1] CRAN (R 3.6.3)
##    base64enc        0.1-3   2015-07-28 [2] CRAN (R 3.6.3)
##    Biobase          2.46.0  2019-10-29 [2] Bioconductor
##    BiocGenerics     0.32.0  2019-10-29 [2] Bioconductor
##    biomformat       1.14.0  2019-10-29 [1] Bioconductor
##    Biostrings       2.54.0  2019-10-29 [1] Bioconductor
##    bit              4.0.4   2020-08-04 [1] CRAN (R 3.6.3)
##    bit64            4.0.5   2020-08-30 [1] CRAN (R 3.6.3)
##    bitops           1.0-7   2021-04-24 [1] CRAN (R 3.6.3)
##    blob             1.2.2   2021-07-23 [1] CRAN (R 3.6.3)
##    bookdown         0.24    2021-09-02 [1] CRAN (R 3.6.3)
##    brio             1.1.3   2021-11-30 [2] CRAN (R 3.6.3)
##    broom            0.7.12  2022-01-28 [1] CRAN (R 3.6.3)
##    bslib            0.3.1   2021-10-06 [1] CRAN (R 3.6.3)
##    cachem           1.0.5   2021-05-15 [1] CRAN (R 3.6.3)
##    callr            3.7.0   2021-04-20 [2] CRAN (R 3.6.3)
##    caTools          1.18.2  2021-03-28 [1] CRAN (R 3.6.3)
##    cellranger       1.1.0   2016-07-27 [1] CRAN (R 3.6.3)
##    checkmate        2.0.0   2020-02-06 [1] CRAN (R 3.6.3)
##    cli              3.1.0   2021-10-27 [1] CRAN (R 3.6.3)
##    cluster          2.1.0   2019-06-19 [2] CRAN (R 3.6.3)
##    codetools        0.2-16  2018-12-24 [2] CRAN (R 3.6.3)
##    colorspace       2.0-2   2021-06-24 [1] CRAN (R 3.6.3)
##    cowplot          1.1.1   2020-12-30 [1] CRAN (R 3.6.3)
##    crayon           1.5.0   2022-02-14 [1] CRAN (R 3.6.3)
##    crosstalk        1.2.0   2021-11-04 [2] CRAN (R 3.6.3)
##    data.table       1.14.0  2021-02-21 [1] CRAN (R 3.6.3)
##    DBI              1.1.1   2021-01-15 [1] CRAN (R 3.6.3)
##    dbplyr           2.1.1   2021-04-06 [1] CRAN (R 3.6.3)
##    desc             1.4.1   2022-03-06 [2] CRAN (R 3.6.3)
##    devtools         2.4.3   2021-11-30 [1] CRAN (R 3.6.3)
##    digest           0.6.29  2021-12-01 [1] CRAN (R 3.6.3)
##    doParallel       1.0.17  2022-02-07 [2] CRAN (R 3.6.3)
##    dplyr          * 1.0.6   2021-05-05 [1] CRAN (R 3.6.3)
##    DT             * 0.23    2022-05-10 [2] CRAN (R 3.6.3)
##    dynamicTreeCut * 1.63-1  2016-03-11 [1] CRAN (R 3.6.3)
##    ellipsis         0.3.2   2021-04-29 [1] CRAN (R 3.6.3)
##    evaluate         0.15    2022-02-18 [2] CRAN (R 3.6.3)
##    fansi            1.0.2   2022-01-14 [1] CRAN (R 3.6.3)
##    farver           2.1.0   2021-02-28 [2] CRAN (R 3.6.3)
##    fastcluster    * 1.2.3   2021-05-24 [1] CRAN (R 3.6.3)
##    fastmap          1.1.0   2021-01-25 [1] CRAN (R 3.6.3)
##    fdrtool          1.2.17  2021-11-13 [1] CRAN (R 3.6.3)
##    forcats        * 0.5.1   2021-01-27 [1] CRAN (R 3.6.3)
##    foreach          1.5.2   2022-02-02 [2] CRAN (R 3.6.3)
##    foreign          0.8-75  2020-01-20 [2] CRAN (R 3.6.3)
##    Formula          1.2-4   2020-10-16 [1] CRAN (R 3.6.3)
##    fs               1.5.2   2021-12-08 [1] CRAN (R 3.6.3)
##    generics         0.1.2   2022-01-31 [1] CRAN (R 3.6.3)
##    ggplot2        * 3.3.5   2021-06-25 [1] CRAN (R 3.6.3)
##    glmnet           4.1-2   2021-06-24 [1] CRAN (R 3.6.3)
##    glue             1.6.1   2022-01-22 [1] CRAN (R 3.6.3)
##    GO.db            3.15.0  2022-09-14 [1] Bioconductor
##    gplots           3.1.1   2020-11-28 [1] CRAN (R 3.6.3)
##    gridExtra        2.3     2017-09-09 [2] CRAN (R 3.6.3)
##    gtable           0.3.0   2019-03-25 [2] CRAN (R 3.6.3)
##    gtools           3.9.2   2021-06-06 [1] CRAN (R 3.6.3)
##    haven            2.4.1   2021-04-23 [1] CRAN (R 3.6.3)
##    highr            0.9     2021-04-16 [1] CRAN (R 3.6.3)
##    Hmisc            4.5-0   2021-02-28 [1] CRAN (R 3.6.3)
##    hms              1.1.1   2021-09-26 [1] CRAN (R 3.6.3)
##    htmlTable        2.3.0   2021-10-12 [1] CRAN (R 3.6.3)
##    htmltools        0.5.2   2021-08-25 [1] CRAN (R 3.6.3)
##    htmlwidgets      1.5.4   2021-09-08 [2] CRAN (R 3.6.3)
##    httr             1.4.3   2022-05-04 [2] CRAN (R 3.6.3)
##    igraph         * 1.3.1   2022-04-20 [2] CRAN (R 3.6.3)
##    IHW              1.14.0  2019-10-29 [1] Bioconductor
##    impute           1.60.0  2019-10-29 [2] Bioconductor
##    IRanges          2.20.2  2020-01-13 [2] Bioconductor
##    iterators        1.0.14  2022-02-05 [2] CRAN (R 3.6.3)
##    jpeg             0.1-9   2021-07-24 [1] CRAN (R 3.6.3)
##    jquerylib        0.1.4   2021-04-26 [1] CRAN (R 3.6.3)
##    jsonlite         1.8.0   2022-02-22 [2] CRAN (R 3.6.3)
##    KEGGREST         1.26.1  2019-11-06 [1] Bioconductor
##    KernSmooth       2.23-16 2019-10-15 [2] CRAN (R 3.6.3)
##    knitr            1.36    2021-09-29 [1] CRAN (R 3.6.3)
##    labeling         0.4.2   2020-10-20 [2] CRAN (R 3.6.3)
##    lattice          0.20-38 2018-11-04 [2] CRAN (R 3.6.3)
##    latticeExtra     0.6-29  2019-12-19 [1] CRAN (R 3.6.3)
##    lifecycle        1.0.1   2021-09-24 [1] CRAN (R 3.6.3)
##    limma            3.42.2  2020-02-03 [2] Bioconductor
##    locfit           1.5-9.4 2020-03-25 [1] CRAN (R 3.6.3)
##    lpsymphony       1.14.0  2019-10-29 [1] Bioconductor (R 3.6.3)
##    lubridate        1.7.10  2021-02-26 [1] CRAN (R 3.6.3)
##    magrittr         2.0.2   2022-01-26 [1] CRAN (R 3.6.3)
##    MASS             7.3-54  2021-05-03 [1] CRAN (R 3.6.3)
##    Matrix           1.3-4   2021-06-01 [1] CRAN (R 3.6.3)
##    matrixStats      0.60.0  2021-07-26 [1] CRAN (R 3.6.3)
##    memoise          2.0.1   2021-11-26 [2] CRAN (R 3.6.3)
##    metagenomeSeq    1.28.2  2020-02-03 [1] Bioconductor
##    mgcv             1.8-31  2019-11-09 [2] CRAN (R 3.6.3)
##    modelr           0.1.8   2020-05-19 [1] CRAN (R 3.6.3)
##    multtest         2.42.0  2019-10-29 [2] Bioconductor
##    munsell          0.5.0   2018-06-12 [2] CRAN (R 3.6.3)
##    nlme             3.1-144 2020-02-06 [2] CRAN (R 3.6.3)
##    nnet             7.3-12  2016-02-02 [2] CRAN (R 3.6.3)
##    permute          0.9-5   2019-03-12 [1] CRAN (R 3.6.3)
##    pheatmap         1.0.12  2019-01-04 [1] CRAN (R 3.6.3)
##    phyloseq       * 1.30.0  2019-10-29 [1] Bioconductor
##    pillar           1.7.0   2022-02-01 [1] CRAN (R 3.6.3)
##    pkgbuild         1.3.1   2021-12-20 [2] CRAN (R 3.6.3)
##    pkgconfig        2.0.3   2019-09-22 [2] CRAN (R 3.6.3)
##    pkgload          1.2.4   2021-11-30 [2] CRAN (R 3.6.3)
##    plyr             1.8.7   2022-03-24 [2] CRAN (R 3.6.3)
##    png              0.1-7   2013-12-03 [1] CRAN (R 3.6.3)
##    preprocessCore   1.48.0  2019-10-29 [2] Bioconductor
##    prettyunits      1.1.1   2020-01-24 [2] CRAN (R 3.6.3)
##    processx         3.5.3   2022-03-25 [2] CRAN (R 3.6.3)
##    ps               1.7.0   2022-04-23 [2] CRAN (R 3.6.3)
##    purrr          * 0.3.4   2020-04-17 [2] CRAN (R 3.6.3)
##    R6               2.5.1   2021-08-19 [1] CRAN (R 3.6.3)
##    RColorBrewer     1.1-3   2022-04-03 [2] CRAN (R 3.6.3)
##    Rcpp             1.0.7   2021-07-07 [1] CRAN (R 3.6.3)
##    readr          * 2.0.0   2021-07-20 [1] CRAN (R 3.6.3)
##    readxl           1.3.1   2019-03-13 [1] CRAN (R 3.6.3)
##    remotes          2.4.2   2021-11-30 [1] CRAN (R 3.6.3)
##    reprex           2.0.1   2021-08-05 [1] CRAN (R 3.6.3)
##    reshape2         1.4.4   2020-04-09 [2] CRAN (R 3.6.3)
##    rhdf5            2.30.1  2019-11-26 [1] Bioconductor
##    Rhdf5lib         1.8.0   2019-10-29 [1] Bioconductor
##    rJava            1.0-5   2021-09-24 [1] CRAN (R 3.6.3)
##  R rlang            1.0.2   &lt;NA&gt;       [2] &lt;NA&gt;
##    rmarkdown        2.11    2021-09-14 [1] CRAN (R 3.6.3)
##    rpart            4.1-15  2019-04-12 [2] CRAN (R 3.6.3)
##    rprojroot        2.0.2   2020-11-15 [1] CRAN (R 3.6.3)
##    RSQLite          2.2.7   2021-04-22 [1] CRAN (R 3.6.3)
##    rstudioapi       0.13    2020-11-12 [2] CRAN (R 3.6.3)
##    rvest            1.0.2   2021-10-16 [1] CRAN (R 3.6.3)
##    S4Vectors        0.24.4  2020-04-09 [2] Bioconductor
##    sass             0.4.0   2021-05-12 [1] CRAN (R 3.6.3)
##    scales           1.2.0   2022-04-13 [2] CRAN (R 3.6.3)
##    sessioninfo      1.2.2   2021-12-06 [2] CRAN (R 3.6.3)
##    shape            1.4.6   2021-05-19 [1] CRAN (R 3.6.3)
##    slam             0.1-49  2021-11-17 [1] CRAN (R 3.6.3)
##    stringi          1.7.4   2021-08-25 [1] CRAN (R 3.6.3)
##    stringr        * 1.4.0   2019-02-10 [2] CRAN (R 3.6.3)
##    survival         3.1-8   2019-12-03 [2] CRAN (R 3.6.3)
##    testthat         3.1.4   2022-04-26 [2] CRAN (R 3.6.3)
##    tibble         * 3.1.6   2021-11-07 [1] CRAN (R 3.6.3)
##    tidyr          * 1.2.0   2022-02-01 [1] CRAN (R 3.6.3)
##    tidyselect       1.1.1   2021-04-30 [1] CRAN (R 3.6.3)
##    tidyverse      * 1.3.1   2021-04-15 [1] CRAN (R 3.6.3)
##    tzdb             0.2.0   2021-10-27 [1] CRAN (R 3.6.3)
##    usethis          2.1.6   2022-05-25 [2] CRAN (R 3.6.3)
##    utf8             1.2.2   2021-07-24 [1] CRAN (R 3.6.3)
##    vctrs            0.3.8   2021-04-29 [1] CRAN (R 3.6.3)
##    vegan            2.5-7   2020-11-28 [1] CRAN (R 3.6.3)
##    WGCNA          * 1.71    2022-04-22 [1] CRAN (R 3.6.3)
##    withr            2.4.3   2021-11-30 [1] CRAN (R 3.6.3)
##    Wrench           1.4.0   2019-10-29 [1] Bioconductor
##    xfun             0.23    2021-05-15 [1] CRAN (R 3.6.3)
##    xlsx           * 0.6.5   2020-11-10 [1] CRAN (R 3.6.3)
##    xlsxjars         0.6.1   2014-08-22 [1] CRAN (R 3.6.3)
##    xml2             1.3.3   2021-11-30 [2] CRAN (R 3.6.3)
##    XVector          0.26.0  2019-10-29 [2] Bioconductor
##    yaml             2.2.2   2022-01-25 [1] CRAN (R 3.6.3)
##    zlibbioc         1.32.0  2019-10-29 [2] Bioconductor
## 
##  [1] /share/home/tongbangzhuo/R/x86_64-pc-linux-gnu-library/3.6
##  [2] /opt/R-3.6.3/lib/R/library
## 
##  R ── Package was removed from disk.
## 
## ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="halla.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/OMicsAnalysis/blob/main/01-WGCNA_network.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OMics analysis workflow.pdf", "OMics analysis workflow.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
